// @telefonica/mistica/dist/accordion.css-mistica.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    chevronContainer: function() {
        return e;
    },
    itemContent: function() {
        return v;
    },
    panel: function() {
        return r;
    },
    panelContainer: function() {
        return a;
    },
    panelTransitionClasses: function() {
        return _;
    },
    touchableBackground: function() {
        return t;
    },
    touchableBackgroundInverse: function() {
        return i;
    }
});
require("./sprinkles.css.ts.vanilla.css-mistica.js");
require("./accordion.css.ts.vanilla.css-mistica.js");
var e = "_8s3szx3", v = "_1y2v1nf7d _1y2v1nf8m _1y2v1nf9v _1y2v1nfb4 _1y2v1nfi7 _1y2v1nfk2 _1y2v1nf7c _1y2v1nfhm _1y2v1nfio", r = "_8s3szx9", a = "_8s3szx4", _ = {
    enter: "_8s3szx5",
    enterActive: "_8s3szx6",
    exit: "_8s3szx7",
    exitActive: "_8s3szx8"
}, t = "_8s3szx1", i = "_8s3szx2";
