// @telefonica/mistica/dist/accordion.js
"use client";
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    Accordion: function() {
        return le;
    },
    AccordionItem: function() {
        return ae;
    },
    BoxedAccordion: function() {
        return ue;
    },
    BoxedAccordionItem: function() {
        return me;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _react = /*#__PURE__*/ _interop_require_wildcard(require("react"));
const _list = require("./list.js");
const _iconchevron = /*#__PURE__*/ _interop_require_default(require("./icons/icon-chevron.js"));
const _box = /*#__PURE__*/ _interop_require_default(require("./box.js"));
const _accordioncssmistica = require("./accordion.css-mistica.js");
const _stack = /*#__PURE__*/ _interop_require_default(require("./stack.js"));
const _touchable = require("./touchable.js");
const _classnames = /*#__PURE__*/ _interop_require_default(require("classnames"));
const _skincontractcssmistica = require("./skins/skin-contract.css-mistica.js");
const _dom = require("./utils/dom.js");
const _divider = /*#__PURE__*/ _interop_require_default(require("./divider.js"));
const _boxed = require("./boxed.js");
const _themevariantcontext = require("./theme-variant-context.js");
const _hooks = require("./hooks.js");
const _reacttransitiongroup = require("react-transition-group");
const _platform = require("./utils/platform.js");
const _sprinklescssmistica = require("./sprinkles.css-mistica.js");
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
function _getRequireWildcardCache(nodeInterop) {
    if (typeof WeakMap !== "function") return null;
    var cacheBabelInterop = new WeakMap();
    var cacheNodeInterop = new WeakMap();
    return (_getRequireWildcardCache = function(nodeInterop) {
        return nodeInterop ? cacheNodeInterop : cacheBabelInterop;
    })(nodeInterop);
}
function _interop_require_wildcard(obj, nodeInterop) {
    if (!nodeInterop && obj && obj.__esModule) {
        return obj;
    }
    if (obj === null || typeof obj !== "object" && typeof obj !== "function") {
        return {
            default: obj
        };
    }
    var cache = _getRequireWildcardCache(nodeInterop);
    if (cache && cache.has(obj)) {
        return cache.get(obj);
    }
    var newObj = {
        __proto__: null
    };
    var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for(var key in obj){
        if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) {
            var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null;
            if (desc && (desc.get || desc.set)) {
                Object.defineProperty(newObj, key, desc);
            } else {
                newObj[key] = obj[key];
            }
        }
    }
    newObj.default = obj;
    if (cache) {
        cache.set(obj, newObj);
    }
    return newObj;
}
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
function _object_without_properties(source, excluded) {
    if (source == null) return {};
    var target = _object_without_properties_loose(source, excluded);
    var key, i;
    if (Object.getOwnPropertySymbols) {
        var sourceSymbolKeys = Object.getOwnPropertySymbols(source);
        for(i = 0; i < sourceSymbolKeys.length; i++){
            key = sourceSymbolKeys[i];
            if (excluded.indexOf(key) >= 0) continue;
            if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
            target[key] = source[key];
        }
    }
    return target;
}
function _object_without_properties_loose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;
    for(i = 0; i < sourceKeys.length; i++){
        key = sourceKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        target[key] = source[key];
    }
    return target;
}
const C = 400, g = /*#__PURE__*/ _react.createContext({
    index: [],
    toggle: ()=>{}
}), H = ()=>_react.useContext(g), b = (param)=>{
    let { value: t, defaultValue: r, onChange: n, singleOpen: i } = param;
    const u = t !== void 0, d = (e)=>e === void 0 ? [] : typeof e == "number" ? [
            e
        ] : e, [c, a] = _react.useState(d(r));
    _react.useEffect(()=>{
        if (c.length > 1 && i) {
            const e = [
                ...c
            ];
            e.splice(1), a(e);
        }
    }, [
        i,
        c
    ]);
    const p = (e, l)=>{
        if (!l) return [
            e
        ];
        const m = l.indexOf(e);
        let I = [
            ...l
        ];
        return m === -1 ? i ? I = [
            e
        ] : I.push(e) : I.splice(m, 1), I;
    }, f = (e)=>{
        if (u || a(p(e, c)), n) {
            const l = (u ? d(t) : c).includes(e);
            n(e, !l);
        }
    };
    return u ? [
        d(t),
        f
    ] : [
        c,
        f
    ];
}, U = (t)=>{
    const r = t == null ? void 0 : t.closest("[data-accordion]");
    if (r) return Array.from(r.querySelectorAll("[data-accordion-item]")).filter((n)=>n.closest("[data-accordion]") === r).findIndex((n)=>n === t);
}, N = /*#__PURE__*/ _react.forwardRef((_param, u)=>{
    var { content: t, dataAttributes: r, trackingEvent: n } = _param, i = _object_without_properties(_param, [
        "content",
        "dataAttributes",
        "trackingEvent"
    ]);
    const d = _react.useRef(null), c = _react.useRef(null), { index: a, toggle: p } = H(), f = (0, _themevariantcontext.useIsInverseVariant)(), e = (0, _hooks.useAriaId)(), l = (0, _hooks.useAriaId)(), [m, I] = _react.useState(), x = m !== void 0 && (a == null ? void 0 : a.includes(m));
    return _react.useEffect(()=>{
        I(U(c.current));
    }, []), /* @__PURE__ */ (0, _jsxruntime.jsxs)("div", _object_spread_props(_object_spread({
        ref: c
    }, (0, _dom.getPrefixedDataAttributes)(_object_spread_props(_object_spread({}, r), {
        "accordion-item": !0
    }))), {
        children: [
            /* @__PURE__ */ (0, _jsxruntime.jsx)(_touchable.BaseTouchable, {
                ref: u,
                className: (0, _classnames.default)(_accordioncssmistica.itemContent, f ? _accordioncssmistica.touchableBackgroundInverse : _accordioncssmistica.touchableBackground),
                onPress: ()=>{
                    m !== void 0 && p(m);
                },
                trackingEvent: n,
                "aria-expanded": x,
                "aria-controls": l,
                children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                    paddingX: 16,
                    children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_list.Content, _object_spread_props(_object_spread({
                        labelId: e
                    }, i), {
                        right: /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                            className: _accordioncssmistica.chevronContainer,
                            children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_iconchevron.default, {
                                size: 24,
                                transitionDuration: C,
                                direction: x ? "up" : "down",
                                color: f ? _skincontractcssmistica.vars.colors.inverse : x ? _skincontractcssmistica.vars.colors.neutralHigh : _skincontractcssmistica.vars.colors.neutralMedium
                            })
                        })
                    }))
                })
            }),
            /* @__PURE__ */ (0, _jsxruntime.jsx)(_reacttransitiongroup.CSSTransition, {
                in: x,
                timeout: (0, _platform.isRunningAcceptanceTest)() ? 0 : C,
                nodeRef: d,
                classNames: _accordioncssmistica.panelTransitionClasses,
                mountOnEnter: !0,
                unmountOnExit: !0,
                children: /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                    className: _accordioncssmistica.panelContainer,
                    ref: d,
                    children: /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                        className: _accordioncssmistica.panel,
                        role: "region",
                        "aria-labelledby": e,
                        id: l,
                        children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                            paddingX: 16,
                            paddingBottom: 16,
                            children: t
                        })
                    })
                })
            })
        ]
    }));
}), ae = /*#__PURE__*/ _react.forwardRef((_param, i)=>{
    var { dataAttributes: t, role: r } = _param, n = _object_without_properties(_param, [
        "dataAttributes",
        "role"
    ]);
    return /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
        role: r,
        className: (0, _sprinklescssmistica.sprinkles)({
            width: "100%"
        }),
        children: /* @__PURE__ */ (0, _jsxruntime.jsx)(N, _object_spread_props(_object_spread({}, n), {
            ref: i,
            dataAttributes: _object_spread({
                "component-name": "AccordionItem"
            }, t)
        }))
    });
}), le = (param)=>{
    let { children: t, dataAttributes: r, index: n, defaultIndex: i, onChange: u, singleOpen: d, role: c } = param;
    const [a, p] = b({
        value: n,
        defaultValue: i,
        onChange: u,
        singleOpen: d
    }), f = _react.Children.toArray(t).filter(Boolean), e = f.length - 1;
    return /* @__PURE__ */ (0, _jsxruntime.jsx)(g.Provider, {
        value: {
            index: a,
            toggle: p
        },
        children: /* @__PURE__ */ (0, _jsxruntime.jsx)("div", _object_spread_props(_object_spread({
            role: c
        }, (0, _dom.getPrefixedDataAttributes)(_object_spread_props(_object_spread({}, r), {
            accordion: !0
        }), "Accordion")), {
            children: f.map((l, m)=>/* @__PURE__ */ (0, _jsxruntime.jsxs)(_react.Fragment, {
                    children: [
                        l,
                        m < e && /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                            paddingX: 16,
                            children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_divider.default, {})
                        })
                    ]
                }, m))
        }))
    });
}, me = /*#__PURE__*/ _react.forwardRef((_param, i)=>{
    var { dataAttributes: t, isInverse: r } = _param, n = _object_without_properties(_param, [
        "dataAttributes",
        "isInverse"
    ]);
    return /* @__PURE__ */ (0, _jsxruntime.jsx)(_boxed.Boxed, {
        isInverse: r,
        ref: i,
        dataAttributes: _object_spread({
            "component-name": "BoxedAccordionItem"
        }, t),
        children: /* @__PURE__ */ (0, _jsxruntime.jsx)(N, _object_spread({}, n))
    });
}), ue = (param)=>{
    let { children: t, dataAttributes: r, index: n, defaultIndex: i, onChange: u, singleOpen: d, role: c } = param;
    const [a, p] = b({
        value: n,
        defaultValue: i,
        onChange: u,
        singleOpen: d
    });
    return /* @__PURE__ */ (0, _jsxruntime.jsx)(g.Provider, {
        value: {
            index: a,
            toggle: p
        },
        children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_stack.default, {
            space: 16,
            role: c,
            dataAttributes: _object_spread({
                "component-name": "BoxedAccordion",
                accordion: !0
            }, r),
            children: t
        })
    });
};
