// @telefonica/mistica/dist/avatar.js
"use client";
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "default", {
    enumerable: true,
    get: function() {
        return O;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _react = /*#__PURE__*/ _interop_require_wildcard(require("react"));
const _badge = /*#__PURE__*/ _interop_require_default(require("./badge.js"));
const _iconuseraccountregular = /*#__PURE__*/ _interop_require_default(require("./generated/mistica-icons/icon-user-account-regular.js"));
const _themevariantcontext = require("./theme-variant-context.js");
const _avatarcssmistica = require("./avatar.css-mistica.js");
const _skincontractcssmistica = require("./skins/skin-contract.css-mistica.js");
const _dom = require("./utils/dom.js");
const _css = require("./utils/css.js");
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
function _getRequireWildcardCache(nodeInterop) {
    if (typeof WeakMap !== "function") return null;
    var cacheBabelInterop = new WeakMap();
    var cacheNodeInterop = new WeakMap();
    return (_getRequireWildcardCache = function(nodeInterop) {
        return nodeInterop ? cacheNodeInterop : cacheBabelInterop;
    })(nodeInterop);
}
function _interop_require_wildcard(obj, nodeInterop) {
    if (!nodeInterop && obj && obj.__esModule) {
        return obj;
    }
    if (obj === null || typeof obj !== "object" && typeof obj !== "function") {
        return {
            default: obj
        };
    }
    var cache = _getRequireWildcardCache(nodeInterop);
    if (cache && cache.has(obj)) {
        return cache.get(obj);
    }
    var newObj = {
        __proto__: null
    };
    var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for(var key in obj){
        if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) {
            var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null;
            if (desc && (desc.get || desc.set)) {
                Object.defineProperty(newObj, key, desc);
            } else {
                newObj[key] = obj[key];
            }
        }
    }
    newObj.default = obj;
    if (cache) {
        cache.set(obj, newObj);
    }
    return newObj;
}
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
function _object_without_properties(source, excluded) {
    if (source == null) return {};
    var target = _object_without_properties_loose(source, excluded);
    var key, i;
    if (Object.getOwnPropertySymbols) {
        var sourceSymbolKeys = Object.getOwnPropertySymbols(source);
        for(i = 0; i < sourceSymbolKeys.length; i++){
            key = sourceSymbolKeys[i];
            if (excluded.indexOf(key) >= 0) continue;
            if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
            target[key] = source[key];
        }
    }
    return target;
}
function _object_without_properties_loose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;
    for(i = 0; i < sourceKeys.length; i++){
        key = sourceKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        target[key] = source[key];
    }
    return target;
}
const $ = (r, o)=>{
    if (!o) return 0;
    const t = r / 2, e = (0, _css.pxToRem)(o === !0 ? 5 : 10);
    return `calc(${t * (1 - Math.sin(Math.PI / 4))}px - ${e})`;
}, k = (r, o)=>{
    if (!o) return null;
    let t;
    return r <= 40 ? t = 14 : r <= 64 ? t = 16 : t = 18, /* @__PURE__ */ (0, _jsxruntime.jsx)("span", {
        style: {
            fontSize: t,
            textTransform: "uppercase"
        },
        children: o
    });
}, y = (_param)=>{
    var { size: r, src: o, Icon: t = _iconuseraccountregular.default, badge: e, initials: c = "", "aria-label": u, dataAttributes: f } = _param, s = _object_without_properties(_param, [
        "size",
        "src",
        "Icon",
        "badge",
        "initials",
        "aria-label",
        "dataAttributes"
    ]);
    var _s_backgroundColor, _s_textColor;
    const l = (0, _themevariantcontext.useIsInverseVariant)(), g = (_s_backgroundColor = s.backgroundColor) !== null && _s_backgroundColor !== void 0 ? _s_backgroundColor : l ? _skincontractcssmistica.vars.colors.brandHigh : _skincontractcssmistica.vars.colors.brandLow, b = (_s_textColor = s.textColor) !== null && _s_textColor !== void 0 ? _s_textColor : l ? _skincontractcssmistica.vars.colors.textPrimaryInverse : _skincontractcssmistica.vars.colors.textBrand, [h, i] = _react.useState(!1);
    _react.useEffect(()=>{
        i(!1);
    }, [
        o
    ]);
    const p = s.border ? `1px solid ${_skincontractcssmistica.vars.colors.borderLow}` : "none", x = c.trim().slice(0, 2), m = $(r, e), v = e === !0 ? void 0 : e || 0, I = !!o && !h, A = r <= 40 ? 16 : 24;
    return /* @__PURE__ */ (0, _jsxruntime.jsx)(_badge.default, {
        value: v,
        top: m,
        right: m,
        children: /* @__PURE__ */ (0, _jsxruntime.jsx)("div", _object_spread_props(_object_spread({
            className: _avatarcssmistica.avatar,
            role: "img",
            "aria-label": u !== null && u !== void 0 ? u : c,
            style: {
                width: r,
                height: r,
                color: b,
                background: g,
                boxSizing: "border-box",
                border: p
            }
        }, (0, _dom.getPrefixedDataAttributes)(f, "Avatar")), {
            children: I ? /* @__PURE__ */ (0, _jsxruntime.jsx)("img", {
                src: o,
                className: _avatarcssmistica.image,
                onError: ()=>i(!0),
                role: "none",
                width: r,
                height: r
            }) : k(r, x) || /* @__PURE__ */ (0, _jsxruntime.jsx)(t, {
                size: A,
                color: "currentColor"
            })
        }))
    });
}, O = y;
