// @telefonica/mistica/dist/boxed.css-mistica.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    boxBorder: function() {
        return e;
    },
    boxed: function() {
        return a;
    },
    desktopOnly: function() {
        return r;
    },
    vars: function() {
        return b;
    }
});
require("./sprinkles.css.ts.vanilla.css-mistica.js");
require("./boxed.css.ts.vanilla.css-mistica.js");
var e = "_1y2v1nfk1", a = "ihbkeqk", r = "ihbkeqm", b = {
    width: "var(--ihbkeq0)",
    height: "var(--ihbkeq1)",
    minHeight: "var(--ihbkeq2)",
    maxWidth: "var(--ihbkeq3)",
    minWidth: "var(--ihbkeq4)",
    mobile: {
        width: "var(--ihbkeq5)",
        height: "var(--ihbkeq6)",
        minHeight: "var(--ihbkeq7)",
        maxWidth: "var(--ihbkeq8)",
        minWidth: "var(--ihbkeq9)"
    },
    tablet: {
        width: "var(--ihbkeqa)",
        height: "var(--ihbkeqb)",
        minHeight: "var(--ihbkeqc)",
        maxWidth: "var(--ihbkeqd)",
        minWidth: "var(--ihbkeqe)"
    },
    desktop: {
        width: "var(--ihbkeqf)",
        height: "var(--ihbkeqg)",
        minHeight: "var(--ihbkeqh)",
        maxWidth: "var(--ihbkeqi)",
        minWidth: "var(--ihbkeqj)"
    }
};
