// @telefonica/mistica/dist/button.js
"use client";
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    ButtonDanger: function() {
        return De;
    },
    ButtonLink: function() {
        return Re;
    },
    ButtonLinkDanger: function() {
        return Be;
    },
    ButtonPrimary: function() {
        return Te;
    },
    ButtonSecondary: function() {
        return Ae;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _react = /*#__PURE__*/ _interop_require_wildcard(require("react"));
const _classnames = /*#__PURE__*/ _interop_require_default(require("classnames"));
const _spinner = /*#__PURE__*/ _interop_require_default(require("./spinner.js"));
const _touchable = require("./touchable.js");
const _themevariantcontext = require("./theme-variant-context.js");
const _formcontext = require("./form-context.js");
const _css = require("./utils/css.js");
const _text = require("./text.js");
const _box = /*#__PURE__*/ _interop_require_default(require("./box.js"));
const _common = require("./utils/common.js");
const _analytics = require("./utils/analytics.js");
const _hooks = require("./hooks.js");
const _utils = require("./skins/utils.js");
const _buttoncssmistica = require("./button.css-mistica.js");
const _constants = require("./skins/constants.js");
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
function _getRequireWildcardCache(nodeInterop) {
    if (typeof WeakMap !== "function") return null;
    var cacheBabelInterop = new WeakMap();
    var cacheNodeInterop = new WeakMap();
    return (_getRequireWildcardCache = function(nodeInterop) {
        return nodeInterop ? cacheNodeInterop : cacheBabelInterop;
    })(nodeInterop);
}
function _interop_require_wildcard(obj, nodeInterop) {
    if (!nodeInterop && obj && obj.__esModule) {
        return obj;
    }
    if (obj === null || typeof obj !== "object" && typeof obj !== "function") {
        return {
            default: obj
        };
    }
    var cache = _getRequireWildcardCache(nodeInterop);
    if (cache && cache.has(obj)) {
        return cache.get(obj);
    }
    var newObj = {
        __proto__: null
    };
    var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for(var key in obj){
        if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) {
            var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null;
            if (desc && (desc.get || desc.set)) {
                Object.defineProperty(newObj, key, desc);
            } else {
                newObj[key] = obj[key];
            }
        }
    }
    newObj.default = obj;
    if (cache) {
        cache.set(obj, newObj);
    }
    return newObj;
}
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
function _object_without_properties(source, excluded) {
    if (source == null) return {};
    var target = _object_without_properties_loose(source, excluded);
    var key, i;
    if (Object.getOwnPropertySymbols) {
        var sourceSymbolKeys = Object.getOwnPropertySymbols(source);
        for(i = 0; i < sourceSymbolKeys.length; i++){
            key = sourceSymbolKeys[i];
            if (excluded.indexOf(key) >= 0) continue;
            if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
            target[key] = source[key];
        }
    }
    return target;
}
function _object_without_properties_loose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;
    for(i = 0; i < sourceKeys.length; i++){
        key = sourceKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        target[key] = source[key];
    }
    return target;
}
const k = (param)=>{
    let { content: e, defaultIconSize: t, renderText: i } = param;
    const l = (0, _utils.flattenChildren)(e), u = l.length, c = [];
    let s = [];
    const f = ()=>{
        c.push(/* @__PURE__ */ (0, _jsxruntime.jsx)(_react.Fragment, {
            children: i(s)
        }, c.length)), s = [];
    };
    return l.forEach((d, m)=>{
        const h = m === 0, o = m === u - 1;
        if (/*#__PURE__*/ _react.isValidElement(d)) {
            s.length && f();
            var _d_props_size;
            const v = (_d_props_size = d.props.size) !== null && _d_props_size !== void 0 ? _d_props_size : t;
            c.push(/* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                style: {
                    display: "flex",
                    alignItems: "center",
                    marginLeft: h ? 0 : _buttoncssmistica.ICON_MARGIN_PX,
                    marginRight: o ? 0 : _buttoncssmistica.ICON_MARGIN_PX
                },
                children: /*#__PURE__*/ _react.cloneElement(d, {
                    size: (0, _css.pxToRem)(v)
                })
            }, c.length));
        } else s.push(d), o && f();
    }), c;
}, he = ()=>{
    const { skinName: e } = (0, _hooks.useTheme)();
    return e === _constants.VIVO_NEW_SKIN ? /* @__PURE__ */ (0, _jsxruntime.jsx)("svg", {
        width: "0.5em",
        height: "0.5em",
        viewBox: "0 0 8 8",
        fill: "none",
        children: /* @__PURE__ */ (0, _jsxruntime.jsx)("path", {
            d: "M5.11111 3.68886L2.73534 1.30383C2.63671 1.21526 2.51339 1.1909 2.4052 1.20285C2.29952 1.21452 2.19613 1.26162 2.12144 1.33458C2.04957 1.40271 2.00869 1.50793 2.00126 1.60783C1.9934 1.71334 2.02104 1.83978 2.12131 1.93639L4.33333 4.07775L2.11406 6.20689L2.11057 6.21056L2.10821 6.21315L2.10381 6.21806C2.01525 6.31669 1.99089 6.44001 2.00284 6.5482C2.01451 6.65388 2.06161 6.75727 2.13456 6.83196C2.20269 6.90383 2.30792 6.94471 2.40782 6.95215C2.51332 6.96 2.63976 6.93237 2.73638 6.83209L5.11111 4.46664C5.32589 4.25186 5.32589 3.90364 5.11111 3.68886Z",
            fill: "currentColor"
        })
    }) : /* @__PURE__ */ (0, _jsxruntime.jsx)("svg", {
        width: "0.5em",
        height: "0.5em",
        viewBox: "0 0 8 8",
        children: /* @__PURE__ */ (0, _jsxruntime.jsx)("path", {
            d: "M6.32595 3.46071L3.03801 0.158595L3.03292 0.153747L3.032 0.152903L3.02931 0.150463L3.02848 0.149738L3.02248 0.144353C2.88533 0.021206 2.71386 -0.0126731 2.56343 0.00394249C2.41648 0.0201739 2.27272 0.0856702 2.16886 0.18711C2.06893 0.281844 2.01209 0.42816 2.00175 0.567065C1.99083 0.71377 2.02925 0.889583 2.16869 1.02392L5.24446 4.00145L2.15859 6.96199L2.15375 6.96708L2.1529 6.968L2.15046 6.97069L2.14974 6.97152L2.14435 6.97752C2.02121 7.11467 1.98733 7.28614 2.00394 7.43657C2.02017 7.58352 2.08567 7.72728 2.18711 7.83114C2.28184 7.93107 2.42816 7.98791 2.56706 7.99825C2.71377 8.00917 2.88958 7.97075 3.02392 7.83132L6.32595 4.5422V4.5422C6.6246 4.24355 6.6246 3.75935 6.32595 3.46071V3.46071Z",
            fill: "currentColor"
        })
    });
}, X = (param)=>{
    let { showSpinner: e, children: t, small: i, loadingText: l, shouldRenderSpinner: u, setShouldRenderSpinner: c, renderText: s, textContentStyle: f, StartIcon: d, EndIcon: m, withChevron: h } = param;
    const o = i ? _buttoncssmistica.SMALL_ICON_SIZE : _buttoncssmistica.ICON_SIZE, r = (0, _css.pxToRem)(i ? _buttoncssmistica.SMALL_SPINNER_SIZE : _buttoncssmistica.SPINNER_SIZE);
    return /* @__PURE__ */ (0, _jsxruntime.jsxs)(_jsxruntime.Fragment, {
        children: [
            /* @__PURE__ */ (0, _jsxruntime.jsxs)("div", {
                "aria-hidden": e ? !0 : void 0,
                className: f,
                children: [
                    d && /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                        style: {
                            display: "flex",
                            alignItems: "center",
                            marginRight: _buttoncssmistica.ICON_MARGIN_PX
                        },
                        children: /* @__PURE__ */ (0, _jsxruntime.jsx)(d, {
                            size: (0, _css.pxToRem)(o),
                            color: "currentColor"
                        })
                    }),
                    /* @__PURE__ */ (0, _jsxruntime.jsxs)("div", {
                        style: {
                            display: "flex",
                            alignItems: "baseline"
                        },
                        children: [
                            k({
                                content: t,
                                defaultIconSize: o,
                                renderText: s
                            }),
                            h && /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                                style: {
                                    display: "flex",
                                    alignItems: "center",
                                    marginLeft: _buttoncssmistica.CHEVRON_MARGIN_LEFT_LINK
                                },
                                children: /* @__PURE__ */ (0, _jsxruntime.jsx)(he, {})
                            })
                        ]
                    }),
                    m && !h && /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                        style: {
                            display: "flex",
                            alignItems: "center",
                            marginLeft: _buttoncssmistica.ICON_MARGIN_PX
                        },
                        children: /* @__PURE__ */ (0, _jsxruntime.jsx)(m, {
                            size: (0, _css.pxToRem)(o),
                            color: "currentColor"
                        })
                    })
                ]
            }),
            /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                className: _buttoncssmistica.loadingFiller,
                "aria-hidden": !0,
                style: l ? {
                    paddingLeft: r,
                    paddingRight: _buttoncssmistica.ICON_MARGIN_PX + 2 * (i ? _buttoncssmistica.X_SMALL_PADDING_PX : _buttoncssmistica.X_PADDING_PX)
                } : void 0,
                children: k({
                    content: l,
                    defaultIconSize: o,
                    renderText: s
                })
            }),
            /* @__PURE__ */ (0, _jsxruntime.jsxs)("div", {
                "aria-hidden": e ? void 0 : !0,
                className: _buttoncssmistica.loadingContent,
                onTransitionEnd: ()=>{
                    e !== u && c(e);
                },
                children: [
                    u ? /* @__PURE__ */ (0, _jsxruntime.jsx)(_spinner.default, {
                        rolePresentation: !!l,
                        color: "currentcolor",
                        delay: "0s",
                        size: r
                    }) : /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                        style: {
                            display: "inline-block",
                            width: r,
                            height: r
                        }
                    }),
                    l ? /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                        paddingLeft: 8,
                        children: k({
                            content: l,
                            defaultIconSize: o,
                            renderText: s
                        })
                    }) : null
                ]
            })
        ]
    });
}, R = /*#__PURE__*/ _react.forwardRef((e, t)=>{
    const { textPresets: i } = (0, _hooks.useTheme)(), { eventFormat: l } = (0, _analytics.useTrackingConfig)(), { formStatus: u, formId: c } = (0, _formcontext.useForm)(), s = (0, _themevariantcontext.useIsInverseVariant)(), { loadingText: f } = e, d = !!e.submit, m = u === "sending", [h, o] = _react.useState(!1), r = e.showSpinner || m && d || h, [v, L] = _react.useState(!!r);
    _react.useEffect(()=>{
        r && !v && L(!0);
    }, [
        r,
        v,
        u
    ]);
    var _e_trackingEvent;
    const N = ()=>l === "google-analytics-4" ? {
            name: _analytics.eventNames.userInteraction,
            component_type: `${e.type}_button`,
            component_copy: (0, _common.getTextFromChildren)(e.children)
        } : {
            category: _analytics.eventCategories.userInteraction,
            action: `${e.type}_button_tapped`,
            label: (0, _common.getTextFromChildren)(e.children)
        }, _ = (I)=>e.small ? /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text, {
            size: 14,
            lineHeight: 20,
            weight: i.button.weight,
            truncate: 1,
            color: "inherit",
            as: "div",
            children: I
        }) : /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text3, {
            weight: i.button.weight,
            truncate: 1,
            color: "inherit",
            as: "div",
            children: I
        }), b = {
        ref: t,
        className: (0, _classnames.default)(s ? _buttoncssmistica.inverseButtonVariants[e.type] : _buttoncssmistica.buttonVariants[e.type], e.className, {
            [_buttoncssmistica.small]: e.small,
            [_buttoncssmistica.isLoading]: r
        }),
        style: _object_spread({
            cursor: e.fake ? "pointer" : void 0
        }, e.style),
        trackingEvent: (_e_trackingEvent = e.trackingEvent) !== null && _e_trackingEvent !== void 0 ? _e_trackingEvent : e.trackEvent ? N() : void 0,
        dataAttributes: e.dataAttributes,
        "aria-label": e["aria-label"],
        "aria-controls": e["aria-controls"],
        "aria-expanded": e["aria-expanded"],
        "aria-haspopup": e["aria-haspopup"],
        tabIndex: e.tabIndex,
        children: X({
            showSpinner: r,
            shouldRenderSpinner: v,
            setShouldRenderSpinner: L,
            children: e.children,
            loadingText: f,
            small: e.small,
            renderText: _,
            textContentStyle: _buttoncssmistica.textContent,
            StartIcon: e.StartIcon,
            EndIcon: e.EndIcon
        }),
        disabled: e.disabled || r || m,
        role: e.role
    };
    if (process.env.NODE_ENV !== "production" && (e.to === "" || e.href === "")) throw Error("to or href props are empty strings");
    if (e.fake) return /* @__PURE__ */ (0, _jsxruntime.jsx)(_touchable.BaseTouchable, _object_spread_props(_object_spread({
        maybe: !0
    }, b), {
        role: "presentation",
        "aria-hidden": "true"
    }));
    if (e.submit) return /* @__PURE__ */ (0, _jsxruntime.jsx)(_touchable.BaseTouchable, _object_spread({
        type: "submit",
        formId: c,
        onPress: ()=>{}
    }, b));
    if (e.onPress) return /* @__PURE__ */ (0, _jsxruntime.jsx)(_touchable.BaseTouchable, _object_spread_props(_object_spread({}, b), {
        onPress: (I)=>{
            const y = e.onPress(I);
            y && (o(!0), y.finally(()=>o(!1)));
        }
    }));
    if (e.to || e.to === "") return /* @__PURE__ */ (0, _jsxruntime.jsx)(_touchable.BaseTouchable, _object_spread_props(_object_spread({}, b), {
        to: e.to,
        newTab: e.newTab,
        fullPageOnWebView: e.fullPageOnWebView,
        onNavigate: e.onNavigate
    }));
    if (e.href || e.href === "") return /* @__PURE__ */ (0, _jsxruntime.jsx)(_touchable.BaseTouchable, _object_spread_props(_object_spread({}, b), {
        href: e.href,
        newTab: e.newTab,
        loadOnTop: e.loadOnTop,
        onNavigate: e.onNavigate
    }));
    if (process.env.NODE_ENV !== "production") throw Error("Bad button props");
    return null;
}), Z = /*#__PURE__*/ _react.forwardRef((_param, i)=>{
    var { type: e } = _param, t = _object_without_properties(_param, [
        "type"
    ]);
    var _t_withChevron;
    const { formStatus: l } = (0, _formcontext.useForm)(), u = (0, _themevariantcontext.useIsInverseVariant)(), { textPresets: c } = (0, _hooks.useTheme)(), { eventFormat: s } = (0, _analytics.useTrackingConfig)(), { isDarkMode: f } = (0, _hooks.useTheme)(), { loadingText: d } = t, m = l === "sending", [h, o] = _react.useState(!1), r = t.showSpinner || h, v = (_t_withChevron = t.withChevron) !== null && _t_withChevron !== void 0 ? _t_withChevron : !!t.href || !!t.to, [L, N] = _react.useState(!!r);
    _react.useEffect(()=>{
        r && !L && N(!0);
    }, [
        r,
        L,
        l
    ]);
    var _t_trackingEvent;
    const _ = ()=>s === "google-analytics-4" ? {
            name: _analytics.eventNames.userInteraction,
            component_type: e === "danger" ? "danger_link" : "link",
            component_copy: (0, _common.getTextFromChildren)(t.children)
        } : {
            category: _analytics.eventCategories.userInteraction,
            action: _analytics.eventActions.linkTapped,
            label: (0, _common.getTextFromChildren)(t.children)
        }, b = (x)=>/* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text2, {
            weight: c.button.weight,
            truncate: 1,
            color: "inherit",
            children: x
        }), I = e === "danger" && f && u ? "dangerDark" : e, y = {
        className: (0, _classnames.default)(u ? _buttoncssmistica.inverseLinkVariants[I] : _buttoncssmistica.linkVariants[I], {
            [_buttoncssmistica.isLoading]: r
        }),
        /**
     * Setting bleed classes with style to override the margin:0 set by the Touchable component.
     * If we set it using className, it may not work depending on the order in which the styles are applied.
     */ style: _object_spread({}, t.bleedLeft ? {
            marginLeft: -_buttoncssmistica.PADDING_X_LINK
        } : void 0, t.bleedRight ? {
            marginRight: -_buttoncssmistica.PADDING_X_LINK
        } : void 0, t.bleedY ? {
            marginTop: -_buttoncssmistica.PADDING_Y_LINK,
            marginBottom: -_buttoncssmistica.PADDING_Y_LINK
        } : void 0),
        trackingEvent: (_t_trackingEvent = t.trackingEvent) !== null && _t_trackingEvent !== void 0 ? _t_trackingEvent : t.trackEvent ? _() : void 0,
        dataAttributes: t.dataAttributes,
        "aria-label": t["aria-label"],
        "aria-controls": t["aria-controls"],
        "aria-expanded": t["aria-expanded"],
        "aria-haspopup": t["aria-haspopup"],
        children: X({
            showSpinner: r,
            shouldRenderSpinner: L,
            setShouldRenderSpinner: N,
            children: t.children,
            loadingText: d,
            small: !0,
            renderText: b,
            textContentStyle: _buttoncssmistica.textContentLink,
            StartIcon: t.StartIcon,
            EndIcon: t.EndIcon,
            withChevron: v
        }),
        disabled: t.disabled || r || m,
        role: t.role
    };
    if (process.env.NODE_ENV !== "production" && (t.to === "" || t.href === "")) throw Error("to or href props are empty strings");
    if (t.onPress) return /* @__PURE__ */ (0, _jsxruntime.jsx)(_touchable.BaseTouchable, _object_spread_props(_object_spread({
        ref: i
    }, y), {
        onPress: (x)=>{
            const B = t.onPress(x);
            B && (o(!0), B.finally(()=>o(!1)));
        }
    }));
    if (t.to || t.to === "") return /* @__PURE__ */ (0, _jsxruntime.jsx)(_touchable.BaseTouchable, _object_spread_props(_object_spread({
        ref: i
    }, y), {
        to: t.to,
        newTab: t.newTab,
        fullPageOnWebView: t.fullPageOnWebView,
        onNavigate: t.onNavigate
    }));
    if (t.href || t.href === "") return /* @__PURE__ */ (0, _jsxruntime.jsx)(_touchable.BaseTouchable, _object_spread_props(_object_spread({
        ref: i
    }, y), {
        href: t.href,
        newTab: t.newTab,
        onNavigate: t.onNavigate
    }));
    if (process.env.NODE_ENV !== "production") throw Error("Bad button props");
    return null;
}), Re = /*#__PURE__*/ _react.forwardRef((_param, i)=>{
    var { dataAttributes: e } = _param, t = _object_without_properties(_param, [
        "dataAttributes"
    ]);
    return /* @__PURE__ */ (0, _jsxruntime.jsx)(Z, _object_spread_props(_object_spread({
        dataAttributes: _object_spread({
            "component-name": "ButtonLink"
        }, e)
    }, t), {
        ref: i,
        type: "default"
    }));
}), Be = /*#__PURE__*/ _react.forwardRef((_param, i)=>{
    var { dataAttributes: e } = _param, t = _object_without_properties(_param, [
        "dataAttributes"
    ]);
    return /* @__PURE__ */ (0, _jsxruntime.jsx)(Z, _object_spread_props(_object_spread({
        dataAttributes: _object_spread({
            "component-name": "ButtonLinkDanger"
        }, e)
    }, t), {
        withChevron: !1,
        ref: i,
        type: "danger"
    }));
}), Te = /*#__PURE__*/ _react.forwardRef((_param, i)=>{
    var { dataAttributes: e } = _param, t = _object_without_properties(_param, [
        "dataAttributes"
    ]);
    return /* @__PURE__ */ (0, _jsxruntime.jsx)(R, _object_spread_props(_object_spread({
        dataAttributes: _object_spread({
            "component-name": "ButtonPrimary"
        }, e)
    }, t), {
        ref: i,
        type: "primary"
    }));
}), Ae = /*#__PURE__*/ _react.forwardRef((_param, i)=>{
    var { dataAttributes: e } = _param, t = _object_without_properties(_param, [
        "dataAttributes"
    ]);
    return /* @__PURE__ */ (0, _jsxruntime.jsx)(R, _object_spread_props(_object_spread({
        dataAttributes: _object_spread({
            "component-name": "ButtonSecondary"
        }, e)
    }, t), {
        ref: i,
        type: "secondary"
    }));
}), De = /*#__PURE__*/ _react.forwardRef((_param, i)=>{
    var { dataAttributes: e } = _param, t = _object_without_properties(_param, [
        "dataAttributes"
    ]);
    return /* @__PURE__ */ (0, _jsxruntime.jsx)(R, _object_spread_props(_object_spread({
        dataAttributes: _object_spread({
            "component-name": "ButtonDanger"
        }, e)
    }, t), {
        ref: i,
        type: "danger"
    }));
});
