// @telefonica/mistica/dist/button-layout.css-mistica.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    alignVariant: function() {
        return y;
    },
    container: function() {
        return l;
    },
    containerWithTwoButtons: function() {
        return i;
    },
    link: function() {
        return k;
    },
    linkInNewLine: function() {
        return r;
    },
    margins: function() {
        return t;
    }
});
require("./sprinkles.css.ts.vanilla.css-mistica.js");
require("./button.css.ts.vanilla.css-mistica.js");
require("./button-layout.css.ts.vanilla.css-mistica.js");
var y = {
    right: "kfxylh8 _1y2v1nfhu",
    left: "kfxylh9 _1y2v1nfhs",
    center: "kfxylha _1y2v1nfht",
    "full-width": "kfxylhb _1y2v1nfht"
}, l = "kfxylh2 _1y2v1nfhk _1y2v1nfhz", i = "kfxylh3", k = "kfxylhe kfxylhd _1y2v1nfhk _1y2v1nfi7", r = "kfxylhf kfxylhd _1y2v1nfhk _1y2v1nfi7", t = "kfxylh0";
