// @telefonica/mistica/dist/button-group.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "default", {
    enumerable: true,
    get: function() {
        return A;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _classnames = /*#__PURE__*/ _interop_require_default(require("classnames"));
const _dom = require("./utils/dom.js");
const _buttongroupcssmistica = require("./button-group.css-mistica.js");
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
const k = (param)=>{
    let { primaryButton: t, secondaryButton: o, link: s, align: e = "left", dataAttributes: m } = param;
    var _e_mobile, _e_tablet, _ref, _e_desktop;
    const d = !!t || !!o || !!s, a = !!t && !!o, l = typeof e == "string" ? {
        mobile: e,
        tablet: e,
        desktop: e
    } : {
        mobile: (_e_mobile = e.mobile) !== null && _e_mobile !== void 0 ? _e_mobile : "left",
        tablet: (_ref = (_e_tablet = e.tablet) !== null && _e_tablet !== void 0 ? _e_tablet : e.mobile) !== null && _ref !== void 0 ? _ref : "left",
        desktop: (_e_desktop = e.desktop) !== null && _e_desktop !== void 0 ? _e_desktop : "left"
    };
    return d ? /* @__PURE__ */ (0, _jsxruntime.jsxs)("div", _object_spread_props(_object_spread({
        className: (0, _classnames.default)(_buttongroupcssmistica.inline, _buttongroupcssmistica.container, {
            [_buttongroupcssmistica.centerInDesktop]: l.desktop === "center",
            [_buttongroupcssmistica.centerInTablet]: l.tablet === "center",
            [_buttongroupcssmistica.centerInMobile]: l.mobile === "center"
        })
    }, (0, _dom.getPrefixedDataAttributes)(m, "ButtonGroup")), {
        children: [
            (t || o) && /* @__PURE__ */ (0, _jsxruntime.jsxs)("div", {
                className: (0, _classnames.default)(_buttongroupcssmistica.inline, _buttongroupcssmistica.buttons),
                children: [
                    t && /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                        className: _buttongroupcssmistica.buttonChild,
                        children: t
                    }),
                    o && /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                        className: _buttongroupcssmistica.buttonChild,
                        children: o
                    })
                ]
            }),
            s && /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                className: _buttongroupcssmistica.buttonChild,
                style: {
                    width: a ? "100%" : "auto"
                },
                children: s
            })
        ]
    })) : null;
}, A = k;
