// @telefonica/mistica/dist/button.css-mistica.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    CHEVRON_MARGIN_LEFT_LINK: function() {
        return _;
    },
    ICON_MARGIN_PX: function() {
        return y;
    },
    ICON_SIZE: function() {
        return f;
    },
    PADDING_X_LINK: function() {
        return r;
    },
    PADDING_Y_LINK: function() {
        return a;
    },
    SMALL_ICON_SIZE: function() {
        return b;
    },
    SMALL_SPINNER_SIZE: function() {
        return k;
    },
    SPINNER_SIZE: function() {
        return e;
    },
    X_PADDING_PX: function() {
        return h;
    },
    X_SMALL_PADDING_PX: function() {
        return p;
    },
    buttonVariants: function() {
        return i;
    },
    inverseButtonVariants: function() {
        return l;
    },
    inverseLinkVariants: function() {
        return t;
    },
    isLoading: function() {
        return o;
    },
    linkVariants: function() {
        return d;
    },
    loadingContent: function() {
        return g;
    },
    loadingFiller: function() {
        return I;
    },
    small: function() {
        return N;
    },
    textContent: function() {
        return m;
    },
    textContentLink: function() {
        return L;
    }
});
require("./sprinkles.css.ts.vanilla.css-mistica.js");
require("./button.css.ts.vanilla.css-mistica.js");
var _ = 2, y = 8, f = 24, r = 12, a = 6, b = 20, k = 16, e = 20, h = 14.5, p = 10.5, i = {
    primary: "rrbrpno rrbrpn2 _1y2v1nf7d _1y2v1nf8m _1y2v1nf9v _1y2v1nfb4 _1y2v1nfho _1y2v1nfhe _1y2v1nfi8 _1y2v1nfk9 _1y2v1nfkl _1y2v1nf2c _1y2v1nf4l",
    secondary: "rrbrpnp rrbrpn2 _1y2v1nf7d _1y2v1nf8m _1y2v1nf9v _1y2v1nfb4 _1y2v1nfho _1y2v1nfhe _1y2v1nfi8 _1y2v1nfk9 _1y2v1nfkl _1y2v1nf2f _1y2v1nf7c",
    danger: "rrbrpnq rrbrpn2 _1y2v1nf7d _1y2v1nf8m _1y2v1nf9v _1y2v1nfb4 _1y2v1nfho _1y2v1nfhe _1y2v1nfi8 _1y2v1nfk9 _1y2v1nfkl _1y2v1nf2c _1y2v1nf4d"
}, l = {
    primary: "rrbrpnr rrbrpn2 _1y2v1nf7d _1y2v1nf8m _1y2v1nf9v _1y2v1nfb4 _1y2v1nfho _1y2v1nfhe _1y2v1nfi8 _1y2v1nfk9 _1y2v1nfkl _1y2v1nf2d _1y2v1nf4m",
    secondary: "rrbrpns rrbrpn2 _1y2v1nf7d _1y2v1nf8m _1y2v1nf9v _1y2v1nfb4 _1y2v1nfho _1y2v1nfhe _1y2v1nfi8 _1y2v1nfk9 _1y2v1nfkl _1y2v1nf2h _1y2v1nf7c",
    danger: "rrbrpnt rrbrpn2 _1y2v1nf7d _1y2v1nf8m _1y2v1nf9v _1y2v1nfb4 _1y2v1nfho _1y2v1nfhe _1y2v1nfi8 _1y2v1nfk9 _1y2v1nfkl _1y2v1nf2c _1y2v1nf4d"
}, t = {
    default: "rrbrpnx rrbrpng _1y2v1nfa7 _1y2v1nfbg _1y2v1nfho _1y2v1nfi8 _1y2v1nfhe _1y2v1nfk9 _1y2v1nfk2 _1y2v1nfkl _1y2v1nfje _1y2v1nf2k _1y2v1nf7c",
    danger: "rrbrpny rrbrpng _1y2v1nfa7 _1y2v1nfbg _1y2v1nfho _1y2v1nfi8 _1y2v1nfhe _1y2v1nfk9 _1y2v1nfk2 _1y2v1nfkl _1y2v1nfje _1y2v1nf2l _1y2v1nf4h",
    dangerDark: "rrbrpnz rrbrpng _1y2v1nfa7 _1y2v1nfbg _1y2v1nfho _1y2v1nfi8 _1y2v1nfhe _1y2v1nfk9 _1y2v1nfk2 _1y2v1nfkl _1y2v1nfje _1y2v1nf2l _1y2v1nf7c"
}, o = "rrbrpn0", d = {
    default: "rrbrpnu rrbrpng _1y2v1nfa7 _1y2v1nfbg _1y2v1nfho _1y2v1nfi8 _1y2v1nfhe _1y2v1nfk9 _1y2v1nfk2 _1y2v1nfkl _1y2v1nfje _1y2v1nf2j _1y2v1nf7c",
    danger: "rrbrpnv rrbrpng _1y2v1nfa7 _1y2v1nfbg _1y2v1nfho _1y2v1nfi8 _1y2v1nfhe _1y2v1nfk9 _1y2v1nfk2 _1y2v1nfkl _1y2v1nfje _1y2v1nf2l _1y2v1nf7c",
    dangerDark: "rrbrpnw rrbrpng _1y2v1nfa7 _1y2v1nfbg _1y2v1nfho _1y2v1nfi8 _1y2v1nfhe _1y2v1nfk9 _1y2v1nfk2 _1y2v1nfkl _1y2v1nfje _1y2v1nf2l _1y2v1nf7c"
}, g = "rrbrpn7 _1y2v1nfhl _1y2v1nfhf _1y2v1nfkr _1y2v1nfm0 _1y2v1nfht _1y2v1nfhz", I = "rrbrpn4 _1y2v1nfhm _1y2v1nfip _1y2v1nfkl", N = "rrbrpn5", m = "rrbrpn9 _1y2v1nfhk _1y2v1nfhz _1y2v1nfht", L = "rrbrpni _1y2v1nfhk _1y2v1nfhz _1y2v1nfht";
