// @telefonica/mistica/dist/callout.js
"use client";
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "default", {
    enumerable: true,
    get: function() {
        return rr;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _stack = /*#__PURE__*/ _interop_require_default(require("./stack.js"));
const _inline = /*#__PURE__*/ _interop_require_default(require("./inline.js"));
const _box = /*#__PURE__*/ _interop_require_default(require("./box.js"));
const _hooks = require("./hooks.js");
const _themevariantcontext = require("./theme-variant-context.js");
const _text = require("./text.js");
const _iconcloseregular = /*#__PURE__*/ _interop_require_default(require("./generated/mistica-icons/icon-close-regular.js"));
const _iconbutton = require("./icon-button.js");
const _classnames = /*#__PURE__*/ _interop_require_default(require("classnames"));
const _buttongroup = /*#__PURE__*/ _interop_require_default(require("./button-group.js"));
const _calloutcssmistica = require("./callout.css-mistica.js");
const _imagecssmistica = require("./image.css-mistica.js");
const _sprinklescssmistica = require("./sprinkles.css-mistica.js");
const _skincontractcssmistica = require("./skins/skin-contract.css-mistica.js");
const _dom = require("./utils/dom.js");
const _css = require("./utils/css.js");
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
const D = (param)=>{
    let { title: a, titleAs: c = "h2", description: d, icon: i, onClose: t, button: l, secondaryButton: s, buttonLink: n, "aria-label": p, dataAttributes: u, role: f } = param;
    const h = (0, _themevariantcontext.useThemeVariant)(), { texts: g } = (0, _hooks.useTheme)();
    return /* @__PURE__ */ (0, _jsxruntime.jsx)("section", _object_spread_props(_object_spread({
        className: (0, _classnames.default)(_calloutcssmistica.container, (0, _sprinklescssmistica.sprinkles)({
            background: {
                inverse: _skincontractcssmistica.vars.colors.backgroundContainer,
                alternative: _skincontractcssmistica.vars.colors.backgroundContainer,
                default: _skincontractcssmistica.vars.colors.backgroundContainerAlternative
            }[h]
        })),
        style: (0, _css.applyCssVars)({
            [_imagecssmistica.vars.mediaBorderRadius]: _skincontractcssmistica.vars.borderRadii.mediaSmall
        }),
        "aria-label": p !== null && p !== void 0 ? p : a,
        role: f
    }, (0, _dom.getPrefixedDataAttributes)(u, "Callout")), {
        children: /* @__PURE__ */ (0, _jsxruntime.jsxs)(_themevariantcontext.ThemeVariant, {
            isInverse: !1,
            children: [
                i && /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                    paddingRight: 16,
                    children: i
                }),
                /* @__PURE__ */ (0, _jsxruntime.jsxs)("div", {
                    className: _calloutcssmistica.content,
                    children: [
                        /* @__PURE__ */ (0, _jsxruntime.jsxs)(_stack.default, {
                            space: 16,
                            children: [
                                /* @__PURE__ */ (0, _jsxruntime.jsxs)(_inline.default, {
                                    fullWidth: !0,
                                    alignItems: "flex-start",
                                    space: "between",
                                    children: [
                                        /* @__PURE__ */ (0, _jsxruntime.jsxs)(_stack.default, {
                                            space: 4,
                                            children: [
                                                /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text3, {
                                                    as: c,
                                                    regular: !0,
                                                    children: a
                                                }),
                                                /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text2, {
                                                    as: "p",
                                                    regular: !0,
                                                    color: _skincontractcssmistica.vars.colors.textSecondary,
                                                    children: d
                                                })
                                            ]
                                        }),
                                        t && // Create empty div in order to fill space that iconButton occupies.
                                        // Without this, the content's vertical alignment can be affected
                                        /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                                            className: _calloutcssmistica.closeButtonContainerSize
                                        })
                                    ]
                                }),
                                (l || s || n) && /* @__PURE__ */ (0, _jsxruntime.jsx)(_buttongroup.default, {
                                    primaryButton: l,
                                    secondaryButton: s,
                                    link: n
                                })
                            ]
                        }),
                        t && /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                            className: _calloutcssmistica.closeButtonContainer,
                            children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_iconbutton.IconButton, {
                                small: !0,
                                bleedY: !0,
                                bleedRight: !0,
                                Icon: _iconcloseregular.default,
                                onPress: t,
                                "aria-label": g.closeButtonLabel
                            })
                        })
                    ]
                })
            ]
        })
    }));
}, rr = D;
