// @telefonica/mistica/dist/card.css-mistica.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    actions: function() {
        return n;
    },
    boxed: function() {
        return e;
    },
    cardContainer: function() {
        return r;
    },
    circularMediaOverlay: function() {
        return y;
    },
    dataCard: function() {
        return f;
    },
    dataCardTopActionsWithoutIcon: function() {
        return s;
    },
    displayCardBackground: function() {
        return i;
    },
    displayCardContainer: function() {
        return d;
    },
    displayCardContent: function() {
        return o;
    },
    displayCardGradient: function() {
        return t;
    },
    flexColumn: function() {
        return h;
    },
    mediaCard: function() {
        return C;
    },
    mediaCardContent: function() {
        return l;
    },
    mediaCardIcon: function() {
        return c;
    },
    nakedCardContent: function() {
        return b;
    },
    snapCard: function() {
        return u;
    },
    touchable: function() {
        return k;
    },
    touchableCardOverlay: function() {
        return p;
    },
    touchableCardOverlayInverse: function() {
        return m;
    },
    touchableCardOverlayMedia: function() {
        return q;
    },
    touchableContainer: function() {
        return O;
    },
    touchableMediaCardOverlay: function() {
        return j;
    },
    touchableNakedMediaOverlay: function() {
        return w;
    },
    vars: function() {
        return x;
    }
});
require("./sprinkles.css.ts.vanilla.css-mistica.js");
require("./icon-button.css.ts.vanilla.css-mistica.js");
require("./card.css.ts.vanilla.css-mistica.js");
var n = "_15e54s91 _1y2v1nfhk _1y2v1nfi6 _1y2v1nfi0", e = "_15e54s93 _1y2v1nfhk _1y2v1nfhe", r = "_15e54s9v", y = "_15e54s98", f = "_15e54s9k _1y2v1nfaa _1y2v1nfbj _1y2v1nf7y _1y2v1nf97 _1y2v1nfhk _1y2v1nfi6 _1y2v1nfhe _1y2v1nfhq _1y2v1nfi7", s = "_15e54s9w", i = "_15e54s9r", d = "_1y2v1nfi7 _1y2v1nfio _1y2v1nfhk _1y2v1nfhe", o = "_15e54s9q", t = "_15e54s9s", h = "_15e54s9x", C = "_1y2v1nfhk _1y2v1nfhq _1y2v1nfio _1y2v1nfi7 _1y2v1nfi6", l = "_15e54s9f _1y2v1nfaa _1y2v1nfbj _1y2v1nfhk _1y2v1nfi6 _1y2v1nfhe _1y2v1nf7s _1y2v1nf97 _1y2v1nfhq _1y2v1nfhw", c = "_15e54s9g", b = "_15e54s9i _1y2v1nfhk _1y2v1nfi6 _1y2v1nfhe _1y2v1nfhq _1y2v1nfhw _1y2v1nf7s _1y2v1nfbj", u = "_15e54s9m _1y2v1nf7s _1y2v1nf91 _1y2v1nfaa _1y2v1nfbj _1y2v1nfhk _1y2v1nfhe _1y2v1nfio _1y2v1nfhq _1y2v1nfhw _1y2v1nfi7 _1y2v1nfi6", k = "_15e54s95", p = "_15e54s9a _15e54s96", m = "_15e54s9b _15e54s96", q = "_15e54s9c _15e54s96", O = "_15e54s94", j = "_15e54s97 _15e54s96", w = "_15e54s99 _15e54s96", x = {
    aspectRatio: "var(--_15e54s9t)",
    topActionsCount: "var(--_15e54s9u)"
};
