// @telefonica/mistica/dist/carousel.js
"use client";
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    Carousel: function() {
        return Ft;
    },
    CarouselContextConsumer: function() {
        return jt;
    },
    CarouselContextProvider: function() {
        return zt;
    },
    CenteredCarousel: function() {
        return Kt;
    },
    PageBullets: function() {
        return xe;
    },
    Slideshow: function() {
        return Yt;
    },
    useCarouselContext: function() {
        return _t;
    },
    useSlideshowContext: function() {
        return Ut;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _react = /*#__PURE__*/ _interop_require_wildcard(require("react"));
const _iconchevronleftregular = /*#__PURE__*/ _interop_require_default(require("./generated/mistica-icons/icon-chevron-left-regular.js"));
const _iconchevronrightregular = /*#__PURE__*/ _interop_require_default(require("./generated/mistica-icons/icon-chevron-right-regular.js"));
const _hooks = require("./hooks.js");
const _inline = /*#__PURE__*/ _interop_require_default(require("./inline.js"));
const _touchable = require("./touchable.js");
const _classnames = /*#__PURE__*/ _interop_require_default(require("classnames"));
const _themevariantcontext = require("./theme-variant-context.js");
const _dom = require("./utils/dom.js");
const _platform = require("./utils/platform.js");
const _documentvisibility = require("./utils/document-visibility.js");
const _carouselcssmistica = require("./carousel.css-mistica.js");
const _imagecssmistica = require("./image.css-mistica.js");
const _desktopcontainertypecontext = require("./desktop-container-type-context.js");
const _constants = require("./skins/constants.js");
const _css = require("./utils/css.js");
const _responsivelayout = require("./responsive-layout.js");
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
function _getRequireWildcardCache(nodeInterop) {
    if (typeof WeakMap !== "function") return null;
    var cacheBabelInterop = new WeakMap();
    var cacheNodeInterop = new WeakMap();
    return (_getRequireWildcardCache = function(nodeInterop) {
        return nodeInterop ? cacheNodeInterop : cacheBabelInterop;
    })(nodeInterop);
}
function _interop_require_wildcard(obj, nodeInterop) {
    if (!nodeInterop && obj && obj.__esModule) {
        return obj;
    }
    if (obj === null || typeof obj !== "object" && typeof obj !== "function") {
        return {
            default: obj
        };
    }
    var cache = _getRequireWildcardCache(nodeInterop);
    if (cache && cache.has(obj)) {
        return cache.get(obj);
    }
    var newObj = {
        __proto__: null
    };
    var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for(var key in obj){
        if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) {
            var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null;
            if (desc && (desc.get || desc.set)) {
                Object.defineProperty(newObj, key, desc);
            } else {
                newObj[key] = obj[key];
            }
        }
    }
    newObj.default = obj;
    if (cache) {
        cache.set(obj, newObj);
    }
    return newObj;
}
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
const ve = (t, e)=>{
    const s = (0, _documentvisibility.useDocumentVisibility)();
    return (0, _hooks.useIsInViewport)(e, !1) && s && !!t;
}, xe = (param)=>{
    let { currentIndex: t, numPages: e, onPress: s } = param;
    var _e_tablet;
    const r = (0, _themevariantcontext.useIsInverseVariant)(), { isDesktopOrBigger: i } = (0, _hooks.useScreenSize)(), h = (u)=>{
        const f = u === t;
        return r ? f ? _carouselcssmistica.bulletActiveInverse : _carouselcssmistica.bulletInverse : f ? _carouselcssmistica.bulletActive : _carouselcssmistica.bullet;
    }, w = typeof e == "number" ? e : Math.max(e.mobile, (_e_tablet = e.tablet) !== null && _e_tablet !== void 0 ? _e_tablet : e.mobile, e.desktop);
    return /* @__PURE__ */ (0, _jsxruntime.jsx)(_inline.default, {
        space: 0,
        alignItems: "center",
        dataAttributes: {
            "component-name": "PageBullets"
        },
        children: Array.from({
            length: w
        }, (u, f)=>/* @__PURE__ */ {
            var _e_tablet;
            return (0, _jsxruntime.jsx)(_touchable.BaseTouchable, {
                className: (0, _classnames.default)(typeof e == "number" ? _carouselcssmistica.bulletButton : {
                    [_carouselcssmistica.bulletButtonMobile]: f < e.mobile,
                    [_carouselcssmistica.bulletButtonTablet]: f < ((_e_tablet = e.tablet) !== null && _e_tablet !== void 0 ? _e_tablet : e.mobile),
                    [_carouselcssmistica.bulletButtonDesktop]: f < e.desktop
                }),
                style: f === 0 ? {
                    paddingLeft: 0
                } : {},
                maybe: !0,
                onPress: i && s ? ()=>s(f) : void 0,
                children: /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                    className: h(f)
                })
            }, f);
        })
    });
}, oe = ()=>{
    throw new Error("You must wrap your component with a CarouselContextProvider to use CarouselContext");
}, Ce = oe, Pe = oe, we = oe, Ie = {
    currentIndex: 0,
    numPages: 0
}, se = /*#__PURE__*/ _react.createContext({
    goPrev: Ce,
    goNext: Pe,
    goToPage: we,
    bulletsProps: Ie
}), re = /*#__PURE__*/ _react.createContext(null), zt = (param)=>{
    let { children: t } = param;
    const [e, s] = _react.useState(Ie), r = _react.useRef(Ce), i = _react.useRef(Pe), h = _react.useRef(we), w = _react.useMemo(()=>({
            goPrev: ()=>{
                r.current();
            },
            goNext: ()=>{
                i.current();
            },
            goToPage: function(u) {
                let f = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : !0;
                h.current(u, f);
            },
            bulletsProps: e
        }), [
        e
    ]);
    return /* @__PURE__ */ (0, _jsxruntime.jsx)(se.Provider, {
        value: w,
        children: /* @__PURE__ */ (0, _jsxruntime.jsx)(re.Provider, {
            value: {
                setGoPrev: (u)=>{
                    r.current = u;
                },
                setGoNext: (u)=>{
                    i.current = u;
                },
                setGoToPage: (u)=>{
                    h.current = u;
                },
                setBulletsProps: s
            },
            children: t
        })
    });
}, _t = ()=>_react.useContext(se), jt = se.Consumer, ae = (t, e, s)=>s ? typeof s == "number" ? s : s[t] || e[t] : e[t], Ct = (t, e)=>{
    const s = {
        mobile: 1,
        tablet: 2,
        desktop: {
            small: 1,
            medium: 2,
            large: 3
        }
    };
    if (!e) return _object_spread_props(_object_spread({}, s), {
        desktop: ae(t, s.desktop)
    });
    if (typeof e == "number") return {
        mobile: e,
        tablet: e,
        desktop: e
    };
    const r = ae(t, s.desktop, e.desktop);
    return _object_spread_props(_object_spread({}, s, e), {
        desktop: r
    });
}, ie = (t, e)=>{
    if (t.length === 0) return [];
    const s = Math.ceil(t.length / e), r = [];
    for(let i = 0; i < t.length; i += s)r.push(t[i]);
    return r[r.length - 1] = t[t.length - s], r;
}, Pt = (t, e)=>{
    const s = [];
    for(let r = 0; r < e.length; r++)r === 0 ? s.push(e[0]) : s.push((e[r] + e[r - 1]) / 2);
    for(let r = s.length - 1; r >= 0; r--)if (t - s[r] >= -1) return r;
    return 0;
}, ke = 5e3, Me = (param)=>{
    let { items: t, itemStyle: e, itemClassName: s, withBullets: r, renderBullets: i, initialActiveItem: h, itemsPerPage: w, itemsToScroll: u, mobilePageOffset: f, gap: C, free: b, centered: N, autoplay: x, onPageChange: y, dataAttributes: W } = param;
    const { texts: I, platformOverrides: S, skinName: H } = (0, _hooks.useTheme)(), J = (0, _desktopcontainertypecontext.useDesktopContainerType)(), a = Ct(J || "large", w), { isDesktopOrBigger: R, isTablet: A } = (0, _hooks.useScreenSize)(), F = A ? a.tablet : a.mobile, P = Math.max(Math.floor(R ? a.desktop : F), 1), l = _react.useRef(null), m = Math.ceil(t.length / Math.max(Math.floor(a.mobile), 1)), v = Math.ceil(t.length / Math.max(Math.floor(a.tablet), 1)), k = Math.ceil(t.length / Math.max(Math.floor(a.desktop), 1)), E = Math.ceil(t.length / P), [{ scrollLeft: D, scrollRight: G }, Re] = _react.useState({
        scrollLeft: 0,
        scrollRight: 0
    }), [Q, Ee] = _react.useState([]), K = _react.useMemo(()=>ie(Q, E), [
        Q,
        E
    ]), U = u ? ie(Q, Math.ceil(t.length / u)) : K, Le = G !== 0, Te = D !== 0;
    _react.useEffect(()=>{
        if (l.current) {
            const c = l.current, d = ()=>{
                const { scrollWidth: q, clientWidth: V } = c, Z = Math.round(c.scrollLeft), $ = Math.round(q - (Z + V));
                Re({
                    scrollLeft: Z,
                    scrollRight: $
                });
            }, p = ()=>{
                const q = c.scrollWidth - c.clientWidth;
                Ee(Array.from(c.querySelectorAll("[data-item]")).map((V, Z)=>{
                    if (Z === t.length - 1) return q;
                    const $ = V.offsetLeft, We = parseInt(getComputedStyle(V).scrollMargin), Ae = N && !R ? $ - V.clientWidth / 2 : $;
                    return Math.min(Ae - We - c.offsetLeft, q);
                }));
            };
            d(), p(), c.addEventListener("scroll", d);
            const g = (0, _dom.listenResize)(c, ()=>{
                d(), p();
            });
            return ()=>{
                c.removeEventListener("scroll", d), g();
            };
        }
        return ()=>{};
    }, [
        a.desktop,
        a.tablet,
        a.mobile,
        E,
        C,
        N,
        R,
        t.length
    ]);
    const L = _react.useCallback(function(c) {
        let d = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : !0;
        const p = l.current;
        if (p) {
            const g = K[c];
            p.scrollTo({
                left: g,
                behavior: d ? "smooth" : "auto"
            });
        }
    }, [
        K
    ]), X = _react.useCallback(()=>{
        const c = l.current;
        if (c) {
            const { scrollLeft: d } = c, p = [
                ...U
            ].reverse().find((g)=>g - d < -1);
            c.scrollTo({
                left: p,
                behavior: "smooth"
            });
        }
    }, [
        U
    ]), O = _react.useCallback(()=>{
        const c = l.current;
        if (c) {
            const { scrollLeft: d } = c, p = U.find((g)=>g - d > 1);
            c.scrollTo({
                left: p,
                behavior: "smooth"
            });
        }
    }, [
        U
    ]), le = ve(!!x, l);
    _react.useEffect(()=>{
        h !== void 0 && L(Math.floor(h / P), !1);
    }, [
        h,
        L,
        P
    ]), _react.useEffect(()=>{
        if (le && x) {
            const c = typeof x == "boolean" ? ke : x.time, d = typeof x == "object" && x.loop, p = setInterval(()=>{
                var g;
                G !== 0 ? O() : d && ((g = l.current) == null || g.scrollTo({
                    left: 0,
                    behavior: "smooth"
                }));
            }, c);
            return ()=>clearInterval(p);
        }
    }, [
        x,
        O,
        G,
        le
    ]);
    const M = Pt(D, K), ne = _react.useRef(!h), ce = _react.useRef(0);
    _react.useEffect(()=>{
        if (y) {
            const c = Math.min((M + 1) * P - 1, t.length - 1), d = [];
            for(let p = 0; p < P; p++){
                const g = c - p;
                g >= 0 && d.unshift(g);
            }
            ne.current ? ce.current !== M && y({
                pageIndex: M,
                shownItemIndexes: d
            }) : ne.current = d.includes(h || 0), ce.current = M;
        }
    }, [
        M,
        t.length,
        P,
        h,
        y
    ]);
    const T = _react.useContext(re), ee = _react.useMemo(()=>({
            currentIndex: M,
            numPages: {
                mobile: m,
                tablet: v,
                desktop: k
            }
        }), [
        M,
        k,
        m,
        v
    ]);
    _react.useEffect(()=>{
        T && (T.setGoPrev(X), T.setGoNext(O), T.setGoToPage(L), T.setBulletsProps(ee));
    }, [
        T,
        O,
        X,
        ee,
        L
    ]);
    let Y = null;
    i ? Y = i({
        numPages: E,
        currentIndex: M,
        onPress: L
    }) : r && (Y = /* @__PURE__ */ (0, _jsxruntime.jsx)(xe, _object_spread_props(_object_spread({}, ee), {
        onPress: L
    })));
    const Be = "64px", ye = "36px";
    return /* @__PURE__ */ (0, _jsxruntime.jsxs)("div", _object_spread_props(_object_spread({}, (0, _dom.getPrefixedDataAttributes)(_object_spread({
        "component-name": "Carousel"
    }, W))), {
        children: [
            /* @__PURE__ */ (0, _jsxruntime.jsxs)("div", {
                className: _carouselcssmistica.carouselContainer,
                children: [
                    /* @__PURE__ */ (0, _jsxruntime.jsx)(_themevariantcontext.ThemeVariant, {
                        isInverse: !1,
                        children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_touchable.BaseTouchable, {
                            className: _carouselcssmistica.carouselPrevArrowButton,
                            "aria-label": I.carouselPrevButton,
                            onPress: X,
                            disabled: !Te,
                            children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_iconchevronleftregular.default, {})
                        })
                    }),
                    /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                        className: (0, _classnames.default)(_carouselcssmistica.carousel, {
                            [_carouselcssmistica.centeredCarousel]: N,
                            [_carouselcssmistica.carouselWithScrollMobile]: m > 1,
                            [_carouselcssmistica.carouselWithScrollTablet]: v > 1
                        }),
                        style: _object_spread_props(_object_spread({}, (0, _css.applyCssVars)(_object_spread({
                            [_carouselcssmistica.vars.itemsPerPageDesktop]: String(a.desktop),
                            [_carouselcssmistica.vars.itemsPerPageTablet]: String(a.tablet),
                            [_carouselcssmistica.vars.itemsPerPageMobile]: String(a.mobile)
                        }, f === "large" ? {
                            [_carouselcssmistica.vars.mobilePageOffset]: Be
                        } : H === _constants.VIVO_NEW_SKIN ? {
                            [_carouselcssmistica.vars.mobilePageOffset]: ye
                        } : {}, C !== void 0 ? {
                            [_carouselcssmistica.vars.gap]: String(C)
                        } : {}))), {
                            scrollSnapType: b ? "initial" : "x mandatory",
                            // Hack to fix https://jira.tid.es/browse/NOVUMCC-8988
                            // there is a webkit rendering bug that causes a half pixel white line to appear at
                            // the bottom of the scrollable area in retina displays when it has a height with
                            // decimals. This extra padding avoids that line to partially cover the carousel
                            // slides border:
                            paddingBottom: (0, _platform.isIos)(S) && !(0, _platform.isRunningAcceptanceTest)(S) ? 0.5 : void 0
                        }),
                        ref: l,
                        children: t.map((c, d)=>/* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                                className: (0, _classnames.default)(_carouselcssmistica.carouselItem, s),
                                style: _object_spread_props(_object_spread({}, e), {
                                    scrollSnapStop: (0, _platform.isAndroid)(S) ? "always" : "normal"
                                }),
                                "data-item": !0,
                                children: c
                            }, d))
                    }),
                    /* @__PURE__ */ (0, _jsxruntime.jsx)(_themevariantcontext.ThemeVariant, {
                        isInverse: !1,
                        children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_touchable.BaseTouchable, {
                            className: _carouselcssmistica.carouselNextArrowButton,
                            "aria-label": I.carouselNextButton,
                            onPress: O,
                            disabled: !Le,
                            children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_iconchevronrightregular.default, {})
                        })
                    })
                ]
            }),
            Y && /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                className: (0, _classnames.default)(_carouselcssmistica.carouselBullets, !i && {
                    [_carouselcssmistica.noCarouselBulletsDesktop]: k <= 1,
                    [_carouselcssmistica.noCarouselBulletsTablet]: v <= 1,
                    [_carouselcssmistica.noCarouselBulletsMobile]: m <= 1
                }),
                children: Y
            })
        ]
    }));
}, Ne = ()=>{
    const [t, e] = _react.useState(1);
    return _react.useEffect(()=>{
        const s = ()=>{
            window.innerWidth !== 0 && (e((r)=>r + 1), window.removeEventListener("resize", s));
        };
        return window.innerWidth === 0 && window.addEventListener("resize", s), ()=>{
            window.removeEventListener("resize", s);
        };
    }, []), t;
}, Ft = (t)=>{
    const e = Ne();
    return /* @__PURE__ */ /*#__PURE__*/ (0, _react.createElement)(Me, _object_spread_props(_object_spread({}, t), {
        key: e
    }));
}, Kt = (param)=>{
    let { items: t, itemStyle: e, itemClassName: s, withBullets: r, renderBullets: i, initialActiveItem: h, onPageChange: w, dataAttributes: u } = param;
    const f = Ne();
    return /* @__PURE__ */ (0, _jsxruntime.jsx)(Me, {
        items: t,
        itemStyle: e,
        itemClassName: s,
        itemsPerPage: {
            mobile: 1,
            tablet: 1,
            desktop: 3
        },
        centered: !0,
        itemsToScroll: 1,
        gap: 0,
        withBullets: r,
        renderBullets: i,
        initialActiveItem: h,
        onPageChange: w,
        dataAttributes: u
    }, f);
}, Se = /*#__PURE__*/ _react.createContext(void 0), Ut = ()=>_react.useContext(Se), Yt = (param)=>{
    let { items: t, withBullets: e, autoplay: s, initialPageIndex: r = 0, onPageChange: i, dataAttributes: h, inverseBullets: w = !0 } = param;
    const { texts: u, platformOverrides: f } = (0, _hooks.useTheme)(), C = _react.useContext(re), b = _react.useRef(null), [{ scrollLeft: N, scrollRight: x }, y] = _react.useState({
        scrollLeft: 0,
        scrollRight: 0
    }), W = _react.useCallback(()=>{
        const l = b.current;
        l && l.scrollBy({
            left: -l.clientWidth,
            behavior: "smooth"
        });
    }, []), I = _react.useCallback(()=>{
        const l = b.current;
        l && l.scrollBy({
            left: l.clientWidth,
            behavior: "smooth"
        });
    }, []), S = _react.useCallback(function(l) {
        let m = arguments.length > 1 && arguments[1] !== void 0 ? arguments[1] : !0;
        const v = b.current;
        v && v.scrollTo({
            left: v.clientWidth * l,
            behavior: m ? "smooth" : "auto"
        });
    }, [
        b
    ]), H = x !== 0, J = N !== 0, a = b.current ? Math.floor((N + b.current.clientWidth / 2) / b.current.clientWidth) : 0;
    _react.useLayoutEffect(()=>{
        const l = b.current;
        if (l) {
            const m = ()=>{
                const { scrollWidth: k, clientWidth: E } = l, D = Math.round(l.scrollLeft), G = Math.round(k - (D + E));
                y({
                    scrollLeft: D,
                    scrollRight: G
                });
            };
            m(), l.addEventListener("scroll", m);
            const v = (0, _dom.listenResize)(l, m);
            return ()=>{
                l.removeEventListener("scroll", m), v();
            };
        }
    }, [
        t.length
    ]);
    const R = ve(!!s, b);
    _react.useEffect(()=>{
        if (R && s) {
            const l = typeof s == "boolean" ? ke : s.time, m = typeof s == "object" && s.loop, v = setInterval(()=>{
                var k;
                x !== 0 ? I() : m && ((k = b.current) == null || k.scrollTo({
                    left: 0,
                    behavior: "smooth"
                }));
            }, l);
            return ()=>clearInterval(v);
        }
    }, [
        s,
        I,
        x,
        R
    ]);
    const A = _react.useRef(!1), F = _react.useRef(0);
    _react.useEffect(()=>{
        i && (A.current ? F.current !== a && i(a) : A.current = r === a), F.current = a;
    }, [
        a,
        r,
        i
    ]), _react.useEffect(()=>{
        const l = b.current;
        r !== void 0 && l && !A.current && l.scrollTo({
            left: l.clientWidth * r
        });
    }, [
        r
    ]);
    const P = _react.useMemo(()=>({
            currentIndex: a,
            numPages: t.length
        }), [
        a,
        t.length
    ]);
    return _react.useEffect(()=>{
        C && (C.setGoPrev(W), C.setGoNext(I), C.setGoToPage(S), C.setBulletsProps(P));
    }, [
        C,
        I,
        W,
        P,
        S
    ]), /* @__PURE__ */ (0, _jsxruntime.jsx)(Se.Provider, {
        value: {
            withBullets: !!e
        },
        children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_responsivelayout.ResetResponsiveLayout, {
            skipDesktop: !0,
            children: /* @__PURE__ */ (0, _jsxruntime.jsxs)("div", _object_spread_props(_object_spread({
                className: (0, _classnames.default)(_carouselcssmistica.slideshowContainer, {
                    [_carouselcssmistica.slideshowWithBullets]: !!e
                })
            }, (0, _dom.getPrefixedDataAttributes)(h, "SlideShow")), {
                children: [
                    /* @__PURE__ */ (0, _jsxruntime.jsx)(_themevariantcontext.ThemeVariant, {
                        isInverse: !1,
                        children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_touchable.BaseTouchable, {
                            className: _carouselcssmistica.slideshowPrevArrowButton,
                            "aria-label": u.carouselPrevButton,
                            onPress: W,
                            disabled: !J,
                            children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_iconchevronleftregular.default, {})
                        })
                    }),
                    /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                        style: (0, _css.applyCssVars)({
                            [_imagecssmistica.vars.mediaBorderRadius]: "0px"
                        }),
                        children: /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                            className: _carouselcssmistica.slideshow,
                            ref: b,
                            children: t.map((l, m)=>/* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                                    className: _carouselcssmistica.slideshowItem,
                                    style: {
                                        scrollSnapStop: (0, _platform.isAndroid)(f) ? "always" : "normal"
                                    },
                                    children: l
                                }, m))
                        })
                    }),
                    /* @__PURE__ */ (0, _jsxruntime.jsx)(_themevariantcontext.ThemeVariant, {
                        isInverse: !1,
                        children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_touchable.BaseTouchable, {
                            className: _carouselcssmistica.slideshowNextArrowButton,
                            "aria-label": u.carouselNextButton,
                            onPress: I,
                            disabled: !H,
                            children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_iconchevronrightregular.default, {})
                        })
                    }),
                    e && t.length > 1 && /* @__PURE__ */ (0, _jsxruntime.jsx)(_themevariantcontext.ThemeVariant, {
                        isInverse: w,
                        children: /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                            className: _carouselcssmistica.slideshowBullets,
                            children: /* @__PURE__ */ (0, _jsxruntime.jsx)(xe, _object_spread({}, P))
                        })
                    })
                ]
            }))
        })
    });
};
