// @telefonica/mistica/dist/carousel.css-mistica.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    bullet: function() {
        return f;
    },
    bulletActive: function() {
        return y;
    },
    bulletActiveInverse: function() {
        return u;
    },
    bulletButton: function() {
        return e;
    },
    bulletButtonDesktop: function() {
        return b;
    },
    bulletButtonMobile: function() {
        return r;
    },
    bulletButtonTablet: function() {
        return c;
    },
    bulletInverse: function() {
        return l;
    },
    carousel: function() {
        return t;
    },
    carouselBullets: function() {
        return o;
    },
    carouselContainer: function() {
        return i;
    },
    carouselItem: function() {
        return s;
    },
    carouselNextArrowButton: function() {
        return h;
    },
    carouselPrevArrowButton: function() {
        return k;
    },
    carouselWithScrollMobile: function() {
        return m;
    },
    carouselWithScrollTablet: function() {
        return d;
    },
    centeredCarousel: function() {
        return B;
    },
    noCarouselBulletsDesktop: function() {
        return w;
    },
    noCarouselBulletsMobile: function() {
        return p;
    },
    noCarouselBulletsTablet: function() {
        return P;
    },
    slideshow: function() {
        return A;
    },
    slideshowBullets: function() {
        return C;
    },
    slideshowContainer: function() {
        return g;
    },
    slideshowItem: function() {
        return j;
    },
    slideshowNextArrowButton: function() {
        return q;
    },
    slideshowPrevArrowButton: function() {
        return z;
    },
    slideshowWithBullets: function() {
        return x;
    },
    vars: function() {
        return I;
    }
});
require("./sprinkles.css.ts.vanilla.css-mistica.js");
require("./responsive-layout.css.ts.vanilla.css-mistica.js");
require("./grid-layout.css.ts.vanilla.css-mistica.js");
require("./carousel.css.ts.vanilla.css-mistica.js");
var f = "_1abcu71 _1y2v1nfic _1y2v1nfis _1y2v1nfk3 _1y2v1nf4y", y = "_1abcu72 _1abcu71 _1y2v1nfic _1y2v1nfis _1y2v1nfk3 _1y2v1nf4z", u = "_1abcu72 _1abcu71 _1y2v1nfic _1y2v1nfis _1y2v1nfk3 _1y2v1nf5i", e = "_1abcu74 _1y2v1nf7d _1y2v1nf8m _1y2v1nfaa _1y2v1nfb4 _1y2v1nfhm _1y2v1nfk2 _1y2v1nf7c", b = "_1abcu78 _1abcu75 _1abcu74 _1y2v1nf7d _1y2v1nf8m _1y2v1nfaa _1y2v1nfb4 _1y2v1nfhm _1y2v1nfk2 _1y2v1nf7c", r = "_1abcu76 _1abcu75 _1abcu74 _1y2v1nf7d _1y2v1nf8m _1y2v1nfaa _1y2v1nfb4 _1y2v1nfhm _1y2v1nfk2 _1y2v1nf7c", c = "_1abcu77 _1abcu75 _1abcu74 _1y2v1nf7d _1y2v1nf8m _1y2v1nfaa _1y2v1nfb4 _1y2v1nfhm _1y2v1nfk2 _1y2v1nf7c", l = "_1abcu7b _1abcu71 _1y2v1nfic _1y2v1nfis _1y2v1nfk3", t = "_1abcu7u _1abcu7j _1y2v1nfhk _1y2v1nfj4", o = "_1abcu714 _1y2v1nfhk _1y2v1nfht _1y2v1nf7y", i = "_1y2v1nfhd _1y2v1nfjh _1y2v1nfhe", s = "_1abcu7y", h = "_1abcu712 _1abcu710 _1abcu7l _1y2v1nfhk _1y2v1nfhz _1y2v1nfht _1y2v1nfk3 _1y2v1nf3q _1y2v1nfii _1y2v1nfiy _1y2v1nfk1 _1y2v1nfhf", k = "_1abcu711 _1abcu710 _1abcu7l _1y2v1nfhk _1y2v1nfhz _1y2v1nfht _1y2v1nfk3 _1y2v1nf3q _1y2v1nfii _1y2v1nfiy _1y2v1nfk1 _1y2v1nfhf", m = "_1abcu7v", d = "_1abcu7w", B = "_1abcu7x", w = "_1abcu715", p = "_1abcu717", P = "_1abcu716", A = "_1abcu719 _1abcu7j _1y2v1nfhk _1y2v1nfj4", C = "_1abcu71h _1y2v1nfhf _1y2v1nfm7 _1y2v1nfhk _1y2v1nfht _1y2v1nfi7", g = "_1abcu7h _1y2v1nfhe", j = "_1abcu71b _1y2v1nfi7 _1y2v1nfi2", q = "_1abcu71f _1abcu710 _1abcu7l _1y2v1nfhk _1y2v1nfhz _1y2v1nfht _1y2v1nfk3 _1y2v1nf3q _1y2v1nfii _1y2v1nfiy _1y2v1nfk1 _1y2v1nfhf _1y2v1nfls", z = "_1abcu71d _1abcu710 _1abcu7l _1y2v1nfhk _1y2v1nfhz _1y2v1nfht _1y2v1nfk3 _1y2v1nf3q _1y2v1nfii _1y2v1nfiy _1y2v1nfk1 _1y2v1nfhf _1y2v1nfld", x = "_1abcu7i", I = {
    itemsPerPageMobile: "var(--_1abcu7o)",
    itemsPerPageTablet: "var(--_1abcu7p)",
    itemsPerPageDesktop: "var(--_1abcu7q)",
    gap: "var(--_1abcu7r)",
    mobilePageOffset: "var(--_1abcu7s)"
};
