// @telefonica/mistica/dist/counter.js
"use client";
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "default", {
    enumerable: true,
    get: function() {
        return vr;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _react = /*#__PURE__*/ _interop_require_wildcard(require("react"));
const _dom = require("./utils/dom.js");
const _countercssmistica = require("./counter.css-mistica.js");
const _themevariantcontext = require("./theme-variant-context.js");
const _skincontractcssmistica = require("./skins/skin-contract.css-mistica.js");
const _inline = /*#__PURE__*/ _interop_require_default(require("./inline.js"));
const _text = require("./text.js");
const _touchable = require("./touchable.js");
const _iconsubtractregular = /*#__PURE__*/ _interop_require_default(require("./generated/mistica-icons/icon-subtract-regular.js"));
const _iconaddmoreregular = /*#__PURE__*/ _interop_require_default(require("./generated/mistica-icons/icon-add-more-regular.js"));
const _icontrashcanregular = /*#__PURE__*/ _interop_require_default(require("./generated/mistica-icons/icon-trash-can-regular.js"));
const _hooks = require("./hooks.js");
const _classnames = /*#__PURE__*/ _interop_require_default(require("classnames"));
const _screenreaderonly = /*#__PURE__*/ _interop_require_default(require("./screen-reader-only.js"));
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
function _getRequireWildcardCache(nodeInterop) {
    if (typeof WeakMap !== "function") return null;
    var cacheBabelInterop = new WeakMap();
    var cacheNodeInterop = new WeakMap();
    return (_getRequireWildcardCache = function(nodeInterop) {
        return nodeInterop ? cacheNodeInterop : cacheBabelInterop;
    })(nodeInterop);
}
function _interop_require_wildcard(obj, nodeInterop) {
    if (!nodeInterop && obj && obj.__esModule) {
        return obj;
    }
    if (obj === null || typeof obj !== "object" && typeof obj !== "function") {
        return {
            default: obj
        };
    }
    var cache = _getRequireWildcardCache(nodeInterop);
    if (cache && cache.has(obj)) {
        return cache.get(obj);
    }
    var newObj = {
        __proto__: null
    };
    var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for(var key in obj){
        if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) {
            var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null;
            if (desc && (desc.get || desc.set)) {
                Object.defineProperty(newObj, key, desc);
            } else {
                newObj[key] = obj[key];
            }
        }
    }
    newObj.default = obj;
    if (cache) {
        cache.set(obj, newObj);
    }
    return newObj;
}
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
const C = 20, Y = (param)=>{
    let { value: h, defaultValue: b, min: m, max: s, onChangeValue: a } = param;
    const i = h !== void 0, p = _react.useCallback((n)=>n === void 0 ? m : Math.max(m, Math.min(s, n)), [
        m,
        s
    ]), [e, v] = _react.useState(p(b)), f = (n)=>{
        i || v(n), a == null || a(n);
    };
    return [
        i ? p(h) : e,
        f
    ];
}, rr = (param)=>{
    let { value: h, defaultValue: b, onChangeValue: m, onRemove: s, min: a, max: i, dataAttributes: p, disabled: e, removeLabel: v, increaseLabel: f, decreaseLabel: n, valueLabel: T } = param;
    const D = (0, _themevariantcontext.useThemeVariant)(), g = (0, _hooks.useAriaId)(), { texts: u } = (0, _hooks.useTheme)(), l = a === void 0 ? 0 : a, d = Math.max(l, i === void 0 ? 999 : i), [t, V] = Y({
        value: h,
        defaultValue: b,
        min: l,
        max: d,
        onChangeValue: m
    }), o = !!s && t === l, $ = ()=>v === void 0 ? u.counterRemoveLabel : v, k = ()=>f === void 0 ? u.counterIncreaseLabel : f, z = ()=>n === void 0 ? u.counterDecreaseLabel : n, L = ()=>`${t}, ${T === void 0 ? u.counterQuantity : T}${a !== void 0 ? `, ${u.counterMinValue} ${a}` : ""}${i !== void 0 ? `, ${u.counterMaxValue} ${i}` : ""}`;
    return /* @__PURE__ */ (0, _jsxruntime.jsx)("div", _object_spread_props(_object_spread({
        className: (0, _classnames.default)(_countercssmistica.counter, {
            [_countercssmistica.disabled]: e
        })
    }, (0, _dom.getPrefixedDataAttributes)(p, "Counter")), {
        style: {
            border: `1px solid ${D === "inverse" ? _skincontractcssmistica.vars.colors.backgroundContainer : _skincontractcssmistica.vars.colors.border}`
        },
        children: /* @__PURE__ */ (0, _jsxruntime.jsxs)(_inline.default, {
            space: 8,
            alignItems: "center",
            children: [
                /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                    className: (0, _classnames.default)(_countercssmistica.buttonContainer, {
                        [_countercssmistica.hasTrashIcon]: o,
                        [_countercssmistica.isButtonDisabled]: e || !o && t === l,
                        [_countercssmistica.disabled]: !e && !o && t === l
                    }),
                    children: /* @__PURE__ */ (0, _jsxruntime.jsxs)(_touchable.BaseTouchable, {
                        className: _countercssmistica.button,
                        disabled: !o && t === l || e,
                        onPress: ()=>o ? s == null ? void 0 : s() : V(t - 1),
                        "aria-label": o ? $() : z(),
                        "aria-controls": g,
                        children: [
                            /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                                className: _countercssmistica.buttonBackground
                            }),
                            /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                                className: _countercssmistica.decreaseButtonIcon,
                                "aria-hidden": o,
                                children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_iconsubtractregular.default, {
                                    size: C,
                                    color: t === l && !o || e ? _skincontractcssmistica.vars.colors.control : _skincontractcssmistica.vars.colors.controlActivated
                                })
                            }),
                            /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                                className: _countercssmistica.trashButtonIcon,
                                "aria-hidden": !o,
                                children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_icontrashcanregular.default, {
                                    size: C,
                                    color: _skincontractcssmistica.vars.colors.controlError
                                })
                            })
                        ]
                    })
                }),
                /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                    className: _countercssmistica.valueContainer,
                    style: {
                        width: `${Math.max(3, String(d).length) * 1.25}ch`
                    },
                    id: g,
                    "aria-live": "polite",
                    children: /* @__PURE__ */ (0, _jsxruntime.jsxs)(_themevariantcontext.ThemeVariant, {
                        variant: "default",
                        children: [
                            /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                                "aria-hidden": !0,
                                children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text3, {
                                    regular: !0,
                                    color: _skincontractcssmistica.vars.colors.textPrimary,
                                    children: t
                                })
                            }),
                            /* @__PURE__ */ (0, _jsxruntime.jsx)(_screenreaderonly.default, {
                                children: /* @__PURE__ */ (0, _jsxruntime.jsx)("span", {
                                    children: L()
                                })
                            })
                        ]
                    })
                }),
                /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                    className: (0, _classnames.default)(_countercssmistica.buttonContainer, {
                        [_countercssmistica.isButtonDisabled]: e || t === d,
                        [_countercssmistica.disabled]: !e && t === d
                    }),
                    children: /* @__PURE__ */ (0, _jsxruntime.jsxs)(_touchable.BaseTouchable, {
                        className: _countercssmistica.button,
                        disabled: t === d || e,
                        onPress: ()=>V(t + 1),
                        "aria-label": k(),
                        "aria-controls": g,
                        children: [
                            /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                                className: _countercssmistica.buttonBackground
                            }),
                            /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                                className: _countercssmistica.defaultButtonIcon,
                                children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_iconaddmoreregular.default, {
                                    size: C,
                                    color: t === d || e ? _skincontractcssmistica.vars.colors.control : _skincontractcssmistica.vars.colors.controlActivated
                                })
                            })
                        ]
                    })
                })
            ]
        })
    }));
}, vr = rr;
