// @telefonica/mistica/dist/counter.css-mistica.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    button: function() {
        return v;
    },
    buttonBackground: function() {
        return s;
    },
    buttonContainer: function() {
        return f;
    },
    counter: function() {
        return a;
    },
    decreaseButtonIcon: function() {
        return i;
    },
    defaultButtonIcon: function() {
        return t;
    },
    disabled: function() {
        return y;
    },
    hasTrashIcon: function() {
        return _;
    },
    isButtonDisabled: function() {
        return o;
    },
    trashButtonIcon: function() {
        return e;
    },
    valueContainer: function() {
        return h;
    }
});
require("./sprinkles.css.ts.vanilla.css-mistica.js");
require("./counter.css.ts.vanilla.css-mistica.js");
var v = "rijsns8 _1y2v1nf7c _1y2v1nfk2 _1y2v1nfhf _1y2v1nfii _1y2v1nfiy", s = "rijsnse _1y2v1nfih _1y2v1nfix _1y2v1nfhf _1y2v1nfk9 _1y2v1nf6l", f = "rijsns6", a = "rijsns3 _1y2v1nfk9 _1y2v1nf3q _1y2v1nfhl _1y2v1nfkl", i = "rijsnsc rijsnsa _1y2v1nfhf", t = "rijsnsa _1y2v1nfhf", y = "rijsns4", _ = "rijsns0", o = "rijsns1", e = "rijsnsb rijsnsa _1y2v1nfhf", h = "_1y2v1nfhe _1y2v1nfhk _1y2v1nfht";
