// @telefonica/mistica/dist/cover-hero.css-mistica.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    centered: function() {
        return n;
    },
    coverHero: function() {
        return o;
    },
    coverHeroContainer: function() {
        return f;
    },
    hasSideExtra: function() {
        return y;
    },
    mainContent: function() {
        return _;
    },
    mediaLayer: function() {
        return m;
    },
    mediaOverlay: function() {
        return t;
    },
    minHeight: function() {
        return h;
    },
    sideExtra: function() {
        return z;
    },
    sixColumns: function() {
        return p;
    },
    vars: function() {
        return s;
    },
    withAspectRatio: function() {
        return c;
    }
});
require("./sprinkles.css.ts.vanilla.css-mistica.js");
require("./responsive-layout.css.ts.vanilla.css-mistica.js");
require("./grid-layout.css.ts.vanilla.css-mistica.js");
require("./carousel.css.ts.vanilla.css-mistica.js");
require("./cover-hero.css.ts.vanilla.css-mistica.js");
var n = "moeazi9 _1y2v1nfhz", o = "moeazi2 _1y2v1nfhe _1y2v1nfhk _1y2v1nfhq _1y2v1nfht _1y2v1nfi7 _1y2v1nfkr _1y2v1nfm0", f = "moeazi4 _1y2v1nfhe _1y2v1nfhk _1y2v1nfi7", y = "moeazi7", _ = "_1y2v1nfhe", m = "_1y2v1nfhf _1y2v1nfmf _1y2v1nfi7 _1y2v1nfio", t = "moeazic _1y2v1nfhf _1y2v1nfkr _1y2v1nfl6 _1y2v1nfi7 _1y2v1nfio", h = "moeazi6", z = "_1y2v1nfhe", p = "moeazie", s = {
    aspectRatio: "var(--moeazi0)"
}, c = "moeazi5";
