// @telefonica/mistica/dist/cover-hero-media.js
"use client";
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "CoverHeroMedia", {
    enumerable: true,
    get: function() {
        return y;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _card = require("./card.js");
const _coverherocssmistica = require("./cover-hero.css-mistica.js");
const _image = /*#__PURE__*/ _interop_require_default(require("./image.js"));
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
const y = (param)=>{
    let { backgroundVideo: o, backgroundImage: e, poster: t, backgroundVideoRef: s } = param;
    const { video: m, videoAction: r } = (0, _card.useVideoWithControls)(o, t, s);
    return /* @__PURE__ */ (0, _jsxruntime.jsxs)("div", {
        className: _coverherocssmistica.mediaLayer,
        children: [
            e && /* @__PURE__ */ (0, _jsxruntime.jsx)(_image.default, {
                width: "100%",
                height: "100%",
                src: e
            }),
            o && m,
            /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                className: _coverherocssmistica.mediaOverlay
            }),
            r && /* @__PURE__ */ (0, _jsxruntime.jsx)(_card.CardActionsGroup, {
                actions: [
                    r
                ],
                type: "media"
            })
        ]
    });
};
