// @telefonica/mistica/dist/credit-card-number-field.js
"use client";
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "default", {
    enumerable: true,
    get: function() {
        return Tr;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _react = /*#__PURE__*/ _interop_require_wildcard(require("react"));
const _formcontext = require("./form-context.js");
const _hooks = require("./hooks.js");
const _creditcard = require("./utils/credit-card.js");
const _textfieldbase = require("./text-field-base.js");
const _iconcreditcardvisaregular = /*#__PURE__*/ _interop_require_default(require("./generated/mistica-icons/icon-credit-card-visa-regular.js"));
const _iconvisa = /*#__PURE__*/ _interop_require_default(require("./icons/icon-visa.js"));
const _iconmastercard = /*#__PURE__*/ _interop_require_default(require("./icons/icon-mastercard.js"));
const _iconamex = /*#__PURE__*/ _interop_require_default(require("./icons/icon-amex.js"));
const _rifm = require("rifm");
const _dom = require("./utils/dom.js");
const _common = require("./utils/common.js");
const _creditcardnumberfieldcssmistica = require("./credit-card-number-field.css-mistica.js");
const _skincontractcssmistica = require("./skins/skin-contract.css-mistica.js");
const _iconbuttoncssmistica = require("./icon-button.css-mistica.js");
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
function _getRequireWildcardCache(nodeInterop) {
    if (typeof WeakMap !== "function") return null;
    var cacheBabelInterop = new WeakMap();
    var cacheNodeInterop = new WeakMap();
    return (_getRequireWildcardCache = function(nodeInterop) {
        return nodeInterop ? cacheNodeInterop : cacheBabelInterop;
    })(nodeInterop);
}
function _interop_require_wildcard(obj, nodeInterop) {
    if (!nodeInterop && obj && obj.__esModule) {
        return obj;
    }
    if (obj === null || typeof obj !== "object" && typeof obj !== "function") {
        return {
            default: obj
        };
    }
    var cache = _getRequireWildcardCache(nodeInterop);
    if (cache && cache.has(obj)) {
        return cache.get(obj);
    }
    var newObj = {
        __proto__: null
    };
    var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for(var key in obj){
        if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) {
            var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null;
            if (desc && (desc.get || desc.set)) {
                Object.defineProperty(newObj, key, desc);
            } else {
                newObj[key] = obj[key];
            }
        }
    }
    newObj.default = obj;
    if (cache) {
        cache.set(obj, newObj);
    }
    return newObj;
}
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
function _object_without_properties(source, excluded) {
    if (source == null) return {};
    var target = _object_without_properties_loose(source, excluded);
    var key, i;
    if (Object.getOwnPropertySymbols) {
        var sourceSymbolKeys = Object.getOwnPropertySymbols(source);
        for(i = 0; i < sourceSymbolKeys.length; i++){
            key = sourceSymbolKeys[i];
            if (excluded.indexOf(key) >= 0) continue;
            if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
            target[key] = source[key];
        }
    }
    return target;
}
function _object_without_properties_loose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;
    for(i = 0; i < sourceKeys.length; i++){
        key = sourceKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        target[key] = source[key];
    }
    return target;
}
const Y = (r)=>{
    var t;
    const n = String(r !== null && r !== void 0 ? r : "").replace(/[^\d]/g, "");
    var _ref;
    return (_ref = (t = n.match(/.{1,4}/g)) == null ? void 0 : t.join(" ")) !== null && _ref !== void 0 ? _ref : n;
}, Z = (_param)=>{
    var { inputRef: r, value: n, defaultValue: t, onChange: e } = _param, a = _object_without_properties(_param, [
        "inputRef",
        "value",
        "defaultValue",
        "onChange"
    ]);
    const [c, T] = _react.useState(t !== null && t !== void 0 ? t : ""), m = _react.useRef(null), f = typeof n < "u", u = f ? n : c, g = _react.useCallback((C)=>{
        f || T(C), m.current && (e == null || e((0, _dom.createChangeEvent)(m.current, C)));
    }, [
        f,
        e
    ]), d = (0, _rifm.useRifm)({
        format: Y,
        value: u,
        onChange: g,
        accept: /[\d]+/g
    });
    return /* @__PURE__ */ (0, _jsxruntime.jsx)("input", _object_spread_props(_object_spread({}, a), {
        type: "text",
        inputMode: "decimal",
        maxLength: (0, _creditcard.getCreditCardNumberLength)(d.value) + 3,
        onChange: d.onChange,
        value: d.value,
        ref: (0, _common.combineRefs)(r, m)
    }));
}, v = (r)=>(0, _creditcard.isVisa)(r) ? /* @__PURE__ */ (0, _jsxruntime.jsx)(_iconvisa.default, {
        size: _iconbuttoncssmistica.iconSize.default
    }) : (0, _creditcard.isMasterCard)(r) ? /* @__PURE__ */ (0, _jsxruntime.jsx)(_iconmastercard.default, {
        size: _iconbuttoncssmistica.iconSize.default
    }) : (0, _creditcard.isAmericanExpress)(r) ? /* @__PURE__ */ (0, _jsxruntime.jsx)(_iconamex.default, {
        size: _iconbuttoncssmistica.iconSize.default
    }) : null, rr = {
    showBackface: !1,
    animationTarget: /* @__PURE__ */ (0, _jsxruntime.jsx)(_iconcreditcardvisaregular.default, {
        size: _iconbuttoncssmistica.iconSize.default
    }),
    isAnimating: !1
}, er = (r, param)=>{
    let { type: n, value: t } = param;
    if (n === "INPUT") {
        const e = v(t);
        if (e && !r.showBackface) return {
            animationTarget: e,
            showBackface: !0,
            isAnimating: !0
        };
        if (!(0, _creditcard.isVisa)(t) && !(0, _creditcard.isMasterCard)(t) && !(0, _creditcard.isAmericanExpress)(t) && r.showBackface) return {
            animationTarget: r.animationTarget,
            showBackface: !1,
            isAnimating: !0
        };
        if (e && r.showBackface) return _object_spread_props(_object_spread({}, r), {
            animationTarget: e
        });
    }
    return n === "TRANSITION_END" ? _object_spread_props(_object_spread({}, r), {
        isAnimating: !1
    }) : r;
}, tr = (param)=>{
    let { value: r } = param;
    const [{ showBackface: n, animationTarget: t, isAnimating: e }, a] = _react.useReducer(er, rr);
    return _react.useEffect(()=>{
        a({
            type: "INPUT",
            value: r
        });
    }, [
        r
    ]), /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
        className: _creditcardnumberfieldcssmistica.flip,
        children: /* @__PURE__ */ (0, _jsxruntime.jsxs)("div", {
            className: _creditcardnumberfieldcssmistica.variants[n ? "backface" : "default"],
            onTransitionEnd: ()=>e && a({
                    type: "TRANSITION_END"
                }),
            children: [
                /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                    className: _creditcardnumberfieldcssmistica.flipFront,
                    children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_iconcreditcardvisaregular.default, {
                        size: _iconbuttoncssmistica.iconSize.default,
                        color: _skincontractcssmistica.vars.colors.neutralMedium
                    })
                }),
                /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                    className: _creditcardnumberfieldcssmistica.flipBack,
                    children: t
                })
            ]
        })
    });
}, ir = (_param)=>{
    var { disabled: r, error: n, helperText: t, name: e, optional: a, validate: c, onChange: T, onChangeValue: m, onBlur: f, acceptedCards: u = {
        americanExpress: !0,
        visa: !0,
        masterCard: !0
    }, value: g, autoComplete: d = "cc-number", defaultValue: C, dataAttributes: F } = _param, R = _object_without_properties(_param, [
        "disabled",
        "error",
        "helperText",
        "name",
        "optional",
        "validate",
        "onChange",
        "onChangeValue",
        "onBlur",
        "acceptedCards",
        "value",
        "autoComplete",
        "defaultValue",
        "dataAttributes"
    ]);
    const { texts: E } = (0, _hooks.useTheme)(), { jumpToNext: z, rawValues: S, setFormError: y } = (0, _formcontext.useForm)(), h = (i, N)=>{
        const s = E.formCreditCardNumberError;
        return i ? (0, _creditcard.isAmericanExpress)(i) && !u.americanExpress || (0, _creditcard.isVisa)(i) && !u.visa || (0, _creditcard.isMasterCard)(i) && !u.masterCard || !(0, _creditcard.isValidCreditCardNumber)(i) || (0, _creditcard.getCreditCardNumberLength)(i) !== i.length ? s : c == null ? void 0 : c(i, N) : a ? "" : E.formFieldErrorIsMandatory;
    }, k = (i)=>i.replace(/\s/g, ""), B = (0, _formcontext.useFieldProps)({
        name: e,
        value: g,
        defaultValue: C,
        processValue: k,
        helperText: t,
        optional: a,
        error: n,
        disabled: r,
        onBlur: f,
        validate: h,
        onChange: T,
        onChangeValue: m
    });
    var _ref;
    return /* @__PURE__ */ (0, _jsxruntime.jsx)(_textfieldbase.TextFieldBaseAutosuggest, _object_spread_props(_object_spread({}, R, B), {
        onChange: (i)=>{
            B.onChange(i);
            const N = i.currentTarget.value, s = k(N);
            if (s.length >= (0, _creditcard.getCreditCardNumberLength)(s)) {
                const V = h == null ? void 0 : h(s, N);
                V ? y({
                    name: e,
                    error: V
                }) : z(e);
            }
        },
        inputComponent: Z,
        autoComplete: d,
        endIcon: /* @__PURE__ */ (0, _jsxruntime.jsx)(tr, {
            value: (_ref = g !== null && g !== void 0 ? g : S[e]) !== null && _ref !== void 0 ? _ref : ""
        }),
        dataAttributes: _object_spread({
            "component-name": "CreditCardNumberField"
        }, F)
    }));
}, Tr = ir;
