// @telefonica/mistica/dist/credit-card-fields.js
"use client";
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "default", {
    enumerable: true,
    get: function() {
        return k;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _formcontext = require("./form-context.js");
const _hooks = require("./hooks.js");
const _creditcard = require("./utils/credit-card.js");
const _doublefield = /*#__PURE__*/ _interop_require_default(require("./double-field.js"));
const _stack = /*#__PURE__*/ _interop_require_default(require("./stack.js"));
const _creditcardexpirationfield = /*#__PURE__*/ _interop_require_default(require("./credit-card-expiration-field.js"));
const _creditcardnumberfield = /*#__PURE__*/ _interop_require_default(require("./credit-card-number-field.js"));
const _cvvfield = /*#__PURE__*/ _interop_require_default(require("./cvv-field.js"));
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
const b = (param)=>{
    let { numberFieldName: t = "ccNum", expirationFieldName: m = "ccExp", cvvFieldName: a = "ccCvv", acceptedCards: o } = param;
    const { values: d } = (0, _formcontext.useForm)(), { texts: r } = (0, _hooks.useTheme)(), l = (0, _creditcard.getCvvLength)(String(d[t]));
    return /* @__PURE__ */ (0, _jsxruntime.jsxs)(_stack.default, {
        space: 16,
        children: [
            /* @__PURE__ */ (0, _jsxruntime.jsx)(_creditcardnumberfield.default, {
                acceptedCards: o,
                name: t,
                label: r.formCreditCardNumberLabel
            }),
            /* @__PURE__ */ (0, _jsxruntime.jsxs)(_doublefield.default, {
                layout: "60/40",
                children: [
                    /* @__PURE__ */ (0, _jsxruntime.jsx)(_creditcardexpirationfield.default, {
                        name: m,
                        label: r.formCreditCardExpirationLabel
                    }),
                    /* @__PURE__ */ (0, _jsxruntime.jsx)(_cvvfield.default, {
                        acceptedCards: o,
                        name: a,
                        label: r.formCreditCardCvvLabel,
                        maxLength: l
                    })
                ]
            })
        ]
    });
}, k = b;
