// @telefonica/mistica/dist/date-time-picker.js
"use client";
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "default", {
    enumerable: true,
    get: function() {
        return ne;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _react = /*#__PURE__*/ _interop_require_wildcard(require("react"));
const _textfieldbase = require("./text-field-base.js");
const _iconcalendarregular = /*#__PURE__*/ _interop_require_default(require("./generated/mistica-icons/icon-calendar-regular.js"));
const _reactdatetime = /*#__PURE__*/ _interop_require_default(require("react-datetime"));
const _overlay = /*#__PURE__*/ _interop_require_default(require("./overlay.js"));
const _textfieldcomponentscssmistica = require("./text-field-components.css-mistica.js");
const _dom = require("./utils/dom.js");
const _hooks = require("./hooks.js");
const _iconcloseregular = /*#__PURE__*/ _interop_require_default(require("./generated/mistica-icons/icon-close-regular.js"));
const _datetimepickercssmistica = require("./date-time-picker.css-mistica.js");
const _skincontractcssmistica = require("./skins/skin-contract.css-mistica.js");
require("moment/locale/es");
require("moment/locale/de");
require("moment/locale/pt-br");
require("moment/locale/en-gb");
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
function _getRequireWildcardCache(nodeInterop) {
    if (typeof WeakMap !== "function") return null;
    var cacheBabelInterop = new WeakMap();
    var cacheNodeInterop = new WeakMap();
    return (_getRequireWildcardCache = function(nodeInterop) {
        return nodeInterop ? cacheNodeInterop : cacheBabelInterop;
    })(nodeInterop);
}
function _interop_require_wildcard(obj, nodeInterop) {
    if (!nodeInterop && obj && obj.__esModule) {
        return obj;
    }
    if (obj === null || typeof obj !== "object" && typeof obj !== "function") {
        return {
            default: obj
        };
    }
    var cache = _getRequireWildcardCache(nodeInterop);
    if (cache && cache.has(obj)) {
        return cache.get(obj);
    }
    var newObj = {
        __proto__: null
    };
    var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for(var key in obj){
        if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) {
            var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null;
            if (desc && (desc.get || desc.set)) {
                Object.defineProperty(newObj, key, desc);
            } else {
                newObj[key] = obj[key];
            }
        }
    }
    newObj.default = obj;
    if (cache) {
        cache.set(obj, newObj);
    }
    return newObj;
}
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
function _object_without_properties(source, excluded) {
    if (source == null) return {};
    var target = _object_without_properties_loose(source, excluded);
    var key, i;
    if (Object.getOwnPropertySymbols) {
        var sourceSymbolKeys = Object.getOwnPropertySymbols(source);
        for(i = 0; i < sourceSymbolKeys.length; i++){
            key = sourceSymbolKeys[i];
            if (excluded.indexOf(key) >= 0) continue;
            if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
            target[key] = source[key];
        }
    }
    return target;
}
function _object_without_properties_loose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;
    for(i = 0; i < sourceKeys.length; i++){
        key = sourceKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        target[key] = source[key];
    }
    return target;
}
const O = {
    es: "es",
    // spanish
    ca: "es",
    // catalan
    eu: "es",
    // euskera
    gl: "es",
    // gallego
    de: "de",
    // german
    pt: "pt-br",
    // portuguese
    en: "en-gb"
}, q = (i)=>{
    const a = i.toLocaleLowerCase().split("-")[0];
    return O[a] || "en-gb";
}, ne = (_param)=>{
    var { withTime: i, mode: a, isValidDate: h, optional: u } = _param, o = _object_without_properties(_param, [
        "withTime",
        "mode",
        "isValidDate",
        "optional"
    ]);
    const [m, y] = _react.useState(!1), { texts: x, i18n: { locale: b } } = (0, _hooks.useTheme)(), r = _react.useRef(null), { height: k, ref: C } = (0, _hooks.useElementDimensions)(), l = (e)=>{
        o.disabled || y(e);
    }, v = ()=>{
        var f;
        const { top: e = 0, bottom: t = 0, left: c = 0 } = ((f = r.current) == null ? void 0 : f.getBoundingClientRect()) || {}, M = 260 + t < window.innerHeight;
        return {
            width: _textfieldcomponentscssmistica.DEFAULT_WIDTH,
            top: M ? t : e - k,
            left: c,
            position: "absolute",
            borderRadius: _skincontractcssmistica.vars.borderRadii.input,
            overflow: "hidden",
            boxShadow: "0px 5px 5px -3px rgba(0,0,0,0.2), 0px 8px 10px 1px rgba(0,0,0,0.14), 0px 3px 14px 2px rgba(0,0,0,0.12)"
        };
    }, s = ()=>{
        var t;
        const e = (t = r.current) == null ? void 0 : t.value;
        return e ? new Date(e) : void 0;
    }, D = (e)=>i ? e.format("yyyy-MM-DD HH:mm") : a === "year-month" ? e.format("yyyy-MM") : e.format("yyyy-MM-DD"), p = (e)=>{
        var c;
        const t = typeof e == "string" ? e : D(e);
        r.current && (r.current.focus(), (c = o.onChange) == null || c.call(o, (0, _dom.createChangeEvent)(r.current, t)));
    }, w = ()=>s() && u ? /* @__PURE__ */ (0, _jsxruntime.jsx)(_textfieldbase.FieldEndIcon, {
            Icon: _iconcloseregular.default,
            "aria-label": x.clearButton,
            hasBackgroundColor: !1,
            onPress: ()=>p("")
        }) : /* @__PURE__ */ (0, _jsxruntime.jsx)(_textfieldbase.FieldEndIcon, {
            Icon: _iconcalendarregular.default,
            "aria-label": "",
            disabled: o.disabled,
            hasBackgroundColor: !1,
            onPress: ()=>l(!m)
        });
    return /* @__PURE__ */ (0, _jsxruntime.jsxs)(_jsxruntime.Fragment, {
        children: [
            /* @__PURE__ */ (0, _jsxruntime.jsx)(_textfieldbase.TextFieldBaseAutosuggest, _object_spread_props(_object_spread({}, o), {
                style: {
                    cursor: "pointer"
                },
                required: !u,
                type: "text",
                autoComplete: "off",
                shrinkLabel: !!s(),
                endIcon: w(),
                inputRef: (e)=>{
                    var t;
                    (t = o == null ? void 0 : o.inputRef) == null || t.call(o, e), r.current = e;
                },
                readOnly: !0,
                onKeyDown: ()=>l(!0),
                onClick: ()=>{
                    l(!0);
                }
            })),
            m && /* @__PURE__ */ (0, _jsxruntime.jsx)(_overlay.default, {
                onPress: (e)=>{
                    (0, _dom.cancelEvent)(e), l(!1);
                },
                disableScroll: !0,
                children: /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                    ref: C,
                    style: v(),
                    className: _datetimepickercssmistica.reactDatePicker,
                    children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_reactdatetime.default, {
                        initialViewMode: a === "year-month" ? "months" : void 0,
                        dateFormat: a === "year-month" ? "YYYY-MM" : void 0,
                        timeFormat: i ? "HH:mm" : !1,
                        initialValue: s(),
                        locale: q(b),
                        input: !1,
                        onChange: p,
                        isValidDate: h
                    })
                })
            })
        ]
    });
};
