// @telefonica/mistica/dist/dialog.css-mistica.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    ANIMATION_DURATION_MS: function() {
        return _;
    },
    closedModalContent: function() {
        return y;
    },
    closedOpactityLayer: function() {
        return f;
    },
    dialogActions: function() {
        return l;
    },
    dialogContent: function() {
        return a;
    },
    iconContainer: function() {
        return r;
    },
    modalCloseButtonContainer: function() {
        return d;
    },
    modalContent: function() {
        return o;
    },
    modalOpacityLayer: function() {
        return t;
    },
    variants: function() {
        return e;
    },
    wrapper: function() {
        return h;
    }
});
require("./sprinkles.css.ts.vanilla.css-mistica.js");
require("./dialog.css.ts.vanilla.css-mistica.js");
var _ = 200, y = "_3dl29lb", f = "_3dl29l6", l = "_3dl29ll _1y2v1nf7y", a = "_3dl29lj _1y2v1nfi6 _1y2v1nfjk", r = "_3dl29lm", d = "_3dl29l8 _1y2v1nf7s _1y2v1nf91 _1y2v1nfaa _1y2v1nfbj _1y2v1nfhf _1y2v1nfkr _1y2v1nfll", o = "_3dl29la _1y2v1nf3p _1y2v1nfkb", t = "_3dl29l5 _1y2v1nfhg _1y2v1nfhk _1y2v1nfht _1y2v1nfhz _1y2v1nfhq _1y2v1nfkr _1y2v1nfll _1y2v1nfm0 _1y2v1nfl6 _1y2v1nf41", e = {
    dialog: "_3dl29ld _1y2v1nfhk _1y2v1nfhq _1y2v1nfhw _1y2v1nfag _1y2v1nfbp _1y2v1nf84 _1y2v1nf97",
    default: "_3dl29ld _1y2v1nfhk _1y2v1nfhq _1y2v1nfhw _1y2v1nf7y _1y2v1nf97 _1y2v1nfag _1y2v1nfbp"
}, h = "_1y2v1nfhe";
