// @telefonica/mistica/dist/dialog.js
"use client";
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "default", {
    enumerable: true,
    get: function() {
        return Ie;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _react = /*#__PURE__*/ _interop_require_wildcard(require("react"));
const _classnames = /*#__PURE__*/ _interop_require_default(require("classnames"));
const _button = require("./button.js");
const _portal = require("./portal.js");
const _focustrap = /*#__PURE__*/ _interop_require_default(require("./focus-trap.js"));
const _iconcloseregular = /*#__PURE__*/ _interop_require_default(require("./generated/mistica-icons/icon-close-regular.js"));
const _iconbutton = require("./icon-button.js");
const _webviewbridge = require("@tef-novum/webview-bridge");
const _hooks = require("./hooks.js");
const _buttonlayout = /*#__PURE__*/ _interop_require_default(require("./button-layout.js"));
const _text = require("./text.js");
const _keys = require("./utils/keys.js");
const _box = /*#__PURE__*/ _interop_require_default(require("./box.js"));
const _platform = require("./utils/platform.js");
const _modalcontextprovider = require("./modal-context-provider.js");
const _stack = /*#__PURE__*/ _interop_require_default(require("./stack.js"));
const _dialogcssmistica = require("./dialog.css-mistica.js");
const _skincontractcssmistica = require("./skins/skin-contract.css-mistica.js");
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
function _getRequireWildcardCache(nodeInterop) {
    if (typeof WeakMap !== "function") return null;
    var cacheBabelInterop = new WeakMap();
    var cacheNodeInterop = new WeakMap();
    return (_getRequireWildcardCache = function(nodeInterop) {
        return nodeInterop ? cacheNodeInterop : cacheBabelInterop;
    })(nodeInterop);
}
function _interop_require_wildcard(obj, nodeInterop) {
    if (!nodeInterop && obj && obj.__esModule) {
        return obj;
    }
    if (obj === null || typeof obj !== "object" && typeof obj !== "function") {
        return {
            default: obj
        };
    }
    var cache = _getRequireWildcardCache(nodeInterop);
    if (cache && cache.has(obj)) {
        return cache.get(obj);
    }
    var newObj = {
        __proto__: null
    };
    var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for(var key in obj){
        if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) {
            var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null;
            if (desc && (desc.get || desc.set)) {
                Object.defineProperty(newObj, key, desc);
            } else {
                newObj[key] = obj[key];
            }
        }
    }
    newObj.default = obj;
    if (cache) {
        cache.set(obj, newObj);
    }
    return newObj;
}
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
function _object_without_properties(source, excluded) {
    if (source == null) return {};
    var target = _object_without_properties_loose(source, excluded);
    var key, i;
    if (Object.getOwnPropertySymbols) {
        var sourceSymbolKeys = Object.getOwnPropertySymbols(source);
        for(i = 0; i < sourceSymbolKeys.length; i++){
            key = sourceSymbolKeys[i];
            if (excluded.indexOf(key) >= 0) continue;
            if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
            target[key] = source[key];
        }
    }
    return target;
}
function _object_without_properties_loose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;
    for(i = 0; i < sourceKeys.length; i++){
        key = sourceKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        target[key] = source[key];
    }
    return target;
}
const ce = ()=>process.env.NODE_ENV !== "test" && !(0, _platform.isRunningAcceptanceTest)(), de = (t)=>{
    const { texts: i } = (0, _hooks.useTheme)(), { className: d, title: r, message: f, icon: o, extra: c, showCancelButton: s, showAcceptButton: l, cancelText: g = i.dialogCancelButton, acceptText: m = i.dialogAcceptButton, onCancel: h, onAccept: C, destructive: B = !1 } = t, u = t.type === "dialog", p = u && !!t.link || l || s, y = {
        onPress: C || (()=>{}),
        children: m,
        // @deprecated - testid should be removed but many webapp tests depend on this
        dataAttributes: {
            testid: "dialog-accept-button"
        }
    };
    return /* @__PURE__ */ (0, _jsxruntime.jsxs)("div", {
        className: (0, _classnames.default)(_dialogcssmistica.variants[u ? "dialog" : "default"], d),
        children: [
            o && /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                paddingBottom: 24,
                children: /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                    className: _dialogcssmistica.iconContainer,
                    children: /*#__PURE__*/ _react.cloneElement(o, {
                        size: "100%"
                    })
                })
            }),
            r && /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                paddingBottom: 16,
                children: u ? /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text5, {
                    as: "h2",
                    children: r
                }) : /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text4, {
                    regular: !0,
                    as: "h2",
                    children: r
                })
            }),
            t.subtitle && /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                paddingBottom: 16,
                children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text4, {
                    regular: !0,
                    as: "h2",
                    children: t.subtitle
                })
            }),
            /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                className: _dialogcssmistica.dialogContent,
                children: /* @__PURE__ */ (0, _jsxruntime.jsxs)(_stack.default, {
                    space: 16,
                    children: [
                        /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text3, {
                            color: _skincontractcssmistica.vars.colors.textSecondary,
                            regular: !0,
                            children: f
                        }),
                        c
                    ]
                })
            }),
            p && /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                className: _dialogcssmistica.dialogActions,
                children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_buttonlayout.default, {
                    link: u ? t.link : void 0,
                    primaryButton: l ? B ? /* @__PURE__ */ (0, _jsxruntime.jsx)(_button.ButtonDanger, _object_spread({
                        tabIndex: 1
                    }, y)) : /* @__PURE__ */ (0, _jsxruntime.jsx)(_button.ButtonPrimary, _object_spread({
                        tabIndex: 1
                    }, y)) : void 0,
                    secondaryButton: s ? /* @__PURE__ */ (0, _jsxruntime.jsx)(_button.ButtonSecondary, {
                        tabIndex: 2,
                        onPress: h || (()=>{}),
                        dataAttributes: {
                            testid: "dialog-cancel-button"
                        },
                        children: g
                    }) : void 0
                })
            })
        ]
    });
}, ue = (param)=>{
    let { type: t, message: i, title: d, acceptText: r, cancelText: f, onAccept: o, onCancel: c, onDestroy: s } = param;
    return t === "confirm" ? (0, _webviewbridge.nativeConfirm)({
        message: i,
        title: d,
        cancelText: f,
        acceptText: r
    }).then((l)=>{
        l ? o == null || o() : c == null || c(), s();
    }) : (0, _webviewbridge.nativeAlert)({
        message: i,
        title: d,
        buttonText: r
    }).then(()=>{
        o == null || o(), s();
    });
}, me = (param)=>{
    let { type: t, onAccept: i, onCancel: d, onDestroy: r, acceptText: f, cancelText: o, message: c, title: s } = param;
    const { texts: l } = (0, _hooks.useTheme)(), g = _react.useRef({
        type: t,
        onAccept: i,
        onCancel: d,
        onDestroy: r,
        acceptText: f || l.dialogAcceptButton,
        cancelText: o || l.dialogCancelButton,
        message: c,
        title: s
    }), m = _react.useRef(!1);
    return _react.useEffect(()=>{
        m.current || (m.current = !0, ue(g.current).finally(()=>{
            m.current = !1;
        }));
    }, []), /* @__PURE__ */ (0, _jsxruntime.jsx)(_jsxruntime.Fragment, {});
}, Ie = (t)=>{
    (0, _modalcontextprovider.useSetModalStateEffect)();
    const i = _react.useRef(null), { texts: d } = (0, _hooks.useTheme)(), [r, f] = _react.useState(!1), o = _react.useRef(!1), c = _react.useRef(!1), [s, l] = _react.useState(!1), g = _react.useRef(!1), m = _react.useRef(ce() ? _dialogcssmistica.ANIMATION_DURATION_MS : 0), h = t.type !== "dialog" && (0, _webviewbridge.isWebViewBridgeAvailable)(), C = t.type === "dialog", B = t.type === "alert", { onAccept: u, onCancel: p, onDestroy: y } = t, I = _object_without_properties(t, [
        "onAccept",
        "onCancel",
        "onDestroy"
    ]);
    _react.useEffect(()=>{
        const a = setTimeout(()=>{
            o.current || l(!0);
        }, m.current);
        return ()=>{
            clearTimeout(a);
        };
    }, []);
    const x = _react.useCallback(()=>{
        c.current || (c.current = !0, g.current ? u == null || u() : p == null || p(), y());
    }, [
        u,
        p,
        y
    ]), R = _react.useCallback(()=>{
        let a;
        return !o.current && s && (o.current = !0, l(!1), f(!0), a = setTimeout(x, m.current)), ()=>{
            a && clearTimeout(a);
        };
    }, [
        x,
        s
    ]), N = _react.useCallback(()=>{
        g.current = !0, R();
    }, [
        R
    ]), b = _react.useCallback(()=>{
        g.current = !1, R();
    }, [
        R
    ]), v = _react.useCallback(()=>{
        o.current || (B ? N() : b());
    }, [
        N,
        b,
        B
    ]), T = _react.useCallback((a)=>{
        a.key === _keys.ESC && (a.stopPropagation(), a.preventDefault(), v());
    }, [
        v
    ]);
    _react.useEffect(()=>{
        if (!h) return document.addEventListener("keydown", T), ()=>{
            document.removeEventListener("keydown", T);
        };
    }, [
        T,
        h
    ]), _react.useEffect(()=>{
        if (!h) return window.addEventListener("popstate", v), ()=>{
            window.removeEventListener("popstate", v);
        };
    }, [
        v,
        h
    ]);
    const P = _react.useCallback((a)=>{
        a.stopPropagation(), C && b();
    }, [
        C,
        b
    ]);
    return h ? /* @__PURE__ */ (0, _jsxruntime.jsx)(me, _object_spread({}, t)) : /* @__PURE__ */ (0, _jsxruntime.jsx)(_portal.Portal, {
        className: _dialogcssmistica.wrapper,
        children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_focustrap.default, {
            children: /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                onClick: P,
                className: (0, _classnames.default)(_dialogcssmistica.modalOpacityLayer, {
                    [_dialogcssmistica.closedOpactityLayer]: r
                }),
                "data-testid": "dialog-overlay",
                "aria-hidden": r || !s,
                children: /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                    role: "dialog",
                    onClick: (a)=>a.stopPropagation(),
                    "data-component-name": "Dialog",
                    children: /* @__PURE__ */ (0, _jsxruntime.jsxs)("div", {
                        ref: i,
                        onAnimationEnd: (a)=>{
                            a.target === i.current && (o.current || l(!0));
                        },
                        onTransitionEnd: (a)=>{
                            a.target === i.current && o.current && x();
                        },
                        className: (0, _classnames.default)(_dialogcssmistica.modalContent, {
                            [_dialogcssmistica.closedModalContent]: r
                        }),
                        children: [
                            C && /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                                className: _dialogcssmistica.modalCloseButtonContainer,
                                children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_iconbutton.InternalIconButton, {
                                    onPress: v,
                                    "aria-label": d.modalClose || d.closeButtonLabel,
                                    bleedLeft: !0,
                                    bleedRight: !0,
                                    bleedY: !0,
                                    Icon: _iconcloseregular.default
                                })
                            }),
                            /* @__PURE__ */ (0, _jsxruntime.jsx)(de, _object_spread_props(_object_spread({}, I), {
                                showAcceptButton: t.type !== "dialog" || !!t.onAccept,
                                showCancelButton: t.type === "confirm" || t.type === "dialog" && !!t.onCancel,
                                onCancel: b,
                                onAccept: N
                            }))
                        ]
                    })
                })
            })
        })
    });
};
