// @telefonica/mistica/dist/empty-state.css-mistica.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    container: function() {
        return n;
    },
    contentVariants: function() {
        return v;
    },
    iconContainer: function() {
        return e;
    },
    inverseBorder: function() {
        return f;
    },
    largeImage: function() {
        return _;
    },
    largeImageContainer: function() {
        return o;
    },
    smallImage: function() {
        return u;
    },
    vars: function() {
        return i;
    }
});
require("./sprinkles.css.ts.vanilla.css-mistica.js");
require("./empty-state.css.ts.vanilla.css-mistica.js");
var n = "_13fzrua2 _1y2v1nfhk _1y2v1nf7c", v = {
    default: "_13fzrua3",
    largeImage: "_13fzrua4"
}, e = "_13fzrua9 _1y2v1nfil _1y2v1nfj1", f = "_13fzruac", _ = "_13fzrua7 _1y2v1nfio", o = "_13fzrua5", u = "_13fzruab _1y2v1nfhm", i = {
    backgroundColor: "var(--_13fzrua0)"
};
