// @telefonica/mistica/dist/empty-state-card.js
"use client";
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "default", {
    enumerable: true,
    get: function() {
        return _;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _box = /*#__PURE__*/ _interop_require_default(require("./box.js"));
const _boxed = require("./boxed.js");
const _hooks = require("./hooks.js");
const _stack = /*#__PURE__*/ _interop_require_default(require("./stack.js"));
const _text = require("./text.js");
const _buttongroup = /*#__PURE__*/ _interop_require_default(require("./button-group.js"));
const _emptystatecardcssmistica = require("./empty-state-card.css-mistica.js");
const _skincontractcssmistica = require("./skins/skin-contract.css-mistica.js");
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
const j = (param)=>{
    let { title: d, titleAs: n = "h3", description: h, button: e, secondaryButton: t, buttonLink: o, icon: a, imageUrl: i, "aria-label": f, dataAttributes: x } = param;
    var s;
    const { textPresets: g } = (0, _hooks.useTheme)();
    let m;
    return i && (m = /* @__PURE__ */ (0, _jsxruntime.jsx)("img", {
        className: _emptystatecardcssmistica.image,
        alt: "",
        src: i
    })), process.env.NODE_ENV !== "production" && e && !((s = e == null ? void 0 : e.props) != null && s.small) && console.error("button property in EmptyStateCard must be a a small Button. Set small prop to true"), /* @__PURE__ */ (0, _jsxruntime.jsx)(_boxed.Boxed, {
        dataAttributes: x,
        children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
            paddingY: {
                mobile: 24,
                desktop: 40
            },
            paddingX: {
                mobile: 16,
                desktop: 40
            },
            children: /* @__PURE__ */ (0, _jsxruntime.jsx)("section", {
                className: _emptystatecardcssmistica.container,
                "aria-label": f,
                children: /* @__PURE__ */ (0, _jsxruntime.jsxs)(_stack.default, {
                    space: 16,
                    children: [
                        m !== null && m !== void 0 ? m : a && /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                            className: _emptystatecardcssmistica.iconContainer,
                            children: a
                        }),
                        /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                            children: /* @__PURE__ */ (0, _jsxruntime.jsxs)(_stack.default, {
                                space: 8,
                                children: [
                                    /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text, _object_spread_props(_object_spread({}, _text.textProps.text4), {
                                        weight: g.cardTitle.weight,
                                        as: n,
                                        children: d
                                    })),
                                    /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text2, {
                                        regular: !0,
                                        color: _skincontractcssmistica.vars.colors.textSecondary,
                                        children: h
                                    })
                                ]
                            })
                        }),
                        (e || t || o) && /* @__PURE__ */ (0, _jsxruntime.jsx)(_buttongroup.default, {
                            primaryButton: e,
                            secondaryButton: t,
                            link: o
                        })
                    ]
                })
            })
        })
    });
}, _ = j;
