// @telefonica/mistica/dist/feedback.css-mistica.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    backgroundBrand: function() {
        return r;
    },
    container: function() {
        return n;
    },
    desktopContainer: function() {
        return t;
    },
    desktopContent: function() {
        return v;
    },
    desktopImage: function() {
        return x;
    },
    feedbackData: function() {
        return h;
    },
    feedbackTextAppearFast: function() {
        return o;
    },
    feedbackTextAppearMedium: function() {
        return g;
    },
    feedbackTextAppearSlow: function() {
        return i;
    },
    iconContainer: function() {
        return p;
    },
    innerContainer: function() {
        return d;
    },
    innerContainerWithButtons: function() {
        return f;
    }
});
require("./sprinkles.css.ts.vanilla.css-mistica.js");
require("./feedback.css.ts.vanilla.css-mistica.js");
var r = "gx6h6af", n = "gx6h6a4 _1y2v1nfhk _1y2v1nfio _1y2v1nfi7 _1y2v1nfhq", t = "gx6h6a0", v = "gx6h6a1", x = "gx6h6a2", h = "gx6h6ae", o = "gx6h6ab gx6h6aa", g = "gx6h6ac gx6h6aa", i = "gx6h6ad gx6h6aa", p = "_1y2v1nfij _1y2v1nfiz", d = "gx6h6a5", f = "gx6h6a6";
