// @telefonica/mistica/dist/fixed-footer-layout.css-mistica.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    absoluteBackgroundLayer: function() {
        return a;
    },
    container: function() {
        return f;
    },
    elevated: function() {
        return n;
    },
    fixedBackgroundLayer: function() {
        return o;
    },
    fixedFooter: function() {
        return j;
    },
    footer: function() {
        return y;
    },
    portal: function() {
        return e;
    },
    vars: function() {
        return _;
    },
    withoutFooter: function() {
        return t;
    }
});
require("./sprinkles.css.ts.vanilla.css-mistica.js");
require("./fixed-footer-layout.css.ts.vanilla.css-mistica.js");
var a = "p5j8j5d _1y2v1nfl6 _1y2v1nfll _1y2v1nfhf", f = "p5j8j57", n = "p5j8j55", o = "p5j8j5b _1y2v1nfkr _1y2v1nfl6 _1y2v1nfll _1y2v1nfm0 _1y2v1nfhg", j = "p5j8j54", y = "p5j8j53 _1y2v1nfi7", e = "p5j8j59 _1y2v1nfkr _1y2v1nfl6 _1y2v1nfll _1y2v1nfm0 _1y2v1nfhg", _ = {
    footerHeight: "var(--p5j8j50)",
    backgroundColor: "var(--p5j8j51)"
}, t = "_1y2v1nfhj";
