// @telefonica/mistica/dist/fixed-footer-layout.js
"use client";
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "default", {
    enumerable: true,
    get: function() {
        return Se;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _react = /*#__PURE__*/ _interop_require_wildcard(require("react"));
const _classnames = /*#__PURE__*/ _interop_require_default(require("classnames"));
const _helpers = require("./utils/helpers.js");
const _platform = require("./utils/platform.js");
const _hooks = require("./hooks.js");
const _dom = require("./utils/dom.js");
const _portal = require("./portal.js");
const _skincontractcssmistica = require("./skins/skin-contract.css-mistica.js");
const _fixedfooterlayoutcssmistica = require("./fixed-footer-layout.css-mistica.js");
const _css = require("./utils/css.js");
const _fixedtotop = require("./fixed-to-top.js");
const _overscrollcolorcontext = require("./overscroll-color-context.js");
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
function _getRequireWildcardCache(nodeInterop) {
    if (typeof WeakMap !== "function") return null;
    var cacheBabelInterop = new WeakMap();
    var cacheNodeInterop = new WeakMap();
    return (_getRequireWildcardCache = function(nodeInterop) {
        return nodeInterop ? cacheNodeInterop : cacheBabelInterop;
    })(nodeInterop);
}
function _interop_require_wildcard(obj, nodeInterop) {
    if (!nodeInterop && obj && obj.__esModule) {
        return obj;
    }
    if (obj === null || typeof obj !== "object" && typeof obj !== "function") {
        return {
            default: obj
        };
    }
    var cache = _getRequireWildcardCache(nodeInterop);
    if (cache && cache.has(obj)) {
        return cache.get(obj);
    }
    var newObj = {
        __proto__: null
    };
    var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for(var key in obj){
        if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) {
            var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null;
            if (desc && (desc.get || desc.set)) {
                Object.defineProperty(newObj, key, desc);
            } else {
                newObj[key] = obj[key];
            }
        }
    }
    newObj.default = obj;
    if (cache) {
        cache.set(obj, newObj);
    }
    return newObj;
}
const ie = (e)=>e === document.documentElement ? window : e, ne = (e)=>{
    const l = setTimeout(e, 0);
    return {
        cancel: ()=>clearTimeout(l)
    };
}, ce = 200, le = (param)=>{
    let { isFooterVisible: e = !0, footer: l, footerHeight: k = "auto", footerBgColor: i = _skincontractcssmistica.vars.colors.background, containerBgColor: F = _skincontractcssmistica.vars.colors.background, children: L, onChangeFooterHeight: n } = param;
    const [$, g] = _react.useState(!1), d = _react.useRef(null), { height: m } = (0, _hooks.useBoundingRect)(d) || {
        height: 0
    }, { isTabletOrSmaller: u } = (0, _hooks.useScreenSize)(), { platformOverrides: h } = (0, _hooks.useTheme)(), { height: r, ref: w } = (0, _hooks.useElementDimensions)({
        includeMargins: !0
    }), { visualHeight: D } = (0, _hooks.useWindowSize)(), s = (0, _fixedtotop.useFixedToTopHeight)(), t = D - s - r > ce, { topColor: I = i } = (0, _overscrollcolorcontext.useOverScrollColor)(), { height: x, ref: N } = (0, _hooks.useElementDimensions)();
    (0, _hooks.useIsomorphicLayoutEffect)(()=>{
        n == null || n(r);
    }, [
        n,
        r
    ]), _react.useEffect(()=>{
        if (!u || (0, _platform.isRunningAcceptanceTest)(h) || !t) {
            g(!1);
            return;
        }
        const f = (0, _dom.getScrollableParentElement)(d.current), A = ()=>(0, _dom.hasScroll)(f) && (0, _dom.getScrollDistanceToBottom)(f) > s + 1, o = (0, _helpers.debounce)(()=>{
            g(A());
        }, 50, {
            leading: !0,
            maxWait: 100
        }), B = ne(o), c = ie(f);
        return (0, _dom.addPassiveEventListener)(c, "resize", o), (0, _dom.addPassiveEventListener)(c, "scroll", o), ()=>{
            o.cancel(), (0, _dom.removePassiveEventListener)(c, "scroll", o), (0, _dom.removePassiveEventListener)(c, "resize", o), B.cancel();
        };
    }, [
        h,
        u,
        t,
        // `topDistance` and `contentHeight` dependencies are needed to recalculate the elevation state
        s,
        m
    ]);
    const R = ()=>/* @__PURE__ */ (0, _jsxruntime.jsxs)(_portal.Portal, {
            className: _fixedfooterlayoutcssmistica.portal,
            children: [
                /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                    className: _fixedfooterlayoutcssmistica.fixedBackgroundLayer,
                    style: {
                        background: `linear-gradient(180deg, ${I} 50%, ${i} 50% 100%)`
                    }
                }),
                /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                    ref: N,
                    className: _fixedfooterlayoutcssmistica.absoluteBackgroundLayer,
                    style: {
                        background: F,
                        // this color could be a gradient
                        top: s - 1,
                        // -1 because the navigationbar could have a 1px transparent background
                        bottom: t ? r : "unset",
                        height: t ? "unset" : m
                    }
                }),
                /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                    className: _fixedfooterlayoutcssmistica.absoluteBackgroundLayer,
                    style: {
                        background: i,
                        top: x + s - 1,
                        height: `calc(${m}px - ${x}px)`
                    }
                })
            ]
        });
    return /* @__PURE__ */ (0, _jsxruntime.jsxs)(_jsxruntime.Fragment, {
        children: [
            /* @__PURE__ */ (0, _jsxruntime.jsxs)("div", {
                ref: d,
                className: _fixedfooterlayoutcssmistica.container,
                style: (0, _css.applyCssVars)({
                    [_fixedfooterlayoutcssmistica.vars.footerHeight]: t ? `${r}px` : "0px"
                }),
                children: [
                    R(),
                    L
                ]
            }),
            /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                className: (0, _classnames.default)(_fixedfooterlayoutcssmistica.footer, {
                    [_fixedfooterlayoutcssmistica.withoutFooter]: !e,
                    [_fixedfooterlayoutcssmistica.elevated]: $,
                    [_fixedfooterlayoutcssmistica.fixedFooter]: t
                }),
                style: {
                    background: u ? i : void 0
                },
                "data-testid": `fixed-footer${e ? "-visible" : "-hidden"}`,
                "data-position-fixed": "bottom",
                children: e && /* @__PURE__ */ (0, _jsxruntime.jsx)("aside", {
                    ref: w,
                    "data-component-name": "FixedFooter",
                    style: {
                        height: k,
                        marginBottom: _css.safeAreaInsetBottom
                    },
                    children: l
                })
            })
        ]
    });
}, Se = le;
