// @telefonica/mistica/dist/grid.css-mistica.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    alignSelf: function() {
        return i;
    },
    columnStart: function() {
        return m;
    },
    fullColumns: function() {
        return f;
    },
    fullRows: function() {
        return y;
    },
    grid: function() {
        return k;
    },
    gridAlignContent: function() {
        return e;
    },
    gridAlignItems: function() {
        return r;
    },
    gridAutoFlow: function() {
        return t;
    },
    gridJustifyContent: function() {
        return a;
    },
    gridJustifyItems: function() {
        return n;
    },
    gridTemplateColumns: function() {
        return s;
    },
    gridTemplateColumnsAutoRepeat: function() {
        return o;
    },
    gridTemplateRows: function() {
        return l;
    },
    gridTemplateRowsAutoRepeat: function() {
        return c;
    },
    justifySelf: function() {
        return d;
    },
    rowStart: function() {
        return u;
    },
    spanColumns: function() {
        return p;
    },
    spanRows: function() {
        return g;
    },
    vars: function() {
        return w;
    }
});
require("./grid.css.ts.vanilla.css-mistica.js");
var i = {
    start: "mfyikv34",
    end: "mfyikv35",
    center: "mfyikv36",
    stretch: "mfyikv37",
    baseline: "mfyikv38"
}, m = {
    1: "mfyikv2c",
    2: "mfyikv2d",
    3: "mfyikv2e",
    4: "mfyikv2f",
    5: "mfyikv2g",
    6: "mfyikv2h",
    7: "mfyikv2i",
    8: "mfyikv2j",
    9: "mfyikv2k",
    10: "mfyikv2l",
    11: "mfyikv2m",
    12: "mfyikv2n"
}, f = "mfyikv1m", y = "mfyikv1n", k = "mfyikv4", e = {
    start: "mfyikv1f",
    end: "mfyikv1g",
    center: "mfyikv1h",
    stretch: "mfyikv1i",
    "space-between": "mfyikv1j",
    "space-around": "mfyikv1k",
    "space-evenly": "mfyikv1l"
}, r = {
    start: "mfyikv13",
    end: "mfyikv14",
    center: "mfyikv15",
    stretch: "mfyikv16",
    baseline: "mfyikv17"
}, t = {
    column: "mfyikvv",
    row: "mfyikvw",
    "row dense": "mfyikvx",
    "column dense": "mfyikvy"
}, a = {
    start: "mfyikv18",
    end: "mfyikv19",
    center: "mfyikv1a",
    stretch: "mfyikv1b",
    "space-between": "mfyikv1c",
    "space-around": "mfyikv1d",
    "space-evenly": "mfyikv1e"
}, n = {
    start: "mfyikvz",
    end: "mfyikv10",
    center: "mfyikv11",
    stretch: "mfyikv12"
}, s = {
    1: "mfyikv7",
    2: "mfyikv8",
    3: "mfyikv9",
    4: "mfyikva",
    5: "mfyikvb",
    6: "mfyikvc",
    7: "mfyikvd",
    8: "mfyikve",
    9: "mfyikvf",
    10: "mfyikvg",
    11: "mfyikvh",
    12: "mfyikvi"
}, o = "mfyikv5", l = {
    1: "mfyikvj",
    2: "mfyikvk",
    3: "mfyikvl",
    4: "mfyikvm",
    5: "mfyikvn",
    6: "mfyikvo",
    7: "mfyikvp",
    8: "mfyikvq",
    9: "mfyikvr",
    10: "mfyikvs",
    11: "mfyikvt",
    12: "mfyikvu"
}, c = "mfyikv6", d = {
    start: "mfyikv30",
    end: "mfyikv31",
    center: "mfyikv32",
    stretch: "mfyikv33"
}, u = {
    1: "mfyikv2o",
    2: "mfyikv2p",
    3: "mfyikv2q",
    4: "mfyikv2r",
    5: "mfyikv2s",
    6: "mfyikv2t",
    7: "mfyikv2u",
    8: "mfyikv2v",
    9: "mfyikv2w",
    10: "mfyikv2x",
    11: "mfyikv2y",
    12: "mfyikv2z"
}, p = {
    1: "mfyikv1o",
    2: "mfyikv1p",
    3: "mfyikv1q",
    4: "mfyikv1r",
    5: "mfyikv1s",
    6: "mfyikv1t",
    7: "mfyikv1u",
    8: "mfyikv1v",
    9: "mfyikv1w",
    10: "mfyikv1x",
    11: "mfyikv1y",
    12: "mfyikv1z"
}, g = {
    1: "mfyikv20",
    2: "mfyikv21",
    3: "mfyikv22",
    4: "mfyikv23",
    5: "mfyikv24",
    6: "mfyikv25",
    7: "mfyikv26",
    8: "mfyikv27",
    9: "mfyikv28",
    10: "mfyikv29",
    11: "mfyikv2a",
    12: "mfyikv2b"
}, w = {
    rowMinSize: "var(--mfyikv0)",
    rowMaxSize: "var(--mfyikv1)",
    columnMinSize: "var(--mfyikv2)",
    columnMaxSize: "var(--mfyikv3)"
};
