// @telefonica/mistica/dist/grid.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    Grid: function() {
        return Y;
    },
    GridItem: function() {
        return Z;
    }
});
const _classnames = /*#__PURE__*/ _interop_require_default(require("classnames"));
const _react = /*#__PURE__*/ _interop_require_wildcard(require("react"));
const _gridcssmistica = require("./grid.css-mistica.js");
const _css = require("./utils/css.js");
const _dom = require("./utils/dom.js");
const _sprinklescssmistica = require("./sprinkles.css-mistica.js");
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
function _getRequireWildcardCache(nodeInterop) {
    if (typeof WeakMap !== "function") return null;
    var cacheBabelInterop = new WeakMap();
    var cacheNodeInterop = new WeakMap();
    return (_getRequireWildcardCache = function(nodeInterop) {
        return nodeInterop ? cacheNodeInterop : cacheBabelInterop;
    })(nodeInterop);
}
function _interop_require_wildcard(obj, nodeInterop) {
    if (!nodeInterop && obj && obj.__esModule) {
        return obj;
    }
    if (obj === null || typeof obj !== "object" && typeof obj !== "function") {
        return {
            default: obj
        };
    }
    var cache = _getRequireWildcardCache(nodeInterop);
    if (cache && cache.has(obj)) {
        return cache.get(obj);
    }
    var newObj = {
        __proto__: null
    };
    var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for(var key in obj){
        if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) {
            var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null;
            if (desc && (desc.get || desc.set)) {
                Object.defineProperty(newObj, key, desc);
            } else {
                newObj[key] = obj[key];
            }
        }
    }
    newObj.default = obj;
    if (cache) {
        cache.set(obj, newObj);
    }
    return newObj;
}
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
const K = (t, e)=>{
    const i = {}, f = (r)=>typeof r == "number" ? `min(${r}px, 100%)` : r, o = (r)=>typeof r == "number" ? `${r}px` : r;
    return t && typeof t != "number" && (typeof t.minSize < "u" && (i[_gridcssmistica.vars.rowMinSize] = f(t.minSize)), typeof t.maxSize < "u" && (i[_gridcssmistica.vars.rowMaxSize] = o(t.maxSize))), e && typeof e != "number" && (typeof e.minSize < "u" && (i[_gridcssmistica.vars.columnMinSize] = f(e.minSize)), typeof e.maxSize < "u" && (i[_gridcssmistica.vars.columnMaxSize] = o(e.maxSize))), i;
}, Y = /*#__PURE__*/ _react.forwardRef((param, x)=>{
    let { rows: t, columns: e, gap: i = 0, flow: f, justifyItems: o, alignItems: r, justifyContent: l, alignContent: n, height: u, minHeight: y, children: s, as: z = "div", dataAttributes: S } = param;
    const [m, a] = Array.isArray(i) ? i : [
        i,
        i
    ], A = {
        columnGap: m,
        rowGap: a,
        gridColumnGap: m,
        gridRowGap: a
    };
    let b = {}, g = "";
    try {
        g = (0, _sprinklescssmistica.sprinkles)(A);
    } catch (e) {
        b = {
            columnGap: typeof m == "object" ? m.mobile : m,
            rowGap: typeof a == "object" ? a.mobile : a,
            // Chrome 57-65 support
            gridColumnGap: typeof m == "object" ? m.mobile : m,
            gridRowGap: typeof a == "object" ? a.mobile : a
        };
    }
    return /*#__PURE__*/ _react.createElement(z, _object_spread({
        ref: x,
        className: (0, _classnames.default)(_gridcssmistica.grid, g, e ? typeof e == "number" ? _gridcssmistica.gridTemplateColumns[e] : _gridcssmistica.gridTemplateColumnsAutoRepeat : "", t ? typeof t == "number" ? _gridcssmistica.gridTemplateRows[t] : _gridcssmistica.gridTemplateRowsAutoRepeat : "", f ? _gridcssmistica.gridAutoFlow[f] : "", o ? _gridcssmistica.gridJustifyItems[o] : "", r ? _gridcssmistica.gridAlignItems[r] : "", l ? _gridcssmistica.gridJustifyContent[l] : "", n ? _gridcssmistica.gridAlignContent[n] : ""),
        style: _object_spread({
            height: u,
            minHeight: y
        }, g ? {} : b, (0, _css.applyCssVars)(_object_spread({}, K(t, e))))
    }, (0, _dom.getPrefixedDataAttributes)(S)), s);
}), Z = /*#__PURE__*/ _react.forwardRef((param, s)=>{
    let { rowStart: t, rowSpan: e, columnStart: i, columnSpan: f, span: o, justifySelf: r, alignSelf: l, children: n, as: u = "div", dataAttributes: y } = param;
    return o && (Array.isArray(o) ? (f = o[0], e = o[1]) : (f = o, e = o)), /*#__PURE__*/ _react.createElement(u, _object_spread({
        ref: s,
        className: (0, _classnames.default)(f ? f === "full" ? _gridcssmistica.fullColumns : _gridcssmistica.spanColumns[f] : "", e ? e === "full" ? _gridcssmistica.fullRows : _gridcssmistica.spanRows[e] : "", i ? _gridcssmistica.columnStart[i] : "", t ? _gridcssmistica.rowStart[t] : "", r ? _gridcssmistica.justifySelf[r] : "", l ? _gridcssmistica.alignSelf[l] : "")
    }, (0, _dom.getPrefixedDataAttributes)(y)), n);
});
