// @telefonica/mistica/dist/header.js
"use client";
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    Header: function() {
        return oo;
    },
    HeaderLayout: function() {
        return eo;
    },
    MainSectionHeader: function() {
        return ro;
    },
    MainSectionHeaderLayout: function() {
        return to;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _box = /*#__PURE__*/ _interop_require_default(require("./box.js"));
const _stack = /*#__PURE__*/ _interop_require_default(require("./stack.js"));
const _themevariantcontext = require("./theme-variant-context.js");
const _responsivelayout = /*#__PURE__*/ _interop_require_default(require("./responsive-layout.js"));
const _gridlayout = /*#__PURE__*/ _interop_require_default(require("./grid-layout.js"));
const _overscrollcolorcontext = require("./overscroll-color-context.js");
const _text = require("./text.js");
const _buttongroup = /*#__PURE__*/ _interop_require_default(require("./button-group.js"));
const _skincontractcssmistica = require("./skins/skin-contract.css-mistica.js");
const _headercssmistica = require("./header.css-mistica.js");
const _dom = require("./utils/dom.js");
const _title = require("./title.js");
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
function _object_without_properties(source, excluded) {
    if (source == null) return {};
    var target = _object_without_properties_loose(source, excluded);
    var key, i;
    if (Object.getOwnPropertySymbols) {
        var sourceSymbolKeys = Object.getOwnPropertySymbols(source);
        for(i = 0; i < sourceSymbolKeys.length; i++){
            key = sourceSymbolKeys[i];
            if (excluded.indexOf(key) >= 0) continue;
            if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
            target[key] = source[key];
        }
    }
    return target;
}
function _object_without_properties_loose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;
    for(i = 0; i < sourceKeys.length; i++){
        key = sourceKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        target[key] = source[key];
    }
    return target;
}
const oo = (param)=>{
    let { headline: r, pretitle: e, pretitleAs: t, title: l, titleAs: d = "h2", description: m, dataAttributes: f, small: s = !1, preamount: c, amount: p, button: g, subtitle: k, isErrorAmount: L, secondaryButton: y } = param;
    const S = (0, _themevariantcontext.useIsInverseVariant)(), b = (h, v)=>{
        if (typeof h == "string") return /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text3, _object_spread_props(_object_spread({
            regular: !0
        }, v), {
            children: h
        }));
        const { text: D } = h, C = _object_without_properties(h, [
            "text"
        ]);
        return /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text3, _object_spread_props(_object_spread({
            regular: !0
        }, v, C), {
            children: h.text
        }));
    };
    return /* @__PURE__ */ (0, _jsxruntime.jsxs)(_stack.default, {
        space: {
            mobile: 24,
            desktop: 32
        },
        dataAttributes: f,
        children: [
            (l || e || m) && /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                paddingRight: 16,
                children: /* @__PURE__ */ (0, _jsxruntime.jsxs)(_stack.default, {
                    space: 8,
                    children: [
                        r,
                        e && b(e, {
                            color: _skincontractcssmistica.vars.colors.textPrimary,
                            as: t
                        }),
                        l && (s ? /* @__PURE__ */ (0, _jsxruntime.jsx)(_title.Title2, {
                            as: d,
                            children: l
                        }) : /* @__PURE__ */ (0, _jsxruntime.jsx)(_title.Title3, {
                            as: d,
                            children: l
                        })),
                        m && (s ? /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text2, {
                            regular: !0,
                            color: _skincontractcssmistica.vars.colors.textSecondary,
                            children: m
                        }) : /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text3, {
                            regular: !0,
                            color: _skincontractcssmistica.vars.colors.textSecondary,
                            children: m
                        }))
                    ]
                })
            }),
            (c || p || g || k) && /* @__PURE__ */ (0, _jsxruntime.jsxs)(_stack.default, {
                space: 16,
                children: [
                    (c || p) && /* @__PURE__ */ (0, _jsxruntime.jsxs)(_stack.default, {
                        space: 8,
                        children: [
                            c && b(c, {
                                color: _skincontractcssmistica.vars.colors.textPrimary
                            }),
                            /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text8, {
                                color: L && !S ? _skincontractcssmistica.vars.colors.highlight : _skincontractcssmistica.vars.colors.textPrimary,
                                children: p
                            })
                        ]
                    }),
                    (g || y) && /* @__PURE__ */ (0, _jsxruntime.jsx)(_buttongroup.default, {
                        primaryButton: g,
                        secondaryButton: y
                    }),
                    k && b(k, {})
                ]
            })
        ]
    });
}, ro = (param)=>{
    let { title: r, titleAs: e = "h1", description: t, button: l } = param;
    return /* @__PURE__ */ (0, _jsxruntime.jsxs)(_stack.default, {
        space: 32,
        children: [
            /* @__PURE__ */ (0, _jsxruntime.jsxs)(_stack.default, {
                space: {
                    mobile: 12,
                    desktop: 16
                },
                children: [
                    r && /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text7, {
                        as: e,
                        children: r
                    }),
                    t && /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text6, {
                        children: t
                    })
                ]
            }),
            l
        ]
    });
}, eo = (param)=>{
    let { isInverse: r = !0, breadcrumbs: e, header: t, extra: l, sideBySideExtraOnDesktop: d = !1, dataAttributes: m, bleed: f = !1, noPaddingY: s = !1 } = param;
    const c = f && r && l, p = /* @__PURE__ */ (0, _jsxruntime.jsxs)("div", {
        children: [
            e && /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                className: _headercssmistica.breadcrumbs,
                children: e
            }),
            t
        ]
    });
    return (0, _overscrollcolorcontext.useSetOverscrollColor)(r ? {
        topColor: _skincontractcssmistica.vars.colors.backgroundBrandTop
    } : {}), /* @__PURE__ */ (0, _jsxruntime.jsxs)("div", _object_spread_props(_object_spread({}, (0, _dom.getPrefixedDataAttributes)(_object_spread({
        "component-name": "HeaderLayout"
    }, m))), {
        children: [
            /* @__PURE__ */ (0, _jsxruntime.jsx)(_responsivelayout.default, {
                isInverse: r,
                children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                    paddingTop: s ? 0 : {
                        mobile: t ? 32 : 0,
                        desktop: e ? 16 : 48
                    },
                    paddingBottom: {
                        mobile: s && !c ? 0 : 24,
                        desktop: c && !d ? 32 : s ? 0 : 48
                    },
                    children: d ? /* @__PURE__ */ (0, _jsxruntime.jsx)(_gridlayout.default, {
                        template: "6+6",
                        left: p,
                        right: /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                            className: c ? _headercssmistica.hideOnTabletOrSmaller : "",
                            children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                                paddingTop: {
                                    mobile: t ? 24 : 0,
                                    desktop: 0
                                },
                                children: l
                            })
                        })
                    }) : /* @__PURE__ */ (0, _jsxruntime.jsxs)(_stack.default, {
                        space: t ? {
                            mobile: 24,
                            desktop: 32
                        } : 0,
                        children: [
                            p,
                            !c && l
                        ]
                    })
                })
            }),
            c && /* @__PURE__ */ (0, _jsxruntime.jsx)(_responsivelayout.default, {
                className: d ? _headercssmistica.hideOnDesktop : "",
                backgroundColor: `linear-gradient(to bottom, ${_skincontractcssmistica.vars.colors.backgroundBrandBottom} 40px, ${_skincontractcssmistica.vars.colors.background} 0%)`,
                children: l
            })
        ]
    }));
}, to = (param)=>{
    let { isInverse: r = !0, children: e, dataAttributes: t } = param;
    return (0, _overscrollcolorcontext.useSetOverscrollColor)(r ? {
        topColor: _skincontractcssmistica.vars.colors.backgroundBrandTop
    } : {}), /* @__PURE__ */ (0, _jsxruntime.jsx)(_responsivelayout.default, {
        isInverse: r,
        dataAttributes: _object_spread({
            "component-name": "MainSectionHeaderLayout"
        }, t),
        children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_gridlayout.default, {
            template: "6+6",
            left: /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                paddingTop: {
                    mobile: 12,
                    desktop: 48
                },
                paddingBottom: {
                    mobile: 24,
                    desktop: 48
                },
                children: e
            }),
            right: null
        })
    });
};
