// @telefonica/mistica/dist/hero.css-mistica.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    actions: function() {
        return t;
    },
    container: function() {
        return o;
    },
    containerDesktop: function() {
        return a;
    },
    containerMinHeight: function() {
        return v;
    },
    containerMobile: function() {
        return i;
    },
    expandedContent: function() {
        return _;
    },
    layout: function() {
        return e;
    },
    vars: function() {
        return p;
    }
});
require("./sprinkles.css.ts.vanilla.css-mistica.js");
require("./hero.css.ts.vanilla.css-mistica.js");
var t = "_16rbpto7 _1y2v1nfhk _1y2v1nfi6 _1y2v1nfi0", o = "_16rbpto2 _1y2v1nfhk", a = "_1y2v1nfhz _1y2v1nfio", v = "_16rbpto3", i = "_1y2v1nfhq", _ = "_16rbpto9", e = "_16rbpto8", p = {
    height: "var(--_16rbpto0)"
};
