// @telefonica/mistica/dist/image.css-mistica.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    FADE_IN_DURATION_MS: function() {
        return v;
    },
    circularBorderRadius: function() {
        return _;
    },
    defaultBorderRadius: function() {
        return t;
    },
    image: function() {
        return i;
    },
    imageWithBorder: function() {
        return d;
    },
    noBorderRadius: function() {
        return f;
    },
    vars: function() {
        return e;
    }
});
require("./sprinkles.css.ts.vanilla.css-mistica.js");
require("./image.css.ts.vanilla.css-mistica.js");
var v = 300, _ = "_17tt25r5", t = "_17tt25r6", i = "_17tt25r3 _1y2v1nfkr _1y2v1nfl6 _1y2v1nfhm _1y2v1nfi7 _1y2v1nfio _1y2v1nfmf _1y2v1nfjz _1y2v1nfk0 _1y2v1nfkb", d = "_17tt25r1", f = "_17tt25r4", e = {
    mediaBorderRadius: "var(--_17tt25r0)"
};
