// @telefonica/mistica/dist/list.js
"use client";
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    BoxedRow: function() {
        return bt;
    },
    BoxedRowList: function() {
        return xt;
    },
    Content: function() {
        return Ye;
    },
    Row: function() {
        return ut;
    },
    RowList: function() {
        return ft;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _react = /*#__PURE__*/ _interop_require_wildcard(require("react"));
const _classnames = /*#__PURE__*/ _interop_require_default(require("classnames"));
const _touchable = require("./touchable.js");
const _text = require("./text.js");
const _box = /*#__PURE__*/ _interop_require_default(require("./box.js"));
const _stack = /*#__PURE__*/ _interop_require_default(require("./stack.js"));
const _badge = /*#__PURE__*/ _interop_require_default(require("./badge.js"));
const _hooks = require("./hooks.js");
const _themevariantcontext = require("./theme-variant-context.js");
const _iconchevron = /*#__PURE__*/ _interop_require_default(require("./icons/icon-chevron.js"));
const _switchcomponent = /*#__PURE__*/ _interop_require_default(require("./switch-component.js"));
const _radiobutton = /*#__PURE__*/ _interop_require_wildcard(require("./radio-button.js"));
const _checkbox = /*#__PURE__*/ _interop_require_default(require("./checkbox.js"));
const _boxed = require("./boxed.js");
const _divider = /*#__PURE__*/ _interop_require_default(require("./divider.js"));
const _dom = require("./utils/dom.js");
const _listcssmistica = require("./list.css-mistica.js");
const _imagecssmistica = require("./image.css-mistica.js");
const _skincontractcssmistica = require("./skins/skin-contract.css-mistica.js");
const _css = require("./utils/css.js");
const _iconbutton = require("./icon-button.js");
const _sprinklescssmistica = require("./sprinkles.css-mistica.js");
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
function _getRequireWildcardCache(nodeInterop) {
    if (typeof WeakMap !== "function") return null;
    var cacheBabelInterop = new WeakMap();
    var cacheNodeInterop = new WeakMap();
    return (_getRequireWildcardCache = function(nodeInterop) {
        return nodeInterop ? cacheNodeInterop : cacheBabelInterop;
    })(nodeInterop);
}
function _interop_require_wildcard(obj, nodeInterop) {
    if (!nodeInterop && obj && obj.__esModule) {
        return obj;
    }
    if (obj === null || typeof obj !== "object" && typeof obj !== "function") {
        return {
            default: obj
        };
    }
    var cache = _getRequireWildcardCache(nodeInterop);
    if (cache && cache.has(obj)) {
        return cache.get(obj);
    }
    var newObj = {
        __proto__: null
    };
    var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for(var key in obj){
        if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) {
            var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null;
            if (desc && (desc.get || desc.set)) {
                Object.defineProperty(newObj, key, desc);
            } else {
                newObj[key] = obj[key];
            }
        }
    }
    newObj.default = obj;
    if (cache) {
        cache.set(obj, newObj);
    }
    return newObj;
}
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
function _object_without_properties(source, excluded) {
    if (source == null) return {};
    var target = _object_without_properties_loose(source, excluded);
    var key, i;
    if (Object.getOwnPropertySymbols) {
        var sourceSymbolKeys = Object.getOwnPropertySymbols(source);
        for(i = 0; i < sourceSymbolKeys.length; i++){
            key = sourceSymbolKeys[i];
            if (excluded.indexOf(key) >= 0) continue;
            if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
            target[key] = source[key];
        }
    }
    return target;
}
function _object_without_properties_loose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;
    for(i = 0; i < sourceKeys.length; i++){
        key = sourceKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        target[key] = source[key];
    }
    return target;
}
const Fe = (e, a)=>typeof e == "function" ? e == null ? void 0 : e({
        centerY: a
    }) : a ? /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
        style: {
            display: "flex",
            alignItems: "center",
            height: "100%"
        },
        children: /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
            children: e
        })
    }) : e, Ye = (param)=>{
    let { withChevron: e, headline: a, headlineRef: r, extraRef: l, title: c, titleAs: v, titleLinesMax: g, subtitle: x, subtitleLinesMax: P, description: w, descriptionLinesMax: D, detail: h, asset: E, danger: T, badge: N, right: y, rightRef: u, extra: k, labelId: W, disabled: s, control: L } = param;
    const A = (0, _themevariantcontext.useIsInverseVariant)(), S = [
        a,
        c,
        x,
        w,
        k
    ].filter(Boolean).length === 1;
    return /* @__PURE__ */ (0, _jsxruntime.jsxs)(_box.default, {
        paddingY: 16,
        className: _listcssmistica.content,
        id: W,
        children: [
            E && /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                paddingRight: 16,
                className: (0, _classnames.default)({
                    [_listcssmistica.center]: S,
                    [_listcssmistica.disabled]: s
                }),
                "aria-hidden": !0,
                children: /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                    className: _listcssmistica.asset,
                    style: (0, _css.applyCssVars)({
                        color: T ? A ? _skincontractcssmistica.vars.colors.textErrorInverse : _skincontractcssmistica.vars.colors.textError : A ? _skincontractcssmistica.vars.colors.textPrimaryInverse : _skincontractcssmistica.vars.colors.textPrimary,
                        [_imagecssmistica.vars.mediaBorderRadius]: _skincontractcssmistica.vars.borderRadii.mediaSmall
                    }),
                    children: E
                })
            }),
            /* @__PURE__ */ (0, _jsxruntime.jsxs)("div", {
                className: (0, _classnames.default)(_listcssmistica.rowBody, {
                    [_listcssmistica.disabled]: s
                }),
                style: {
                    justifyContent: S ? "center" : "flex-start"
                },
                children: [
                    /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text3, {
                        regular: !0,
                        color: T ? _skincontractcssmistica.vars.colors.textError : _skincontractcssmistica.vars.colors.textPrimary,
                        truncate: g,
                        hyphens: "auto",
                        as: v,
                        children: c
                    }),
                    a && /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                        ref: r,
                        style: {
                            order: -1,
                            paddingBottom: 4
                        },
                        children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text1, {
                            regular: !0,
                            color: _skincontractcssmistica.vars.colors.textPrimary,
                            hyphens: "auto",
                            children: a
                        })
                    }),
                    x && /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                        paddingTop: 2,
                        children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text2, {
                            regular: !0,
                            color: _skincontractcssmistica.vars.colors.textPrimary,
                            truncate: P,
                            hyphens: "auto",
                            children: x
                        })
                    }),
                    w && /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                        paddingTop: 2,
                        children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text2, {
                            regular: !0,
                            color: _skincontractcssmistica.vars.colors.textSecondary,
                            truncate: D,
                            hyphens: "auto",
                            children: w
                        })
                    }),
                    k && /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                        ref: l,
                        paddingTop: 2,
                        children: k
                    })
                ]
            }),
            N && /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                paddingLeft: 16,
                children: /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                    className: (0, _classnames.default)(_listcssmistica.badge, {
                        [_listcssmistica.disabled]: s
                    }),
                    children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_badge.default, {
                        value: N === !0 ? void 0 : N
                    })
                })
            }),
            (h || y || e || L) && /* @__PURE__ */ (0, _jsxruntime.jsxs)("div", {
                className: (0, _classnames.default)(_listcssmistica.rightContent, {
                    [_listcssmistica.rightRestrictedWidth]: !!h
                }),
                children: [
                    h && /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                        className: (0, _classnames.default)(_listcssmistica.detail, {
                            [_listcssmistica.disabled]: s
                        }),
                        children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text2, {
                            regular: !0,
                            color: _skincontractcssmistica.vars.colors.textSecondary,
                            hyphens: "auto",
                            children: h
                        })
                    }),
                    y && /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                        className: (0, _classnames.default)({
                            [_listcssmistica.detailRight]: !!h,
                            [_listcssmistica.disabled]: s
                        }),
                        ref: u,
                        children: Fe(y, S)
                    }),
                    e && /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                        paddingLeft: h || y ? 4 : 0,
                        className: (0, _classnames.default)(_listcssmistica.center, {
                            [_listcssmistica.disabled]: s
                        }),
                        children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_iconchevron.default, {
                            color: A ? _skincontractcssmistica.vars.colors.inverse : _skincontractcssmistica.vars.colors.neutralMedium,
                            direction: "right"
                        })
                    }),
                    L && /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                        paddingLeft: h || y ? 8 : 0,
                        className: _listcssmistica.center,
                        children: L
                    })
                ]
            })
        ]
    });
}, qe = (param)=>{
    let { value: e, defaultValue: a, onChange: r } = param;
    const l = e !== void 0, [c, v] = _react.useState(!!a);
    return [
        l ? !!e : c,
        ()=>{
            l || v(!c), r == null || r(l ? !e : !c);
        }
    ];
}, ze = (e)=>[
        "switch",
        "checkbox",
        "radioValue",
        "iconButton"
    ].some((a)=>e[a] !== void 0), le = /*#__PURE__*/ _react.forwardRef((e, a)=>{
    var p, ee;
    const r = (0, _hooks.useAriaId)(), l = (0, _themevariantcontext.useIsInverseVariant)(), { asset: c, headline: v, title: g, titleAs: x, titleLinesMax: P, subtitle: w, subtitleLinesMax: D, description: h, descriptionLinesMax: E, detail: T, danger: N, badge: y, role: u, extra: k, withChevron: W, dataAttributes: s, right: L, "aria-label": A } = e, [Q, S] = _react.useState(""), [de, ce] = _react.useState(""), [se, he] = _react.useState(""), me = [
        g,
        Q,
        w,
        h,
        de,
        T,
        se
    ].filter(Boolean).join(" "), B = A !== null && A !== void 0 ? A : e.href || e.to ? me : void 0, ue = (0, _radiobutton.useRadioContext)(), f = e.disabled || e.radioValue !== void 0 && ue.disabled, j = !f && !l, M = !f && l, U = ze(e), V = !!e.onPress || !!e.href || !!e.to, fe = U ? !1 : W !== null && W !== void 0 ? W : V, Z = {
        href: e.href,
        newTab: e.newTab,
        loadOnTop: e.loadOnTop,
        to: e.to,
        fullPageOnWebView: e.fullPageOnWebView,
        replace: e.replace,
        onNavigate: e.onNavigate,
        onPress: e.onPress,
        trackingEvent: e.trackingEvent
    }, [_, $] = qe(e.switch || e.checkbox || {}), R = (i)=>/* @__PURE__ */ (0, _jsxruntime.jsx)(Ye, {
            asset: c,
            headline: v,
            headlineRef: (n)=>{
                n && S(n.textContent || "");
            },
            title: g,
            titleAs: x,
            subtitle: w,
            description: h,
            badge: y,
            titleLinesMax: P,
            subtitleLinesMax: D,
            descriptionLinesMax: E,
            detail: T,
            danger: N,
            right: L,
            rightRef: (n)=>{
                n && he(n.innerText || n.textContent || "");
            },
            control: i == null ? void 0 : i.control,
            extra: k,
            extraRef: (n)=>{
                n && ce(n.innerText || n.textContent || "");
            },
            labelId: i == null ? void 0 : i.labelId,
            disabled: f,
            withChevron: fe
        });
    if (V && !U) return /* @__PURE__ */ (0, _jsxruntime.jsx)(_touchable.BaseTouchable, _object_spread_props(_object_spread({
        ref: a,
        className: (0, _classnames.default)(_listcssmistica.rowContent, {
            [_listcssmistica.touchableBackground]: j,
            [_listcssmistica.touchableBackgroundInverse]: M,
            [_listcssmistica.pointer]: !f
        })
    }, Z), {
        role: u,
        dataAttributes: s,
        disabled: f,
        "aria-label": B,
        children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
            paddingX: 16,
            "aria-hidden": !!e.to || !!e.href || void 0,
            children: R()
        })
    }));
    const O = (i)=>/* @__PURE__ */ (0, _jsxruntime.jsxs)("div", _object_spread_props(_object_spread({
            className: _listcssmistica.dualActionContainer,
            ref: a
        }, (0, _dom.getPrefixedDataAttributes)(s)), {
            children: [
                /* @__PURE__ */ (0, _jsxruntime.jsx)(_touchable.BaseTouchable, _object_spread_props(_object_spread({
                    disabled: f
                }, Z), {
                    role: u,
                    className: (0, _classnames.default)(_listcssmistica.dualActionLeft, {
                        [_listcssmistica.touchableBackground]: j,
                        [_listcssmistica.touchableBackgroundInverse]: M
                    }),
                    "aria-label": B,
                    children: R({
                        labelId: r
                    })
                })),
                /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                    className: _listcssmistica.dualActionDivider
                }),
                i
            ]
        })), H = (i, n)=>/* @__PURE__ */ (0, _jsxruntime.jsx)("div", _object_spread_props(_object_spread({
            className: (0, _classnames.default)(_listcssmistica.rowContent, {
                [_listcssmistica.touchableBackground]: j && n,
                [_listcssmistica.touchableBackgroundInverse]: M && n,
                [_listcssmistica.pointer]: !f && n
            }),
            ref: a
        }, (0, _dom.getPrefixedDataAttributes)(s)), {
            children: i
        }));
    if (e.switch || e.checkbox) {
        var _ref, _ref1;
        const i = e.switch ? _switchcomponent.default : _checkbox.default, n = (_ref1 = (_ref = (p = e.switch) == null ? void 0 : p.name) !== null && _ref !== void 0 ? _ref : (ee = e.checkbox) == null ? void 0 : ee.name) !== null && _ref1 !== void 0 ? _ref1 : r;
        return V ? O(/* @__PURE__ */ (0, _jsxruntime.jsx)(i, {
            disabled: f,
            name: n,
            checked: _,
            "aria-label": B,
            "aria-labelledby": r,
            onChange: $,
            render: (param)=>{
                let { controlElement: F } = param;
                return /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                    className: _listcssmistica.dualActionRight,
                    children: F
                });
            }
        })) : H(/* @__PURE__ */ (0, _jsxruntime.jsx)(i, {
            disabled: f,
            name: n,
            checked: _,
            "aria-label": B,
            "aria-labelledby": r,
            onChange: $,
            render: (param)=>{
                let { controlElement: F, labelId: be } = param;
                return /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                    paddingX: 16,
                    role: u,
                    children: R({
                        labelId: be,
                        control: /* @__PURE__ */ (0, _jsxruntime.jsx)(_stack.default, {
                            space: "around",
                            children: F
                        })
                    })
                });
            }
        }), !0);
    }
    return e.radioValue ? V ? O(/* @__PURE__ */ (0, _jsxruntime.jsx)(_radiobutton.default, {
        value: e.radioValue,
        "aria-label": B,
        "aria-labelledby": r,
        render: (param)=>{
            let { controlElement: i } = param;
            return /* @__PURE__ */ (0, _jsxruntime.jsx)(_stack.default, {
                space: "around",
                children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                    paddingX: 16,
                    children: i
                })
            });
        }
    })) : H(/* @__PURE__ */ (0, _jsxruntime.jsx)(_radiobutton.default, {
        value: e.radioValue,
        "aria-label": B,
        "aria-labelledby": r,
        render: (param)=>{
            let { controlElement: i } = param;
            return /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                paddingX: 16,
                role: u,
                children: R({
                    labelId: r,
                    control: /* @__PURE__ */ (0, _jsxruntime.jsx)(_stack.default, {
                        space: "around",
                        children: i
                    })
                })
            });
        }
    }), !0) : e.iconButton ? V ? O(/* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
        padding: 16,
        children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_stack.default, {
            space: "around",
            children: e.iconButton.Icon ? /* @__PURE__ */ (0, _jsxruntime.jsx)(_iconbutton.IconButton, _object_spread_props(_object_spread({}, e.iconButton), {
                disabled: e.disabled
            })) : /* @__PURE__ */ (0, _jsxruntime.jsx)(_iconbutton.ToggleIconButton, _object_spread_props(_object_spread({}, e.iconButton), {
                disabled: e.disabled
            }))
        })
    })) : H(/* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
        paddingX: 16,
        children: R({
            labelId: r,
            control: /* @__PURE__ */ (0, _jsxruntime.jsx)(_stack.default, {
                space: "around",
                children: e.iconButton.Icon ? /* @__PURE__ */ (0, _jsxruntime.jsx)(_iconbutton.IconButton, _object_spread_props(_object_spread({}, e.iconButton), {
                    disabled: e.disabled,
                    role: u
                })) : /* @__PURE__ */ (0, _jsxruntime.jsx)(_iconbutton.ToggleIconButton, _object_spread_props(_object_spread({}, e.iconButton), {
                    disabled: e.disabled,
                    role: u
                }))
            })
        })
    })) : /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
        paddingX: 16,
        className: _listcssmistica.rowContent,
        role: u,
        dataAttributes: s,
        ref: a,
        children: R()
    });
}), ut = /*#__PURE__*/ _react.forwardRef((_param, l)=>{
    var { dataAttributes: e, role: a = "listitem" } = _param, r = _object_without_properties(_param, [
        "dataAttributes",
        "role"
    ]);
    return /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
        role: a,
        className: (0, _sprinklescssmistica.sprinkles)({
            width: "100%"
        }),
        children: /* @__PURE__ */ (0, _jsxruntime.jsx)(le, _object_spread_props(_object_spread({}, r), {
            ref: l,
            dataAttributes: _object_spread({
                "component-name": "Row"
            }, e)
        }))
    });
}), ft = (param)=>{
    let { children: e, ariaLabelledby: a, role: r = "list", dataAttributes: l } = param;
    const c = _react.Children.toArray(e).filter(Boolean), v = c.length - 1;
    return /* @__PURE__ */ (0, _jsxruntime.jsx)("div", _object_spread_props(_object_spread({
        role: r,
        "aria-labelledby": a
    }, (0, _dom.getPrefixedDataAttributes)(l, "RowList")), {
        children: c.map((g, x)=>/* @__PURE__ */ (0, _jsxruntime.jsxs)(_react.Fragment, {
                children: [
                    g,
                    x < v && /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                        paddingX: 16,
                        children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_divider.default, {})
                    })
                ]
            }, x))
    }));
}, bt = /*#__PURE__*/ _react.forwardRef((_param, r)=>{
    var { dataAttributes: e } = _param, a = _object_without_properties(_param, [
        "dataAttributes"
    ]);
    return /* @__PURE__ */ (0, _jsxruntime.jsx)(_boxed.Boxed, {
        isInverse: a.isInverse,
        ref: r,
        dataAttributes: _object_spread({
            "component-name": "BoxedRow"
        }, e),
        children: /* @__PURE__ */ (0, _jsxruntime.jsx)(le, _object_spread({}, a))
    });
}), xt = (param)=>{
    let { children: e, ariaLabelledby: a, role: r = "list", dataAttributes: l } = param;
    return /* @__PURE__ */ (0, _jsxruntime.jsx)(_stack.default, {
        space: 16,
        role: r,
        "aria-labelledby": a,
        dataAttributes: _object_spread({
            "component-name": "BoxedRowList"
        }, l),
        children: e
    });
};
