// @telefonica/mistica/dist/loading-screen.css-mistica.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    inOutAnimationMs: function() {
        return r;
    },
    loadingScreen: function() {
        return a;
    },
    loadingScreenChildren: function() {
        return f;
    },
    loadingScreenText: function() {
        return v;
    },
    loadingScreenTextAnimated: function() {
        return l;
    },
    loadingScreenTextAnimatedOut: function() {
        return i;
    },
    logo: function() {
        return o;
    },
    pulseLogo: function() {
        return k;
    },
    screenBackground: function() {
        return t;
    },
    screenBackgroundAnimated: function() {
        return d;
    },
    screenBackgroundFadeOut: function() {
        return c;
    }
});
require("./sprinkles.css.ts.vanilla.css-mistica.js");
require("./loading-screen.css.ts.vanilla.css-mistica.js");
var r = 500, a = "flke8b1 _1y2v1nfhk _1y2v1nfhq _1y2v1nfhz", f = "_1y2v1nfi6 _1y2v1nfhk _1y2v1nfhq _1y2v1nfhz _1y2v1nfht", v = "flke8b9 _1y2v1nf7y", l = "flke8bb", i = "flke8bc", o = "flke8bf", k = "flke8be", t = {
    default: "flke8b4",
    inverse: "flke8b5"
}, d = "flke8b3", c = "flke8b6";
