// @telefonica/mistica/dist/maybe-dismissable.js
"use client";
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    default: function() {
        return M;
    },
    useIsDismissable: function() {
        return j;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _react = /*#__PURE__*/ _interop_require_wildcard(require("react"));
const _iconcloseregular = /*#__PURE__*/ _interop_require_default(require("./generated/mistica-icons/icon-close-regular.js"));
const _hooks = require("./hooks.js");
const _iconbutton = require("./icon-button.js");
const _maybedismissablecssmistica = require("./maybe-dismissable.css-mistica.js");
const _themevariantcontext = require("./theme-variant-context.js");
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
function _getRequireWildcardCache(nodeInterop) {
    if (typeof WeakMap !== "function") return null;
    var cacheBabelInterop = new WeakMap();
    var cacheNodeInterop = new WeakMap();
    return (_getRequireWildcardCache = function(nodeInterop) {
        return nodeInterop ? cacheNodeInterop : cacheBabelInterop;
    })(nodeInterop);
}
function _interop_require_wildcard(obj, nodeInterop) {
    if (!nodeInterop && obj && obj.__esModule) {
        return obj;
    }
    if (obj === null || typeof obj !== "object" && typeof obj !== "function") {
        return {
            default: obj
        };
    }
    var cache = _getRequireWildcardCache(nodeInterop);
    if (cache && cache.has(obj)) {
        return cache.get(obj);
    }
    var newObj = {
        __proto__: null
    };
    var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for(var key in obj){
        if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) {
            var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null;
            if (desc && (desc.get || desc.set)) {
                Object.defineProperty(newObj, key, desc);
            } else {
                newObj[key] = obj[key];
            }
        }
    }
    newObj.default = obj;
    if (cache) {
        cache.set(obj, newObj);
    }
    return newObj;
}
const r = /*#__PURE__*/ _react.createContext(!1), j = ()=>_react.useContext(r), I = (param)=>{
    let { children: s, width: i, onClose: t, "aria-label": o, isOverMedia: m, isInverse: l } = param;
    const { texts: n } = (0, _hooks.useTheme)();
    return t ? /* @__PURE__ */ (0, _jsxruntime.jsxs)("section", {
        className: _maybedismissablecssmistica.dismissableContainer,
        "aria-label": o,
        style: {
            width: i || "100%",
            minHeight: "100%"
        },
        children: [
            /* @__PURE__ */ (0, _jsxruntime.jsx)(r.Provider, {
                value: !0,
                children: s
            }),
            /* @__PURE__ */ (0, _jsxruntime.jsx)(_themevariantcontext.ThemeVariant, {
                isInverse: l,
                children: /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                    className: _maybedismissablecssmistica.dismissableButton,
                    children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_iconbutton.InternalIconButton, {
                        onPress: t,
                        "aria-label": n.closeButtonLabel,
                        small: !0,
                        isOverMedia: m,
                        Icon: _iconcloseregular.default
                    })
                })
            })
        ]
    }) : /* @__PURE__ */ (0, _jsxruntime.jsx)(_jsxruntime.Fragment, {
        children: s
    });
}, M = I;
