// @telefonica/mistica/dist/media-queries.css-mistica.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    desktop: function() {
        return e;
    },
    desktopOrBigger: function() {
        return n;
    },
    largeDesktop: function() {
        return i;
    },
    mediaQueriesConfig: function() {
        return a;
    },
    mobile: function() {
        return t;
    },
    supportsHover: function() {
        return r;
    },
    tablet: function() {
        return d;
    },
    tabletOrBigger: function() {
        return h;
    },
    tabletOrSmaller: function() {
        return p;
    },
    touchableOnly: function() {
        return o;
    }
});
var e = "only screen and (min-height: 550px) and (min-width: 1024px) and (max-width: 1367px)", n = "only screen and (min-height: 550px) and (min-width: 1024px)", i = "only screen and (min-height: 550px) and (min-width: 1368px)", a = {
    tabletMinWidth: 768,
    desktopMinWidth: 1024,
    largeDesktopMinWidth: 1368,
    desktopOrTabletMinHeight: 550
}, t = "only screen and (max-width: 767px), (max-height: 549px)", r = "(pointer: fine), (pointer: none)", d = "only screen and (min-height: 550px) and (min-width: 768px) and (max-width: 1023px)", h = "only screen and (min-height: 550px) and (min-width: 768px)", p = "only screen and (max-width: 1023px), (max-height: 549px)", o = "(pointer: coarse)";
