// @telefonica/mistica/dist/menu.js
"use client";
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    Menu: function() {
        return Je;
    },
    MenuItem: function() {
        return qe;
    },
    MenuSection: function() {
        return Ve;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _react = /*#__PURE__*/ _interop_require_wildcard(require("react"));
const _classnames = /*#__PURE__*/ _interop_require_default(require("classnames"));
const _keys = require("./utils/keys.js");
const _dom = require("./utils/dom.js");
const _overlay = /*#__PURE__*/ _interop_require_default(require("./overlay.js"));
const _menucssmistica = require("./menu.css-mistica.js");
const _hooks = require("./hooks.js");
const _portal = require("./portal.js");
const _box = /*#__PURE__*/ _interop_require_default(require("./box.js"));
const _inline = /*#__PURE__*/ _interop_require_default(require("./inline.js"));
const _touchable = /*#__PURE__*/ _interop_require_default(require("./touchable.js"));
const _text = require("./text.js");
const _skincontractcssmistica = require("./skins/skin-contract.css-mistica.js");
const _divider = /*#__PURE__*/ _interop_require_default(require("./divider.js"));
const _checkbox = /*#__PURE__*/ _interop_require_default(require("./checkbox.js"));
const _reacttransitiongroup = require("react-transition-group");
const _common = require("./utils/common.js");
const _css = require("./utils/css.js");
const _platform = require("./utils/platform.js");
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
function _getRequireWildcardCache(nodeInterop) {
    if (typeof WeakMap !== "function") return null;
    var cacheBabelInterop = new WeakMap();
    var cacheNodeInterop = new WeakMap();
    return (_getRequireWildcardCache = function(nodeInterop) {
        return nodeInterop ? cacheNodeInterop : cacheBabelInterop;
    })(nodeInterop);
}
function _interop_require_wildcard(obj, nodeInterop) {
    if (!nodeInterop && obj && obj.__esModule) {
        return obj;
    }
    if (obj === null || typeof obj !== "object" && typeof obj !== "function") {
        return {
            default: obj
        };
    }
    var cache = _getRequireWildcardCache(nodeInterop);
    if (cache && cache.has(obj)) {
        return cache.get(obj);
    }
    var newObj = {
        __proto__: null
    };
    var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for(var key in obj){
        if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) {
            var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null;
            if (desc && (desc.get || desc.set)) {
                Object.defineProperty(newObj, key, desc);
            } else {
                newObj[key] = obj[key];
            }
        }
    }
    newObj.default = obj;
    if (cache) {
        cache.set(obj, newObj);
    }
    return newObj;
}
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
const ke = 120, Q = /*#__PURE__*/ _react.createContext({
    focusedItem: null,
    isMenuOpen: !1,
    setFocusedItem: ()=>{},
    closeMenu: ()=>{}
}), Ne = ()=>_react.useContext(Q), D = (r)=>r ? Array.from(r.querySelectorAll("[role=menuitem],[role=menuitemcheckbox]")) : [], Re = (r, f)=>{
    if (!f) return null;
    const d = D(r).indexOf(f);
    return d < 0 ? null : d;
}, qe = (param)=>{
    let { label: r, Icon: f, destructive: d, disabled: u, onPress: C, controlType: s, checked: p } = param;
    const { focusedItem: n, setFocusedItem: v, closeMenu: i, isMenuOpen: k } = Ne(), c = _react.useRef(null), h = d ? _skincontractcssmistica.vars.colors.textLinkDanger : _skincontractcssmistica.vars.colors.neutralHigh, x = c == null ? void 0 : c.current, M = (x == null ? void 0 : x.closest("[role=menu]")) || null, m = Re(M, x), l = ()=>s === "checkbox" ? /* @__PURE__ */ (0, _jsxruntime.jsx)(_checkbox.default, {
            ref: c,
            name: r,
            checked: p,
            onChange: ()=>{
                k && m !== null && (C(m), i());
            },
            disabled: u,
            role: "menuitemcheckbox",
            render: (param)=>{
                let { controlElement: N } = param;
                return /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                    paddingX: 8,
                    paddingY: 12,
                    children: /* @__PURE__ */ (0, _jsxruntime.jsxs)(_inline.default, {
                        space: "between",
                        alignItems: "center",
                        children: [
                            /* @__PURE__ */ (0, _jsxruntime.jsxs)("div", {
                                className: _menucssmistica.itemContent,
                                children: [
                                    f && /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                                        className: _menucssmistica.iconContainer,
                                        children: /* @__PURE__ */ (0, _jsxruntime.jsx)(f, {
                                            size: 24,
                                            color: h
                                        })
                                    }),
                                    /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text3, {
                                        regular: !0,
                                        color: h,
                                        children: r
                                    })
                                ]
                            }),
                            /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                                paddingLeft: 16,
                                children: N
                            })
                        ]
                    })
                });
            }
        }) : /* @__PURE__ */ (0, _jsxruntime.jsx)(_touchable.default, {
            ref: c,
            onPress: ()=>{
                k && m !== null && (C(m), i());
            },
            disabled: u,
            role: "menuitem",
            children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                paddingX: 8,
                paddingY: 12,
                children: /* @__PURE__ */ (0, _jsxruntime.jsxs)("div", {
                    className: _menucssmistica.itemContent,
                    children: [
                        f && /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                            className: _menucssmistica.iconContainer,
                            children: /* @__PURE__ */ (0, _jsxruntime.jsx)(f, {
                                size: 24,
                                color: h
                            })
                        }),
                        /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text3, {
                            regular: !0,
                            color: h,
                            children: r
                        })
                    ]
                })
            })
        });
    return /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
        className: (0, _classnames.default)(_menucssmistica.menuItem, {
            [_menucssmistica.menuItemEnabled]: !u,
            [_menucssmistica.menuItemDisabled]: u,
            [_menucssmistica.menuItemHovered]: !u && m !== null && n === m
        }),
        onMouseMove: ()=>v(u ? null : m),
        onMouseLeave: ()=>v(null),
        children: l()
    });
}, Ve = (param)=>{
    let { children: r } = param;
    return r ? /* @__PURE__ */ (0, _jsxruntime.jsxs)(_jsxruntime.Fragment, {
        children: [
            r,
            /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                paddingY: 8,
                className: _menucssmistica.menuSectionDivider,
                children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_divider.default, {})
            })
        ]
    }) : /* @__PURE__ */ (0, _jsxruntime.jsx)(_jsxruntime.Fragment, {});
}, V = 12, J = 8, Je = (param)=>{
    let { renderTarget: r, renderMenu: f, width: d, position: u = "left", dataAttributes: C } = param;
    const [s, p] = _react.useState(!1), [n, v] = _react.useState(null), [i, k] = _react.useState(null), [c, h] = _react.useState(null), [x, M] = _react.useState(!1), m = _react.useRef(null), [l, N] = _react.useState(null), R = (0, _hooks.useWindowSize)();
    _react.useEffect(()=>{
        const a = n == null ? void 0 : n.getBoundingClientRect();
        if (!i || !a || !s) return;
        const { top: t, right: O, left: g, bottom: A } = a, T = i.scrollHeight, I = u === "left" ? g : void 0, F = u === "right" ? R.width - O : void 0, L = t - J, $ = A + J, H = R.height - $ - V, P = L - V, B = H > T, z = P > T, te = B || !z && H > P;
        N(te ? {
            left: I,
            right: F,
            top: `${$}px`,
            bottom: "auto",
            maxHeight: B ? void 0 : H,
            transformOrigin: "center top"
        } : {
            left: I,
            right: F,
            top: "auto",
            bottom: `${R.height - L}px`,
            maxHeight: z ? void 0 : P,
            transformOrigin: "center bottom"
        });
    }, [
        u,
        s,
        i,
        n,
        d,
        R
    ]);
    const Z = _react.useMemo(()=>({
            ref: v,
            onPress: ()=>{
                p(!s);
            }
        }), [
        v,
        s
    ]), ee = {
        ref: (0, _common.combineRefs)(k, m),
        className: _menucssmistica.menuContainer,
        close: ()=>p(!1)
    }, w = _react.useCallback(()=>{
        const t = D(i).findIndex((O)=>!O.getAttribute("aria-disabled"));
        h(t < 0 ? null : t);
    }, [
        i
    ]), _ = _react.useCallback((a)=>{
        var T;
        const t = D(i);
        a && t.reverse();
        const O = c === null ? -1 : a ? t.length - 1 - c : c;
        let g = t.findIndex((I, F)=>!I.getAttribute("aria-disabled") && F > O);
        g === -1 && (g = t.findIndex((I)=>!I.getAttribute("aria-disabled")));
        const A = a && g !== -1 ? t.length - 1 - g : g;
        h(A < 0 ? null : A), (T = t[g]) == null || T.focus();
    }, [
        c,
        i
    ]);
    return _react.useEffect(()=>{
        s ? x && i && (w(), M(!1)) : h(null);
    }, [
        s,
        w,
        x,
        i
    ]), _react.useEffect(()=>{
        const a = (t)=>{
            if (s) switch(t.key){
                case _keys.RIGHT:
                case _keys.DOWN:
                    (0, _dom.cancelEvent)(t), _();
                    break;
                case _keys.LEFT:
                case _keys.UP:
                    (0, _dom.cancelEvent)(t), _(!0);
                    break;
                case _keys.ESC:
                    p(!1);
                    break;
                case _keys.SPACE:
                case _keys.ENTER:
                    (0, _dom.cancelEvent)(t), c !== null && D(i)[c].click();
                    break;
                case _keys.TAB:
                    (0, _dom.cancelEvent)(t);
                    break;
            }
            else switch(t.key){
                case _keys.ENTER:
                case _keys.SPACE:
                    M(!0);
                    break;
                case _keys.DOWN:
                    n === document.activeElement && (M(!0), p(!0));
                    break;
            }
        };
        return document.addEventListener("keydown", a, !1), ()=>{
            document.removeEventListener("keydown", a, !1);
        };
    }), _react.useEffect(()=>{
        n == null || n.setAttribute("aria-haspopup", "menu"), n == null || n.setAttribute("aria-expanded", String(s));
    }, [
        n,
        s
    ]), /* @__PURE__ */ (0, _jsxruntime.jsxs)("div", _object_spread_props(_object_spread({}, (0, _dom.getPrefixedDataAttributes)(C, "Menu")), {
        children: [
            r(_object_spread_props(_object_spread({}, Z), {
                isMenuOpen: s
            })),
            /* @__PURE__ */ (0, _jsxruntime.jsx)(_portal.Portal, {
                children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_reacttransitiongroup.CSSTransition, {
                    in: s,
                    nodeRef: m,
                    timeout: (0, _platform.isRunningAcceptanceTest)() ? 0 : ke,
                    classNames: _menucssmistica.menuTransitionClasses,
                    mountOnEnter: !0,
                    unmountOnExit: !0,
                    onExit: ()=>n == null ? void 0 : n.focus(),
                    children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_overlay.default, {
                        onPress: (a)=>{
                            (0, _dom.cancelEvent)(a), p(!1);
                        },
                        disableScroll: !0,
                        children: /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                            style: _object_spread({}, (0, _css.applyCssVars)(_object_spread({}, l ? _object_spread({
                                [_menucssmistica.vars.top]: l.top,
                                [_menucssmistica.vars.bottom]: l.bottom,
                                [_menucssmistica.vars.transformOrigin]: l.transformOrigin
                            }, l.left !== void 0 && {
                                [_menucssmistica.vars.left]: `${l.left}px`
                            }, l.right !== void 0 && {
                                [_menucssmistica.vars.right]: `${l.right}px`
                            }, l.maxHeight !== void 0 && {
                                [_menucssmistica.vars.maxHeight]: `${l.maxHeight}px`
                            }, d !== void 0 && {
                                [_menucssmistica.vars.width]: d ? `${d}px` : ""
                            }) : {}))),
                            role: "menu",
                            children: /* @__PURE__ */ (0, _jsxruntime.jsx)(Q.Provider, {
                                value: {
                                    isMenuOpen: s,
                                    focusedItem: c,
                                    setFocusedItem: h,
                                    closeMenu: ()=>p(!1)
                                },
                                children: f(ee)
                            })
                        })
                    })
                })
            })
        ]
    }));
};
