// @telefonica/mistica/dist/menu.css-mistica.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    iconContainer: function() {
        return r;
    },
    itemContent: function() {
        return v;
    },
    menuContainer: function() {
        return n;
    },
    menuItem: function() {
        return a;
    },
    menuItemDisabled: function() {
        return t;
    },
    menuItemEnabled: function() {
        return _;
    },
    menuItemHovered: function() {
        return u;
    },
    menuSectionDivider: function() {
        return i;
    },
    menuTransitionClasses: function() {
        return m;
    },
    vars: function() {
        return o;
    }
});
require("./sprinkles.css.ts.vanilla.css-mistica.js");
require("./menu.css.ts.vanilla.css-mistica.js");
var r = "_2k0k3uj", v = "_2k0k3ui", n = "_2k0k3u8 _1y2v1nf7m _1y2v1nf8v _1y2v1nfa4 _1y2v1nfbd _1y2v1nf3q _1y2v1nfkd _1y2v1nfhk _1y2v1nfhq _1y2v1nfhf", a = "_2k0k3ud", t = "_2k0k3ug", _ = "_2k0k3uf", u = "_2k0k3ue", i = "_2k0k3uh", m = {
    enter: "_2k0k3u9",
    enterActive: "_2k0k3ua",
    exit: "_2k0k3ub",
    exitActive: "_2k0k3uc"
}, o = {
    top: "var(--_2k0k3u0)",
    bottom: "var(--_2k0k3u1)",
    left: "var(--_2k0k3u2)",
    right: "var(--_2k0k3u4)",
    width: "var(--_2k0k3u3)",
    maxHeight: "var(--_2k0k3u5)",
    transformOrigin: "var(--_2k0k3u6)"
};
