// @telefonica/mistica/dist/navigation-bar.js
"use client";
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    FunnelNavigationBar: function() {
        return pn;
    },
    MainNavigationBar: function() {
        return mn;
    },
    NavigationBar: function() {
        return un;
    },
    NavigationBarAction: function() {
        return gn;
    },
    NavigationBarActionGroup: function() {
        return hn;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _react = /*#__PURE__*/ _interop_require_wildcard(require("react"));
const _reacttransitiongroup = require("react-transition-group");
const _classnames = /*#__PURE__*/ _interop_require_default(require("classnames"));
const _responsivelayout = /*#__PURE__*/ _interop_require_default(require("./responsive-layout.js"));
const _inline = /*#__PURE__*/ _interop_require_default(require("./inline.js"));
const _touchable = /*#__PURE__*/ _interop_require_wildcard(require("./touchable.js"));
const _text = require("./text.js");
const _hooks = require("./hooks.js");
const _iconmenuregular = /*#__PURE__*/ _interop_require_default(require("./generated/mistica-icons/icon-menu-regular.js"));
const _iconcloseregular = /*#__PURE__*/ _interop_require_default(require("./generated/mistica-icons/icon-close-regular.js"));
const _iconchevronleftregular = /*#__PURE__*/ _interop_require_default(require("./generated/mistica-icons/icon-chevron-left-regular.js"));
const _iconbutton = require("./icon-button.js");
const _negativebox = /*#__PURE__*/ _interop_require_default(require("./negative-box.js"));
const _list = require("./list.js");
const _themevariantcontext = require("./theme-variant-context.js");
const _focustrap = /*#__PURE__*/ _interop_require_default(require("./focus-trap.js"));
const _portal = require("./portal.js");
const _gridlayout = /*#__PURE__*/ _interop_require_default(require("./grid-layout.js"));
const _modalcontextprovider = require("./modal-context-provider.js");
const _logo = require("./logo.js");
const _skincontractcssmistica = require("./skins/skin-contract.css-mistica.js");
const _navigationbarcssmistica = require("./navigation-bar.css-mistica.js");
const _sprinklescssmistica = require("./sprinkles.css-mistica.js");
const _dom = require("./utils/dom.js");
const _stack = /*#__PURE__*/ _interop_require_default(require("./stack.js"));
const _box = /*#__PURE__*/ _interop_require_default(require("./box.js"));
const _platform = require("./utils/platform.js");
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
function _getRequireWildcardCache(nodeInterop) {
    if (typeof WeakMap !== "function") return null;
    var cacheBabelInterop = new WeakMap();
    var cacheNodeInterop = new WeakMap();
    return (_getRequireWildcardCache = function(nodeInterop) {
        return nodeInterop ? cacheNodeInterop : cacheBabelInterop;
    })(nodeInterop);
}
function _interop_require_wildcard(obj, nodeInterop) {
    if (!nodeInterop && obj && obj.__esModule) {
        return obj;
    }
    if (obj === null || typeof obj !== "object" && typeof obj !== "function") {
        return {
            default: obj
        };
    }
    var cache = _getRequireWildcardCache(nodeInterop);
    if (cache && cache.has(obj)) {
        return cache.get(obj);
    }
    var newObj = {
        __proto__: null
    };
    var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for(var key in obj){
        if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) {
            var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null;
            if (desc && (desc.get || desc.set)) {
                Object.defineProperty(newObj, key, desc);
            } else {
                newObj[key] = obj[key];
            }
        }
    }
    newObj.default = obj;
    if (cache) {
        cache.set(obj, newObj);
    }
    return newObj;
}
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
function _object_without_properties(source, excluded) {
    if (source == null) return {};
    var target = _object_without_properties_loose(source, excluded);
    var key, i;
    if (Object.getOwnPropertySymbols) {
        var sourceSymbolKeys = Object.getOwnPropertySymbols(source);
        for(i = 0; i < sourceSymbolKeys.length; i++){
            key = sourceSymbolKeys[i];
            if (excluded.indexOf(key) >= 0) continue;
            if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
            target[key] = source[key];
        }
    }
    return target;
}
function _object_without_properties_loose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;
    for(i = 0; i < sourceKeys.length; i++){
        key = sourceKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        target[key] = source[key];
    }
    return target;
}
const we = (param)=>{
    let { isOpen: n } = param;
    return /* @__PURE__ */ (0, _jsxruntime.jsxs)("div", {
        className: _navigationbarcssmistica.burgerIconContainer,
        role: "presentation",
        "data-component-name": "BurgerMenuIcon",
        children: [
            /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                className: n ? _navigationbarcssmistica.iconCloseOpen : _navigationbarcssmistica.iconCloseHidden,
                children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_iconcloseregular.default, {})
            }),
            /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                className: n ? _navigationbarcssmistica.iconMenuHidden : _navigationbarcssmistica.iconMenuOpen,
                children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_iconmenuregular.default, {})
            })
        ]
    });
}, De = 300, A = (param)=>{
    let { children: n, topFixed: a, withBorder: t, isMenuOpen: r, isInverse: o, dataAttributes: s } = param;
    const { isDarkMode: c } = (0, _hooks.useTheme)();
    return /* @__PURE__ */ (0, _jsxruntime.jsx)("header", _object_spread_props(_object_spread({
        className: (0, _classnames.default)((()=>o && !c || !t ? _navigationbarcssmistica.navbarBorderColorVariants.noBorder : r ? _navigationbarcssmistica.navbarBorderColorVariants.menuOpen : _navigationbarcssmistica.navbarBorderColorVariants.default)(), {
            [_navigationbarcssmistica.topFixed]: a
        }),
        style: {
            background: o ? _skincontractcssmistica.vars.colors.navigationBarBackground : _skincontractcssmistica.vars.colors.background
        }
    }, (0, _dom.getPrefixedDataAttributes)(s)), {
        children: n
    }));
}, v = (param)=>{
    let { right: n, children: a, desktopOnly: t } = param;
    return /* @__PURE__ */ (0, _jsxruntime.jsxs)("div", {
        className: (0, _classnames.default)(_navigationbarcssmistica.navigationBarContent, {
            [_navigationbarcssmistica.desktopOnly]: t
        }),
        children: [
            a,
            n && /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                className: _navigationbarcssmistica.navigationBarContentRight,
                children: n
            })
        ]
    });
}, mn = (param)=>{
    let { sections: n = [], selectedIndex: a, right: t, isInverse: r = !1, topFixed: o = !0, withBorder: s = !0, burgerMenuExtra: c, logo: l, large: p = !1 } = param;
    const { texts: m, isDarkMode: q } = (0, _hooks.useTheme)(), [d, I] = _react.useState(!1), [O, h] = _react.useState("closed"), R = (0, _hooks.useAriaId)(), J = q ? 1 : 0.2, { isTabletOrSmaller: K } = (0, _hooks.useScreenSize)(), y = (0, _modalcontextprovider.useSetModalState)();
    l = l !== null && l !== void 0 ? l : /* @__PURE__ */ (0, _jsxruntime.jsx)(_logo.Logo, {
        size: {
            mobile: 40,
            desktop: 48
        }
    });
    const E = ()=>/* @__PURE__ */ (0, _jsxruntime.jsx)("nav", {
            className: _navigationbarcssmistica.desktopOnly,
            children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_inline.default, {
                space: 32,
                children: n.map((_param, g)=>{
                    var { title: B } = _param, M = _object_without_properties(_param, [
                        "title"
                    ]);
                    return /* @__PURE__ */ /*#__PURE__*/ (0, _react.createElement)(_touchable.BaseTouchable, _object_spread_props(_object_spread({}, M), {
                        key: g,
                        className: (0, _classnames.default)(_navigationbarcssmistica.section, {
                            [_navigationbarcssmistica.selectedSectionVariantes[r ? "inverse" : "default"]]: g === a
                        }, _navigationbarcssmistica.textWrapperVariants[r ? "inverse" : "default"])
                    }), /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text3, {
                        regular: !0,
                        color: "inherit",
                        children: B
                    }));
                })
            })
        }), b = p && n.length > 0, Q = ()=>{
        I(!0), y({
            isModalOpen: !0
        });
    }, L = ()=>{
        I(!1), y({
            isModalOpen: !1
        });
    }, w = O !== "open", D = n.length > 1, V = /* @__PURE__ */ (0, _jsxruntime.jsxs)(_themevariantcontext.ThemeVariant, {
        isInverse: r,
        children: [
            /* @__PURE__ */ (0, _jsxruntime.jsx)(A, {
                topFixed: o,
                withBorder: s,
                isMenuOpen: d,
                isInverse: r,
                dataAttributes: {
                    "component-name": "MainNavigationBar"
                },
                children: /* @__PURE__ */ (0, _jsxruntime.jsxs)(_responsivelayout.default, {
                    children: [
                        /* @__PURE__ */ (0, _jsxruntime.jsx)(v, {
                            right: t,
                            children: /* @__PURE__ */ (0, _jsxruntime.jsxs)("div", {
                                className: _navigationbarcssmistica.mainNavbarContent,
                                children: [
                                    D && /* @__PURE__ */ (0, _jsxruntime.jsx)(_touchable.default, {
                                        className: _navigationbarcssmistica.burgerMenuButton,
                                        "aria-live": "polite",
                                        "aria-label": d ? m.closeNavigationMenu : m.openNavigationMenu,
                                        "aria-expanded": d,
                                        "aria-controls": R,
                                        onPress: d ? L : Q,
                                        children: /* @__PURE__ */ (0, _jsxruntime.jsx)(we, {
                                            isOpen: d
                                        })
                                    }),
                                    /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                                        className: _navigationbarcssmistica.logoContainer,
                                        children: l
                                    }),
                                    !b && E()
                                ]
                            })
                        }),
                        b && /* @__PURE__ */ (0, _jsxruntime.jsx)(v, {
                            desktopOnly: !0,
                            children: E()
                        })
                    ]
                })
            }),
            o && /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                className: b ? _navigationbarcssmistica.spacerLarge : _navigationbarcssmistica.spacer
            })
        ]
    });
    return K ? /* @__PURE__ */ (0, _jsxruntime.jsxs)(_jsxruntime.Fragment, {
        children: [
            /* @__PURE__ */ (0, _jsxruntime.jsx)(_focustrap.default, {
                disabled: w,
                group: "burger-menu-lock",
                children: V
            }),
            D && /* @__PURE__ */ (0, _jsxruntime.jsx)(_portal.Portal, {
                children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_focustrap.default, {
                    disabled: w,
                    group: "burger-menu-lock",
                    children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_reacttransitiongroup.Transition, {
                        onEntering: ()=>{
                            h("opening");
                        },
                        onEntered: ()=>{
                            h("open");
                        },
                        onExiting: ()=>{
                            h("closing");
                        },
                        onExited: ()=>{
                            h("closed");
                        },
                        in: d,
                        timeout: (0, _platform.isRunningAcceptanceTest)() ? 0 : De,
                        unmountOnExit: !0,
                        children: (B)=>/* @__PURE__ */ (0, _jsxruntime.jsx)(_jsxruntime.Fragment, {
                                children: /* @__PURE__ */ (0, _jsxruntime.jsx)("nav", {
                                    className: (0, _classnames.default)(_navigationbarcssmistica.burgerMenu, _navigationbarcssmistica.burgerMenuTransition[B]),
                                    style: {
                                        boxShadow: O !== "closed" ? `6px 0 4px -4px rgba(0, 0, 0, ${J})` : "none"
                                    },
                                    id: R,
                                    onClick: ()=>{
                                        L();
                                    },
                                    children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_responsivelayout.default, {
                                        children: /* @__PURE__ */ (0, _jsxruntime.jsxs)(_stack.default, {
                                            space: 16,
                                            children: [
                                                /* @__PURE__ */ (0, _jsxruntime.jsx)(_negativebox.default, {
                                                    children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_list.RowList, {
                                                        children: n.map((M, g)=>/* @__PURE__ */ (0, _jsxruntime.jsx)(_list.Row, _object_spread({}, M), g))
                                                    })
                                                }),
                                                c && /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                                                    paddingBottom: 16,
                                                    children: c
                                                })
                                            ]
                                        })
                                    })
                                })
                            })
                    })
                })
            })
        ]
    }) : V;
}, un = (param)=>{
    let { onBack: n, title: a, titleAs: t, right: r, isInverse: o = !1, topFixed: s = !0, paddingX: c = 0, withBorder: l = !0 } = param;
    const { texts: p } = (0, _hooks.useTheme)(), m = /* @__PURE__ */ (0, _jsxruntime.jsx)(v, {
        right: r,
        children: /* @__PURE__ */ (0, _jsxruntime.jsxs)(_inline.default, {
            space: 24,
            alignItems: "center",
            children: [
                n && /* @__PURE__ */ (0, _jsxruntime.jsx)(_iconbutton.IconButton, {
                    "aria-label": p.backNavigationBar,
                    onPress: n,
                    Icon: _iconchevronleftregular.default,
                    bleedLeft: !0,
                    bleedRight: !0
                }),
                /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text3, {
                    regular: !0,
                    truncate: !0,
                    as: t,
                    children: a
                })
            ]
        })
    });
    return /* @__PURE__ */ (0, _jsxruntime.jsxs)(_themevariantcontext.ThemeVariant, {
        isInverse: o,
        children: [
            /* @__PURE__ */ (0, _jsxruntime.jsx)(A, {
                topFixed: s,
                withBorder: l,
                isInverse: o,
                dataAttributes: {
                    "component-name": "NavigationBar"
                },
                children: s ? /* @__PURE__ */ (0, _jsxruntime.jsx)(_responsivelayout.default, {
                    children: m
                }) : /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                    className: (0, _sprinklescssmistica.sprinkles)({
                        width: "100%"
                    }),
                    style: {
                        padding: `0 ${c}px`
                    },
                    children: m
                })
            }),
            s && /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                className: _navigationbarcssmistica.spacer
            })
        ]
    });
}, pn = (param)=>{
    let { logo: n, right: a, isInverse: t = !1, topFixed: r = !0, withBorder: o = !0 } = param;
    return n = n !== null && n !== void 0 ? n : /* @__PURE__ */ (0, _jsxruntime.jsx)(_logo.Logo, {
        size: {
            mobile: 40,
            desktop: 48
        }
    }), /* @__PURE__ */ (0, _jsxruntime.jsxs)(_themevariantcontext.ThemeVariant, {
        isInverse: t,
        children: [
            /* @__PURE__ */ (0, _jsxruntime.jsx)(A, {
                topFixed: r,
                withBorder: o,
                isInverse: t,
                dataAttributes: {
                    "component-name": "FunnelNavigationBar"
                },
                children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_responsivelayout.default, {
                    children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_gridlayout.default, {
                        template: "10",
                        children: /* @__PURE__ */ (0, _jsxruntime.jsx)(v, {
                            right: a,
                            children: n
                        })
                    })
                })
            }),
            r && /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                className: _navigationbarcssmistica.spacer
            })
        ]
    });
}, hn = (param)=>{
    let { children: n } = param;
    return /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
        className: _navigationbarcssmistica.lineHeightFix,
        "data-component-name": "NavigationBarActionGroup",
        children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_inline.default, {
            space: 24,
            alignItems: "center",
            children: n
        })
    });
}, gn = (_param)=>{
    var { children: n } = _param, a = _object_without_properties(_param, [
        "children"
    ]);
    const t = (0, _themevariantcontext.useIsInverseVariant)();
    return /* @__PURE__ */ (0, _jsxruntime.jsx)(_touchable.BaseTouchable, _object_spread_props(_object_spread({}, a), {
        className: (0, _classnames.default)((0, _sprinklescssmistica.sprinkles)({
            border: "none",
            background: "transparent",
            padding: 0
        }), _navigationbarcssmistica.lineHeightFix, _navigationbarcssmistica.textWrapperVariants[t ? "inverse" : "default"]),
        dataAttributes: {
            "component-name": "NavigationBarAction"
        },
        children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_inline.default, {
            space: 16,
            alignItems: "center",
            children: _react.Children.map(n, (r)=>typeof r == "string" ? /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text2, {
                    regular: !0,
                    color: "inherit",
                    children: r
                }) : r)
        })
    }));
};
