// @telefonica/mistica/dist/navigation-breadcrumbs.js
"use client";
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "default", {
    enumerable: true,
    get: function() {
        return D;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _text = require("./text.js");
const _dom = require("./utils/dom.js");
const _textlink = /*#__PURE__*/ _interop_require_default(require("./text-link.js"));
const _themevariantcontext = require("./theme-variant-context.js");
const _navigationbreadcrumbscssmistica = require("./navigation-breadcrumbs.css-mistica.js");
const _skincontractcssmistica = require("./skins/skin-contract.css-mistica.js");
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
const N = " / ", b = (param)=>{
    let { title: s, breadcrumbs: n, dataAttributes: c, "aria-label": m = "Breadcrumb" } = param;
    const i = (0, _themevariantcontext.useIsInverseVariant)();
    return /* @__PURE__ */ (0, _jsxruntime.jsx)("nav", _object_spread_props(_object_spread({
        "aria-label": m
    }, (0, _dom.getPrefixedDataAttributes)(c, "NavigationBreadcrumbs")), {
        children: /* @__PURE__ */ (0, _jsxruntime.jsxs)("ol", {
            className: _navigationbreadcrumbscssmistica.list,
            children: [
                n.map((param, u)=>{
                    let { title: a, url: d } = param;
                    return /* @__PURE__ */ (0, _jsxruntime.jsxs)("li", {
                        className: _navigationbreadcrumbscssmistica.listItem,
                        children: [
                            /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text1, {
                                regular: !0,
                                children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_textlink.default, {
                                    to: d,
                                    style: {
                                        color: i ? _skincontractcssmistica.vars.colors.textPrimaryInverse : _skincontractcssmistica.vars.colors.textPrimary
                                    },
                                    className: _navigationbreadcrumbscssmistica.link,
                                    children: a
                                })
                            }),
                            /* @__PURE__ */ (0, _jsxruntime.jsx)("span", {
                                role: "presentation",
                                children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text1, {
                                    regular: !0,
                                    children: N
                                })
                            })
                        ]
                    }, u);
                }),
                /* @__PURE__ */ (0, _jsxruntime.jsx)("li", {
                    className: _navigationbreadcrumbscssmistica.listItem,
                    children: /* @__PURE__ */ (0, _jsxruntime.jsx)("a", {
                        "aria-current": "page",
                        href: "#",
                        className: _navigationbreadcrumbscssmistica.current,
                        onClick: (a)=>{
                            a.preventDefault();
                        },
                        children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text1, {
                            regular: !0,
                            color: i ? _skincontractcssmistica.vars.colors.textSecondaryInverse : _skincontractcssmistica.vars.colors.textSecondary,
                            children: s
                        })
                    })
                })
            ]
        })
    }));
}, D = b;
