// @telefonica/mistica/dist/overscroll-color-context.js
"use client";
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    OverscrollColorProvider: function() {
        return P;
    },
    default: function() {
        return L;
    },
    useOverScrollColor: function() {
        return M;
    },
    useSetOverscrollColor: function() {
        return j;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _react = /*#__PURE__*/ _interop_require_wildcard(require("react"));
const _themevariantcontext = require("./theme-variant-context.js");
const _hooks = require("./hooks.js");
const _nestablecontext = require("./nestable-context.js");
const _platform = require("./utils/platform.js");
const _skincontractcssmistica = require("./skins/skin-contract.css-mistica.js");
const _dom = require("./utils/dom.js");
function _getRequireWildcardCache(nodeInterop) {
    if (typeof WeakMap !== "function") return null;
    var cacheBabelInterop = new WeakMap();
    var cacheNodeInterop = new WeakMap();
    return (_getRequireWildcardCache = function(nodeInterop) {
        return nodeInterop ? cacheNodeInterop : cacheBabelInterop;
    })(nodeInterop);
}
function _interop_require_wildcard(obj, nodeInterop) {
    if (!nodeInterop && obj && obj.__esModule) {
        return obj;
    }
    if (obj === null || typeof obj !== "object" && typeof obj !== "function") {
        return {
            default: obj
        };
    }
    var cache = _getRequireWildcardCache(nodeInterop);
    if (cache && cache.has(obj)) {
        return cache.get(obj);
    }
    var newObj = {
        __proto__: null
    };
    var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for(var key in obj){
        if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) {
            var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null;
            if (desc && (desc.get || desc.set)) {
                Object.defineProperty(newObj, key, desc);
            } else {
                newObj[key] = obj[key];
            }
        }
    }
    newObj.default = obj;
    if (cache) {
        cache.set(obj, newObj);
    }
    return newObj;
}
const { Provider: b, useSetValue: d, useValue: u } = (0, _nestablecontext.createNestableContext)({}), M = u, i = (o)=>(0, _platform.getPlatform)(o) === "ios", O = ()=>{
    const { topColor: o } = u(), { isDarkMode: e } = (0, _hooks.useTheme)();
    return _react.useEffect(()=>{
        if (!o) return;
        const t = document.head.querySelectorAll("meta[name=theme-color]");
        t.forEach((n)=>{
            n.remove();
        });
        const r = document.createElement("meta");
        return r.name = "theme-color", r.media = e ? "(prefers-color-scheme: dark)" : "(prefers-color-scheme: light)", r.content = (0, _dom.isCssVar)(o) ? (0, _dom.getCssVarValue)(o) : o, document.head.appendChild(r), ()=>{
            r.remove(), t.forEach((n)=>{
                document.head.appendChild(n);
            });
        };
    }, [
        e,
        o
    ]), null;
}, y = ()=>{
    const [o, e] = _react.useState("transparent");
    return _react.useEffect(()=>{
        const t = window.matchMedia("(prefers-color-scheme: dark)");
        let r;
        const n = ()=>{
            r = requestAnimationFrame(()=>{
                const f = getComputedStyle(document.body).backgroundColor;
                e(f);
            });
        };
        return t.addListener(n), n(), ()=>{
            t.removeListener(n), cancelAnimationFrame(r);
        };
    }, []), o;
}, B = ()=>{
    const { topColor: o, bottomColor: e } = u(), t = y();
    if (!e && !o) return null;
    const r = e !== null && e !== void 0 ? e : t;
    return r === o || r === "transparent" ? null : /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
        style: {
            position: "fixed",
            zIndex: -1,
            background: r,
            width: "100%",
            height: 300,
            left: 0,
            right: 0,
            bottom: 0
        }
    });
}, P = (param)=>{
    let { children: o } = param;
    const { platformOverrides: e } = (0, _hooks.useTheme)();
    return i(e) ? /* @__PURE__ */ (0, _jsxruntime.jsxs)(b, {
        children: [
            /* @__PURE__ */ (0, _jsxruntime.jsx)(O, {}),
            o,
            /* @__PURE__ */ (0, _jsxruntime.jsx)(B, {})
        ]
    }) : /* @__PURE__ */ (0, _jsxruntime.jsx)(_jsxruntime.Fragment, {
        children: o
    });
}, x = (o)=>{
    switch(o){
        case "default":
            return _skincontractcssmistica.vars.colors.background;
        case "alternative":
            return _skincontractcssmistica.vars.colors.backgroundAlternative;
        case "inverse":
            return _skincontractcssmistica.vars.colors.backgroundBrand;
        default:
            const e = o;
            throw new Error(`Invalid variant: ${e}`);
    }
}, w = ()=>{
    const o = (0, _themevariantcontext.useThemeVariant)();
    return d({
        topColor: x(o)
    }), null;
}, E = ()=>i((0, _hooks.useTheme)().platformOverrides) ? /* @__PURE__ */ (0, _jsxruntime.jsx)(w, {}) : /* @__PURE__ */ (0, _jsxruntime.jsx)(_jsxruntime.Fragment, {}), j = (o)=>{
    d(o);
}, L = E;
