// @telefonica/mistica/dist/password-field.js
"use client";
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "default", {
    enumerable: true,
    get: function() {
        return D;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _react = /*#__PURE__*/ _interop_require_wildcard(require("react"));
const _formcontext = require("./form-context.js");
const _textfieldbase = require("./text-field-base.js");
const _hooks = require("./hooks.js");
const _iconeyeoffregular = /*#__PURE__*/ _interop_require_default(require("./generated/mistica-icons/icon-eye-off-regular.js"));
const _iconeyeregular = /*#__PURE__*/ _interop_require_default(require("./generated/mistica-icons/icon-eye-regular.js"));
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
function _getRequireWildcardCache(nodeInterop) {
    if (typeof WeakMap !== "function") return null;
    var cacheBabelInterop = new WeakMap();
    var cacheNodeInterop = new WeakMap();
    return (_getRequireWildcardCache = function(nodeInterop) {
        return nodeInterop ? cacheNodeInterop : cacheBabelInterop;
    })(nodeInterop);
}
function _interop_require_wildcard(obj, nodeInterop) {
    if (!nodeInterop && obj && obj.__esModule) {
        return obj;
    }
    if (obj === null || typeof obj !== "object" && typeof obj !== "function") {
        return {
            default: obj
        };
    }
    var cache = _getRequireWildcardCache(nodeInterop);
    if (cache && cache.has(obj)) {
        return cache.get(obj);
    }
    var newObj = {
        __proto__: null
    };
    var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for(var key in obj){
        if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) {
            var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null;
            if (desc && (desc.get || desc.set)) {
                Object.defineProperty(newObj, key, desc);
            } else {
                newObj[key] = obj[key];
            }
        }
    }
    newObj.default = obj;
    if (cache) {
        cache.set(obj, newObj);
    }
    return newObj;
}
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
function _object_without_properties(source, excluded) {
    if (source == null) return {};
    var target = _object_without_properties_loose(source, excluded);
    var key, i;
    if (Object.getOwnPropertySymbols) {
        var sourceSymbolKeys = Object.getOwnPropertySymbols(source);
        for(i = 0; i < sourceSymbolKeys.length; i++){
            key = sourceSymbolKeys[i];
            if (excluded.indexOf(key) >= 0) continue;
            if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
            target[key] = source[key];
        }
    }
    return target;
}
function _object_without_properties_loose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;
    for(i = 0; i < sourceKeys.length; i++){
        key = sourceKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        target[key] = source[key];
    }
    return target;
}
const j = (param)=>{
    let { isVisible: n, setVisibility: i, focus: c } = param;
    const { texts: t } = (0, _hooks.useTheme)();
    return /* @__PURE__ */ (0, _jsxruntime.jsx)(_textfieldbase.FieldEndIcon, {
        checkedProps: {
            Icon: _iconeyeoffregular.default,
            "aria-label": t.disablePasswordVisibility
        },
        uncheckedProps: {
            Icon: _iconeyeregular.default,
            "aria-label": t.enablePasswordVisibility
        },
        checked: n,
        onChange: (a)=>{
            i(a), c();
        }
    });
}, q = (_param)=>{
    var { disabled: n, error: i, helperText: c, name: t, optional: a, validate: f, onChange: m, onChangeValue: p, onBlur: P, value: b, autoComplete: R = "current-password", defaultValue: w, dataAttributes: F } = _param, I = _object_without_properties(_param, [
        "disabled",
        "error",
        "helperText",
        "name",
        "optional",
        "validate",
        "onChange",
        "onChangeValue",
        "onBlur",
        "value",
        "autoComplete",
        "defaultValue",
        "dataAttributes"
    ]);
    const [u, y] = _react.useState(!1), s = _react.useRef(0), o = _react.useRef(null), V = (e)=>e, g = ()=>{
        const e = o.current;
        e && (e.selectionStart !== null && (s.current = e.selectionStart), e.focus());
    };
    _react.useEffect(()=>{
        const e = o.current;
        if (e) {
            const h = requestAnimationFrame(()=>{
                e.selectionStart = s.current, e.selectionEnd = s.current;
            });
            return ()=>{
                cancelAnimationFrame(h);
            };
        }
    }, [
        u,
        s,
        o
    ]);
    const d = (0, _formcontext.useFieldProps)({
        name: t,
        value: b,
        defaultValue: w,
        processValue: V,
        helperText: c,
        optional: a,
        error: i,
        disabled: n,
        onBlur: P,
        validate: f,
        onChange: m,
        onChangeValue: p
    });
    return /* @__PURE__ */ (0, _jsxruntime.jsx)(_textfieldbase.TextFieldBaseAutosuggest, _object_spread_props(_object_spread({}, I, d), {
        type: u ? "text" : "password",
        inputRef: (e)=>{
            d.inputRef(e), o.current = e;
        },
        autoComplete: R,
        endIcon: /* @__PURE__ */ (0, _jsxruntime.jsx)(j, {
            focus: g,
            isVisible: u,
            setVisibility: y
        }),
        dataAttributes: _object_spread({
            "component-name": "PasswordField"
        }, F)
    }));
}, D = q;
