// @telefonica/mistica/dist/progress-bar.js
"use client";
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    ProgressBar: function() {
        return R;
    },
    ProgressBarStepped: function() {
        return T;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _react = /*#__PURE__*/ _interop_require_wildcard(require("react"));
const _hooks = require("./hooks.js");
const _skincontractcssmistica = require("./skins/skin-contract.css-mistica.js");
const _progressbarcssmistica = require("./progress-bar.css-mistica.js");
const _dom = require("./utils/dom.js");
const _classnames = /*#__PURE__*/ _interop_require_default(require("classnames"));
const _inline = /*#__PURE__*/ _interop_require_default(require("./inline.js"));
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
function _getRequireWildcardCache(nodeInterop) {
    if (typeof WeakMap !== "function") return null;
    var cacheBabelInterop = new WeakMap();
    var cacheNodeInterop = new WeakMap();
    return (_getRequireWildcardCache = function(nodeInterop) {
        return nodeInterop ? cacheNodeInterop : cacheBabelInterop;
    })(nodeInterop);
}
function _interop_require_wildcard(obj, nodeInterop) {
    if (!nodeInterop && obj && obj.__esModule) {
        return obj;
    }
    if (obj === null || typeof obj !== "object" && typeof obj !== "function") {
        return {
            default: obj
        };
    }
    var cache = _getRequireWildcardCache(nodeInterop);
    if (cache && cache.has(obj)) {
        return cache.get(obj);
    }
    var newObj = {
        __proto__: null
    };
    var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for(var key in obj){
        if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) {
            var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null;
            if (desc && (desc.get || desc.set)) {
                Object.defineProperty(newObj, key, desc);
            } else {
                newObj[key] = obj[key];
            }
        }
    }
    newObj.default = obj;
    if (cache) {
        cache.set(obj, newObj);
    }
    return newObj;
}
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
const R = (param)=>{
    let { progressPercent: e, color: s, "aria-label": p, "aria-labelledby": n, "aria-hidden": t, dataAttributes: c, reverse: l = !1 } = param;
    const { texts: m } = (0, _hooks.useTheme)(), a = Math.max(0, Math.min(100, e)), i = t && t !== "false" ? {
        "aria-hidden": t
    } : {
        role: "progressbar",
        "aria-valuenow": a,
        "aria-valuemin": 0,
        "aria-valuemax": 100,
        "aria-label": n ? void 0 : (()=>`${p || m.loading}, ${a}% ${m.progressBarCompletedLabel}`)(),
        "aria-labelledby": n
    };
    return /* @__PURE__ */ (0, _jsxruntime.jsx)("div", _object_spread_props(_object_spread(_object_spread_props(_object_spread({}, (0, _dom.getPrefixedDataAttributes)(c, "ProgressBar")), {
        className: _progressbarcssmistica.barBackground
    }), i), {
        children: /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
            className: (0, _classnames.default)(_progressbarcssmistica.bar, l ? _progressbarcssmistica.inverse : _progressbarcssmistica.normal),
            style: {
                maxWidth: `${a}%`,
                backgroundColor: s !== null && s !== void 0 ? s : _skincontractcssmistica.vars.colors.controlActivated
            }
        })
    }));
}, T = (param)=>{
    let { steps: e, currentStep: s = 0, color: p, dataAttributes: n, "aria-label": t, "aria-labelledby": c, "aria-hidden": l } = param;
    const { texts: m } = (0, _hooks.useTheme)(), [a, g] = _react.useState(Math.ceil(s)), [i, k] = _react.useState(!1);
    _react.useEffect(()=>{
        const r = Math.ceil(s);
        a !== r && (k(r < a), g(r));
    }, [
        s,
        e,
        a
    ]);
    const A = l && l !== "false" ? {
        "aria-hidden": l
    } : {
        role: "progressbar",
        "aria-valuenow": a,
        "aria-valuemin": 0,
        "aria-valuemax": e,
        "aria-label": c ? void 0 : (()=>{
            const r = m.progressBarStepLabel.replace("1$s", String(a)).replace("2$s", String(e));
            return t ? `${t}, ${r.toLowerCase()}` : r;
        })(),
        "aria-labelledby": c
    };
    return /* @__PURE__ */ (0, _jsxruntime.jsx)("div", _object_spread_props(_object_spread(_object_spread_props(_object_spread({}, (0, _dom.getPrefixedDataAttributes)(n, "ProgressBarStepped")), {
        role: "progressbar"
    }), A), {
        className: _progressbarcssmistica.progressBarSteppedContainer,
        children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_inline.default, {
            space: 8,
            fullWidth: !0,
            children: Array.from({
                length: e
            }, (r, d)=>{
                const u = d === a, h = d < a, f = d === a - 1;
                return /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                    className: _progressbarcssmistica.barBackground,
                    "aria-hidden": "true",
                    children: (h || u) && /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                        className: (0, _classnames.default)(_progressbarcssmistica.bar, {
                            [_progressbarcssmistica.normal]: f && !i,
                            [_progressbarcssmistica.inverse]: u && i
                        }),
                        style: {
                            backgroundColor: p !== null && p !== void 0 ? p : _skincontractcssmistica.vars.colors.controlActivated,
                            maxWidth: h || f && !i ? "100%" : "0"
                        }
                    })
                }, d);
            })
        })
    }));
};
