// @telefonica/mistica/dist/radio-button.js
"use client";
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    RadioGroup: function() {
        return re;
    },
    default: function() {
        return le;
    },
    useRadioContext: function() {
        return J;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _react = /*#__PURE__*/ _interop_require_wildcard(require("react"));
const _keys = require("./utils/keys.js");
const _formcontext = require("./form-context.js");
const _common = require("./utils/common.js");
const _text = require("./text.js");
const _inline = /*#__PURE__*/ _interop_require_default(require("./inline.js"));
const _classnames = /*#__PURE__*/ _interop_require_default(require("classnames"));
const _hooks = require("./hooks.js");
const _dom = require("./utils/dom.js");
const _radiobuttoncssmistica = require("./radio-button.css-mistica.js");
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
function _getRequireWildcardCache(nodeInterop) {
    if (typeof WeakMap !== "function") return null;
    var cacheBabelInterop = new WeakMap();
    var cacheNodeInterop = new WeakMap();
    return (_getRequireWildcardCache = function(nodeInterop) {
        return nodeInterop ? cacheNodeInterop : cacheBabelInterop;
    })(nodeInterop);
}
function _interop_require_wildcard(obj, nodeInterop) {
    if (!nodeInterop && obj && obj.__esModule) {
        return obj;
    }
    if (obj === null || typeof obj !== "object" && typeof obj !== "function") {
        return {
            default: obj
        };
    }
    var cache = _getRequireWildcardCache(nodeInterop);
    if (cache && cache.has(obj)) {
        return cache.get(obj);
    }
    var newObj = {
        __proto__: null
    };
    var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for(var key in obj){
        if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) {
            var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null;
            if (desc && (desc.get || desc.set)) {
                Object.defineProperty(newObj, key, desc);
            } else {
                newObj[key] = obj[key];
            }
        }
    }
    newObj.default = obj;
    if (cache) {
        cache.set(obj, newObj);
    }
    return newObj;
}
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
function _object_without_properties(source, excluded) {
    if (source == null) return {};
    var target = _object_without_properties_loose(source, excluded);
    var key, i;
    if (Object.getOwnPropertySymbols) {
        var sourceSymbolKeys = Object.getOwnPropertySymbols(source);
        for(i = 0; i < sourceSymbolKeys.length; i++){
            key = sourceSymbolKeys[i];
            if (excluded.indexOf(key) >= 0) continue;
            if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
            target[key] = source[key];
        }
    }
    return target;
}
function _object_without_properties_loose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;
    for(i = 0; i < sourceKeys.length; i++){
        key = sourceKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        target[key] = source[key];
    }
    return target;
}
const N = /*#__PURE__*/ _react.createContext({
    disabled: !1,
    selectedValue: null,
    focusableValue: null,
    select: ()=>{},
    selectNext: ()=>{},
    selectPrev: ()=>{}
}), J = ()=>_react.useContext(N), L = (_param)=>{
    var { value: t, id: s, dataAttributes: h, "aria-labelledby": v, "aria-label": x } = _param, u = _object_without_properties(_param, [
        "value",
        "id",
        "dataAttributes",
        "aria-labelledby",
        "aria-label"
    ]);
    const { disabled: l, selectedValue: C, focusableValue: p, select: c, selectNext: R, selectPrev: g } = J(), a = (0, _hooks.useAriaId)(v), y = _react.useRef(null), d = t === C, V = p === t ? 0 : -1, { isIos: e } = (0, _hooks.useTheme)(), r = (o)=>{
        switch(o.key){
            case _keys.SPACE:
                c(t), o.preventDefault(), o.stopPropagation();
                break;
            case _keys.RIGHT:
            case _keys.DOWN:
                R(), o.preventDefault(), o.stopPropagation();
                break;
            case _keys.LEFT:
            case _keys.UP:
                g(), o.preventDefault(), o.stopPropagation();
                break;
        }
    }, f = /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
        className: (0, _classnames.default)(e ? _radiobuttoncssmistica.outerCircleVariants.ios : _radiobuttoncssmistica.outerCircleVariants.default, {
            [_radiobuttoncssmistica.outerCircleCheckedVariants[e ? "ios" : "default"]]: d,
            [_radiobuttoncssmistica.disabled]: l
        }),
        children: !e && /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
            className: (0, _classnames.default)(_radiobuttoncssmistica.innerCircle, {
                [_radiobuttoncssmistica.innerCircleChecked]: d
            })
        })
    });
    return /* @__PURE__ */ (0, _jsxruntime.jsx)("span", _object_spread_props(_object_spread({
        ref: y,
        id: s,
        tabIndex: l ? void 0 : V,
        role: "radio",
        "data-value": t,
        "aria-checked": d,
        "aria-disabled": l,
        "aria-label": x,
        "aria-labelledby": x ? void 0 : a,
        onClick: (o)=>{
            o.stopPropagation(), l || c(t);
        },
        onKeyDown: l ? void 0 : r,
        className: l ? _radiobuttoncssmistica.radioButtonContainerDisabled : _radiobuttoncssmistica.radioButton
    }, (0, _dom.getPrefixedDataAttributes)(h, "RadioButton")), {
        children: u.render ? u.render({
            controlElement: f,
            disabled: !!l,
            checked: d,
            labelId: a
        }) : /* @__PURE__ */ (0, _jsxruntime.jsxs)(_inline.default, {
            space: 16,
            children: [
                /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text3, {
                    regular: !0,
                    as: "div",
                    children: /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                        style: {
                            position: "relative",
                            top: -2
                        },
                        children: f
                    })
                }),
                /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text3, {
                    regular: !0,
                    as: "div",
                    id: a,
                    children: /* @__PURE__ */ (0, _jsxruntime.jsx)("span", {
                        className: l ? _radiobuttoncssmistica.disabled : "",
                        children: u.children
                    })
                })
            ]
        })
    }));
}, re = (t)=>{
    var _ref;
    const { value: s, defaultValue: h, onChange: v, focusableRef: x, disabled: u } = (0, _formcontext.useControlProps)({
        name: t.name,
        value: t.value,
        defaultValue: t.defaultValue,
        onChange: t.onChange,
        disabled: t.disabled
    }), l = typeof s < "u", [C, p] = _react.useState((_ref = s !== null && s !== void 0 ? s : h) !== null && _ref !== void 0 ? _ref : "");
    _react.useEffect(()=>{
        s !== void 0 && p(s);
    }, [
        s
    ]);
    const c = (e)=>{
        l || p(e), v(e);
    }, [R, g] = _react.useState(null), a = _react.useRef(null), y = ()=>{
        if (a.current) {
            const e = a.current.querySelector("[role=radio][aria-checked=true]"), r = Array.from(a.current.querySelectorAll("[role=radio]"));
            if (r.length === 0) return;
            const o = ((e ? r.indexOf(e) : 0) + 1) % r.length, m = r[o], b = m.dataset.value;
            b && (m.focus(), c(b));
        }
    }, d = ()=>{
        if (a.current) {
            const e = a.current.querySelector("[role=radio][aria-checked=true]"), r = Array.from(a.current.querySelectorAll("[role=radio]"));
            if (r.length === 0) return;
            const f = e ? r.indexOf(e) : 0, o = (r.length + f - 1) % r.length, m = r[o], b = m.dataset.value;
            b && (m.focus(), c(b));
        }
    };
    _react.useEffect(()=>{
        if (a.current) {
            const e = a.current.querySelector("[role=radio]");
            var _e_dataset_value;
            e && g((_e_dataset_value = e.dataset.value) !== null && _e_dataset_value !== void 0 ? _e_dataset_value : null);
        }
    }, []);
    var _ref1;
    const V = (_ref1 = C !== null && C !== void 0 ? C : R) !== null && _ref1 !== void 0 ? _ref1 : null;
    return /* @__PURE__ */ (0, _jsxruntime.jsx)("div", _object_spread_props(_object_spread({
        ref: (0, _common.combineRefs)(a, x),
        role: "radiogroup",
        "aria-labelledby": t["aria-labelledby"]
    }, (0, _dom.getPrefixedDataAttributes)(t.dataAttributes, "RadioGroup")), {
        children: /* @__PURE__ */ (0, _jsxruntime.jsx)(N.Provider, {
            value: {
                disabled: u,
                selectedValue: C !== null && C !== void 0 ? C : h,
                focusableValue: V,
                select: c,
                selectNext: y,
                selectPrev: d
            },
            children: t.children
        })
    }));
}, le = L;
