// @telefonica/mistica/dist/radio-button.css-mistica.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    disabled: function() {
        return d;
    },
    innerCircle: function() {
        return r;
    },
    innerCircleChecked: function() {
        return c;
    },
    outerCircleCheckedVariants: function() {
        return a;
    },
    outerCircleVariants: function() {
        return f;
    },
    radioButton: function() {
        return i;
    },
    radioButtonContainerDisabled: function() {
        return s;
    }
});
require("./sprinkles.css.ts.vanilla.css-mistica.js");
require("./radio-button.css.ts.vanilla.css-mistica.js");
var d = "c6dmsud", r = "c6dmsu5 _1y2v1nfhk _1y2v1nfk3", c = "c6dmsu7 _1y2v1nf4z", a = {
    default: "c6dmsub",
    ios: "c6dmsuc"
}, f = {
    default: "c6dmsu2 c6dmsu1 _1y2v1nfhl _1y2v1nfhz _1y2v1nfht _1y2v1nfkk _1y2v1nfi2 _1y2v1nfk3 _1y2v1nf3p",
    ios: "c6dmsu3 c6dmsu1 _1y2v1nfhl _1y2v1nfhz _1y2v1nfht _1y2v1nfkk _1y2v1nfi2 _1y2v1nfk3 _1y2v1nf3p"
}, i = "c6dmsu9 c6dmsu8", s = "c6dmsua c6dmsu8";
