// @telefonica/mistica/dist/responsive-layout.css-mistica.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    MOBILE_SIDE_MARGIN: function() {
        return e;
    },
    SMALL_DESKTOP_SIDE_MARGIN: function() {
        return a;
    },
    TABLET_SIDE_MARGIN: function() {
        return n;
    },
    backgroundVariant: function() {
        return t;
    },
    desktopContainer: function() {
        return i;
    },
    forcedMarginDesktopContainer: function() {
        return v;
    },
    forcedMarginMobileContainer: function() {
        return _;
    },
    fullWidth: function() {
        return f;
    },
    mobileContainer: function() {
        return l;
    },
    resetContainerDesktop: function() {
        return g;
    },
    resetContainerMobile: function() {
        return s;
    },
    resetDesktop: function() {
        return M;
    },
    resetMobile: function() {
        return p;
    },
    responsiveLayout: function() {
        return C;
    },
    responsiveLayoutContainer: function() {
        return D;
    }
});
require("./sprinkles.css.ts.vanilla.css-mistica.js");
require("./responsive-layout.css.ts.vanilla.css-mistica.js");
var e = 16, a = 40, n = 32, t = {
    inverse: "_1y2v1nf3z",
    alternative: "_1y2v1nf44"
}, i = "_17lofg77", v = "_17lofg78", _ = "_17lofg7a", f = "_17lofg7f _1y2v1nfi7", l = "_17lofg79", g = "_17lofg73", s = "_17lofg72", M = "_17lofg75", p = "_17lofg74", C = "_17lofg7d", D = "_17lofg76";
