// @telefonica/mistica/dist/responsive-layout.js
"use client";
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    InternalResponsiveLayout: function() {
        return k;
    },
    ResetResponsiveLayout: function() {
        return q;
    },
    default: function() {
        return w;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _classnames = /*#__PURE__*/ _interop_require_default(require("classnames"));
const _dom = require("./utils/dom.js");
const _responsivelayoutcssmistica = require("./responsive-layout.css-mistica.js");
const _themevariantcontext = require("./theme-variant-context.js");
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
function _object_without_properties(source, excluded) {
    if (source == null) return {};
    var target = _object_without_properties_loose(source, excluded);
    var key, i;
    if (Object.getOwnPropertySymbols) {
        var sourceSymbolKeys = Object.getOwnPropertySymbols(source);
        for(i = 0; i < sourceSymbolKeys.length; i++){
            key = sourceSymbolKeys[i];
            if (excluded.indexOf(key) >= 0) continue;
            if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
            target[key] = source[key];
        }
    }
    return target;
}
function _object_without_properties_loose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;
    for(i = 0; i < sourceKeys.length; i++){
        key = sourceKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        target[key] = source[key];
    }
    return target;
}
const k = (param)=>{
    let { children: a, isInverse: e = !1, variant: r, backgroundColor: s, className: d, fullWidth: i, dataAttributes: m, shouldExpandWhenNested: l = !1, innerDivClassName: c } = param;
    const u = (0, _themevariantcontext.useThemeVariant)(), n = r || e && "inverse" || void 0;
    return /* @__PURE__ */ (0, _jsxruntime.jsx)(_themevariantcontext.ThemeVariant, {
        variant: n !== null && n !== void 0 ? n : u,
        children: /* @__PURE__ */ (0, _jsxruntime.jsx)("div", _object_spread_props(_object_spread({
            className: (0, _classnames.default)(d, n && n !== "default" && _responsivelayoutcssmistica.backgroundVariant[n], ...i ? [] : [
                l ? _responsivelayoutcssmistica.desktopContainer : _responsivelayoutcssmistica.forcedMarginDesktopContainer,
                l === !0 ? _responsivelayoutcssmistica.mobileContainer : _responsivelayoutcssmistica.forcedMarginMobileContainer,
                _responsivelayoutcssmistica.responsiveLayoutContainer
            ]),
            style: s ? {
                background: s
            } : void 0
        }, (0, _dom.getPrefixedDataAttributes)(m)), {
            children: /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                className: (0, _classnames.default)(i ? _responsivelayoutcssmistica.fullWidth : _responsivelayoutcssmistica.responsiveLayout, c),
                children: a
            })
        }))
    });
}, T = (_param)=>{
    var { children: a } = _param, e = _object_without_properties(_param, [
        "children"
    ]);
    return /* @__PURE__ */ (0, _jsxruntime.jsx)(k, _object_spread_props(_object_spread({}, e), {
        shouldExpandWhenNested: !0,
        children: a
    }));
}, q = (param)=>{
    let { children: a, skipMobile: e = !1, skipDesktop: r = !1 } = param;
    return /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
        className: (0, _classnames.default)({
            [_responsivelayoutcssmistica.resetContainerMobile]: !e,
            [_responsivelayoutcssmistica.resetContainerDesktop]: !r
        }),
        children: /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
            className: (0, _classnames.default)({
                [_responsivelayoutcssmistica.resetMobile]: !e,
                [_responsivelayoutcssmistica.resetDesktop]: !r
            }),
            children: a
        })
    });
}, w = T;
