// @telefonica/mistica/dist/sheet.js
"use client";
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    ActionsListSheet: function() {
        return tt;
    },
    ActionsSheet: function() {
        return nt;
    },
    InfoSheet: function() {
        return rt;
    },
    RadioListSheet: function() {
        return et;
    },
    SheetBody: function() {
        return T;
    },
    default: function() {
        return ot;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _classnames = /*#__PURE__*/ _interop_require_default(require("classnames"));
const _react = /*#__PURE__*/ _interop_require_wildcard(require("react"));
const _sheetcssmistica = require("./sheet.css-mistica.js");
const _focustrap = /*#__PURE__*/ _interop_require_default(require("./focus-trap.js"));
const _hooks = require("./hooks.js");
const _modalcontextprovider = require("./modal-context-provider.js");
const _portal = require("./portal.js");
const _text = require("./text.js");
const _skincontractcssmistica = require("./skins/skin-contract.css-mistica.js");
const _radiobutton = require("./radio-button.js");
const _list = require("./list.js");
const _negativebox = /*#__PURE__*/ _interop_require_default(require("./negative-box.js"));
const _stack = /*#__PURE__*/ _interop_require_default(require("./stack.js"));
const _box = /*#__PURE__*/ _interop_require_default(require("./box.js"));
const _touchable = /*#__PURE__*/ _interop_require_default(require("./touchable.js"));
const _inline = /*#__PURE__*/ _interop_require_default(require("./inline.js"));
const _circle = /*#__PURE__*/ _interop_require_default(require("./circle.js"));
const _divider = /*#__PURE__*/ _interop_require_default(require("./divider.js"));
const _dom = require("./utils/dom.js");
const _button = require("./button.js");
const _iconcloseregular = /*#__PURE__*/ _interop_require_default(require("./generated/mistica-icons/icon-close-regular.js"));
const _iconbutton = require("./icon-button.js");
const _buttonlayout = /*#__PURE__*/ _interop_require_default(require("./button-layout.js"));
const _image = /*#__PURE__*/ _interop_require_default(require("./image.js"));
const _css = require("./utils/css.js");
const _responsivelayoutcssmistica = require("./responsive-layout.css-mistica.js");
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
function _getRequireWildcardCache(nodeInterop) {
    if (typeof WeakMap !== "function") return null;
    var cacheBabelInterop = new WeakMap();
    var cacheNodeInterop = new WeakMap();
    return (_getRequireWildcardCache = function(nodeInterop) {
        return nodeInterop ? cacheNodeInterop : cacheBabelInterop;
    })(nodeInterop);
}
function _interop_require_wildcard(obj, nodeInterop) {
    if (!nodeInterop && obj && obj.__esModule) {
        return obj;
    }
    if (obj === null || typeof obj !== "object" && typeof obj !== "function") {
        return {
            default: obj
        };
    }
    var cache = _getRequireWildcardCache(nodeInterop);
    if (cache && cache.has(obj)) {
        return cache.get(obj);
    }
    var newObj = {
        __proto__: null
    };
    var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for(var key in obj){
        if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) {
            var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null;
            if (desc && (desc.get || desc.set)) {
                Object.defineProperty(newObj, key, desc);
            } else {
                newObj[key] = obj[key];
            }
        }
    }
    newObj.default = obj;
    if (cache) {
        cache.set(obj, newObj);
    }
    return newObj;
}
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
function _object_without_properties(source, excluded) {
    if (source == null) return {};
    var target = _object_without_properties_loose(source, excluded);
    var key, i;
    if (Object.getOwnPropertySymbols) {
        var sourceSymbolKeys = Object.getOwnPropertySymbols(source);
        for(i = 0; i < sourceSymbolKeys.length; i++){
            key = sourceSymbolKeys[i];
            if (excluded.indexOf(key) >= 0) continue;
            if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
            target[key] = source[key];
        }
    }
    return target;
}
function _object_without_properties_loose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;
    for(i = 0; i < sourceKeys.length; i++){
        key = sourceKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        target[key] = source[key];
    }
    return target;
}
const be = process.env.NODE_ENV === "test" ? 0 : _sheetcssmistica.transitionDuration, P = (n)=>"touches" in n ? n.touches[0].clientY : n.clientY, Te = (param)=>{
    let { closeModal: n } = param;
    const [s, l] = _react.useState(0), g = _react.useRef(!1), f = _react.useRef(0), r = _react.useRef(0), o = _react.useRef(0), u = _react.useRef(0), { isDesktopOrBigger: m } = (0, _hooks.useScreenSize)(), t = _react.useCallback((h)=>{
        g.current = !0, f.current = 0, r.current = Date.now(), o.current = P(h);
    }, []), c = _react.useCallback(()=>{
        g.current = !1, l(0);
    }, []);
    return _react.useEffect(()=>{
        if (m) return;
        const h = (a)=>{
            if (g.current) {
                if (f.current < 3) {
                    f.current++;
                    return;
                }
                u.current = P(a), l(u.current - o.current);
            }
        }, d = ()=>{
            if (!g.current) return;
            const a = Date.now() - r.current, p = u.current - o.current, D = p / a;
            g.current = !1, l(0), p > 50 && (u.current > window.innerHeight * 0.75 || D > 0.5) && n();
        };
        return document.addEventListener("touchmove", h), document.addEventListener("touchend", d), document.addEventListener("mousemove", h), document.addEventListener("mouseup", d), ()=>{
            document.removeEventListener("touchmove", h), document.removeEventListener("touchend", d), document.removeEventListener("mousemove", h), document.removeEventListener("mouseup", d);
        };
    }, [
        n,
        m
    ]), m ? {} : {
        onTouchStart: t,
        onMouseDown: t,
        style: s ? {
            transform: `translateY(${s}px)`,
            transition: "none"
        } : void 0,
        onScroll: c,
        overlayStyle: s ? {
            // decrease opacity when dragging down the sheet
            opacity: 0.25 + 1 - s / (window.innerHeight - o.current),
            transition: "none"
        } : void 0
    };
}, ke = {
    closed: {
        open: "opening"
    },
    opening: {
        close: "closed",
        transitionEnd: "open"
    },
    open: {
        close: "closing"
    },
    closing: {
        transitionEnd: "closed"
    }
}, Le = (n, s)=>ke[n][s] || n, E = /*#__PURE__*/ _react.forwardRef((param, g)=>{
    let { onClose: n, children: s, dataAttributes: l } = param;
    const { texts: f } = (0, _hooks.useTheme)(), [r, o] = _react.useReducer(Le, "closed"), u = _react.useRef(!1), m = (0, _hooks.useAriaId)(), t = _react.useCallback((p)=>{
        p.target === p.currentTarget && o("transitionEnd");
    }, []), c = ()=>{
        r === "open" && o("close");
    };
    _react.useEffect(()=>{
        if (r === "opening" || r === "closing") {
            const p = setTimeout(()=>{
                o("transitionEnd");
            }, be);
            return ()=>clearTimeout(p);
        }
    }, [
        r
    ]), _react.useEffect(()=>{
        r === "closed" ? u.current ? n == null || n() : o("open") : u.current = !0;
    }, [
        r,
        n
    ]);
    const _Te = Te({
        closeModal: c
    }), { onScroll: h, overlayStyle: d } = _Te, a = _object_without_properties(_Te, [
        "onScroll",
        "overlayStyle"
    ]);
    return (0, _modalcontextprovider.useSetModalStateEffect)(), (0, _hooks.useDisableBodyScroll)(r !== "closed"), r === "closed" ? null : /* @__PURE__ */ (0, _jsxruntime.jsx)(_portal.Portal, {
        children: /* @__PURE__ */ (0, _jsxruntime.jsxs)(_focustrap.default, {
            children: [
                /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                    className: (0, _classnames.default)(_sheetcssmistica.overlay, {
                        [_sheetcssmistica.closingOverlay]: r === "closing"
                    }),
                    style: d,
                    onClick: c
                }),
                /* @__PURE__ */ (0, _jsxruntime.jsx)("div", _object_spread_props(_object_spread({
                    className: (0, _classnames.default)(_sheetcssmistica.SheetContainer, {
                        [_sheetcssmistica.closingSheet]: r === "closing"
                    }),
                    onTransitionEnd: t,
                    onAnimationEnd: t
                }, a, (0, _dom.getPrefixedDataAttributes)(l, "Sheet")), {
                    ref: g,
                    children: /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                        className: _sheetcssmistica.Sheet,
                        children: /* @__PURE__ */ (0, _jsxruntime.jsxs)("div", {
                            className: _sheetcssmistica.SheetContent,
                            children: [
                                /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                                    className: _sheetcssmistica.handleContainer,
                                    children: /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                                        className: _sheetcssmistica.handle
                                    })
                                }),
                                /* @__PURE__ */ (0, _jsxruntime.jsxs)("section", {
                                    role: "dialog",
                                    "aria-modal": "true",
                                    "aria-labelledby": m,
                                    onScroll: h,
                                    className: _sheetcssmistica.children,
                                    style: {
                                        paddingBottom: _css.safeAreaInsetBottom
                                    },
                                    children: [
                                        typeof s == "function" ? s({
                                            closeModal: c,
                                            modalTitleId: m
                                        }) : s,
                                        /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                                            className: _sheetcssmistica.modalCloseButton,
                                            children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_iconbutton.InternalIconButton, {
                                                onPress: c,
                                                "aria-label": f.modalClose || f.closeButtonLabel,
                                                Icon: _iconcloseregular.default,
                                                bleedLeft: !0,
                                                bleedRight: !0,
                                                bleedY: !0
                                            })
                                        })
                                    ]
                                })
                            ]
                        })
                    })
                }))
            ]
        })
    });
}), k = {
    mobile: _responsivelayoutcssmistica.MOBILE_SIDE_MARGIN,
    tablet: _responsivelayoutcssmistica.TABLET_SIDE_MARGIN,
    desktop: _responsivelayoutcssmistica.SMALL_DESKTOP_SIDE_MARGIN
}, T = (param)=>{
    let { title: n, subtitle: s, description: l, modalTitleId: g, button: f, secondaryButton: r, link: o, children: u } = param;
    const m = _react.useRef(null), t = _react.useRef(null), c = _react.useRef(null);
    _react.useEffect(()=>{
        t.current && (c.current = (0, _dom.getScrollableParentElement)(t.current));
    }, []);
    const h = !(0, _hooks.useIsInViewport)(m, !0, {
        root: c.current
    }), d = !(0, _hooks.useIsInViewport)(t, !0, {
        rootMargin: "1px",
        // bottomScrollSignal div has 0px height so we need a 1px margin to trigger the intersection observer
        root: c.current
    }), a = !!f || !!r || !!o;
    return /* @__PURE__ */ (0, _jsxruntime.jsxs)(_jsxruntime.Fragment, {
        children: [
            /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                ref: m
            }),
            /* @__PURE__ */ (0, _jsxruntime.jsxs)("div", {
                className: _sheetcssmistica.stickyTitle,
                children: [
                    n ? /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                        paddingBottom: 8,
                        paddingTop: {
                            mobile: 0,
                            desktop: 40
                        },
                        paddingX: k,
                        children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text5, {
                            as: "h2",
                            id: g,
                            truncate: !0,
                            children: n
                        })
                    }) : /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                        paddingTop: {
                            mobile: 0,
                            desktop: 40
                        }
                    }),
                    h && /* @__PURE__ */ (0, _jsxruntime.jsx)(_divider.default, {})
                ]
            }),
            /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                className: _sheetcssmistica.bodyContent,
                children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                    paddingBottom: a ? 0 : {
                        desktop: 40,
                        mobile: 0
                    },
                    paddingX: k,
                    children: /* @__PURE__ */ (0, _jsxruntime.jsxs)(_stack.default, {
                        space: 8,
                        children: [
                            s || l ? /* @__PURE__ */ (0, _jsxruntime.jsxs)(_stack.default, {
                                space: {
                                    mobile: 8,
                                    desktop: 16
                                },
                                children: [
                                    s && /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text3, {
                                        as: "p",
                                        regular: !0,
                                        children: s
                                    }),
                                    l && (Array.isArray(l) ? /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text2, {
                                        as: "div",
                                        regular: !0,
                                        color: _skincontractcssmistica.vars.colors.textSecondary,
                                        children: l.map((p, D)=>/* @__PURE__ */ (0, _jsxruntime.jsx)("p", {
                                                style: {
                                                    margin: 0,
                                                    marginBottom: D < l.length - 1 ? "1em" : void 0
                                                },
                                                children: p
                                            }, D))
                                    }) : /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text2, {
                                        as: "p",
                                        regular: !0,
                                        color: _skincontractcssmistica.vars.colors.textSecondary,
                                        children: l
                                    }))
                                ]
                            }) : null,
                            u
                        ]
                    })
                })
            }),
            a && /* @__PURE__ */ (0, _jsxruntime.jsxs)("div", {
                className: _sheetcssmistica.stickyButtons,
                children: [
                    d && /* @__PURE__ */ (0, _jsxruntime.jsx)(_divider.default, {}),
                    /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                        paddingY: {
                            mobile: 16,
                            desktop: 40
                        },
                        paddingX: k,
                        children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_buttonlayout.default, {
                            align: "full-width",
                            link: o,
                            primaryButton: f,
                            secondaryButton: r
                        })
                    })
                ]
            }),
            /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                ref: t
            })
        ]
    });
}, et = /*#__PURE__*/ _react.forwardRef((param, t)=>{
    let { title: n, subtitle: s, description: l, items: g, selectedId: f, onClose: r, onSelect: o, button: u, dataAttributes: m } = param;
    const [c, h] = _react.useState(f), d = _react.useRef(!1), { isDesktopOrBigger: a } = (0, _hooks.useScreenSize)(), { texts: p } = (0, _hooks.useTheme)();
    return /* @__PURE__ */ (0, _jsxruntime.jsx)(E, {
        onClose: r,
        ref: t,
        dataAttributes: _object_spread({
            "component-name": "RadioListSheet"
        }, m),
        children: (param)=>/* @__PURE__ */ {
            let { closeModal: D, modalTitleId: B } = param;
            var _ref;
            return (0, _jsxruntime.jsx)(T, {
                title: n,
                subtitle: s,
                description: l,
                modalTitleId: B,
                button: a ? /* @__PURE__ */ (0, _jsxruntime.jsx)(_button.ButtonPrimary, {
                    onPress: ()=>{
                        d.current && (o == null || o(c !== null && c !== void 0 ? c : "")), D();
                    },
                    children: (_ref = u == null ? void 0 : u.text) !== null && _ref !== void 0 ? _ref : p.sheetConfirmButton
                }) : void 0,
                children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_negativebox.default, {
                    children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_radiobutton.RadioGroup, {
                        "aria-labelledby": B,
                        name: "sheetselection",
                        value: c,
                        onChange: (R)=>{
                            h(R), d.current = !0, !a && (o == null || o(R), setTimeout(()=>{
                                D();
                            }, 200));
                        },
                        children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_list.RowList, {
                            children: g.map((R)=>/* @__PURE__ */ {
                                var _R_title;
                                return (0, _jsxruntime.jsx)(_list.Row, {
                                    title: (_R_title = R.title) !== null && _R_title !== void 0 ? _R_title : "",
                                    description: R.description,
                                    asset: R.asset,
                                    radioValue: R.id
                                }, R.id);
                            })
                        })
                    })
                })
            });
        }
    });
}), tt = /*#__PURE__*/ _react.forwardRef((param, u)=>{
    let { title: n, subtitle: s, description: l, items: g, onClose: f, onSelect: r, dataAttributes: o } = param;
    const { isDarkMode: m } = (0, _hooks.useTheme)();
    return /* @__PURE__ */ (0, _jsxruntime.jsx)(E, {
        onClose: f,
        ref: u,
        dataAttributes: _object_spread({
            "component-name": "ActionsListSheet"
        }, o),
        children: (param)=>{
            let { closeModal: t, modalTitleId: c } = param;
            return /* @__PURE__ */ (0, _jsxruntime.jsx)(T, {
                title: n,
                subtitle: s,
                description: l,
                modalTitleId: c,
                children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_negativebox.default, {
                    children: g.map((param)=>{
                        let { id: h, style: d, title: a, icon: p } = param;
                        return /* @__PURE__ */ (0, _jsxruntime.jsx)(_touchable.default, {
                            onPress: ()=>{
                                r == null || r(h), t();
                            },
                            children: /* @__PURE__ */ (0, _jsxruntime.jsxs)("div", {
                                className: _sheetcssmistica.sheetActionRow,
                                children: [
                                    p && /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                                        paddingRight: 16,
                                        children: p.Icon ? /* @__PURE__ */ (0, _jsxruntime.jsx)(p.Icon, {
                                            size: 24,
                                            color: d === "destructive" ? _skincontractcssmistica.vars.colors.textLinkDanger : _skincontractcssmistica.vars.colors.neutralHigh
                                        }) : /* @__PURE__ */ (0, _jsxruntime.jsx)(_image.default, {
                                            circular: !0,
                                            src: m && p.urlDark ? p.urlDark : p.url,
                                            width: 40
                                        })
                                    }),
                                    /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text3, {
                                        regular: !0,
                                        color: d === "destructive" ? _skincontractcssmistica.vars.colors.textLinkDanger : _skincontractcssmistica.vars.colors.textPrimary,
                                        children: a
                                    })
                                ]
                            })
                        }, h);
                    })
                })
            });
        }
    });
}), rt = /*#__PURE__*/ _react.forwardRef((param, o)=>{
    let { title: n, subtitle: s, description: l, items: g, onClose: f, dataAttributes: r } = param;
    const { isDarkMode: u } = (0, _hooks.useTheme)();
    return /* @__PURE__ */ (0, _jsxruntime.jsx)(E, {
        onClose: f,
        ref: o,
        dataAttributes: _object_spread({
            "component-name": "InfoSheet"
        }, r),
        children: (param)=>{
            let { modalTitleId: m } = param;
            return /* @__PURE__ */ (0, _jsxruntime.jsx)(T, {
                title: n,
                subtitle: s,
                description: l,
                modalTitleId: m,
                children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                    paddingBottom: 16,
                    children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_stack.default, {
                        space: 16,
                        role: "list",
                        children: g.map((t, c)=>/* @__PURE__ */ (0, _jsxruntime.jsxs)(_inline.default, {
                                space: 8,
                                children: [
                                    /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                                        className: _sheetcssmistica.infoItemIcon,
                                        children: t.icon.type === "bullet" ? /* @__PURE__ */ (0, _jsxruntime.jsx)(_circle.default, {
                                            size: 8,
                                            backgroundColor: _skincontractcssmistica.vars.colors.textPrimary
                                        }) : t.icon.Icon ? /* @__PURE__ */ (0, _jsxruntime.jsx)(t.icon.Icon, {
                                            size: t.icon.type === "small" ? 16 : 24
                                        }) : /* @__PURE__ */ (0, _jsxruntime.jsx)(_image.default, {
                                            src: u && t.icon.urlDark ? t.icon.urlDark : t.icon.url,
                                            width: t.icon.type === "small" ? 16 : 24,
                                            height: t.icon.type === "small" ? 16 : 24
                                        })
                                    }),
                                    /* @__PURE__ */ (0, _jsxruntime.jsxs)(_stack.default, {
                                        space: 2,
                                        children: [
                                            /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text3, {
                                                regular: !0,
                                                children: t.title
                                            }),
                                            t.description && /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text2, {
                                                regular: !0,
                                                color: _skincontractcssmistica.vars.colors.textSecondary,
                                                children: t.description
                                            })
                                        ]
                                    })
                                ]
                            }, t.id || c))
                    })
                })
            });
        }
    });
}), nt = /*#__PURE__*/ _react.forwardRef((param, t)=>{
    let { title: n, subtitle: s, description: l, button: g, secondaryButton: f, buttonLink: r, onClose: o, dataAttributes: u, onPressButton: m } = param;
    const c = (d, a)=>()=>{
            m == null || m(a), d();
        }, h = (_param)=>{
        var { text: d } = _param, a = _object_without_properties(_param, [
            "text"
        ]);
        return _object_spread({
            children: d
        }, a);
    };
    return /* @__PURE__ */ (0, _jsxruntime.jsx)(E, {
        onClose: o,
        ref: t,
        dataAttributes: _object_spread({
            "component-name": "ActionsSheet"
        }, u),
        children: (param)=>{
            let { modalTitleId: d, closeModal: a } = param;
            return /* @__PURE__ */ (0, _jsxruntime.jsx)(T, {
                title: n,
                subtitle: s,
                description: l,
                modalTitleId: d,
                button: /* @__PURE__ */ (0, _jsxruntime.jsx)(_button.ButtonPrimary, _object_spread_props(_object_spread({}, h(g)), {
                    onPress: c(a, "PRIMARY")
                })),
                secondaryButton: f ? /* @__PURE__ */ (0, _jsxruntime.jsx)(_button.ButtonSecondary, _object_spread_props(_object_spread({}, h(f)), {
                    onPress: c(a, "SECONDARY")
                })) : void 0,
                link: r ? /* @__PURE__ */ (0, _jsxruntime.jsx)(_button.ButtonLink, _object_spread_props(_object_spread({}, h(r)), {
                    onPress: c(a, "LINK")
                })) : void 0
            });
        }
    });
}), ot = E;
