// @telefonica/mistica/dist/sheet.css-mistica.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    Sheet: function() {
        return y;
    },
    SheetContainer: function() {
        return f;
    },
    SheetContent: function() {
        return _;
    },
    bodyContent: function() {
        return r;
    },
    children: function() {
        return l;
    },
    closingOverlay: function() {
        return a;
    },
    closingSheet: function() {
        return t;
    },
    handle: function() {
        return h;
    },
    handleContainer: function() {
        return e;
    },
    infoItemIcon: function() {
        return i;
    },
    modalCloseButton: function() {
        return o;
    },
    overlay: function() {
        return c;
    },
    sheetActionRow: function() {
        return k;
    },
    stickyButtons: function() {
        return z;
    },
    stickyTitle: function() {
        return s;
    },
    transitionDuration: function() {
        return m;
    }
});
require("./sprinkles.css.ts.vanilla.css-mistica.js");
require("./sheet.css.ts.vanilla.css-mistica.js");
var y = "cylz3v6 _1y2v1nf3p", f = "cylz3v4 _1y2v1nfhg _1y2v1nfl6 _1y2v1nfll _1y2v1nfm0", _ = "cylz3v9 _1y2v1nf81 _1y2v1nfhk _1y2v1nfhq", r = "cylz3vo", l = "_1y2v1nfkq _1y2v1nfi6 _1y2v1nfhk _1y2v1nfhq", a = "cylz3vl", t = "cylz3v7", h = "_1y2v1nf4y _1y2v1nfig _1y2v1nfir _1y2v1nfk4", e = "cylz3vc _1y2v1nf7m _1y2v1nf8v _1y2v1nfhf _1y2v1nfkr _1y2v1nfj1 _1y2v1nfi7 _1y2v1nfhk _1y2v1nfht", i = "_1y2v1nfhk _1y2v1nfhz _1y2v1nfiw", o = "cylz3vf _1y2v1nf7y _1y2v1nf97 _1y2v1nfag _1y2v1nfbp _1y2v1nfhf _1y2v1nfkr _1y2v1nfll", c = "cylz3vk _1y2v1nfhg _1y2v1nf41 _1y2v1nfkr _1y2v1nfl6 _1y2v1nfll _1y2v1nfm0", k = "cylz3vq _1y2v1nf7s _1y2v1nf91 _1y2v1nfaa _1y2v1nfbj _1y2v1nfhk _1y2v1nfjx _1y2v1nfhz", z = "_1y2v1nfhi _1y2v1nfm0 _1y2v1nf3p", s = "_1y2v1nfhi _1y2v1nfkr _1y2v1nf3p", m = 400;
