// @telefonica/mistica/dist/skeleton-base.js
"use client";
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "default", {
    enumerable: true,
    get: function() {
        return v;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _themevariantcontext = require("./theme-variant-context.js");
const _classnames = /*#__PURE__*/ _interop_require_default(require("classnames"));
const _sprinklescssmistica = require("./sprinkles.css-mistica.js");
const _skincontractcssmistica = require("./skins/skin-contract.css-mistica.js");
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
const d = (param)=>{
    let { width: e = "100%", height: o = 8, radius: s = _skincontractcssmistica.vars.borderRadii.container, className: a, noBorderRadius: n = !1 } = param;
    const t = (0, _themevariantcontext.useIsInverseVariant)();
    return /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
        className: (0, _classnames.default)(a, (0, _sprinklescssmistica.sprinkles)({
            background: t ? _skincontractcssmistica.vars.colors.backgroundSkeletonInverse : _skincontractcssmistica.vars.colors.backgroundSkeleton
        })),
        style: {
            borderRadius: n ? 0 : s,
            width: e,
            height: o
        },
        "aria-hidden": !0
    });
}, v = d;
