// @telefonica/mistica/dist/slider.js
"use client";
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "default", {
    enumerable: true,
    get: function() {
        return De;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _react = /*#__PURE__*/ _interop_require_wildcard(require("react"));
const _slidercssmistica = require("./slider.css-mistica.js");
const _skincontractcssmistica = require("./skins/skin-contract.css-mistica.js");
const _environment = require("./utils/environment.js");
const _classnames = /*#__PURE__*/ _interop_require_default(require("classnames"));
const _dom = require("./utils/dom.js");
const _hooks = require("./hooks.js");
const _tooltip = /*#__PURE__*/ _interop_require_default(require("./tooltip.js"));
const _box = /*#__PURE__*/ _interop_require_default(require("./box.js"));
const _formcontext = require("./form-context.js");
const _common = require("./utils/common.js");
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
function _getRequireWildcardCache(nodeInterop) {
    if (typeof WeakMap !== "function") return null;
    var cacheBabelInterop = new WeakMap();
    var cacheNodeInterop = new WeakMap();
    return (_getRequireWildcardCache = function(nodeInterop) {
        return nodeInterop ? cacheNodeInterop : cacheBabelInterop;
    })(nodeInterop);
}
function _interop_require_wildcard(obj, nodeInterop) {
    if (!nodeInterop && obj && obj.__esModule) {
        return obj;
    }
    if (obj === null || typeof obj !== "object" && typeof obj !== "function") {
        return {
            default: obj
        };
    }
    var cache = _getRequireWildcardCache(nodeInterop);
    if (cache && cache.has(obj)) {
        return cache.get(obj);
    }
    var newObj = {
        __proto__: null
    };
    var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for(var key in obj){
        if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) {
            var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null;
            if (desc && (desc.get || desc.set)) {
                Object.defineProperty(newObj, key, desc);
            } else {
                newObj[key] = obj[key];
            }
        }
    }
    newObj.default = obj;
    if (cache) {
        cache.set(obj, newObj);
    }
    return newObj;
}
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function _object_without_properties(source, excluded) {
    if (source == null) return {};
    var target = _object_without_properties_loose(source, excluded);
    var key, i;
    if (Object.getOwnPropertySymbols) {
        var sourceSymbolKeys = Object.getOwnPropertySymbols(source);
        for(i = 0; i < sourceSymbolKeys.length; i++){
            key = sourceSymbolKeys[i];
            if (excluded.indexOf(key) >= 0) continue;
            if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
            target[key] = source[key];
        }
    }
    return target;
}
function _object_without_properties_loose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;
    for(i = 0; i < sourceKeys.length; i++){
        key = sourceKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        target[key] = source[key];
    }
    return target;
}
const Re = 48, me = 20, Ce = 28, Ve = (n, t, o)=>t >= o ? 0 : (n - t) / (o - t), I = (n, t, o, a, g)=>{
    const v = (i)=>{
        if (i === void 0) return t;
        const B = n ? t + (o - t) * i : i;
        return Math.max(t, Math.min(o, B));
    };
    if (t >= o) return t;
    const h = v(g), P = t + Math.floor((h - t) / a) * a, R = t + Math.ceil((h - t) / a) * a;
    return R <= o && R - h <= h - P ? R : P;
}, w = (n, t, o)=>{
    if (!o) return t;
    if (t === void 0) return n;
    let a = 0;
    return o.forEach((g, v)=>{
        o && Math.abs(g - t) <= Math.abs(o[a] - t) && (a = v);
    }), a;
}, Pe = /*#__PURE__*/ _react.forwardRef((_param, B)=>{
    var { values: n, step: t = 1, min: o = 0, max: a = 100, "aria-label": g, "aria-labelledby": v, id: h, dataAttributes: P, tooltip: R } = _param, i = _object_without_properties(_param, [
        "values",
        "step",
        "min",
        "max",
        "aria-label",
        "aria-labelledby",
        "id",
        "dataAttributes",
        "tooltip"
    ]);
    n && (n.length === 0 ? n = void 0 : a = n.length - 1), t = t | 0, t = t <= 0 ? 1 : t;
    const { defaultValue: Z, value: D, onChange: H, focusableRef: x, disabled: U, name: z } = (0, _formcontext.useControlProps)({
        name: i.name,
        value: i.value !== void 0 ? I(!1, o, a, t, w(o, i.value, n)) : void 0,
        defaultValue: i.defaultValue !== void 0 ? I(!1, o, a, t, w(o, i.defaultValue, n)) : void 0,
        onChange: i.onChangeValue,
        disabled: i.disabled
    }), [q, G] = _react.useState(D !== null && D !== void 0 ? D : I(!1, o, a, t, w(o, Z, n))), u = D !== null && D !== void 0 ? D : q, $ = _react.useRef(u), T = _react.useCallback((e, r)=>{
        const c = I(r, o, a, t, e);
        $.current !== c && (H(n ? n[c] : c), G(c), $.current = c);
    }, [
        o,
        a,
        t,
        n,
        H
    ]);
    _react.useEffect(()=>{
        T($.current, !1);
    }, [
        T
    ]);
    const _ = _react.useRef(null), m = _react.useRef(null), N = _react.useRef(null), J = _react.useRef(null), [p, E] = _react.useState(!1), [y, F] = _react.useState(!1), [K, L] = _react.useState(!1), { isIos: S } = (0, _hooks.useTheme)(), A = (e, r, c)=>{
        const d = e == null ? void 0 : e.getBoundingClientRect();
        return !!d && d.left <= r && r <= d.right && d.top <= c && c <= d.bottom;
    }, l = (0, _environment.isTouchableDevice)(), C = S ? Ce : me, f = l ? Re : C, M = (e)=>{
        const r = _.current;
        if (r) {
            const c = r.getBoundingClientRect().left + C / 2, d = r.getBoundingClientRect().right - C / 2;
            T((e - c) / (d - c), !0);
        }
    }, Q = (e)=>{
        (0, _dom.cancelEvent)(e), M(e.clientX);
    }, W = (e)=>{
        const r = m.current;
        r && (r.onpointermove = Q, r.setPointerCapture(e.pointerId));
    }, ee = (e)=>{
        const r = m.current;
        r && (r.onpointermove = null, r.releasePointerCapture(e.pointerId));
    }, k = `calc(${Ve(u, o, a)} * (100% - ${C}px) - ${(f - C) / 2}px)`, O = `calc(${k} + ${f / 2}px)`, X = /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
        className: (0, _classnames.default)(S ? _slidercssmistica.iosThumb : _slidercssmistica.defaultThumb, {
            [_slidercssmistica.thumbHover]: !S && y && !p,
            [_slidercssmistica.thumbActive]: !S && p
        })
    });
    return /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
        paddingY: 8,
        dataAttributes: _object_spread({
            "component-name": "Slider"
        }, P),
        children: /* @__PURE__ */ (0, _jsxruntime.jsxs)("div", {
            className: (0, _classnames.default)(_slidercssmistica.container, {
                [_slidercssmistica.disabled]: U
            }),
            style: {
                height: f
            },
            ref: N,
            onPointerDown: (e)=>{
                const r = e.clientX, c = e.clientY;
                !l && A(N.current, r, c) ? (A(m.current, r, c) || M(r), E(!0), W(e)) : (0, _dom.cancelEvent)(e);
            },
            onPointerUp: (e)=>{
                l || (E(!1), ee(e));
            },
            onTouchStart: (e)=>{
                if ((0, _dom.cancelEvent)(e), l) {
                    const r = e.nativeEvent.touches[0].clientX, c = e.nativeEvent.touches[0].clientY;
                    A(m.current, r, c) || M(r), E(!0);
                }
            },
            onTouchEnd: (e)=>{
                (0, _dom.cancelEvent)(e), l && E(!1);
            },
            onTouchMove: (e)=>{
                (0, _dom.cancelEvent)(e), l && M(e.nativeEvent.touches[0].clientX);
            },
            children: [
                /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                    className: _slidercssmistica.track,
                    ref: _,
                    style: {
                        background: `linear-gradient(to right, ${_skincontractcssmistica.vars.colors.controlActivated} ${O}, ${_skincontractcssmistica.vars.colors.barTrack} ${O}`
                    }
                }),
                /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                    className: _slidercssmistica.thumbContainer,
                    ref: m,
                    style: {
                        cursor: p ? "grabbing" : y ? "grab" : "auto",
                        left: k,
                        width: f,
                        height: f
                    },
                    onPointerEnter: ()=>{
                        l || F(!0);
                    },
                    onPointerLeave: ()=>{
                        l || F(!1);
                    },
                    children: R ? /* @__PURE__ */ (0, _jsxruntime.jsx)(_tooltip.default, {
                        target: X,
                        open: p || K || y ? !0 : void 0,
                        description: String(n ? n[u] : u),
                        centerContent: !0,
                        delay: !1
                    }) : X
                }),
                /* @__PURE__ */ (0, _jsxruntime.jsx)("input", {
                    type: "range",
                    min: o,
                    max: a,
                    ref: (0, _common.combineRefs)(B, J, x),
                    step: t,
                    "aria-label": g,
                    "aria-labelledby": v,
                    id: h,
                    className: _slidercssmistica.input,
                    "aria-valuetext": String(n ? n[u] : u),
                    style: {
                        left: k,
                        width: f,
                        height: f
                    },
                    name: z,
                    value: u,
                    disabled: U,
                    onChange: (e)=>T(+e.target.value, !1),
                    onFocus: ()=>{
                        L(!0);
                    },
                    onBlur: ()=>{
                        L(!1);
                    }
                })
            ]
        })
    });
}), De = Pe;
