// @telefonica/mistica/dist/slider.css-mistica.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    container: function() {
        return n;
    },
    defaultThumb: function() {
        return f;
    },
    disabled: function() {
        return h;
    },
    input: function() {
        return y;
    },
    iosThumb: function() {
        return r;
    },
    thumbActive: function() {
        return k;
    },
    thumbContainer: function() {
        return a;
    },
    thumbHover: function() {
        return i;
    },
    track: function() {
        return t;
    }
});
require("./sprinkles.css.ts.vanilla.css-mistica.js");
require("./slider.css.ts.vanilla.css-mistica.js");
var n = "_1j1wlhk1 _1y2v1nfi7 _1y2v1nfhe _1y2v1nfhk _1y2v1nfhz _1y2v1nfkk", f = "_1j1wlhk7 _1y2v1nfif _1y2v1nfiv _1y2v1nfk3 _1y2v1nf4z", h = "_1j1wlhk2", y = "_1j1wlhkd _1y2v1nfhf _1y2v1nfkr", r = "_1j1wlhk9 _1y2v1nf59 _1y2v1nfk3", k = "_1j1wlhkb", a = "_1j1wlhk5 _1y2v1nfhf _1y2v1nfhk _1y2v1nfhz _1y2v1nfht", i = "_1j1wlhka", t = "_1y2v1nfi7 _1y2v1nfir _1y2v1nfkh";
