// @telefonica/mistica/dist/snackbar.css-mistica.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    TRANSITION_TIME_IN_MS: function() {
        return f;
    },
    button: function() {
        return r;
    },
    content: function() {
        return _;
    },
    dismissButton: function() {
        return y;
    },
    dismissIcon: function() {
        return a;
    },
    longButton: function() {
        return o;
    },
    snackbar: function() {
        return b;
    },
    snackbarContainer: function() {
        return t;
    },
    snackbarOpen: function() {
        return h;
    },
    wrapper: function() {
        return e;
    },
    wrapperCritical: function() {
        return p;
    },
    wrapperInfo: function() {
        return i;
    },
    wrapperOpen: function() {
        return k;
    }
});
require("./sprinkles.css.ts.vanilla.css-mistica.js");
require("./snackbar.css.ts.vanilla.css-mistica.js");
var f = 300, r = "o7bjb0c", _ = "_1y2v1nfhk _1y2v1nfhw", y = {
    centered: "o7bjb0g _1y2v1nfhk _1y2v1nfhf _1y2v1nfhz _1y2v1nfht _1y2v1nflo",
    topRight: "_1y2v1nfhk _1y2v1nfhf _1y2v1nfhz _1y2v1nfht _1y2v1nflo _1y2v1nfku"
}, a = "o7bjb0j _1y2v1nfih _1y2v1nfix _1y2v1nfhk _1y2v1nfhz _1y2v1nfht _1y2v1nfk3", o = "o7bjb0d", b = "o7bjb01 _1y2v1nfi7 _1y2v1nfhk _1y2v1nfht", t = "o7bjb03 _1y2v1nfhe", h = "o7bjb04", e = "o7bjb06 _1y2v1nfhg _1y2v1nfkd", p = "_1y2v1nf5e", i = "_1y2v1nf5f", k = "o7bjb09";
