// @telefonica/mistica/dist/snackbar.js
"use client";
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "default", {
    enumerable: true,
    get: function() {
        return he;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _react = /*#__PURE__*/ _interop_require_wildcard(require("react"));
const _touchable = /*#__PURE__*/ _interop_require_wildcard(require("./touchable.js"));
const _classnames = /*#__PURE__*/ _interop_require_default(require("classnames"));
const _webviewbridge = require("@tef-novum/webview-bridge");
const _hooks = require("./hooks.js");
const _text = require("./text.js");
const _snackbarcssmistica = require("./snackbar.css-mistica.js");
const _sprinklescssmistica = require("./sprinkles.css-mistica.js");
const _skincontractcssmistica = require("./skins/skin-contract.css-mistica.js");
const _dom = require("./utils/dom.js");
const _portal = require("./portal.js");
const _iconcloseregular = /*#__PURE__*/ _interop_require_default(require("./generated/mistica-icons/icon-close-regular.js"));
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
function _getRequireWildcardCache(nodeInterop) {
    if (typeof WeakMap !== "function") return null;
    var cacheBabelInterop = new WeakMap();
    var cacheNodeInterop = new WeakMap();
    return (_getRequireWildcardCache = function(nodeInterop) {
        return nodeInterop ? cacheNodeInterop : cacheBabelInterop;
    })(nodeInterop);
}
function _interop_require_wildcard(obj, nodeInterop) {
    if (!nodeInterop && obj && obj.__esModule) {
        return obj;
    }
    if (obj === null || typeof obj !== "object" && typeof obj !== "function") {
        return {
            default: obj
        };
    }
    var cache = _getRequireWildcardCache(nodeInterop);
    if (cache && cache.has(obj)) {
        return cache.get(obj);
    }
    var newObj = {
        __proto__: null
    };
    var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for(var key in obj){
        if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) {
            var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null;
            if (desc && (desc.get || desc.set)) {
                Object.defineProperty(newObj, key, desc);
            } else {
                newObj[key] = obj[key];
            }
        }
    }
    newObj.default = obj;
    if (cache) {
        cache.set(obj, newObj);
    }
    return newObj;
}
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
const Z = [
    "DISMISS",
    "TIMEOUT",
    "CONSECUTIVE",
    "BUTTON"
], ee = 5e3, re = 1e4, te = /*#__PURE__*/ _react.forwardRef((param, I)=>{
    let { message: m, buttonText: n, buttonAccessibilityLabel: u, duration: t, onClose: o, type: i, withDismiss: d = !1, dataAttributes: O } = param;
    const { texts: p } = (0, _hooks.useTheme)(), [s, a] = _react.useState(!1), { width: R, ref: c } = (0, _hooks.useElementDimensions)(), { isDesktopOrBigger: D } = (0, _hooks.useScreenSize)(), l = R > (D ? 160 : 128), E = _react.useRef(null), g = t === 1 / 0 && !n || d, T = _react.useRef(o);
    _react.useEffect(()=>{
        T.current = o;
    }, [
        o
    ]);
    const f = _react.useCallback((v)=>{
        a(!1), setTimeout(()=>{
            var h;
            (h = T.current) == null || h.call(T, v);
        }, process.env.NODE_ENV === "test" ? 0 : _snackbarcssmistica.TRANSITION_TIME_IN_MS);
    }, []);
    return _react.useImperativeHandle(I, ()=>_object_spread_props(_object_spread({}, E), {
            close: f
        }), [
        f
    ]), _react.useEffect(()=>{
        const v = setTimeout(()=>{
            a(!0);
        }, 50), h = t !== 1 / 0 ? setTimeout(()=>f({
                action: "TIMEOUT"
            }), t) : void 0;
        return ()=>{
            clearTimeout(v), clearTimeout(h);
        };
    }, [
        f,
        t
    ]), /* @__PURE__ */ (0, _jsxruntime.jsx)(_portal.Portal, {
        className: _snackbarcssmistica.snackbarContainer,
        children: /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
            ref: E,
            className: (0, _classnames.default)(_snackbarcssmistica.snackbar, {
                [_snackbarcssmistica.snackbarOpen]: s
            }),
            children: /* @__PURE__ */ (0, _jsxruntime.jsxs)("div", _object_spread_props(_object_spread({
                role: "alert",
                className: (0, _classnames.default)(_snackbarcssmistica.wrapper, i === "CRITICAL" ? _snackbarcssmistica.wrapperCritical : _snackbarcssmistica.wrapperInfo, {
                    [_snackbarcssmistica.wrapperOpen]: s
                })
            }, (0, _dom.getPrefixedDataAttributes)(O, "SnackBar")), {
                children: [
                    /* @__PURE__ */ (0, _jsxruntime.jsxs)("div", {
                        className: (0, _classnames.default)(_snackbarcssmistica.content, (0, _sprinklescssmistica.sprinkles)({
                            flexDirection: l ? "column" : "row",
                            alignItems: l ? void 0 : "center"
                        })),
                        style: g && !l ? {
                            paddingRight: 32
                        } : void 0,
                        children: [
                            /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                                style: g && l ? {
                                    paddingRight: 32
                                } : void 0,
                                children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text2, {
                                    regular: !0,
                                    color: _skincontractcssmistica.vars.colors.textPrimaryInverse,
                                    children: m
                                })
                            }),
                            n && /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                                className: (0, _classnames.default)(_snackbarcssmistica.button, {
                                    [_snackbarcssmistica.longButton]: l
                                }),
                                children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_touchable.BaseTouchable, {
                                    className: (0, _sprinklescssmistica.sprinkles)({
                                        paddingY: 4,
                                        paddingX: 8,
                                        border: "none",
                                        padding: 0,
                                        background: "transparent"
                                    }),
                                    ref: c,
                                    onPress: ()=>{
                                        f({
                                            action: "BUTTON"
                                        });
                                    },
                                    "aria-label": u,
                                    children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text3, {
                                        medium: !0,
                                        forceMobileSizes: !0,
                                        truncate: !0,
                                        color: i === "CRITICAL" ? _skincontractcssmistica.vars.colors.textPrimaryInverse : _skincontractcssmistica.vars.colors.textLinkSnackbar,
                                        children: n
                                    })
                                })
                            })
                        ]
                    }),
                    g ? /* @__PURE__ */ (0, _jsxruntime.jsx)(_touchable.default, {
                        onPress: ()=>{
                            f({
                                action: "DISMISS"
                            });
                        },
                        "aria-label": p.closeButtonLabel,
                        className: _snackbarcssmistica.dismissButton[l ? "topRight" : "centered"],
                        style: {
                            display: "flex",
                            width: 32,
                            height: 32
                        },
                        children: /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                            className: _snackbarcssmistica.dismissIcon,
                            children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_iconcloseregular.default, {
                                color: _skincontractcssmistica.vars.colors.inverse,
                                size: 20
                            })
                        })
                    }) : null
                ]
            }))
        })
    });
}), ne = /*#__PURE__*/ _react.forwardRef((param, O)=>{
    let { message: m, buttonText: n, buttonAccessibilityLabel: u, duration: t, onClose: o = ()=>{}, type: i = "INFORMATIVE", withDismiss: d } = param;
    const I = n ? re : ee;
    t = Math.max(t !== null && t !== void 0 ? t : I, I);
    const p = (0, _webviewbridge.isWebViewBridgeAvailable)(), s = _react.useRef(o), a = _react.useRef(!1);
    return _react.useEffect(()=>{
        s.current = o;
    }, [
        o
    ]), _react.useEffect(()=>{
        p && !a.current && (a.current = !0, (0, _webviewbridge.nativeMessage)({
            message: m,
            // @ts-expect-error duration can be 'PERSISTENT' in new webview-bridge lib versions, and old apps will ignore it
            duration: t === 1 / 0 ? "PERSISTENT" : void 0,
            buttonText: n,
            buttonAccessibilityLabel: u,
            type: i,
            withDismiss: d
        }).then((R)=>{
            const c = R;
            c != null && c.action && Z.includes(c.action) ? s.current({
                action: c.action
            }) : s.current({
                action: "DISMISS"
            });
        }).finally(()=>{
            a.current = !1;
        }));
    }, [
        u,
        n,
        t,
        m,
        p,
        i,
        d
    ]), p ? null : /* @__PURE__ */ (0, _jsxruntime.jsx)(te, {
        ref: O,
        message: m,
        duration: t,
        buttonText: n,
        buttonAccessibilityLabel: u,
        type: i,
        onClose: s.current,
        withDismiss: d
    });
}), he = ne;
