// @telefonica/mistica/dist/snackbar-context.js
"use client";
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    SnackbarRoot: function() {
        return R;
    },
    useSnackbar: function() {
        return g;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _react = /*#__PURE__*/ _interop_require_wildcard(require("react"));
const _webviewbridge = require("@tef-novum/webview-bridge");
function _getRequireWildcardCache(nodeInterop) {
    if (typeof WeakMap !== "function") return null;
    var cacheBabelInterop = new WeakMap();
    var cacheNodeInterop = new WeakMap();
    return (_getRequireWildcardCache = function(nodeInterop) {
        return nodeInterop ? cacheNodeInterop : cacheBabelInterop;
    })(nodeInterop);
}
function _interop_require_wildcard(obj, nodeInterop) {
    if (!nodeInterop && obj && obj.__esModule) {
        return obj;
    }
    if (obj === null || typeof obj !== "object" && typeof obj !== "function") {
        return {
            default: obj
        };
    }
    var cache = _getRequireWildcardCache(nodeInterop);
    if (cache && cache.has(obj)) {
        return cache.get(obj);
    }
    var newObj = {
        __proto__: null
    };
    var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for(var key in obj){
        if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) {
            var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null;
            if (desc && (desc.get || desc.set)) {
                Object.defineProperty(newObj, key, desc);
            } else {
                newObj[key] = obj[key];
            }
        }
    }
    newObj.default = obj;
    if (cache) {
        cache.set(obj, newObj);
    }
    return newObj;
}
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
const k = /*#__PURE__*/ _react.createContext({
    snackbars: [],
    setSnackbars: ()=>{}
}), p = /*#__PURE__*/ _react.lazy(()=>Promise.resolve().then(()=>/*#__PURE__*/ _interop_require_wildcard(require(/* webpackChunkName: "snackbar" */ "./snackbar.js")))), R = (param)=>{
    let { children: l } = param;
    const [s, n] = _react.useState([]), o = _react.useRef(null), a = _react.useRef(!1), r = (0, _webviewbridge.isWebViewBridgeAvailable)();
    _react.useEffect(()=>{
        var c;
        s.length > 1 && !a.current && (a.current = !0, r ? n((i)=>i.slice(1)) : (c = o.current) == null || c.close({
            action: "CONSECUTIVE"
        }));
    }, [
        s,
        r
    ]);
    const S = (param)=>{
        let { action: c } = param;
        var i, b;
        a.current = !1, n(r && c === "CONSECUTIVE" ? (u)=>u.slice(0) : (u)=>u.slice(1)), (b = (i = s[0]).onClose) == null || b.call(i, {
            action: c
        });
    }, m = _react.useMemo(()=>({
            snackbars: s,
            setSnackbars: n
        }), [
        s
    ]), t = s[0];
    return /* @__PURE__ */ (0, _jsxruntime.jsxs)(k.Provider, {
        value: m,
        children: [
            l,
            !!t && /* @__PURE__ */ (0, _jsxruntime.jsx)(_react.Suspense, {
                fallback: null,
                children: /* @__PURE__ */ (0, _jsxruntime.jsx)(p, {
                    ref: o,
                    message: t.message,
                    buttonText: t.buttonText,
                    buttonAccessibilityLabel: t.buttonAccessibilityLabel,
                    duration: t.duration,
                    type: t.type,
                    withDismiss: t.withDismiss,
                    onClose: S
                }, t.id)
            })
        ]
    });
}, g = ()=>{
    const { snackbars: l, setSnackbars: s } = _react.useContext(k);
    return {
        openSnackbar: _react.useCallback((o)=>{
            const a = Date.now() + "-" + Math.random();
            s((r)=>[
                    ...r,
                    _object_spread_props(_object_spread({}, o), {
                        id: a
                    })
                ]);
        }, [
            s
        ]),
        snackbars: l
    };
};
