// @telefonica/mistica/dist/stacking-group.js
"use client";
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "default", {
    enumerable: true,
    get: function() {
        return B;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _react = /*#__PURE__*/ _interop_require_wildcard(require("react"));
const _themevariantcontext = require("./theme-variant-context.js");
const _skincontractcssmistica = require("./skins/skin-contract.css-mistica.js");
const _inline = /*#__PURE__*/ _interop_require_default(require("./inline.js"));
const _text = require("./text.js");
const _stackinggroupcssmistica = require("./stacking-group.css-mistica.js");
const _imagecssmistica = require("./image.css-mistica.js");
const _css = require("./utils/css.js");
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
function _getRequireWildcardCache(nodeInterop) {
    if (typeof WeakMap !== "function") return null;
    var cacheBabelInterop = new WeakMap();
    var cacheNodeInterop = new WeakMap();
    return (_getRequireWildcardCache = function(nodeInterop) {
        return nodeInterop ? cacheNodeInterop : cacheBabelInterop;
    })(nodeInterop);
}
function _interop_require_wildcard(obj, nodeInterop) {
    if (!nodeInterop && obj && obj.__esModule) {
        return obj;
    }
    if (obj === null || typeof obj !== "object" && typeof obj !== "function") {
        return {
            default: obj
        };
    }
    var cache = _getRequireWildcardCache(nodeInterop);
    if (cache && cache.has(obj)) {
        return cache.get(obj);
    }
    var newObj = {
        __proto__: null
    };
    var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for(var key in obj){
        if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) {
            var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null;
            if (desc && (desc.get || desc.set)) {
                Object.defineProperty(newObj, key, desc);
            } else {
                newObj[key] = obj[key];
            }
        }
    }
    newObj.default = obj;
    if (cache) {
        cache.set(obj, newObj);
    }
    return newObj;
}
const C = (param)=>{
    let { moreItemsStyle: s, stacked: t = !0, maxItems: o = 1 / 0, children: n } = param;
    const c = (0, _themevariantcontext.useIsInverseVariant)(), i = _react.Children.count(n), m = i - o + 1, p = t ? -8 : 8, a = s.size, d = s.type === "circle" ? "50%" : _skincontractcssmistica.vars.borderRadii.mediaSmall;
    return /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
        style: (0, _css.applyCssVars)({
            [_imagecssmistica.vars.mediaBorderRadius]: d
        }),
        children: /* @__PURE__ */ (0, _jsxruntime.jsxs)(_inline.default, {
            space: p,
            children: [
                _react.Children.toArray(n).slice(0, i > o ? o - 1 : o),
                i > o && /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                    className: _stackinggroupcssmistica.moreItems,
                    style: {
                        width: a,
                        height: a,
                        borderRadius: d,
                        backgroundColor: c ? _skincontractcssmistica.vars.colors.brandHigh : _skincontractcssmistica.vars.colors.brandLow,
                        border: t ? `1px solid ${_skincontractcssmistica.vars.colors.borderLow}` : "none"
                    },
                    children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text4, {
                        regular: !0,
                        color: c ? _skincontractcssmistica.vars.colors.textPrimaryInverse : _skincontractcssmistica.vars.colors.textBrand,
                        children: "+" + m
                    })
                })
            ]
        })
    });
}, B = C;
