// @telefonica/mistica/dist/stepper.js
"use client";
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "default", {
    enumerable: true,
    get: function() {
        return re;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _react = /*#__PURE__*/ _interop_require_wildcard(require("react"));
const _classnames = /*#__PURE__*/ _interop_require_default(require("classnames"));
const _text = require("./text.js");
const _hooks = require("./hooks.js");
const _iconsuccess = /*#__PURE__*/ _interop_require_default(require("./icons/icon-success.js"));
const _steppercssmistica = require("./stepper.css-mistica.js");
const _css = require("./utils/css.js");
const _skincontractcssmistica = require("./skins/skin-contract.css-mistica.js");
const _dom = require("./utils/dom.js");
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
function _getRequireWildcardCache(nodeInterop) {
    if (typeof WeakMap !== "function") return null;
    var cacheBabelInterop = new WeakMap();
    var cacheNodeInterop = new WeakMap();
    return (_getRequireWildcardCache = function(nodeInterop) {
        return nodeInterop ? cacheNodeInterop : cacheBabelInterop;
    })(nodeInterop);
}
function _interop_require_wildcard(obj, nodeInterop) {
    if (!nodeInterop && obj && obj.__esModule) {
        return obj;
    }
    if (obj === null || typeof obj !== "object" && typeof obj !== "function") {
        return {
            default: obj
        };
    }
    var cache = _getRequireWildcardCache(nodeInterop);
    if (cache && cache.has(obj)) {
        return cache.get(obj);
    }
    var newObj = {
        __proto__: null
    };
    var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for(var key in obj){
        if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) {
            var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null;
            if (desc && (desc.get || desc.set)) {
                Object.defineProperty(newObj, key, desc);
            } else {
                newObj[key] = obj[key];
            }
        }
    }
    newObj.default = obj;
    if (cache) {
        cache.set(obj, newObj);
    }
    return newObj;
}
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
const J = (param)=>{
    let { steps: n, currentIndex: m, "aria-label": v, dataAttributes: f } = param;
    const { textPresets: g } = (0, _hooks.useTheme)(), { isDesktopOrBigger: S } = (0, _hooks.useScreenSize)(), { height: b, ref: N } = (0, _hooks.useElementDimensions)(), x = b, [a, A] = _react.useState(Math.ceil(m)), [i, y] = _react.useState(!1);
    return _react.useEffect(()=>{
        const s = Math.ceil(m);
        a !== s && (y(s < a), A(s));
    }, [
        m,
        n,
        a
    ]), /* @__PURE__ */ (0, _jsxruntime.jsx)("div", _object_spread_props(_object_spread({
        className: _steppercssmistica.stepper,
        style: (0, _css.applyCssVars)({
            [_steppercssmistica.vars.stepperMinHeight]: (0, _css.pxToRem)(40 + x)
        })
    }, (0, _dom.getPrefixedDataAttributes)(f, "Stepper")), {
        children: n.map((s, t)=>{
            const r = t === a, C = t === n.length - 1, c = t < a, p = t === a - 1;
            return /* @__PURE__ */ (0, _jsxruntime.jsxs)(_react.Fragment, {
                children: [
                    /* @__PURE__ */ (0, _jsxruntime.jsxs)("div", {
                        className: _steppercssmistica.step,
                        role: "progressbar",
                        "aria-valuenow": r ? t + 1 : void 0,
                        "aria-valuemin": 1,
                        "aria-valuemax": n.length,
                        "aria-valuetext": s,
                        "aria-label": v,
                        children: [
                            c ? /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                                className: (0, _classnames.default)(_steppercssmistica.stepIconNumber, {
                                    [_steppercssmistica.iconAnimation]: p && !i
                                }),
                                "aria-hidden": "true",
                                children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_iconsuccess.default, {
                                    color: _skincontractcssmistica.vars.colors.controlActivated,
                                    size: "100%",
                                    skipAnimation: !p || i
                                })
                            }) : /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                                className: (0, _classnames.default)(_steppercssmistica.stepIconNumber, _steppercssmistica.number, {
                                    [_steppercssmistica.currentNumber]: r || r
                                }),
                                children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text1, {
                                    as: "span",
                                    weight: g.indicator.weight,
                                    color: r ? _skincontractcssmistica.vars.colors.textPrimaryInverse : _skincontractcssmistica.vars.colors.textSecondary,
                                    "aria-hidden": "true",
                                    children: t + 1
                                })
                            }),
                            S && /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                                className: _steppercssmistica.textContainer,
                                ref: N,
                                "aria-hidden": "true",
                                children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text2, {
                                    as: "span",
                                    regular: !0,
                                    color: c || r ? _skincontractcssmistica.vars.colors.textPrimary : _skincontractcssmistica.vars.colors.textSecondary,
                                    children: s
                                })
                            })
                        ]
                    }),
                    !C && /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                        className: _steppercssmistica.bar,
                        "aria-hidden": "true",
                        children: (c || r) && /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                            className: (0, _classnames.default)({
                                [_steppercssmistica.barFilled]: i && r || c,
                                [_steppercssmistica.barFilledAnimation]: p && !i,
                                [_steppercssmistica.barFilledReverseAnimation]: r && i
                            })
                        })
                    })
                ]
            }, t);
        })
    }));
}, re = J;
