// @telefonica/mistica/dist/switch-component.js
"use client";
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "default", {
    enumerable: true,
    get: function() {
        return U;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _react = /*#__PURE__*/ _interop_require_wildcard(require("react"));
const _helpers = require("./utils/helpers.js");
const _keys = require("./utils/keys.js");
const _formcontext = require("./form-context.js");
const _text = require("./text.js");
const _inline = /*#__PURE__*/ _interop_require_default(require("./inline.js"));
const _hooks = require("./hooks.js");
const _dom = require("./utils/dom.js");
const _switchcomponentcssmistica = require("./switch-component.css-mistica.js");
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
function _getRequireWildcardCache(nodeInterop) {
    if (typeof WeakMap !== "function") return null;
    var cacheBabelInterop = new WeakMap();
    var cacheNodeInterop = new WeakMap();
    return (_getRequireWildcardCache = function(nodeInterop) {
        return nodeInterop ? cacheNodeInterop : cacheBabelInterop;
    })(nodeInterop);
}
function _interop_require_wildcard(obj, nodeInterop) {
    if (!nodeInterop && obj && obj.__esModule) {
        return obj;
    }
    if (obj === null || typeof obj !== "object" && typeof obj !== "function") {
        return {
            default: obj
        };
    }
    var cache = _getRequireWildcardCache(nodeInterop);
    if (cache && cache.has(obj)) {
        return cache.get(obj);
    }
    var newObj = {
        __proto__: null
    };
    var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for(var key in obj){
        if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) {
            var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null;
            if (desc && (desc.get || desc.set)) {
                Object.defineProperty(newObj, key, desc);
            } else {
                newObj[key] = obj[key];
            }
        }
    }
    newObj.default = obj;
    if (cache) {
        cache.set(obj, newObj);
    }
    return newObj;
}
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
const L = (e)=>{
    const { isIos: d, isDarkMode: f } = (0, _hooks.useTheme)(), n = (0, _hooks.useAriaId)(e["aria-labelledby"]), { defaultValue: u, value: l, onChange: a, focusableRef: k, disabled: i } = (0, _formcontext.useControlProps)({
        name: e.name,
        value: e.checked,
        defaultValue: e.defaultChecked,
        onChange: e.onChange,
        disabled: e.disabled
    }), [c, v] = _react.useState(!!u), r = l !== null && l !== void 0 ? l : c, N = _react.useMemo(()=>process.env.NODE_ENV === "test" ? (t)=>a == null ? void 0 : a(t) : (0, _helpers.debounce)((t)=>{
            a == null || a(t);
        }, 300), [
        a
    ]), o = ()=>{
        l !== void 0 ? a == null || a(!l) : (v(!c), N(!c));
    }, w = (t)=>{
        t.key === _keys.SPACE && (t.preventDefault(), t.stopPropagation(), o());
    }, m = /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
        className: _switchcomponentcssmistica.checkboxVariants[d ? i ? "disabledIos" : "ios" : i ? "disabled" : "default"],
        children: /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
            className: _switchcomponentcssmistica.switchCheckboxContainerVariants[d ? "ios" : "default"],
            children: /* @__PURE__ */ (0, _jsxruntime.jsxs)("div", {
                className: _switchcomponentcssmistica.switchCheckboxLabel,
                children: [
                    /* @__PURE__ */ (0, _jsxruntime.jsx)("span", {
                        className: _switchcomponentcssmistica.barVariants[d ? r ? "checkedIos" : f ? "iosDark" : "ios" : r ? "checked" : "default"]
                    }),
                    /* @__PURE__ */ (0, _jsxruntime.jsx)("span", {
                        className: _switchcomponentcssmistica.ballVariants[d ? r ? "checkedIos" : "ios" : r ? "checked" : "default"]
                    })
                ]
            })
        })
    });
    return(// eslint-disable-next-line jsx-a11y/interactive-supports-focus
    /* @__PURE__ */ (0, _jsxruntime.jsx)("span", _object_spread_props(_object_spread({
        role: "switch",
        "aria-checked": l !== null && l !== void 0 ? l : c,
        onClick: (t)=>{
            t.stopPropagation(), i || o();
        },
        onKeyDown: i ? void 0 : w,
        tabIndex: i ? void 0 : 0,
        ref: k,
        className: i ? _switchcomponentcssmistica.containerDisabled : _switchcomponentcssmistica.container,
        "aria-disabled": i,
        "aria-label": e["aria-label"],
        "aria-labelledby": e["aria-label"] ? void 0 : n
    }, (0, _dom.getPrefixedDataAttributes)(e.dataAttributes, "Switch")), {
        children: e.render ? /* @__PURE__ */ (0, _jsxruntime.jsx)(_jsxruntime.Fragment, {
            children: e.render({
                controlElement: m,
                labelId: n,
                disabled: !!i,
                checked: l !== null && l !== void 0 ? l : c
            })
        }) : /* @__PURE__ */ (0, _jsxruntime.jsxs)(_inline.default, {
            space: 16,
            alignItems: "center",
            className: _switchcomponentcssmistica.interactiveArea,
            children: [
                m,
                e.children && /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text3, {
                    regular: !0,
                    as: "div",
                    id: n,
                    children: /* @__PURE__ */ (0, _jsxruntime.jsx)("span", {
                        className: i ? _switchcomponentcssmistica.disabled : "",
                        children: e.children
                    })
                })
            ]
        })
    })));
}, U = L;
