// @telefonica/mistica/dist/table.css-mistica.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    actionsHeaderText: function() {
        return r;
    },
    actionsTableCell: function() {
        return l;
    },
    boxed: function() {
        return t;
    },
    cellTextAlign: function() {
        return k;
    },
    collapsedRowHeaderItem: function() {
        return _;
    },
    collapsedRowTitle: function() {
        return v;
    },
    collapsedRowsInMobile: function() {
        return i;
    },
    fullWidth: function() {
        return s;
    },
    hiddenHeadersInDesktop: function() {
        return d;
    },
    hiddenHeadersInMobile: function() {
        return n;
    },
    mobileCellHeading: function() {
        return c;
    },
    rowFirstItem: function() {
        return p;
    },
    rowLastCollapsedItem: function() {
        return m;
    },
    rowLastItem: function() {
        return b;
    },
    scrollContainer: function() {
        return I;
    },
    scrollOverResponsiveLayout: function() {
        return g;
    },
    table: function() {
        return h;
    },
    topActions: function() {
        return w;
    },
    vars: function() {
        return H;
    },
    verticalAlign: function() {
        return x;
    }
});
require("./sprinkles.css.ts.vanilla.css-mistica.js");
require("./responsive-layout.css.ts.vanilla.css-mistica.js");
require("./table.css.ts.vanilla.css-mistica.js");
var r = "_107ok88f", l = "_107ok88g", t = "_107ok886", k = {
    left: "_107ok888 _107ok887",
    right: "_107ok889 _107ok887",
    center: "_107ok88a _107ok887"
}, _ = "_107ok88i", v = "_107ok88k", i = "_107ok883", s = "_107ok882", d = "_107ok885", n = "_107ok884", c = "_107ok88j", p = "_107ok88l", m = "_107ok88n", b = "_107ok88m", I = "_107ok88b", g = "_107ok88c", h = "_107ok881", w = "_107ok88h", H = {
    maxHeight: "var(--_107ok880)"
}, x = {
    top: "_107ok88d",
    middle: "_107ok88e"
};
