// @telefonica/mistica/dist/tabs.js
"use client";
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "default", {
    enumerable: true,
    get: function() {
        return ie;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _react = /*#__PURE__*/ _interop_require_wildcard(require("react"));
const _classnames = /*#__PURE__*/ _interop_require_default(require("classnames"));
const _touchable = require("./touchable.js");
const _responsivelayout = /*#__PURE__*/ _interop_require_default(require("./responsive-layout.js"));
const _hooks = require("./hooks.js");
const _text = require("./text.js");
const _platform = require("./utils/platform.js");
const _dom = require("./utils/dom.js");
const _tabscssmistica = require("./tabs.css-mistica.js");
const _inline = /*#__PURE__*/ _interop_require_default(require("./inline.js"));
const _themevariantcontext = require("./theme-variant-context.js");
const _skincontractcssmistica = require("./skins/skin-contract.css-mistica.js");
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
function _getRequireWildcardCache(nodeInterop) {
    if (typeof WeakMap !== "function") return null;
    var cacheBabelInterop = new WeakMap();
    var cacheNodeInterop = new WeakMap();
    return (_getRequireWildcardCache = function(nodeInterop) {
        return nodeInterop ? cacheNodeInterop : cacheBabelInterop;
    })(nodeInterop);
}
function _interop_require_wildcard(obj, nodeInterop) {
    if (!nodeInterop && obj && obj.__esModule) {
        return obj;
    }
    if (obj === null || typeof obj !== "object" && typeof obj !== "function") {
        return {
            default: obj
        };
    }
    var cache = _getRequireWildcardCache(nodeInterop);
    if (cache && cache.has(obj)) {
        return cache.get(obj);
    }
    var newObj = {
        __proto__: null
    };
    var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for(var key in obj){
        if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) {
            var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null;
            if (desc && (desc.get || desc.set)) {
                Object.defineProperty(newObj, key, desc);
            } else {
                newObj[key] = obj[key];
            }
        }
    }
    newObj.default = obj;
    if (cache) {
        cache.set(obj, newObj);
    }
    return newObj;
}
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
const u = (0, _platform.isRunningAcceptanceTest)() ? 0 : 300, F = (o)=>{
    switch(o){
        case 1:
        case 2:
        case 3:
            return "fullWidth";
        default:
            return "default";
    }
}, U = (param)=>{
    let { selectedIndex: o, onChange: I, tabs: b, dataAttributes: T } = param;
    const { textPresets: n } = (0, _hooks.useTheme)(), m = (0, _hooks.useAriaId)(), { ref: N } = (0, _hooks.useElementDimensions)(), h = _react.useRef(null), p = _react.useRef(null), [v, g] = _react.useState(!1), r = (0, _themevariantcontext.useIsInverseVariant)(), S = (c, d)=>{
        const s = document.querySelector(`[id='${m}'] [data-tabindex="${c}"]`), l = document.querySelector(`[id='${m}'] [data-tabindex="${d}"]`), e = h.current, a = p.current;
        s && l && e && a && (g(!0), e.style.display = "block", e.style.width = `${s.offsetWidth}px`, e.style.transform = `translate(${s.offsetLeft - a.scrollLeft}px, 0)`, Promise.resolve().then(()=>{
            e.style.transition = `transform ${u}ms, width ${u}ms`, e.style.width = `${l.offsetWidth}px`, e.style.transform = `translate(${l.offsetLeft - a.scrollLeft}px, 0)`;
        }), setTimeout(()=>{
            e.style.transition = "", e.style.display = "none", g(!1);
        }, u));
    };
    return /* @__PURE__ */ (0, _jsxruntime.jsx)("div", _object_spread_props(_object_spread({
        id: m,
        role: "tablist",
        ref: N,
        className: r ? _tabscssmistica.outerBorderInverse : _tabscssmistica.outerBorder
    }, (0, _dom.getPrefixedDataAttributes)(T, "Tabs")), {
        children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_responsivelayout.default, {
            fullWidth: !0,
            children: /* @__PURE__ */ (0, _jsxruntime.jsxs)("div", {
                className: _tabscssmistica.outer,
                children: [
                    /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                        ref: p,
                        className: _tabscssmistica.inner,
                        children: /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                            className: _tabscssmistica.tabsContainer,
                            children: b.map((param, e)=>{
                                let { text: c, trackingEvent: d, icon: s, "aria-controls": l } = param;
                                const a = e === o;
                                return /* @__PURE__ */ (0, _jsxruntime.jsx)(_touchable.BaseTouchable, {
                                    dataAttributes: {
                                        "component-name": "Tab",
                                        tabindex: e
                                    },
                                    className: (0, _classnames.default)(_tabscssmistica.tabVariants[F(b.length)], a ? v ? r ? _tabscssmistica.tabSelectionVariants.selectedAnimatingInverse : _tabscssmistica.tabSelectionVariants.selectedAnimating : r ? _tabscssmistica.tabSelectionVariants.selectedInverse : _tabscssmistica.tabSelectionVariants.selected : r ? _tabscssmistica.tabSelectionVariants.noSelectedInverse : _tabscssmistica.tabSelectionVariants.noSelected, r ? _tabscssmistica.tabHover.inverse : _tabscssmistica.tabHover.default),
                                    onPress: ()=>{
                                        !v && o !== e && (I(e), S(o, e));
                                    },
                                    trackingEvent: d,
                                    role: "tab",
                                    "aria-controls": l,
                                    "aria-selected": a ? "true" : "false",
                                    children: /* @__PURE__ */ (0, _jsxruntime.jsxs)(_inline.default, {
                                        space: s && c ? 8 : 0,
                                        alignItems: "center",
                                        children: [
                                            s && /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                                                className: _tabscssmistica.icon,
                                                children: s
                                            }),
                                            /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text, {
                                                as: "div",
                                                desktopSize: n.tabsLabel.size.desktop,
                                                mobileSize: n.tabsLabel.size.mobile,
                                                desktopLineHeight: n.tabsLabel.lineHeight.desktop,
                                                mobileLineHeight: n.tabsLabel.lineHeight.mobile,
                                                weight: n.tabsLabel.weight,
                                                color: "inherit",
                                                wordBreak: !1,
                                                textAlign: "center",
                                                hyphens: "auto",
                                                children: c
                                            })
                                        ]
                                    })
                                }, e);
                            })
                        })
                    }),
                    /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                        ref: h,
                        className: _tabscssmistica.animatedLine,
                        style: {
                            background: r ? _skincontractcssmistica.vars.colors.controlActivatedInverse : _skincontractcssmistica.vars.colors.controlActivated
                        }
                    })
                ]
            })
        })
    }));
}, ie = U;
