// @telefonica/mistica/dist/tag.js
"use client";
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "default", {
    enumerable: true,
    get: function() {
        return I;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _box = /*#__PURE__*/ _interop_require_default(require("./box.js"));
const _text = require("./text.js");
const _themevariantcontext = require("./theme-variant-context.js");
const _css = require("./utils/css.js");
const _dom = require("./utils/dom.js");
const _tagcssmistica = require("./tag.css-mistica.js");
const _sprinklescssmistica = require("./sprinkles.css-mistica.js");
const _skincontractcssmistica = require("./skins/skin-contract.css-mistica.js");
const _classnames = /*#__PURE__*/ _interop_require_default(require("classnames"));
const _hooks = require("./hooks.js");
const _badge = /*#__PURE__*/ _interop_require_default(require("./badge.js"));
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
const { colors: t } = _skincontractcssmistica.vars, V = (param)=>{
    let { Icon: e, children: i, dataAttributes: m, type: l = "promo", badge: n } = param;
    const { textPresets: d } = (0, _hooks.useTheme)(), p = (0, _themevariantcontext.useThemeVariant)(), o = n === !0 ? void 0 : n || 0;
    if (!i) return null;
    const u = {
        // [textColor, backgroundColor]
        promo: [
            t.tagTextPromo,
            t.tagBackgroundPromo
        ],
        active: [
            t.tagTextActive,
            t.tagBackgroundActive
        ],
        inactive: [
            t.tagTextInactive,
            // TODO: remove logic for alternative variant (https://jira.tid.es/browse/WEB-1803)
            p === "alternative" ? t.neutralLowAlternative : t.tagBackgroundInactive
        ],
        success: [
            t.tagTextSuccess,
            t.tagBackgroundSuccess
        ],
        warning: [
            t.tagTextWarning,
            t.tagBackgroundWarning
        ],
        error: [
            t.tagTextError,
            t.tagBackgroundError
        ]
    }, [s, f] = u[l];
    return /* @__PURE__ */ (0, _jsxruntime.jsxs)("span", _object_spread_props(_object_spread({}, (0, _dom.getPrefixedDataAttributes)(m, "Tag")), {
        className: (0, _classnames.default)(_tagcssmistica.tag, (0, _sprinklescssmistica.sprinkles)({
            paddingLeft: e ? 8 : 12,
            paddingRight: o !== 0 ? 8 : 12,
            background: f
        })),
        children: [
            e && /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                paddingRight: 4,
                children: /* @__PURE__ */ (0, _jsxruntime.jsx)(e, {
                    color: s,
                    size: (0, _css.pxToRem)(16),
                    className: (0, _sprinklescssmistica.sprinkles)({
                        display: "block"
                    })
                })
            }),
            /* @__PURE__ */ (0, _jsxruntime.jsxs)(_themevariantcontext.ThemeVariant, {
                isInverse: !1,
                children: [
                    /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text, {
                        color: s,
                        size: 14,
                        lineHeight: 20,
                        weight: d.indicator.weight,
                        truncate: !0,
                        children: i
                    }),
                    o !== 0 && /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                        paddingLeft: 4,
                        className: (0, _sprinklescssmistica.sprinkles)({
                            display: "inline-flex"
                        }),
                        children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_badge.default, {
                            value: o
                        })
                    })
                ]
            })
        ]
    }));
}, I = V;
