// @telefonica/mistica/dist/text-field-components.js
"use client";
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    FieldContainer: function() {
        return M;
    },
    HelperText: function() {
        return K;
    },
    Label: function() {
        return J;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _react = /*#__PURE__*/ _interop_require_wildcard(require("react"));
const _classnames = /*#__PURE__*/ _interop_require_default(require("classnames"));
const _themevariantcontext = require("./theme-variant-context.js");
const _hooks = require("./hooks.js");
const _text = require("./text.js");
const _textfieldcomponentscssmistica = require("./text-field-components.css-mistica.js");
const _sprinklescssmistica = require("./sprinkles.css-mistica.js");
const _skincontractcssmistica = require("./skins/skin-contract.css-mistica.js");
const _dom = require("./utils/dom.js");
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
function _getRequireWildcardCache(nodeInterop) {
    if (typeof WeakMap !== "function") return null;
    var cacheBabelInterop = new WeakMap();
    var cacheNodeInterop = new WeakMap();
    return (_getRequireWildcardCache = function(nodeInterop) {
        return nodeInterop ? cacheNodeInterop : cacheBabelInterop;
    })(nodeInterop);
}
function _interop_require_wildcard(obj, nodeInterop) {
    if (!nodeInterop && obj && obj.__esModule) {
        return obj;
    }
    if (obj === null || typeof obj !== "object" && typeof obj !== "function") {
        return {
            default: obj
        };
    }
    var cache = _getRequireWildcardCache(nodeInterop);
    if (cache && cache.has(obj)) {
        return cache.get(obj);
    }
    var newObj = {
        __proto__: null
    };
    var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for(var key in obj){
        if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) {
            var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null;
            if (desc && (desc.get || desc.set)) {
                Object.defineProperty(newObj, key, desc);
            } else {
                newObj[key] = obj[key];
            }
        }
    }
    newObj.default = obj;
    if (cache) {
        cache.set(obj, newObj);
    }
    return newObj;
}
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
const J = (param)=>{
    let { shrinkLabel: o, forId: s, inputState: r, error: l, children: i, style: n, optional: a } = param;
    const f = o || r === "focused" || r === "filled", [u, h] = _react.useState("initial"), { texts: d } = (0, _hooks.useTheme)();
    _react.useEffect(()=>{
        const g = setTimeout(()=>{
            process.env.NODE_ENV !== "test" && h("transform 150ms, width 150ms");
        });
        return ()=>{
            clearTimeout(g);
        };
    }, []);
    let x = _skincontractcssmistica.vars.colors.textSecondary;
    return l && r !== "default" ? x = _skincontractcssmistica.vars.colors.textError : r === "focused" && (x = _skincontractcssmistica.vars.colors.textActivated), /* @__PURE__ */ (0, _jsxruntime.jsxs)("label", {
        className: (0, _classnames.default)(_textfieldcomponentscssmistica.labelContainer, {
            [_textfieldcomponentscssmistica.shrinked]: f
        }, (0, _sprinklescssmistica.sprinkles)({
            color: x
        })),
        htmlFor: s,
        style: _object_spread_props(_object_spread({}, n), {
            transition: u
        }),
        children: [
            /* @__PURE__ */ (0, _jsxruntime.jsx)("span", {
                className: _textfieldcomponentscssmistica.labelText,
                children: i
            }),
            a ? /* @__PURE__ */ (0, _jsxruntime.jsxs)("span", {
                children: [
                    "\xa0(",
                    d.formFieldOptionalLabelSuffix,
                    ")"
                ]
            }) : null
        ]
    });
}, K = (param)=>{
    let { leftText: o, rightText: s, error: r, id: l } = param;
    const i = (0, _themevariantcontext.useIsInverseVariant)(), n = i ? _skincontractcssmistica.vars.colors.textPrimaryInverse : r ? _skincontractcssmistica.vars.colors.textError : _skincontractcssmistica.vars.colors.textSecondary, a = i ? _skincontractcssmistica.vars.colors.textPrimaryInverse : _skincontractcssmistica.vars.colors.textSecondary;
    return /* @__PURE__ */ (0, _jsxruntime.jsxs)(_jsxruntime.Fragment, {
        children: [
            o && /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                className: (0, _classnames.default)(_textfieldcomponentscssmistica.helperText, _textfieldcomponentscssmistica.leftText),
                children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text1, {
                    color: n,
                    regular: !0,
                    as: "p",
                    id: l,
                    children: o
                })
            }),
            s && /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                className: (0, _classnames.default)(_textfieldcomponentscssmistica.helperText, _textfieldcomponentscssmistica.rightText),
                children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text1, {
                    color: a,
                    regular: !0,
                    as: "p",
                    children: s
                })
            })
        ]
    });
}, M = (param)=>{
    let { multiline: o, disabled: s, children: r, helperText: l, className: i, fieldRef: n, fullWidth: a, readOnly: f, dataAttributes: u } = param;
    return /* @__PURE__ */ (0, _jsxruntime.jsxs)("div", _object_spread_props(_object_spread({
        className: (0, _classnames.default)(_textfieldcomponentscssmistica.fieldContainer, a ? _textfieldcomponentscssmistica.fullWidth : _textfieldcomponentscssmistica.normalWidth, {
            [_textfieldcomponentscssmistica.disabled]: s
        }),
        onClick: (h)=>{
            var d;
            (d = h.currentTarget.querySelector(o ? "textarea" : "input")) == null || d.focus();
        }
    }, (0, _dom.getPrefixedDataAttributes)(u)), {
        children: [
            /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                className: (0, _classnames.default)(_textfieldcomponentscssmistica.field, (0, _sprinklescssmistica.sprinkles)({
                    background: f ? _skincontractcssmistica.vars.colors.neutralLow : _skincontractcssmistica.vars.colors.backgroundContainer
                }), i),
                ref: n,
                children: r
            }),
            l && /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                className: _textfieldcomponentscssmistica.helperContainer,
                children: l
            })
        ]
    }));
};
