// @telefonica/mistica/dist/text.js
"use client";
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    Text: function() {
        return o;
    },
    Text1: function() {
        return He;
    },
    Text10: function() {
        return he;
    },
    Text2: function() {
        return ke;
    },
    Text3: function() {
        return fe;
    },
    Text4: function() {
        return Se;
    },
    Text5: function() {
        return Te;
    },
    Text6: function() {
        return be;
    },
    Text7: function() {
        return de;
    },
    Text8: function() {
        return ue;
    },
    Text9: function() {
        return ge;
    },
    default: function() {
        return Le;
    },
    mapToWeight: function() {
        return $;
    },
    textProps: function() {
        return x;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _react = /*#__PURE__*/ _interop_require_wildcard(require("react"));
const _classnames = /*#__PURE__*/ _interop_require_default(require("classnames"));
const _themevariantcontext = require("./theme-variant-context.js");
const _css = require("./utils/css.js");
const _dom = require("./utils/dom.js");
const _hooks = require("./hooks.js");
const _skincontractcssmistica = require("./skins/skin-contract.css-mistica.js");
const _textcssmistica = require("./text.css-mistica.js");
const _constants = require("./skins/constants.js");
const _screenreaderonly = /*#__PURE__*/ _interop_require_default(require("./screen-reader-only.js"));
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
function _getRequireWildcardCache(nodeInterop) {
    if (typeof WeakMap !== "function") return null;
    var cacheBabelInterop = new WeakMap();
    var cacheNodeInterop = new WeakMap();
    return (_getRequireWildcardCache = function(nodeInterop) {
        return nodeInterop ? cacheNodeInterop : cacheBabelInterop;
    })(nodeInterop);
}
function _interop_require_wildcard(obj, nodeInterop) {
    if (!nodeInterop && obj && obj.__esModule) {
        return obj;
    }
    if (obj === null || typeof obj !== "object" && typeof obj !== "function") {
        return {
            default: obj
        };
    }
    var cache = _getRequireWildcardCache(nodeInterop);
    if (cache && cache.has(obj)) {
        return cache.get(obj);
    }
    var newObj = {
        __proto__: null
    };
    var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for(var key in obj){
        if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) {
            var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null;
            if (desc && (desc.get || desc.set)) {
                Object.defineProperty(newObj, key, desc);
            } else {
                newObj[key] = obj[key];
            }
        }
    }
    newObj.default = obj;
    if (cache) {
        cache.set(obj, newObj);
    }
    return newObj;
}
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
function _object_without_properties(source, excluded) {
    if (source == null) return {};
    var target = _object_without_properties_loose(source, excluded);
    var key, i;
    if (Object.getOwnPropertySymbols) {
        var sourceSymbolKeys = Object.getOwnPropertySymbols(source);
        for(i = 0; i < sourceSymbolKeys.length; i++){
            key = sourceSymbolKeys[i];
            if (excluded.indexOf(key) >= 0) continue;
            if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
            target[key] = source[key];
        }
    }
    return target;
}
function _object_without_properties_loose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;
    for(i = 0; i < sourceKeys.length; i++){
        key = sourceKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        target[key] = source[key];
    }
    return target;
}
const $ = {
    light: 300,
    regular: 400,
    medium: 500,
    bold: 700
}, ee = (e)=>e === !0 ? 1 : e || "initial", d = "Ħ", te = /* @__PURE__ */ (0, _jsxruntime.jsxs)(_jsxruntime.Fragment, {
    children: [
        /* @__PURE__ */ (0, _jsxruntime.jsx)("span", {
            "aria-hidden": !0,
            children: d
        }),
        /* @__PURE__ */ (0, _jsxruntime.jsx)(_screenreaderonly.default, {
            children: /* @__PURE__ */ (0, _jsxruntime.jsx)("span", {
                children: "Vivo"
            })
        })
    ]
}), ie = (e)=>_react.Children.map(e, (t)=>typeof t != "string" || !t.includes(d) ? t : /* @__PURE__ */ (0, _jsxruntime.jsx)(_jsxruntime.Fragment, {
            children: t.split(d).map((i, r)=>/* @__PURE__ */ (0, _jsxruntime.jsxs)(_react.Fragment, {
                    children: [
                        r > 0 && te,
                        i
                    ]
                }, r))
        })), o = (param)=>{
    let { weight: e, color: t = _skincontractcssmistica.vars.colors.textPrimary, decoration: i, truncate: r, transform: g, wordBreak: b = !0, hyphens: T = "auto", as: S = "span", children: l, size: f, mobileSize: k = f, desktopSize: H = f, lineHeight: L, mobileLineHeight: w = L, desktopLineHeight: z = L, letterSpacing: P, textAlign: V, textShadow: y, id: C, role: W, "aria-level": R, dataAttributes: N } = param;
    const { skinName: E } = (0, _hooks.useTheme)(), F = (0, _themevariantcontext.useIsInverseVariant)(), O = ee(r), _ = {
        [_skincontractcssmistica.vars.colors.textPrimary]: _skincontractcssmistica.vars.colors.textPrimaryInverse,
        [_skincontractcssmistica.vars.colors.textSecondary]: _skincontractcssmistica.vars.colors.textSecondaryInverse,
        [_skincontractcssmistica.vars.colors.textLink]: _skincontractcssmistica.vars.colors.textLinkInverse,
        [_skincontractcssmistica.vars.colors.textError]: _skincontractcssmistica.vars.colors.textErrorInverse
    };
    if (!l && l !== 0) return null;
    const j = (0, _classnames.default)(_textcssmistica.text, b ? _textcssmistica.withWordBreak : _textcssmistica.withoutWordBreak, {
        [_textcssmistica.truncate]: r,
        [_textcssmistica.truncateToOneLine]: r === 1 || r === !0
    }), D = (0, _css.applyCssVars)({
        [_textcssmistica.vars.mobileSize]: k ? (0, _css.pxToRem)(k) : "inherit",
        [_textcssmistica.vars.mobileLineHeight]: w ? (0, _css.pxToRem)(w) : "inherit",
        [_textcssmistica.vars.desktopSize]: H ? (0, _css.pxToRem)(H) : "inherit",
        [_textcssmistica.vars.desktopLineHeight]: z ? (0, _css.pxToRem)(z) : "inherit"
    }), B = r ? (0, _css.applyCssVars)({
        [_textcssmistica.vars.lineClamp]: String(O)
    }) : {};
    var __t;
    return /*#__PURE__*/ _react.createElement(S, _object_spread_props(_object_spread({
        className: j,
        id: C,
        role: W,
        "aria-level": R
    }, (0, _dom.getPrefixedDataAttributes)(N, "Text")), {
        style: _object_spread_props(_object_spread({}, D, B), {
            hyphens: T,
            WebkitHyphens: T,
            letterSpacing: P,
            fontWeight: e ? $[e] : "inherit",
            textTransform: g || "inherit",
            textDecoration: i !== null && i !== void 0 ? i : "inherit",
            overflowWrap: b ? "anywhere" : "inherit",
            color: F ? (__t = _[t]) !== null && __t !== void 0 ? __t : t : t,
            textAlign: V,
            textShadow: y,
            // When rendering as <pre/>, spaces are preserved and we don't want to remove them
            whiteSpace: S === "pre" ? void 0 : "pre-line"
        })
    }), E === _constants.VIVO_NEW_SKIN ? ie(l) : l);
}, h = (e)=>{
    if (e.light) return "light";
    if (e.regular) return "regular";
    if (e.medium) return "medium";
    if (e.weight) return e.weight;
}, s = (param)=>{
    let { forceMobileSizes: e, mobileSize: t, mobileLineHeight: i, desktopSize: r, desktopLineHeight: g } = param;
    return e ? {
        size: t,
        lineHeight: i
    } : {
        mobileSize: t,
        mobileLineHeight: i,
        desktopSize: r,
        desktopLineHeight: g
    };
}, x = {
    text1: {
        mobileSize: 12,
        mobileLineHeight: "16px",
        desktopSize: 14,
        desktopLineHeight: "20px"
    },
    text2: {
        mobileSize: 14,
        mobileLineHeight: "20px",
        desktopSize: 16,
        desktopLineHeight: "24px"
    },
    text3: {
        mobileSize: 16,
        mobileLineHeight: "24px",
        desktopSize: 18,
        desktopLineHeight: "24px"
    },
    text4: {
        mobileSize: 18,
        mobileLineHeight: "24px",
        desktopSize: 20,
        desktopLineHeight: "28px"
    },
    text5: {
        mobileSize: 20,
        mobileLineHeight: "24px",
        desktopSize: 28,
        desktopLineHeight: "32px"
    },
    text6: {
        mobileSize: 24,
        mobileLineHeight: "32px",
        desktopSize: 32,
        desktopLineHeight: "40px"
    },
    text7: {
        mobileSize: 28,
        mobileLineHeight: "32px",
        desktopSize: 40,
        desktopLineHeight: "48px"
    },
    text8: {
        mobileSize: 32,
        mobileLineHeight: "40px",
        desktopSize: 48,
        desktopLineHeight: "56px"
    },
    text9: {
        mobileSize: 40,
        mobileLineHeight: "48px",
        desktopSize: 56,
        desktopLineHeight: "64px"
    },
    text10: {
        mobileSize: 48,
        mobileLineHeight: "56px",
        desktopSize: 64,
        desktopLineHeight: "72px"
    }
}, he = (_param)=>{
    var { dataAttributes: e, forceMobileSizes: t } = _param, i = _object_without_properties(_param, [
        "dataAttributes",
        "forceMobileSizes"
    ]);
    const { textPresets: r } = (0, _hooks.useTheme)();
    return /* @__PURE__ */ (0, _jsxruntime.jsx)(o, _object_spread(_object_spread_props(_object_spread({}, s(_object_spread({
        forceMobileSizes: t
    }, x.text10))), {
        weight: r.text10.weight,
        dataAttributes: _object_spread({
            "component-name": "Text10"
        }, e)
    }), i));
}, ge = (_param)=>{
    var { dataAttributes: e, forceMobileSizes: t } = _param, i = _object_without_properties(_param, [
        "dataAttributes",
        "forceMobileSizes"
    ]);
    const { textPresets: r } = (0, _hooks.useTheme)();
    return /* @__PURE__ */ (0, _jsxruntime.jsx)(o, _object_spread(_object_spread_props(_object_spread({}, s(_object_spread({
        forceMobileSizes: t
    }, x.text9))), {
        weight: r.text9.weight,
        dataAttributes: _object_spread({
            "component-name": "Text9"
        }, e)
    }), i));
}, ue = (_param)=>{
    var { dataAttributes: e, forceMobileSizes: t } = _param, i = _object_without_properties(_param, [
        "dataAttributes",
        "forceMobileSizes"
    ]);
    const { textPresets: r } = (0, _hooks.useTheme)();
    return /* @__PURE__ */ (0, _jsxruntime.jsx)(o, _object_spread(_object_spread_props(_object_spread({}, s(_object_spread({
        forceMobileSizes: t
    }, x.text8))), {
        weight: r.text8.weight,
        dataAttributes: _object_spread({
            "component-name": "Text8"
        }, e)
    }), i));
}, de = (_param)=>{
    var { dataAttributes: e, forceMobileSizes: t } = _param, i = _object_without_properties(_param, [
        "dataAttributes",
        "forceMobileSizes"
    ]);
    const { textPresets: r } = (0, _hooks.useTheme)();
    return /* @__PURE__ */ (0, _jsxruntime.jsx)(o, _object_spread(_object_spread_props(_object_spread({}, s(_object_spread({
        forceMobileSizes: t
    }, x.text7))), {
        weight: r.text7.weight,
        dataAttributes: _object_spread({
            "component-name": "Text7"
        }, e)
    }), i));
}, be = (_param)=>{
    var { dataAttributes: e, forceMobileSizes: t } = _param, i = _object_without_properties(_param, [
        "dataAttributes",
        "forceMobileSizes"
    ]);
    const { textPresets: r } = (0, _hooks.useTheme)();
    return /* @__PURE__ */ (0, _jsxruntime.jsx)(o, _object_spread(_object_spread_props(_object_spread({}, s(_object_spread({
        forceMobileSizes: t
    }, x.text6))), {
        weight: r.text6.weight,
        dataAttributes: _object_spread({
            "component-name": "Text6"
        }, e)
    }), i));
}, Te = (_param)=>{
    var { dataAttributes: e, forceMobileSizes: t } = _param, i = _object_without_properties(_param, [
        "dataAttributes",
        "forceMobileSizes"
    ]);
    const { textPresets: r } = (0, _hooks.useTheme)();
    return /* @__PURE__ */ (0, _jsxruntime.jsx)(o, _object_spread(_object_spread_props(_object_spread({}, s(_object_spread({
        forceMobileSizes: t
    }, x.text5))), {
        weight: r.text5.weight,
        dataAttributes: _object_spread({
            "component-name": "Text5"
        }, e)
    }), i));
}, Se = (_param)=>{
    var { dataAttributes: e, forceMobileSizes: t } = _param, i = _object_without_properties(_param, [
        "dataAttributes",
        "forceMobileSizes"
    ]);
    return /* @__PURE__ */ (0, _jsxruntime.jsx)(o, _object_spread(_object_spread_props(_object_spread({}, s(_object_spread({
        forceMobileSizes: t
    }, x.text4))), {
        weight: h(i),
        dataAttributes: _object_spread({
            "component-name": "Text4"
        }, e)
    }), i));
}, fe = (_param)=>{
    var { dataAttributes: e, forceMobileSizes: t } = _param, i = _object_without_properties(_param, [
        "dataAttributes",
        "forceMobileSizes"
    ]);
    return /* @__PURE__ */ (0, _jsxruntime.jsx)(o, _object_spread(_object_spread_props(_object_spread({}, s(_object_spread({
        forceMobileSizes: t
    }, x.text3))), {
        weight: h(i),
        dataAttributes: _object_spread({
            "component-name": "Text3"
        }, e)
    }), i));
}, ke = (_param)=>{
    var { dataAttributes: e, forceMobileSizes: t } = _param, i = _object_without_properties(_param, [
        "dataAttributes",
        "forceMobileSizes"
    ]);
    return /* @__PURE__ */ (0, _jsxruntime.jsx)(o, _object_spread(_object_spread_props(_object_spread({}, s(_object_spread({
        forceMobileSizes: t
    }, x.text2))), {
        weight: h(i),
        dataAttributes: _object_spread({
            "component-name": "Text2"
        }, e)
    }), i));
}, He = (_param)=>{
    var { dataAttributes: e, forceMobileSizes: t } = _param, i = _object_without_properties(_param, [
        "dataAttributes",
        "forceMobileSizes"
    ]);
    return /* @__PURE__ */ (0, _jsxruntime.jsx)(o, _object_spread(_object_spread_props(_object_spread({}, s(_object_spread({
        forceMobileSizes: t
    }, x.text1))), {
        weight: h(i),
        dataAttributes: _object_spread({
            "component-name": "Text1"
        }, e)
    }), i));
}, Le = o;
