// @telefonica/mistica/dist/text-field-base.js
"use client";
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    FieldEndIcon: function() {
        return Ze;
    },
    TextFieldBase: function() {
        return W;
    },
    TextFieldBaseAutosuggest: function() {
        return Ve;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _react = /*#__PURE__*/ _interop_require_wildcard(require("react"));
const _textfieldcomponents = require("./text-field-components.js");
const _textfieldcomponentscssmistica = require("./text-field-components.css-mistica.js");
const _text = require("./text.js");
const _platform = require("./utils/platform.js");
const _hooks = require("./hooks.js");
const _classnames = /*#__PURE__*/ _interop_require_default(require("classnames"));
const _common = require("./utils/common.js");
const _textfieldbasecssmistica = require("./text-field-base.css-mistica.js");
const _skincontractcssmistica = require("./skins/skin-contract.css-mistica.js");
const _iconbutton = require("./icon-button.js");
const _themevariantcontext = require("./theme-variant-context.js");
const _iconbuttoncssmistica = require("./icon-button.css-mistica.js");
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
function _getRequireWildcardCache(nodeInterop) {
    if (typeof WeakMap !== "function") return null;
    var cacheBabelInterop = new WeakMap();
    var cacheNodeInterop = new WeakMap();
    return (_getRequireWildcardCache = function(nodeInterop) {
        return nodeInterop ? cacheNodeInterop : cacheBabelInterop;
    })(nodeInterop);
}
function _interop_require_wildcard(obj, nodeInterop) {
    if (!nodeInterop && obj && obj.__esModule) {
        return obj;
    }
    if (obj === null || typeof obj !== "object" && typeof obj !== "function") {
        return {
            default: obj
        };
    }
    var cache = _getRequireWildcardCache(nodeInterop);
    if (cache && cache.has(obj)) {
        return cache.get(obj);
    }
    var newObj = {
        __proto__: null
    };
    var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for(var key in obj){
        if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) {
            var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null;
            if (desc && (desc.get || desc.set)) {
                Object.defineProperty(newObj, key, desc);
            } else {
                newObj[key] = obj[key];
            }
        }
    }
    newObj.default = obj;
    if (cache) {
        cache.set(obj, newObj);
    }
    return newObj;
}
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
function _object_without_properties(source, excluded) {
    if (source == null) return {};
    var target = _object_without_properties_loose(source, excluded);
    var key, i;
    if (Object.getOwnPropertySymbols) {
        var sourceSymbolKeys = Object.getOwnPropertySymbols(source);
        for(i = 0; i < sourceSymbolKeys.length; i++){
            key = sourceSymbolKeys[i];
            if (excluded.indexOf(key) >= 0) continue;
            if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
            target[key] = source[key];
        }
    }
    return target;
}
function _object_without_properties_loose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;
    for(i = 0; i < sourceKeys.length; i++){
        key = sourceKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        target[key] = source[key];
    }
    return target;
}
const Be = (o, f)=>{
    if (!f) return !0;
    const i = document.createElement("input");
    return i.type = f, i.value = o || "", i.value !== "";
}, Ze = (param)=>{
    let { hasBackgroundColor: o = !0, onPress: f, onChange: i, disabled: m, Icon: g, checkedProps: r, uncheckedProps: a, "aria-label": h } = param;
    return /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
        className: _textfieldbasecssmistica.fieldEndIconContainer,
        children: r ? /* @__PURE__ */ (0, _jsxruntime.jsx)(_iconbutton.InternalToggleIconButton, {
            checkedProps: _object_spread_props(_object_spread({}, r), {
                "aria-label": r["aria-label"] || ""
            }),
            uncheckedProps: _object_spread_props(_object_spread({}, a), {
                "aria-label": a["aria-label"] || ""
            }),
            onChange: i,
            hasOverlay: o,
            disabled: m
        }) : /* @__PURE__ */ (0, _jsxruntime.jsx)(_iconbutton.InternalIconButton, {
            Icon: g,
            disabled: m,
            "aria-label": h || "",
            onPress: f,
            hasOverlay: o
        })
    });
}, W = /*#__PURE__*/ _react.forwardRef((_param, U)=>{
    var { error: o, helperText: f, label: i, inputProps: m, inputRef: g, defaultValue: r, value: a, onFocus: h, onBlur: x, inputComponent: p, prefix: b, startIcon: t, endIcon: u, endIconOverlay: S, shrinkLabel: N, multiline: R = !1, focus: y, fieldRef: H, maxLength: C, id: K, autoComplete: M, fullWidth: J, dataAttributes: Q } = _param, l = _object_without_properties(_param, [
        "error",
        "helperText",
        "label",
        "inputProps",
        "inputRef",
        "defaultValue",
        "value",
        "onFocus",
        "onBlur",
        "inputComponent",
        "prefix",
        "startIcon",
        "endIcon",
        "endIconOverlay",
        "shrinkLabel",
        "multiline",
        "focus",
        "fieldRef",
        "maxLength",
        "id",
        "autoComplete",
        "fullWidth",
        "dataAttributes"
    ]);
    var _ref;
    const w = (0, _hooks.useAriaId)(K), P = (0, _hooks.useAriaId)(), [s, c] = _react.useState(r != null && r.length || a != null && a.length ? "filled" : "default"), { isTabletOrSmaller: X } = (0, _hooks.useScreenSize)(), [O, B] = _react.useState((_ref = r == null ? void 0 : r.length) !== null && _ref !== void 0 ? _ref : 0), A = !!i || !l.required, E = l.type === "date" || l.type === "datetime-local" || l.type === "month", D = _react.useRef(void 0);
    (0, _hooks.useIsomorphicLayoutEffect)(()=>{
        const e = E && !Be(a, l.type) ? "" : a;
        D.current !== a && E && !(e != null && e.length) && s === "filled" && y === void 0 && c("default"), D.current = a, s !== "focused" && e != null && e.length && (B(e.length), c("filled")), y && c("focused"), y === !1 && !(e != null && e.length) && c("default"), y === !1 && e != null && e.length && c("filled");
    }, [
        s,
        a,
        y,
        E,
        l.type
    ]), _react.useEffect(()=>{
        l.autoFocus && c("focused");
    }, [
        l.autoFocus
    ]);
    const Y = R ? "textarea" : "input", Z = p ? {
        inputRef: g
    } : {
        ref: (0, _common.combineRefs)(U, g)
    }, I = _object_spread(_object_spread_props(_object_spread({}, l), {
        maxLength: C,
        autoComplete: M
    }), m), $ = `calc(${_iconbuttoncssmistica.iconSize.default} + ${_textfieldbasecssmistica.fieldElementsGap}px)`, q = `calc(${_textfieldbasecssmistica.iconButtonSize} + ${_textfieldbasecssmistica.fieldEndIconGap}px)`, V = N || s === "focused" || s === "filled" ? X ? _textfieldcomponentscssmistica.LABEL_SCALE_MOBILE : _textfieldcomponentscssmistica.LABEL_SCALE_DESKTOP : 1, ee = {
        left: `calc(${_textfieldbasecssmistica.fieldLeftPadding}px + ${t ? $ : "0px"})`,
        // shrinking means applying a scale transformation, so width will be proportionally reduced.
        // Let's keep the original width.
        width: `calc((100% - ${_textfieldbasecssmistica.fieldLeftPadding}px - ${t ? $ : "0px"} - ${u || S ? q : `${_textfieldbasecssmistica.fieldRightPadding}px`}) / ${V})`
    }, te = l.type === "password" && O > 0 ? "Lucida Grande, Arial, sans-serif" : "inherit";
    return /* @__PURE__ */ (0, _jsxruntime.jsx)(_textfieldcomponents.FieldContainer, {
        disabled: l.disabled,
        helperText: /* @__PURE__ */ (0, _jsxruntime.jsx)(_textfieldcomponents.HelperText, {
            error: o,
            leftText: f,
            id: P,
            rightText: R && C ? `${O}/${C}` : void 0
        }),
        multiline: R,
        fullWidth: J,
        fieldRef: H,
        readOnly: l.readOnly,
        dataAttributes: Q,
        children: /* @__PURE__ */ (0, _jsxruntime.jsxs)(_themevariantcontext.ThemeVariant, {
            isInverse: !1,
            children: [
                t && /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                    className: _textfieldbasecssmistica.startIcon,
                    children: t
                }),
                b && /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                    className: (0, _classnames.default)(_textfieldbasecssmistica.prefix, A ? _textfieldbasecssmistica.inputWithLabel : _textfieldbasecssmistica.inputWithoutLabel),
                    style: {
                        opacity: s === "default" ? 0 : 1
                    },
                    children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text3, {
                        color: _skincontractcssmistica.vars.colors.textSecondary,
                        regular: !0,
                        wordBreak: !1,
                        children: b
                    })
                }),
                /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                    className: _textfieldbasecssmistica.fullWidth,
                    children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text3, {
                        as: "div",
                        regular: !0,
                        children: /*#__PURE__*/ _react.createElement(p || Y, _object_spread(_object_spread_props(_object_spread({}, Z, I), {
                            id: w,
                            style: _object_spread_props(_object_spread({
                                paddingRight: u ? 0 : S ? q : _textfieldbasecssmistica.fieldRightPadding,
                                paddingLeft: b ? 0 : t ? `calc(${$} + ${_textfieldbasecssmistica.fieldLeftPadding}px)` : _textfieldbasecssmistica.fieldLeftPadding
                            }, I.style), {
                                fontFamily: te
                            }),
                            className: R ? (0, _classnames.default)(_textfieldbasecssmistica.textArea, A ? _textfieldbasecssmistica.textAreaWithLabel : _textfieldbasecssmistica.textAreaWithoutLabel) : (0, _classnames.default)(_textfieldbasecssmistica.input, A ? _textfieldbasecssmistica.inputWithLabel : _textfieldbasecssmistica.inputWithoutLabel, {
                                [_textfieldbasecssmistica.inputFirefoxStyles]: (0, _platform.isFirefox)(),
                                // Hide webkit placeholder when label is not shrinked and value is empty
                                [_textfieldbasecssmistica.hiddenDatePlaceholder]: E && s === "default",
                                // Force height of input when value is empty to avoid field from having height 0 in iOS
                                [_textfieldbasecssmistica.emptyDateValue]: E && s !== "filled"
                            }),
                            onFocus: (e)=>{
                                c("focused"), h == null || h(e);
                            },
                            onBlur: (e)=>{
                                e.target.value.length > 0 ? c("filled") : c("default"), x == null || x(e);
                            },
                            onChange: (e)=>{
                                var z;
                                C === void 0 || e.target.value.length <= C ? (B(e.target.value.length), e.target.value.length > 0 && s !== "focused" && c(e.target.value.length > 0 ? "filled" : "default"), (z = I.onChange) == null || z.call(I, e)) : (e.stopPropagation(), e.preventDefault());
                            },
                            defaultValue: r,
                            value: a
                        }), o && {
                            "aria-invalid": !0
                        }, f && {
                            "aria-describedby": P
                        }))
                    })
                }),
                i && /* @__PURE__ */ (0, _jsxruntime.jsx)(_textfieldcomponents.Label, {
                    style: ee,
                    error: o,
                    forId: w,
                    inputState: s,
                    shrinkLabel: N,
                    optional: !l.required,
                    children: i
                }),
                u && /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                    className: _textfieldbasecssmistica.endIconContainer,
                    children: u
                }),
                S
            ]
        })
    });
}), De = /*#__PURE__*/ _react.lazy(()=>Promise.resolve().then(()=>/*#__PURE__*/ _interop_require_wildcard(require(/* webpackChunkName: "react-autosuggest" */ "react-autosuggest")))), Ve = /*#__PURE__*/ _react.forwardRef((_param, m)=>{
    var { getSuggestions: o, id: f } = _param, i = _object_without_properties(_param, [
        "getSuggestions",
        "id"
    ]);
    const [g, r] = _react.useState([]), a = _react.useRef(null), { platformOverrides: h, texts: x } = (0, _hooks.useTheme)(), p = (0, _hooks.useAriaId)(f), b = (0, _hooks.useAriaId)();
    if (o && (i.value === void 0 || i.defaultValue !== void 0)) throw Error("Fields with suggestions must be used in controlled mode");
    return o ? /* @__PURE__ */ (0, _jsxruntime.jsx)(_react.Suspense, {
        fallback: /* @__PURE__ */ (0, _jsxruntime.jsx)(W, _object_spread_props(_object_spread({}, i), {
            label: (0, _platform.isRunningAcceptanceTest)(h) ? "" : i.label,
            autoComplete: "off",
            ref: m,
            id: p
        })),
        children: /* @__PURE__ */ (0, _jsxruntime.jsx)(De, {
            id: b,
            inputProps: _object_spread_props(_object_spread({}, i), {
                id: p,
                autoComplete: "off",
                // @ts-expect-error Autosuggest expects slightly different types
                onChange: (t, param)=>{
                    let { newValue: u } = param;
                    var S;
                    t.target = _object_spread_props(_object_spread({}, t.target), {
                        value: u
                    }), t.currentTarget = _object_spread_props(_object_spread({}, t.currentTarget), {
                        value: u
                    }), (S = i.onChange) == null || S.call(i, t);
                }
            }),
            renderInputComponent: (t)=>/* @__PURE__ */ (0, _jsxruntime.jsx)(W, _object_spread_props(_object_spread({}, t), {
                    inputRef: (0, _common.combineRefs)(a, i.inputRef, m)
                })),
            suggestions: g,
            onSuggestionsFetchRequested: (param)=>{
                let { value: t } = param;
                return r(o(t));
            },
            onSuggestionsClearRequested: ()=>r([]),
            getSuggestionValue: (t)=>t,
            renderSuggestion: (t, param)=>{
                let { isHighlighted: u } = param;
                return /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                    role: "menuitem",
                    className: (0, _classnames.default)(_textfieldbasecssmistica.menuItem, {
                        [_textfieldbasecssmistica.menuItemSelected]: u
                    }),
                    children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text3, {
                        regular: !0,
                        children: t
                    })
                });
            },
            renderSuggestionsContainer: (t)=>/* @__PURE__ */ (0, _jsxruntime.jsx)("div", _object_spread_props(_object_spread({}, t.containerProps), {
                    style: {
                        width: a.current ? a.current.clientWidth + 2 : 0
                    },
                    className: _textfieldbasecssmistica.suggestionsContainer,
                    "aria-label": `${i.label} ${x.menuLabelSuffix}`,
                    children: t.children
                }))
        })
    }) : /* @__PURE__ */ (0, _jsxruntime.jsx)(W, _object_spread_props(_object_spread({}, i), {
        id: p,
        ref: m
    }));
});
