// @telefonica/mistica/dist/text-field-base.css-mistica.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    emptyDateValue: function() {
        return n;
    },
    endIconContainer: function() {
        return v;
    },
    fieldElementsGap: function() {
        return y;
    },
    fieldEndIconContainer: function() {
        return a;
    },
    fieldEndIconGap: function() {
        return f;
    },
    fieldLeftPadding: function() {
        return h;
    },
    fieldRightPadding: function() {
        return i;
    },
    fullWidth: function() {
        return _;
    },
    hiddenDatePlaceholder: function() {
        return d;
    },
    iconButtonSize: function() {
        return l;
    },
    input: function() {
        return o;
    },
    inputFirefoxStyles: function() {
        return p;
    },
    inputWithLabel: function() {
        return u;
    },
    inputWithoutLabel: function() {
        return m;
    },
    menuItem: function() {
        return c;
    },
    menuItemSelected: function() {
        return b;
    },
    prefix: function() {
        return g;
    },
    startIcon: function() {
        return k;
    },
    suggestionsContainer: function() {
        return x;
    },
    textArea: function() {
        return s;
    },
    textAreaWithLabel: function() {
        return I;
    },
    textAreaWithoutLabel: function() {
        return L;
    }
});
require("./sprinkles.css.ts.vanilla.css-mistica.js");
require("./icon-button.css.ts.vanilla.css-mistica.js");
require("./text-field-base.css.ts.vanilla.css-mistica.js");
var n = "rhey4t8", v = "rhey4th _1y2v1nfa1 _1y2v1nfhk _1y2v1nfhz", y = 12, a = "rhey4tr", f = 4, h = 11, i = 15, _ = "rhey4t4", d = "rhey4t7", l = "calc(1.5rem + 24px)", o = "rhey4ta _1y2v1nfhe rhey4t1 _1y2v1nfk2 _1y2v1nfj5 _1y2v1nf26 _1y2v1nfi7", p = "rhey4tb", u = "rhey4tc", m = "rhey4td", c = "rhey4tn _1y2v1nfhk _1y2v1nfhz _1y2v1nfkk", b = "_1y2v1nf44", g = "rhey4tl _1y2v1nfbg", k = "rhey4tj _1y2v1nfbg _1y2v1nfhk _1y2v1nfhz _1y2v1nfio _1y2v1nfhf", x = "rhey4tq _1y2v1nfhf", s = "rhey4t6 _1y2v1nf7d _1y2v1nf8m rhey4t1 _1y2v1nfk2 _1y2v1nfj5 _1y2v1nf26 _1y2v1nfi7", I = "rhey4te", L = "rhey4tf";
