// @telefonica/mistica/dist/text-link.js
"use client";
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "default", {
    enumerable: true,
    get: function() {
        return A;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _touchable = require("./touchable.js");
const _classnames = /*#__PURE__*/ _interop_require_default(require("classnames"));
const _hooks = require("./hooks.js");
const _themevariantcontext = require("./theme-variant-context.js");
const _formcontext = require("./form-context.js");
const _common = require("./utils/common.js");
const _analytics = require("./utils/analytics.js");
const _textlinkcssmistica = require("./text-link.css-mistica.js");
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
function _object_without_properties(source, excluded) {
    if (source == null) return {};
    var target = _object_without_properties_loose(source, excluded);
    var key, i;
    if (Object.getOwnPropertySymbols) {
        var sourceSymbolKeys = Object.getOwnPropertySymbols(source);
        for(i = 0; i < sourceSymbolKeys.length; i++){
            key = sourceSymbolKeys[i];
            if (excluded.indexOf(key) >= 0) continue;
            if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
            target[key] = source[key];
        }
    }
    return target;
}
function _object_without_properties_loose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;
    for(i = 0; i < sourceKeys.length; i++){
        key = sourceKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        target[key] = source[key];
    }
    return target;
}
const C = (_param)=>{
    var { children: e, className: r = "", disabled: a, style: s, trackEvent: i } = _param, t = _object_without_properties(_param, [
        "children",
        "className",
        "disabled",
        "style",
        "trackEvent"
    ]);
    const m = (0, _themevariantcontext.useIsInverseVariant)(), { isDarkMode: c } = (0, _hooks.useTheme)(), { formStatus: f } = (0, _formcontext.useForm)(), { eventFormat: u } = (0, _analytics.useTrackingConfig)(), p = ()=>u === "google-analytics-4" ? {
            name: _analytics.eventNames.userInteraction,
            component_type: "link",
            component_copy: (0, _common.getTextFromChildren)(e)
        } : {
            category: _analytics.eventCategories.userInteraction,
            action: _analytics.eventActions.linkTapped,
            label: (0, _common.getTextFromChildren)(e)
        };
    var _t_trackingEvent;
    return /* @__PURE__ */ (0, _jsxruntime.jsx)(_touchable.BaseTouchable, _object_spread_props(_object_spread({}, t), {
        stopPropagation: !0,
        as: t.onPress ? "a" : void 0,
        trackingEvent: (_t_trackingEvent = t.trackingEvent) !== null && _t_trackingEvent !== void 0 ? _t_trackingEvent : i ? p() : void 0,
        disabled: a || f === "sending",
        className: (0, _classnames.default)(m ? c ? _textlinkcssmistica.variants.inverseDark : _textlinkcssmistica.variants.inverseLight : _textlinkcssmistica.variants.default, r),
        style: s,
        children: e
    }));
}, A = C;
