// @telefonica/mistica/dist/text-field-components.css-mistica.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    DEFAULT_WIDTH: function() {
        return a;
    },
    LABEL_SCALE_DESKTOP: function() {
        return f;
    },
    LABEL_SCALE_MOBILE: function() {
        return e;
    },
    disabled: function() {
        return k;
    },
    field: function() {
        return i;
    },
    fieldContainer: function() {
        return y;
    },
    fullWidth: function() {
        return t;
    },
    helperContainer: function() {
        return l;
    },
    helperText: function() {
        return h;
    },
    labelContainer: function() {
        return u;
    },
    labelText: function() {
        return o;
    },
    leftText: function() {
        return d;
    },
    normalWidth: function() {
        return p;
    },
    rightText: function() {
        return L;
    },
    shrinked: function() {
        return E;
    }
});
require("./sprinkles.css.ts.vanilla.css-mistica.js");
require("./icon-button.css.ts.vanilla.css-mistica.js");
require("./text-field-base.css.ts.vanilla.css-mistica.js");
require("./text-field-components.css.ts.vanilla.css-mistica.js");
var a = 328, f = 0.78, e = 0.75, k = "_15k6ur97", i = "_1y2v1nfkl _1y2v1nfk1 _1y2v1nfhk _1y2v1nfka _1y2v1nfhe", y = "_15k6ur96 _1y2v1nfhk _1y2v1nfhq", t = "_1y2v1nfi7", l = "_1y2v1nfa7 _1y2v1nfbj _1y2v1nfhk", h = "_15k6ur9d", u = "_15k6ur91 _1y2v1nfhf _1y2v1nfhk _1y2v1nfhp", o = "_15k6ur93 _1y2v1nfkl _1y2v1nfi3", d = "_15k6ur9e", p = "_15k6ur99 _1y2v1nfi7", L = "_15k6ur9f", E = "_15k6ur94";
