// @telefonica/mistica/dist/timer.css-mistica.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    boxedTimerDisplayValue: function() {
        return n;
    },
    boxedTimerValueContainer: function() {
        return y;
    },
    boxedTimerValueContainerInverse: function() {
        return a;
    },
    inlineText: function() {
        return f;
    },
    shortLabelText: function() {
        return _;
    },
    timerDisplayValue: function() {
        return r;
    },
    timerWrapper: function() {
        return i;
    },
    unitContainer: function() {
        return t;
    }
});
require("./sprinkles.css.ts.vanilla.css-mistica.js");
require("./timer.css.ts.vanilla.css-mistica.js");
var n = "_18fye848", y = "_18fye84a _1y2v1nfa1 _1y2v1nfba _1y2v1nf7m _1y2v1nf8v _1y2v1nfkb _1y2v1nf6l", a = "_18fye84a _1y2v1nfa1 _1y2v1nfba _1y2v1nf7m _1y2v1nf8v _1y2v1nfkb _1y2v1nf3q", f = "_18fye841", _ = "_18fye841 _1y2v1nfho", r = "_1y2v1nfhk _1y2v1nfhq _1y2v1nfhz", i = "_1y2v1nfho", t = "_18fye841 _1y2v1nfhl _1y2v1nfht";
