// @telefonica/mistica/dist/tooltip.css-mistica.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    CONTENT_MIN_WIDTH: function() {
        return a;
    },
    arrow: function() {
        return v;
    },
    arrowContainer: function() {
        return _;
    },
    closeButtonIcon: function() {
        return o;
    },
    container: function() {
        return r;
    },
    content: function() {
        return f;
    },
    contentContainer: function() {
        return y;
    },
    tooltip: function() {
        return e;
    },
    tooltipCenter: function() {
        return i;
    },
    tooltipTransitionClasses: function() {
        return p;
    }
});
require("./sprinkles.css.ts.vanilla.css-mistica.js");
require("./tooltip.css.ts.vanilla.css-mistica.js");
var a = 40, v = "_1xhatbpb _1y2v1nfid _1y2v1nfit _1y2v1nfhf _1y2v1nfkr _1y2v1nf3q", _ = "_1xhatbp9 _1y2v1nfhf _1y2v1nfif _1y2v1nfkl", o = "_1xhatbpd _1y2v1nfhf", r = "_1xhatbp1 _1y2v1nfhf", f = "_1xhatbp6", y = "_1xhatbp5 _1y2v1nfhe _1y2v1nfkl", e = "_1xhatbp3 _1y2v1nfhe _1y2v1nfkd _1y2v1nf3q", i = "_1y2v1nfhk _1y2v1nfhq _1y2v1nfhz", p = {
    entering: {
        opacity: 1
    },
    entered: {
        opacity: 1
    },
    exiting: {
        opacity: 0
    },
    exited: {
        opacity: 0
    },
    unmounted: {}
};
