// @telefonica/mistica/dist/utils/platform.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    getPlatform: function() {
        return p;
    },
    isAndroid: function() {
        return o;
    },
    isChrome: function() {
        return a;
    },
    isFirefox: function() {
        return d;
    },
    isInsideNovumNativeApp: function() {
        return r;
    },
    isIos: function() {
        return u;
    },
    isRunningAcceptanceTest: function() {
        return f;
    }
});
const _webviewbridge = require("@tef-novum/webview-bridge");
const t = function() {
    let e = arguments.length > 0 && arguments[0] !== void 0 ? arguments[0] : {};
    var n;
    var _e_userAgent, _ref;
    return typeof self > "u" ? (_e_userAgent = e.userAgent) !== null && _e_userAgent !== void 0 ? _e_userAgent : "" : (_ref = (n = self.navigator) == null ? void 0 : n.userAgent) !== null && _ref !== void 0 ? _ref : "";
}, r = function() {
    let e = arguments.length > 0 && arguments[0] !== void 0 ? arguments[0] : {};
    return typeof e.insideNovumNativeApp < "u" ? e.insideNovumNativeApp : (0, _webviewbridge.isWebViewBridgeAvailable)();
}, f = function() {
    let e = arguments.length > 0 && arguments[0] !== void 0 ? arguments[0] : {};
    return t(e).includes("acceptance-test") && !process.env.SSR_TEST;
}, i = !!(typeof self < "u" && self.MSStream), o = (e)=>t(e).toLowerCase().includes("android") && !i, u = (e)=>!!(/iPad|iPhone|iPod/.test(t(e)) && !i || r(e) && !o(e)), d = (e)=>!!t(e).match(/Firefox\/([0-9]+)\./), a = (e)=>!!t(e).match(/Chrom(e|ium)\/([0-9]+)\./), p = function() {
    let e = arguments.length > 0 && arguments[0] !== void 0 ? arguments[0] : {};
    const n = e.platform;
    return n || (u(e) ? "ios" : o(e) ? "android" : "desktop");
};
