// @telefonica/mistica/dist-es/skins/blau.js
import { BLAU_SKIN as a } from "./constants.js";
import { applyAlpha as r } from "../utils/color.js";
const e = {
    blauBluePrimary: "#00B6F1",
    blauBluePrimary10: "#F7FDFF",
    blauBluePrimary20: "#E5F6FD",
    blauBluePrimary30: "#B3E9FB",
    blauBlueSecondary: "#0072BC",
    blauBlueSecondary10: "#E5F1F9",
    blauBlueSecondary20: "#B2D4EC",
    blauBlueSecondary30: "#80B7DF",
    blauBlueSecondary60: "#005A99",
    blauPurple: "#7814B3",
    blauPurple10: "#F1E7F7",
    blauPurple30: "#BB89D9",
    blauYellow: "#FFA922",
    blauYellow10: "#FFF6E9",
    blauYellow40: "#FFC364",
    blauYellow60: "#F09500",
    blauYellow70: "#996614",
    blauGreen: "#30D300",
    blauGreen10: "#EAFBE5",
    blauGreen30: "#97E980",
    blauGreen70: "#1D7F00",
    blauRed: "#F64417",
    blauRed10: "#FEECE8",
    blauRed20: "#FCC7B9",
    blauRed30: "#FA9E87",
    blauRed40: "#F97C5D",
    blauRed70: "#C93712",
    grey1: "#F5F9FA",
    grey2: "#E7E7E7",
    grey3: "#B8B8B8",
    grey4: "#A0A0A0",
    grey5: "#808285",
    grey6: "#000000",
    white: "#FFFFFF",
    darkModeBlack: "#191919",
    darkModeGrey: "#242424",
    darkModeGrey6: "#313235"
}, d = ()=>({
        name: a,
        colors: {
            backgroundBrand: e.blauBluePrimary,
            backgroundBrandSecondary: e.blauBluePrimary,
            appBarBackground: e.white,
            background: e.white,
            backgroundContainer: e.white,
            backgroundContainerError: e.blauRed10,
            backgroundContainerHover: r(e.blauBluePrimary, 0.03),
            backgroundContainerPressed: r(e.blauBluePrimary, 0.05),
            backgroundContainerBrand: e.blauBluePrimary,
            backgroundContainerBrandHover: r(e.darkModeBlack, 0.2),
            backgroundContainerBrandPressed: r(e.darkModeBlack, 0.4),
            backgroundContainerBrandOverInverse: e.blauBlueSecondary,
            backgroundContainerAlternative: e.blauBluePrimary20,
            backgroundOverlay: r(e.blauBlueSecondary, 0.75),
            backgroundSkeleton: e.grey2,
            backgroundSkeletonInverse: e.blauBlueSecondary,
            navigationBarBackground: e.blauBluePrimary,
            backgroundAlternative: e.blauBluePrimary20,
            backgroundBrandTop: e.blauBluePrimary,
            backgroundBrandBottom: e.blauBluePrimary,
            skeletonWave: e.grey2,
            borderLow: e.grey1,
            border: e.grey2,
            borderHigh: e.grey5,
            borderSelected: e.blauBlueSecondary60,
            coverBackgroundHover: r(e.darkModeBlack, 0.25),
            coverBackgroundPressed: r(e.darkModeBlack, 0.35),
            buttonDangerBackground: e.blauRed,
            buttonDangerBackgroundSelected: e.blauRed70,
            buttonDangerBackgroundHover: e.blauRed70,
            buttonLinkDangerBackgroundSelected: e.blauRed10,
            buttonLinkDangerBackgroundInverse: e.white,
            buttonLinkDangerBackgroundInverseSelected: e.blauRed10,
            buttonLinkBackgroundSelected: e.blauPurple10,
            buttonLinkBackgroundInverseSelected: r(e.white, 0.15),
            buttonPrimaryBackground: e.blauBlueSecondary,
            buttonPrimaryBackgroundInverse: e.white,
            buttonPrimaryBackgroundSelected: e.blauBlueSecondary60,
            buttonPrimaryBackgroundHover: e.blauBlueSecondary60,
            buttonPrimaryBackgroundInverseSelected: e.blauBluePrimary30,
            buttonSecondaryBorder: e.blauBlueSecondary,
            buttonSecondaryBorderSelected: e.blauBlueSecondary60,
            buttonSecondaryBorderInverse: e.white,
            buttonSecondaryBackgroundHover: e.blauBlueSecondary10,
            buttonSecondaryBackgroundSelected: e.blauBlueSecondary10,
            buttonSecondaryBorderInverseSelected: e.white,
            buttonSecondaryBackgroundInverseHover: r(e.white, 0.2),
            buttonSecondaryBackgroundInverseSelected: r(e.white, 0.2),
            textButtonPrimary: e.white,
            textButtonPrimaryInverse: e.blauBlueSecondary,
            textButtonPrimaryInverseSelected: e.blauBlueSecondary60,
            textButtonSecondary: e.blauBlueSecondary,
            textButtonSecondarySelected: e.blauBlueSecondary60,
            textButtonSecondaryInverse: e.white,
            textButtonSecondaryInverseSelected: e.white,
            textLink: e.blauPurple,
            textLinkInverse: e.white,
            textLinkDanger: e.blauRed,
            textLinkSnackbar: e.blauPurple30,
            textActivated: e.blauBlueSecondary,
            textBrand: e.blauBluePrimary,
            control: e.grey2,
            controlActivated: e.blauBlueSecondary,
            controlInverse: e.blauBluePrimary30,
            controlActivatedInverse: e.white,
            controlError: e.blauRed,
            barTrack: e.grey2,
            loadingBar: e.blauBlueSecondary,
            loadingBarBackground: e.blauBlueSecondary10,
            toggleAndroidInactive: e.grey2,
            toggleAndroidBackgroundActive: e.blauBlueSecondary10,
            iosControlKnob: e.white,
            divider: e.grey2,
            dividerInverse: r(e.white, 0.2),
            navigationBarDivider: e.blauBluePrimary,
            badge: e.blauRed,
            feedbackErrorBackground: e.blauRed,
            feedbackInfoBackground: e.grey6,
            brand: e.blauBluePrimary,
            brandHigh: e.blauBlueSecondary,
            inverse: e.white,
            neutralHigh: e.grey6,
            neutralMedium: e.grey5,
            neutralMediumInverse: e.grey5,
            neutralLow: e.grey1,
            neutralLowAlternative: e.grey2,
            textPrimary: e.grey6,
            textPrimaryInverse: e.white,
            textSecondary: e.grey5,
            textSecondaryInverse: e.blauBluePrimary20,
            error: e.blauRed,
            textError: e.blauRed,
            textErrorInverse: e.white,
            success: e.blauGreen,
            warning: e.blauYellow,
            promo: e.blauPurple,
            highlight: e.blauBluePrimary,
            successLow: e.blauGreen10,
            warningLow: e.blauYellow10,
            errorLow: e.blauRed10,
            promoLow: e.blauPurple10,
            brandLow: e.blauBlueSecondary10,
            successHigh: e.blauGreen70,
            warningHigh: e.blauYellow70,
            errorHigh: e.blauRed70,
            promoHigh: e.blauPurple,
            successHighInverse: e.blauGreen70,
            warningHighInverse: e.blauYellow70,
            errorHighInverse: e.blauRed70,
            promoHighInverse: e.blauPurple,
            textNavigationBarPrimary: e.white,
            textNavigationBarSecondary: e.blauBlueSecondary20,
            textNavigationSearchBarHint: e.blauBlueSecondary20,
            textNavigationSearchBarText: e.white,
            textAppBar: e.grey5,
            textAppBarSelected: e.blauBlueSecondary60,
            customTabsBackground: e.blauBluePrimary,
            tagTextPromo: e.blauPurple,
            tagTextActive: e.blauBluePrimary,
            tagTextInactive: e.grey5,
            tagTextSuccess: e.blauGreen70,
            tagTextWarning: e.blauYellow70,
            tagTextError: e.blauRed70,
            tagBackgroundPromo: e.blauPurple10,
            tagBackgroundActive: e.blauBlueSecondary10,
            tagBackgroundInactive: e.grey1,
            tagBackgroundSuccess: e.blauGreen10,
            tagBackgroundWarning: e.blauYellow10,
            tagBackgroundError: e.blauRed10,
            cardContentOverlay: `linear-gradient(180deg, ${r(e.grey6, 0)} 0%, ${r(e.grey6, 0.4)} 30%, ${r(e.grey6, 0.7)} 100%)`
        },
        darkModeColors: {
            background: e.darkModeBlack,
            backgroundAlternative: e.darkModeBlack,
            backgroundBrand: e.darkModeBlack,
            backgroundBrandSecondary: e.darkModeBlack,
            backgroundContainer: e.darkModeGrey,
            backgroundContainerError: e.darkModeGrey,
            backgroundContainerHover: r(e.white, 0.03),
            backgroundContainerPressed: r(e.white, 0.05),
            backgroundContainerBrand: e.darkModeGrey,
            backgroundContainerBrandHover: r(e.white, 0.03),
            backgroundContainerBrandPressed: r(e.white, 0.05),
            backgroundContainerBrandOverInverse: e.darkModeGrey,
            backgroundContainerAlternative: e.darkModeGrey,
            backgroundOverlay: r(e.darkModeGrey, 0.8),
            backgroundSkeleton: e.darkModeGrey,
            backgroundSkeletonInverse: e.darkModeGrey,
            backgroundBrandTop: e.darkModeBlack,
            backgroundBrandBottom: e.darkModeBlack,
            appBarBackground: e.darkModeGrey,
            navigationBarBackground: e.darkModeBlack,
            skeletonWave: e.grey5,
            borderLow: e.darkModeBlack,
            border: e.darkModeGrey,
            borderHigh: e.grey5,
            borderSelected: e.blauBlueSecondary60,
            coverBackgroundHover: r(e.darkModeBlack, 0.25),
            coverBackgroundPressed: r(e.darkModeBlack, 0.35),
            buttonDangerBackground: e.blauRed,
            buttonDangerBackgroundSelected: e.blauRed70,
            buttonDangerBackgroundHover: e.blauRed70,
            buttonLinkDangerBackgroundSelected: r(e.blauPurple, 0.3),
            buttonLinkDangerBackgroundInverse: r(e.white, 0),
            buttonLinkDangerBackgroundInverseSelected: r(e.blauPurple, 0.3),
            buttonLinkBackgroundSelected: r(e.blauPurple, 0.3),
            buttonLinkBackgroundInverseSelected: r(e.blauPurple, 0.3),
            buttonPrimaryBackground: e.blauBlueSecondary,
            buttonPrimaryBackgroundInverse: e.blauBluePrimary,
            buttonPrimaryBackgroundSelected: e.blauBlueSecondary60,
            buttonPrimaryBackgroundHover: e.blauBlueSecondary60,
            buttonPrimaryBackgroundInverseSelected: e.blauBlueSecondary60,
            buttonSecondaryBorder: e.white,
            buttonSecondaryBorderSelected: e.white,
            buttonSecondaryBackgroundHover: r(e.white, 0.15),
            buttonSecondaryBackgroundSelected: r(e.white, 0.15),
            buttonSecondaryBorderInverse: e.white,
            buttonSecondaryBorderInverseSelected: e.white,
            buttonSecondaryBackgroundInverseHover: r(e.white, 0.15),
            buttonSecondaryBackgroundInverseSelected: r(e.white, 0.15),
            textButtonPrimary: e.grey2,
            textButtonPrimaryInverse: e.grey2,
            textButtonPrimaryInverseSelected: e.grey2,
            textButtonSecondary: e.grey2,
            textButtonSecondarySelected: e.grey2,
            textButtonSecondaryInverse: e.grey2,
            textButtonSecondaryInverseSelected: e.grey2,
            textLink: e.blauPurple30,
            textLinkInverse: e.blauPurple30,
            textLinkDanger: e.blauRed,
            textLinkSnackbar: e.blauPurple30,
            textActivated: e.blauBlueSecondary,
            textBrand: e.blauBluePrimary,
            control: e.darkModeGrey6,
            controlActivated: e.blauBlueSecondary,
            controlInverse: e.darkModeGrey6,
            controlActivatedInverse: e.blauBlueSecondary,
            controlError: e.blauRed,
            barTrack: e.darkModeGrey6,
            loadingBar: e.blauBluePrimary,
            loadingBarBackground: e.darkModeGrey,
            toggleAndroidInactive: e.grey5,
            toggleAndroidBackgroundActive: e.blauBlueSecondary30,
            iosControlKnob: e.grey2,
            divider: r(e.white, 0.05),
            dividerInverse: r(e.white, 0.05),
            navigationBarDivider: e.darkModeBlack,
            badge: e.blauRed,
            feedbackErrorBackground: e.blauRed,
            feedbackInfoBackground: e.darkModeGrey,
            brand: e.blauBluePrimary,
            brandHigh: r(e.white, 0.05),
            inverse: e.grey2,
            neutralHigh: e.grey2,
            neutralMedium: e.grey5,
            neutralMediumInverse: e.grey5,
            neutralLow: e.darkModeGrey6,
            neutralLowAlternative: e.darkModeGrey6,
            textPrimary: e.grey2,
            textPrimaryInverse: e.grey2,
            textSecondary: e.grey4,
            textSecondaryInverse: e.grey4,
            error: e.blauRed,
            textError: e.blauRed,
            textErrorInverse: e.blauRed,
            success: e.blauGreen,
            warning: e.blauYellow,
            promo: e.blauPurple,
            highlight: e.blauBluePrimary,
            successLow: e.darkModeGrey6,
            warningLow: e.darkModeGrey6,
            errorLow: e.darkModeGrey6,
            promoLow: e.darkModeGrey6,
            brandLow: e.darkModeGrey6,
            successHigh: e.blauGreen30,
            warningHigh: e.blauYellow40,
            errorHigh: e.blauRed40,
            promoHigh: e.blauPurple30,
            successHighInverse: e.blauGreen70,
            warningHighInverse: e.blauYellow70,
            errorHighInverse: e.blauRed70,
            promoHighInverse: e.blauPurple,
            textNavigationBarPrimary: e.grey2,
            textNavigationBarSecondary: e.grey4,
            textNavigationSearchBarHint: e.grey4,
            textNavigationSearchBarText: e.grey2,
            textAppBar: e.grey5,
            textAppBarSelected: e.grey2,
            customTabsBackground: e.darkModeBlack,
            tagTextPromo: e.blauPurple30,
            tagTextActive: e.blauBluePrimary,
            tagTextInactive: e.grey5,
            tagTextSuccess: e.blauGreen30,
            tagTextWarning: e.blauYellow40,
            tagTextError: e.blauRed40,
            tagBackgroundPromo: e.darkModeGrey6,
            tagBackgroundActive: e.darkModeGrey6,
            tagBackgroundInactive: e.darkModeGrey6,
            tagBackgroundSuccess: e.darkModeGrey6,
            tagBackgroundWarning: e.darkModeGrey6,
            tagBackgroundError: e.darkModeGrey6,
            cardContentOverlay: `linear-gradient(180deg, ${r(e.grey6, 0)} 0%, ${r(e.grey6, 0.4)} 30%, ${r(e.grey6, 0.7)} 100%)`
        },
        borderRadii: {
            avatar: "50%",
            bar: "999px",
            button: "4px",
            checkbox: "2px",
            container: "8px",
            indicator: "999px",
            input: "8px",
            legacyDisplay: "16px",
            popup: "8px",
            sheet: "8px",
            mediaSmall: "8px"
        },
        textPresets: {
            cardTitle: {
                weight: "regular"
            },
            button: {
                weight: "medium"
            },
            tabsLabel: {
                weight: "medium",
                size: {
                    mobile: 16,
                    desktop: 18
                },
                lineHeight: {
                    mobile: 24,
                    desktop: 24
                }
            },
            link: {
                weight: "medium"
            },
            title1: {
                weight: "medium"
            },
            title2: {
                weight: "light",
                size: {
                    mobile: 20,
                    desktop: 28
                },
                lineHeight: {
                    mobile: 24,
                    desktop: 32
                }
            },
            indicator: {
                weight: "medium"
            },
            navigationBar: {
                weight: "medium"
            },
            text1: {
                size: {
                    mobile: 12,
                    desktop: 14
                },
                lineHeight: {
                    mobile: 16,
                    desktop: 20
                }
            },
            text2: {
                size: {
                    mobile: 14,
                    desktop: 16
                },
                lineHeight: {
                    mobile: 20,
                    desktop: 24
                }
            },
            text3: {
                size: {
                    mobile: 16,
                    desktop: 18
                },
                lineHeight: {
                    mobile: 24,
                    desktop: 24
                }
            },
            text4: {
                size: {
                    mobile: 18,
                    desktop: 20
                },
                lineHeight: {
                    mobile: 24,
                    desktop: 28
                }
            },
            text5: {
                weight: "light",
                size: {
                    mobile: 20,
                    desktop: 28
                },
                lineHeight: {
                    mobile: 24,
                    desktop: 32
                }
            },
            text6: {
                weight: "light",
                size: {
                    mobile: 24,
                    desktop: 32
                },
                lineHeight: {
                    mobile: 32,
                    desktop: 40
                }
            },
            text7: {
                weight: "light",
                size: {
                    mobile: 28,
                    desktop: 40
                },
                lineHeight: {
                    mobile: 32,
                    desktop: 48
                }
            },
            text8: {
                weight: "light",
                size: {
                    mobile: 32,
                    desktop: 48
                },
                lineHeight: {
                    mobile: 40,
                    desktop: 56
                }
            },
            text9: {
                weight: "light",
                size: {
                    mobile: 40,
                    desktop: 56
                },
                lineHeight: {
                    mobile: 48,
                    desktop: 64
                }
            },
            text10: {
                weight: "light",
                size: {
                    mobile: 48,
                    desktop: 64
                },
                lineHeight: {
                    mobile: 56,
                    desktop: 72
                }
            }
        }
    });
export { d as getBlauSkin, e as palette };
