// @telefonica/mistica/dist-es/skins/telefonica.js
import { TELEFONICA_SKIN as t } from "./constants.js";
import { applyAlpha as r } from "../utils/color.js";
const e = {
    telefonicaBlue: "#0066FF",
    telefonicaBlue10: "#E5F0FF",
    telefonicaBlue20: "#B2D1FF",
    telefonicaBlue30: "#80B3FF",
    telefonicaBlue70: "#0356C9",
    ambar: "#EAC344",
    ambar10: "#FDF9EC",
    ambar40: "#F0D57C",
    ambar70: "#69581F",
    coral: "#E66C64",
    coral10: "#FDF0EF",
    coral40: "#E3A19A",
    coral70: "#D50000",
    coral80: "#912C31",
    orchid: "#C466EF",
    orchid10: "#F9F0FD",
    orchid40: "#D694F4",
    orchid70: "#8A1A93",
    turquoise: "#59C2C9",
    turquoise10: "#EEF9FA",
    turquoise40: "#8BD4D9",
    turquoise70: "#3E888D",
    grey1: "#F2F4FF",
    grey2: "#D1D5E4",
    grey3: "#B0B6CA",
    grey4: "#8F97AF",
    grey5: "#6E7894",
    grey6: "#58617A",
    grey7: "#414B61",
    grey8: "#2B3447",
    grey9: "#031A34",
    white: "#FFFFFF",
    black: "#000000",
    darkModeBlack: "#191919",
    darkModeGrey: "#242424",
    darkModeGrey6: "#313235"
}, i = ()=>({
        name: t,
        colors: {
            appBarBackground: e.white,
            background: e.white,
            backgroundContainer: e.white,
            backgroundContainerError: e.coral10,
            backgroundContainerHover: r(e.telefonicaBlue, 0.03),
            backgroundContainerPressed: r(e.telefonicaBlue, 0.05),
            backgroundContainerBrand: e.telefonicaBlue,
            backgroundContainerBrandHover: r(e.darkModeBlack, 0.2),
            backgroundContainerBrandPressed: r(e.darkModeBlack, 0.4),
            backgroundContainerBrandOverInverse: e.telefonicaBlue70,
            backgroundContainerAlternative: e.grey1,
            backgroundBrand: e.telefonicaBlue,
            backgroundBrandSecondary: e.telefonicaBlue,
            backgroundOverlay: r(e.grey6, 0.8),
            backgroundSkeleton: e.grey2,
            backgroundSkeletonInverse: e.telefonicaBlue70,
            navigationBarBackground: e.telefonicaBlue,
            backgroundAlternative: e.grey1,
            backgroundBrandTop: e.telefonicaBlue,
            backgroundBrandBottom: e.telefonicaBlue,
            skeletonWave: e.grey2,
            borderLow: e.grey1,
            border: e.grey2,
            borderHigh: e.grey5,
            borderSelected: e.telefonicaBlue,
            coverBackgroundHover: r(e.darkModeBlack, 0.25),
            coverBackgroundPressed: r(e.darkModeBlack, 0.35),
            buttonDangerBackground: e.coral,
            buttonDangerBackgroundSelected: e.coral80,
            buttonDangerBackgroundHover: e.coral80,
            buttonLinkDangerBackgroundSelected: e.coral10,
            buttonLinkDangerBackgroundInverse: e.white,
            buttonLinkDangerBackgroundInverseSelected: e.coral10,
            buttonLinkBackgroundSelected: e.grey1,
            buttonLinkBackgroundInverseSelected: r(e.white, 0.08),
            buttonPrimaryBackground: e.telefonicaBlue,
            buttonPrimaryBackgroundInverse: e.white,
            buttonPrimaryBackgroundSelected: e.telefonicaBlue70,
            buttonPrimaryBackgroundHover: e.telefonicaBlue70,
            buttonPrimaryBackgroundInverseSelected: e.telefonicaBlue30,
            buttonSecondaryBorder: e.telefonicaBlue,
            buttonSecondaryBorderSelected: e.telefonicaBlue70,
            buttonSecondaryBackgroundHover: e.telefonicaBlue10,
            buttonSecondaryBackgroundSelected: e.telefonicaBlue10,
            buttonSecondaryBorderInverse: e.white,
            buttonSecondaryBorderInverseSelected: e.white,
            buttonSecondaryBackgroundInverseHover: r(e.white, 0.1),
            buttonSecondaryBackgroundInverseSelected: r(e.white, 0.1),
            textButtonPrimary: e.white,
            textButtonPrimaryInverse: e.telefonicaBlue,
            textButtonPrimaryInverseSelected: e.telefonicaBlue,
            textButtonSecondary: e.telefonicaBlue,
            textButtonSecondarySelected: e.telefonicaBlue70,
            textButtonSecondaryInverse: e.white,
            textButtonSecondaryInverseSelected: e.white,
            textLink: e.telefonicaBlue,
            textLinkInverse: e.white,
            textLinkDanger: e.coral,
            textLinkSnackbar: e.telefonicaBlue30,
            textActivated: e.telefonicaBlue,
            textBrand: e.telefonicaBlue,
            control: e.grey3,
            controlActivated: e.telefonicaBlue,
            controlInverse: e.grey1,
            controlActivatedInverse: e.white,
            controlError: e.coral,
            barTrack: e.grey3,
            loadingBar: e.telefonicaBlue30,
            loadingBarBackground: e.telefonicaBlue70,
            toggleAndroidInactive: e.grey2,
            toggleAndroidBackgroundActive: e.grey2,
            iosControlKnob: e.white,
            divider: e.grey2,
            dividerInverse: r(e.white, 0.2),
            navigationBarDivider: e.telefonicaBlue,
            badge: e.coral70,
            feedbackErrorBackground: e.coral,
            feedbackInfoBackground: e.grey9,
            brand: e.telefonicaBlue,
            brandHigh: e.telefonicaBlue70,
            inverse: e.white,
            neutralHigh: e.grey9,
            neutralMedium: e.grey5,
            neutralMediumInverse: e.grey5,
            neutralLow: e.grey1,
            neutralLowAlternative: e.grey2,
            textPrimary: e.grey9,
            textPrimaryInverse: e.white,
            textSecondary: e.grey5,
            textSecondaryInverse: e.telefonicaBlue10,
            error: e.coral,
            textError: e.coral,
            textErrorInverse: e.white,
            success: e.turquoise,
            warning: e.ambar,
            promo: e.orchid,
            highlight: e.coral40,
            successLow: e.turquoise10,
            warningLow: e.ambar10,
            errorLow: e.coral10,
            promoLow: e.orchid10,
            brandLow: e.telefonicaBlue10,
            successHigh: e.turquoise70,
            warningHigh: e.ambar70,
            errorHigh: e.coral70,
            promoHigh: e.orchid70,
            successHighInverse: e.turquoise70,
            warningHighInverse: e.ambar70,
            errorHighInverse: e.coral70,
            promoHighInverse: e.orchid70,
            textNavigationBarPrimary: e.white,
            textNavigationBarSecondary: e.grey1,
            textNavigationSearchBarHint: e.grey1,
            textNavigationSearchBarText: e.white,
            textAppBar: e.grey4,
            textAppBarSelected: e.telefonicaBlue,
            customTabsBackground: e.telefonicaBlue,
            tagTextPromo: e.orchid70,
            tagTextActive: e.telefonicaBlue,
            tagTextInactive: e.grey5,
            tagTextSuccess: e.turquoise70,
            tagTextWarning: e.ambar70,
            tagTextError: e.coral70,
            tagBackgroundPromo: e.orchid10,
            tagBackgroundActive: e.telefonicaBlue10,
            tagBackgroundInactive: e.grey1,
            tagBackgroundSuccess: e.turquoise10,
            tagBackgroundWarning: e.ambar10,
            tagBackgroundError: e.coral10,
            cardContentOverlay: `linear-gradient(180deg, ${r(e.black, 0)} 0%, ${r(e.black, 0.4)} 30%, ${r(e.black, 0.7)} 100%)`
        },
        darkModeColors: {
            background: e.darkModeBlack,
            backgroundAlternative: e.darkModeBlack,
            backgroundBrand: e.darkModeBlack,
            backgroundBrandSecondary: e.darkModeBlack,
            backgroundContainer: e.darkModeGrey,
            backgroundContainerError: e.darkModeGrey,
            backgroundContainerHover: r(e.white, 0.03),
            backgroundContainerPressed: r(e.white, 0.05),
            backgroundContainerBrand: e.darkModeGrey,
            backgroundContainerBrandHover: r(e.white, 0.03),
            backgroundContainerBrandPressed: r(e.white, 0.05),
            backgroundContainerBrandOverInverse: e.darkModeGrey,
            backgroundContainerAlternative: e.darkModeGrey,
            backgroundOverlay: r(e.darkModeGrey, 0.8),
            backgroundSkeleton: e.grey6,
            backgroundSkeletonInverse: e.grey6,
            backgroundBrandTop: e.darkModeBlack,
            backgroundBrandBottom: e.darkModeBlack,
            appBarBackground: e.darkModeGrey,
            navigationBarBackground: e.darkModeBlack,
            skeletonWave: e.grey5,
            borderLow: e.darkModeBlack,
            border: e.darkModeGrey,
            borderHigh: e.grey5,
            borderSelected: e.telefonicaBlue,
            coverBackgroundHover: r(e.darkModeBlack, 0.25),
            coverBackgroundPressed: r(e.darkModeBlack, 0.35),
            buttonDangerBackground: e.coral,
            buttonDangerBackgroundSelected: e.coral80,
            buttonDangerBackgroundHover: e.coral80,
            buttonLinkDangerBackgroundSelected: r(e.white, 0.08),
            buttonLinkDangerBackgroundInverse: r(e.white, 0),
            buttonLinkDangerBackgroundInverseSelected: r(e.white, 0.08),
            buttonLinkBackgroundSelected: r(e.white, 0.08),
            buttonLinkBackgroundInverseSelected: r(e.white, 0.08),
            buttonPrimaryBackground: e.telefonicaBlue,
            buttonPrimaryBackgroundInverse: e.telefonicaBlue,
            buttonPrimaryBackgroundSelected: e.telefonicaBlue70,
            buttonPrimaryBackgroundHover: e.telefonicaBlue70,
            buttonPrimaryBackgroundInverseSelected: e.telefonicaBlue70,
            buttonSecondaryBackgroundSelected: r(e.white, 0.15),
            buttonSecondaryBorder: e.white,
            buttonSecondaryBackgroundHover: r(e.white, 0.15),
            buttonSecondaryBorderSelected: e.white,
            buttonSecondaryBorderInverse: e.white,
            buttonSecondaryBorderInverseSelected: e.white,
            buttonSecondaryBackgroundInverseHover: r(e.white, 0.15),
            buttonSecondaryBackgroundInverseSelected: r(e.white, 0.15),
            textButtonPrimary: e.grey2,
            textButtonPrimaryInverse: e.grey2,
            textButtonPrimaryInverseSelected: e.grey2,
            textButtonSecondary: e.grey2,
            textButtonSecondarySelected: e.grey2,
            textButtonSecondaryInverse: e.grey2,
            textButtonSecondaryInverseSelected: e.grey2,
            textLink: e.telefonicaBlue,
            textLinkInverse: e.telefonicaBlue,
            textLinkDanger: e.coral,
            textLinkSnackbar: e.telefonicaBlue30,
            textActivated: e.telefonicaBlue,
            textBrand: e.telefonicaBlue,
            control: e.darkModeGrey6,
            controlActivated: e.telefonicaBlue,
            controlInverse: e.darkModeGrey6,
            controlActivatedInverse: e.telefonicaBlue,
            controlError: e.coral,
            barTrack: e.darkModeGrey6,
            loadingBar: e.telefonicaBlue,
            loadingBarBackground: e.grey6,
            toggleAndroidInactive: e.grey4,
            toggleAndroidBackgroundActive: e.telefonicaBlue20,
            iosControlKnob: e.grey2,
            divider: r(e.white, 0.05),
            dividerInverse: r(e.white, 0.05),
            navigationBarDivider: e.darkModeBlack,
            badge: e.coral70,
            feedbackErrorBackground: e.coral,
            feedbackInfoBackground: e.grey8,
            brand: e.telefonicaBlue,
            brandHigh: r(e.white, 0.05),
            inverse: e.grey2,
            neutralHigh: e.grey2,
            neutralMedium: e.grey5,
            neutralMediumInverse: e.grey5,
            neutralLow: e.darkModeGrey6,
            neutralLowAlternative: e.darkModeGrey6,
            textPrimary: e.grey2,
            textPrimaryInverse: e.grey2,
            textSecondary: e.grey4,
            textSecondaryInverse: e.grey4,
            error: e.coral,
            textError: e.coral,
            textErrorInverse: e.coral,
            success: e.turquoise,
            warning: e.ambar,
            promo: e.orchid,
            highlight: e.coral40,
            successLow: e.darkModeGrey6,
            warningLow: e.darkModeGrey6,
            errorLow: e.darkModeGrey6,
            promoLow: e.darkModeGrey6,
            brandLow: e.darkModeGrey6,
            successHigh: e.turquoise40,
            warningHigh: e.ambar40,
            errorHigh: e.coral40,
            promoHigh: e.orchid40,
            successHighInverse: e.turquoise70,
            warningHighInverse: e.ambar70,
            errorHighInverse: e.coral70,
            promoHighInverse: e.orchid70,
            textNavigationBarPrimary: e.grey2,
            textNavigationBarSecondary: e.grey4,
            textNavigationSearchBarHint: e.grey4,
            textNavigationSearchBarText: e.grey2,
            textAppBar: e.grey5,
            textAppBarSelected: e.grey2,
            customTabsBackground: e.darkModeBlack,
            tagTextPromo: e.orchid40,
            tagTextActive: e.telefonicaBlue,
            tagTextInactive: e.grey5,
            tagTextSuccess: e.turquoise40,
            tagTextWarning: e.ambar40,
            tagTextError: e.coral40,
            tagBackgroundPromo: e.darkModeGrey6,
            tagBackgroundActive: e.darkModeGrey6,
            tagBackgroundInactive: e.darkModeGrey6,
            tagBackgroundSuccess: e.darkModeGrey6,
            tagBackgroundWarning: e.darkModeGrey6,
            tagBackgroundError: e.darkModeGrey6,
            cardContentOverlay: `linear-gradient(180deg, ${r(e.black, 0)} 0%, ${r(e.black, 0.4)} 30%, ${r(e.black, 0.7)} 100%)`
        },
        borderRadii: {
            avatar: "50%",
            bar: "0px",
            button: "999px",
            checkbox: "0px",
            container: "0px",
            indicator: "999px",
            input: "0px",
            legacyDisplay: "0px",
            popup: "0px",
            sheet: "0px",
            mediaSmall: "0px"
        },
        textPresets: {
            cardTitle: {
                weight: "regular"
            },
            button: {
                weight: "medium"
            },
            tabsLabel: {
                weight: "medium",
                size: {
                    mobile: 16,
                    desktop: 18
                },
                lineHeight: {
                    mobile: 24,
                    desktop: 24
                }
            },
            link: {
                weight: "medium"
            },
            title1: {
                weight: "medium"
            },
            title2: {
                weight: "regular",
                size: {
                    mobile: 20,
                    desktop: 28
                },
                lineHeight: {
                    mobile: 24,
                    desktop: 32
                }
            },
            indicator: {
                weight: "medium"
            },
            navigationBar: {
                weight: "medium"
            },
            text1: {
                size: {
                    mobile: 12,
                    desktop: 14
                },
                lineHeight: {
                    mobile: 16,
                    desktop: 20
                }
            },
            text2: {
                size: {
                    mobile: 14,
                    desktop: 16
                },
                lineHeight: {
                    mobile: 20,
                    desktop: 24
                }
            },
            text3: {
                size: {
                    mobile: 16,
                    desktop: 18
                },
                lineHeight: {
                    mobile: 24,
                    desktop: 24
                }
            },
            text4: {
                size: {
                    mobile: 18,
                    desktop: 20
                },
                lineHeight: {
                    mobile: 24,
                    desktop: 28
                }
            },
            text5: {
                weight: "regular",
                size: {
                    mobile: 20,
                    desktop: 28
                },
                lineHeight: {
                    mobile: 24,
                    desktop: 32
                }
            },
            text6: {
                weight: "regular",
                size: {
                    mobile: 24,
                    desktop: 32
                },
                lineHeight: {
                    mobile: 32,
                    desktop: 40
                }
            },
            text7: {
                weight: "regular",
                size: {
                    mobile: 28,
                    desktop: 40
                },
                lineHeight: {
                    mobile: 32,
                    desktop: 48
                }
            },
            text8: {
                weight: "regular",
                size: {
                    mobile: 32,
                    desktop: 48
                },
                lineHeight: {
                    mobile: 40,
                    desktop: 56
                }
            },
            text9: {
                weight: "regular",
                size: {
                    mobile: 40,
                    desktop: 56
                },
                lineHeight: {
                    mobile: 48,
                    desktop: 64
                }
            },
            text10: {
                weight: "regular",
                size: {
                    mobile: 48,
                    desktop: 64
                },
                lineHeight: {
                    mobile: 56,
                    desktop: 72
                }
            }
        }
    });
export { i as getTelefonicaSkin, e as palette };
