// @telefonica/mistica/dist/skins/skin-contract.css-mistica.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
Object.defineProperty(exports, "vars", {
    enumerable: true,
    get: function() {
        return r;
    }
});
var r = {
    colors: {
        appBarBackground: "var(--_1vqcj1i0)",
        background: "var(--_1vqcj1i1)",
        backgroundContainer: "var(--_1vqcj1i2)",
        backgroundContainerError: "var(--_1vqcj1i3)",
        backgroundContainerHover: "var(--_1vqcj1i4)",
        backgroundContainerPressed: "var(--_1vqcj1i5)",
        backgroundContainerBrand: "var(--_1vqcj1i6)",
        backgroundContainerBrandHover: "var(--_1vqcj1i7)",
        backgroundContainerBrandPressed: "var(--_1vqcj1i8)",
        backgroundContainerBrandOverInverse: "var(--_1vqcj1i9)",
        backgroundContainerAlternative: "var(--_1vqcj1ia)",
        backgroundBrand: "var(--_1vqcj1ib)",
        backgroundBrandSecondary: "var(--_1vqcj1ic)",
        backgroundOverlay: "var(--_1vqcj1id)",
        backgroundSkeleton: "var(--_1vqcj1ie)",
        backgroundSkeletonInverse: "var(--_1vqcj1if)",
        backgroundAlternative: "var(--_1vqcj1ig)",
        backgroundBrandTop: "var(--_1vqcj1ih)",
        backgroundBrandBottom: "var(--_1vqcj1ii)",
        navigationBarBackground: "var(--_1vqcj1ij)",
        skeletonWave: "var(--_1vqcj1ik)",
        border: "var(--_1vqcj1il)",
        borderLow: "var(--_1vqcj1im)",
        borderHigh: "var(--_1vqcj1in)",
        borderSelected: "var(--_1vqcj1io)",
        buttonDangerBackground: "var(--_1vqcj1ip)",
        buttonDangerBackgroundSelected: "var(--_1vqcj1iq)",
        buttonDangerBackgroundHover: "var(--_1vqcj1ir)",
        buttonLinkDangerBackgroundSelected: "var(--_1vqcj1is)",
        buttonLinkDangerBackgroundInverse: "var(--_1vqcj1it)",
        buttonLinkDangerBackgroundInverseSelected: "var(--_1vqcj1iu)",
        buttonLinkBackgroundSelected: "var(--_1vqcj1iv)",
        buttonLinkBackgroundInverseSelected: "var(--_1vqcj1iw)",
        buttonPrimaryBackground: "var(--_1vqcj1ix)",
        buttonPrimaryBackgroundInverse: "var(--_1vqcj1iy)",
        buttonPrimaryBackgroundSelected: "var(--_1vqcj1iz)",
        buttonPrimaryBackgroundInverseSelected: "var(--_1vqcj1i10)",
        buttonPrimaryBackgroundHover: "var(--_1vqcj1i11)",
        buttonSecondaryBorder: "var(--_1vqcj1i12)",
        buttonSecondaryBorderSelected: "var(--_1vqcj1i13)",
        buttonSecondaryBorderInverse: "var(--_1vqcj1i14)",
        buttonSecondaryBorderInverseSelected: "var(--_1vqcj1i15)",
        buttonSecondaryBackgroundHover: "var(--_1vqcj1i16)",
        buttonSecondaryBackgroundSelected: "var(--_1vqcj1i17)",
        buttonSecondaryBackgroundInverseHover: "var(--_1vqcj1i18)",
        buttonSecondaryBackgroundInverseSelected: "var(--_1vqcj1i19)",
        control: "var(--_1vqcj1i1a)",
        controlActivated: "var(--_1vqcj1i1b)",
        controlInverse: "var(--_1vqcj1i1c)",
        controlActivatedInverse: "var(--_1vqcj1i1d)",
        controlError: "var(--_1vqcj1i1e)",
        barTrack: "var(--_1vqcj1i1f)",
        loadingBar: "var(--_1vqcj1i1g)",
        loadingBarBackground: "var(--_1vqcj1i1h)",
        textActivated: "var(--_1vqcj1i1i)",
        toggleAndroidInactive: "var(--_1vqcj1i1j)",
        toggleAndroidBackgroundActive: "var(--_1vqcj1i1k)",
        iosControlKnob: "var(--_1vqcj1i1l)",
        divider: "var(--_1vqcj1i1m)",
        dividerInverse: "var(--_1vqcj1i1n)",
        navigationBarDivider: "var(--_1vqcj1i1o)",
        badge: "var(--_1vqcj1i1p)",
        feedbackErrorBackground: "var(--_1vqcj1i1q)",
        feedbackInfoBackground: "var(--_1vqcj1i1r)",
        brand: "var(--_1vqcj1i1s)",
        brandHigh: "var(--_1vqcj1i1t)",
        inverse: "var(--_1vqcj1i1u)",
        neutralHigh: "var(--_1vqcj1i1v)",
        neutralLow: "var(--_1vqcj1i1w)",
        neutralLowAlternative: "var(--_1vqcj1i1x)",
        neutralMedium: "var(--_1vqcj1i1y)",
        neutralMediumInverse: "var(--_1vqcj1i1z)",
        promo: "var(--_1vqcj1i20)",
        textBrand: "var(--_1vqcj1i21)",
        error: "var(--_1vqcj1i22)",
        highlight: "var(--_1vqcj1i23)",
        success: "var(--_1vqcj1i24)",
        warning: "var(--_1vqcj1i25)",
        textPrimary: "var(--_1vqcj1i26)",
        textPrimaryInverse: "var(--_1vqcj1i27)",
        textSecondary: "var(--_1vqcj1i28)",
        textSecondaryInverse: "var(--_1vqcj1i29)",
        textError: "var(--_1vqcj1i2a)",
        textErrorInverse: "var(--_1vqcj1i2b)",
        textButtonPrimary: "var(--_1vqcj1i2c)",
        textButtonPrimaryInverse: "var(--_1vqcj1i2d)",
        textButtonPrimaryInverseSelected: "var(--_1vqcj1i2e)",
        textButtonSecondary: "var(--_1vqcj1i2f)",
        textButtonSecondarySelected: "var(--_1vqcj1i2g)",
        textButtonSecondaryInverse: "var(--_1vqcj1i2h)",
        textButtonSecondaryInverseSelected: "var(--_1vqcj1i2i)",
        textLink: "var(--_1vqcj1i2j)",
        textLinkInverse: "var(--_1vqcj1i2k)",
        textLinkDanger: "var(--_1vqcj1i2l)",
        textLinkSnackbar: "var(--_1vqcj1i2m)",
        textNavigationBarPrimary: "var(--_1vqcj1i2n)",
        textNavigationBarSecondary: "var(--_1vqcj1i2o)",
        textNavigationSearchBarHint: "var(--_1vqcj1i2p)",
        textNavigationSearchBarText: "var(--_1vqcj1i2q)",
        textAppBar: "var(--_1vqcj1i2r)",
        textAppBarSelected: "var(--_1vqcj1i2s)",
        successLow: "var(--_1vqcj1i2t)",
        warningLow: "var(--_1vqcj1i2u)",
        errorLow: "var(--_1vqcj1i2v)",
        promoLow: "var(--_1vqcj1i2w)",
        brandLow: "var(--_1vqcj1i2x)",
        successHigh: "var(--_1vqcj1i2y)",
        warningHigh: "var(--_1vqcj1i2z)",
        errorHigh: "var(--_1vqcj1i30)",
        promoHigh: "var(--_1vqcj1i31)",
        successHighInverse: "var(--_1vqcj1i32)",
        warningHighInverse: "var(--_1vqcj1i33)",
        errorHighInverse: "var(--_1vqcj1i34)",
        promoHighInverse: "var(--_1vqcj1i35)",
        tagTextPromo: "var(--_1vqcj1i36)",
        tagTextActive: "var(--_1vqcj1i37)",
        tagTextInactive: "var(--_1vqcj1i38)",
        tagTextSuccess: "var(--_1vqcj1i39)",
        tagTextWarning: "var(--_1vqcj1i3a)",
        tagTextError: "var(--_1vqcj1i3b)",
        tagBackgroundPromo: "var(--_1vqcj1i3c)",
        tagBackgroundActive: "var(--_1vqcj1i3d)",
        tagBackgroundInactive: "var(--_1vqcj1i3e)",
        tagBackgroundSuccess: "var(--_1vqcj1i3f)",
        tagBackgroundWarning: "var(--_1vqcj1i3g)",
        tagBackgroundError: "var(--_1vqcj1i3h)",
        customTabsBackground: "var(--_1vqcj1i3i)",
        coverBackgroundHover: "var(--_1vqcj1i3j)",
        coverBackgroundPressed: "var(--_1vqcj1i3k)",
        cardContentOverlay: "var(--_1vqcj1i3l)"
    },
    rawColors: {
        appBarBackground: "var(--_1vqcj1i3m)",
        background: "var(--_1vqcj1i3n)",
        backgroundContainer: "var(--_1vqcj1i3o)",
        backgroundContainerError: "var(--_1vqcj1i3p)",
        backgroundContainerHover: "var(--_1vqcj1i3q)",
        backgroundContainerPressed: "var(--_1vqcj1i3r)",
        backgroundContainerBrand: "var(--_1vqcj1i3s)",
        backgroundContainerBrandHover: "var(--_1vqcj1i3t)",
        backgroundContainerBrandPressed: "var(--_1vqcj1i3u)",
        backgroundContainerBrandOverInverse: "var(--_1vqcj1i3v)",
        backgroundContainerAlternative: "var(--_1vqcj1i3w)",
        backgroundBrand: "var(--_1vqcj1i3x)",
        backgroundBrandSecondary: "var(--_1vqcj1i3y)",
        backgroundOverlay: "var(--_1vqcj1i3z)",
        backgroundSkeleton: "var(--_1vqcj1i40)",
        backgroundSkeletonInverse: "var(--_1vqcj1i41)",
        backgroundAlternative: "var(--_1vqcj1i42)",
        backgroundBrandTop: "var(--_1vqcj1i43)",
        backgroundBrandBottom: "var(--_1vqcj1i44)",
        navigationBarBackground: "var(--_1vqcj1i45)",
        skeletonWave: "var(--_1vqcj1i46)",
        border: "var(--_1vqcj1i47)",
        borderLow: "var(--_1vqcj1i48)",
        borderHigh: "var(--_1vqcj1i49)",
        borderSelected: "var(--_1vqcj1i4a)",
        buttonDangerBackground: "var(--_1vqcj1i4b)",
        buttonDangerBackgroundSelected: "var(--_1vqcj1i4c)",
        buttonDangerBackgroundHover: "var(--_1vqcj1i4d)",
        buttonLinkDangerBackgroundSelected: "var(--_1vqcj1i4e)",
        buttonLinkDangerBackgroundInverse: "var(--_1vqcj1i4f)",
        buttonLinkDangerBackgroundInverseSelected: "var(--_1vqcj1i4g)",
        buttonLinkBackgroundSelected: "var(--_1vqcj1i4h)",
        buttonLinkBackgroundInverseSelected: "var(--_1vqcj1i4i)",
        buttonPrimaryBackground: "var(--_1vqcj1i4j)",
        buttonPrimaryBackgroundInverse: "var(--_1vqcj1i4k)",
        buttonPrimaryBackgroundSelected: "var(--_1vqcj1i4l)",
        buttonPrimaryBackgroundInverseSelected: "var(--_1vqcj1i4m)",
        buttonPrimaryBackgroundHover: "var(--_1vqcj1i4n)",
        buttonSecondaryBorder: "var(--_1vqcj1i4o)",
        buttonSecondaryBorderSelected: "var(--_1vqcj1i4p)",
        buttonSecondaryBorderInverse: "var(--_1vqcj1i4q)",
        buttonSecondaryBorderInverseSelected: "var(--_1vqcj1i4r)",
        buttonSecondaryBackgroundHover: "var(--_1vqcj1i4s)",
        buttonSecondaryBackgroundSelected: "var(--_1vqcj1i4t)",
        buttonSecondaryBackgroundInverseHover: "var(--_1vqcj1i4u)",
        buttonSecondaryBackgroundInverseSelected: "var(--_1vqcj1i4v)",
        control: "var(--_1vqcj1i4w)",
        controlActivated: "var(--_1vqcj1i4x)",
        controlInverse: "var(--_1vqcj1i4y)",
        controlActivatedInverse: "var(--_1vqcj1i4z)",
        controlError: "var(--_1vqcj1i50)",
        barTrack: "var(--_1vqcj1i51)",
        loadingBar: "var(--_1vqcj1i52)",
        loadingBarBackground: "var(--_1vqcj1i53)",
        textActivated: "var(--_1vqcj1i54)",
        toggleAndroidInactive: "var(--_1vqcj1i55)",
        toggleAndroidBackgroundActive: "var(--_1vqcj1i56)",
        iosControlKnob: "var(--_1vqcj1i57)",
        divider: "var(--_1vqcj1i58)",
        dividerInverse: "var(--_1vqcj1i59)",
        navigationBarDivider: "var(--_1vqcj1i5a)",
        badge: "var(--_1vqcj1i5b)",
        feedbackErrorBackground: "var(--_1vqcj1i5c)",
        feedbackInfoBackground: "var(--_1vqcj1i5d)",
        brand: "var(--_1vqcj1i5e)",
        brandHigh: "var(--_1vqcj1i5f)",
        inverse: "var(--_1vqcj1i5g)",
        neutralHigh: "var(--_1vqcj1i5h)",
        neutralLow: "var(--_1vqcj1i5i)",
        neutralLowAlternative: "var(--_1vqcj1i5j)",
        neutralMedium: "var(--_1vqcj1i5k)",
        neutralMediumInverse: "var(--_1vqcj1i5l)",
        promo: "var(--_1vqcj1i5m)",
        textBrand: "var(--_1vqcj1i5n)",
        error: "var(--_1vqcj1i5o)",
        highlight: "var(--_1vqcj1i5p)",
        success: "var(--_1vqcj1i5q)",
        warning: "var(--_1vqcj1i5r)",
        textPrimary: "var(--_1vqcj1i5s)",
        textPrimaryInverse: "var(--_1vqcj1i5t)",
        textSecondary: "var(--_1vqcj1i5u)",
        textSecondaryInverse: "var(--_1vqcj1i5v)",
        textError: "var(--_1vqcj1i5w)",
        textErrorInverse: "var(--_1vqcj1i5x)",
        textButtonPrimary: "var(--_1vqcj1i5y)",
        textButtonPrimaryInverse: "var(--_1vqcj1i5z)",
        textButtonPrimaryInverseSelected: "var(--_1vqcj1i60)",
        textButtonSecondary: "var(--_1vqcj1i61)",
        textButtonSecondarySelected: "var(--_1vqcj1i62)",
        textButtonSecondaryInverse: "var(--_1vqcj1i63)",
        textButtonSecondaryInverseSelected: "var(--_1vqcj1i64)",
        textLink: "var(--_1vqcj1i65)",
        textLinkInverse: "var(--_1vqcj1i66)",
        textLinkDanger: "var(--_1vqcj1i67)",
        textLinkSnackbar: "var(--_1vqcj1i68)",
        textNavigationBarPrimary: "var(--_1vqcj1i69)",
        textNavigationBarSecondary: "var(--_1vqcj1i6a)",
        textNavigationSearchBarHint: "var(--_1vqcj1i6b)",
        textNavigationSearchBarText: "var(--_1vqcj1i6c)",
        textAppBar: "var(--_1vqcj1i6d)",
        textAppBarSelected: "var(--_1vqcj1i6e)",
        successLow: "var(--_1vqcj1i6f)",
        warningLow: "var(--_1vqcj1i6g)",
        errorLow: "var(--_1vqcj1i6h)",
        promoLow: "var(--_1vqcj1i6i)",
        brandLow: "var(--_1vqcj1i6j)",
        successHigh: "var(--_1vqcj1i6k)",
        warningHigh: "var(--_1vqcj1i6l)",
        errorHigh: "var(--_1vqcj1i6m)",
        promoHigh: "var(--_1vqcj1i6n)",
        successHighInverse: "var(--_1vqcj1i6o)",
        warningHighInverse: "var(--_1vqcj1i6p)",
        errorHighInverse: "var(--_1vqcj1i6q)",
        promoHighInverse: "var(--_1vqcj1i6r)",
        tagTextPromo: "var(--_1vqcj1i6s)",
        tagTextActive: "var(--_1vqcj1i6t)",
        tagTextInactive: "var(--_1vqcj1i6u)",
        tagTextSuccess: "var(--_1vqcj1i6v)",
        tagTextWarning: "var(--_1vqcj1i6w)",
        tagTextError: "var(--_1vqcj1i6x)",
        tagBackgroundPromo: "var(--_1vqcj1i6y)",
        tagBackgroundActive: "var(--_1vqcj1i6z)",
        tagBackgroundInactive: "var(--_1vqcj1i70)",
        tagBackgroundSuccess: "var(--_1vqcj1i71)",
        tagBackgroundWarning: "var(--_1vqcj1i72)",
        tagBackgroundError: "var(--_1vqcj1i73)",
        customTabsBackground: "var(--_1vqcj1i74)",
        coverBackgroundHover: "var(--_1vqcj1i75)",
        coverBackgroundPressed: "var(--_1vqcj1i76)",
        cardContentOverlay: "var(--_1vqcj1i77)"
    },
    borderRadii: {
        button: "var(--_1vqcj1i78)",
        input: "var(--_1vqcj1i79)",
        container: "var(--_1vqcj1i7a)",
        legacyDisplay: "var(--_1vqcj1i7b)",
        popup: "var(--_1vqcj1i7c)",
        checkbox: "var(--_1vqcj1i7d)",
        indicator: "var(--_1vqcj1i7e)",
        sheet: "var(--_1vqcj1i7f)",
        bar: "var(--_1vqcj1i7g)",
        avatar: "var(--_1vqcj1i7h)",
        mediaSmall: "var(--_1vqcj1i7i)"
    },
    textPresets: {
        cardTitle: {
            weight: "var(--_1vqcj1i7j)"
        },
        button: {
            weight: "var(--_1vqcj1i7k)"
        },
        link: {
            weight: "var(--_1vqcj1i7l)"
        },
        title1: {
            weight: "var(--_1vqcj1i7m)"
        },
        title2: {
            weight: "var(--_1vqcj1i7n)",
            size: "var(--_1vqcj1i7o)",
            lineHeight: "var(--_1vqcj1i7p)"
        },
        indicator: {
            weight: "var(--_1vqcj1i7q)"
        },
        tabsLabel: {
            weight: "var(--_1vqcj1i7r)",
            size: "var(--_1vqcj1i7s)",
            lineHeight: "var(--_1vqcj1i7t)"
        },
        navigationBar: {
            weight: "var(--_1vqcj1i7u)"
        },
        text1: {
            size: "var(--_1vqcj1i7v)",
            lineHeight: "var(--_1vqcj1i7w)"
        },
        text2: {
            size: "var(--_1vqcj1i7x)",
            lineHeight: "var(--_1vqcj1i7y)"
        },
        text3: {
            size: "var(--_1vqcj1i7z)",
            lineHeight: "var(--_1vqcj1i80)"
        },
        text4: {
            size: "var(--_1vqcj1i81)",
            lineHeight: "var(--_1vqcj1i82)"
        },
        text5: {
            weight: "var(--_1vqcj1i83)",
            size: "var(--_1vqcj1i84)",
            lineHeight: "var(--_1vqcj1i85)"
        },
        text6: {
            weight: "var(--_1vqcj1i86)",
            size: "var(--_1vqcj1i87)",
            lineHeight: "var(--_1vqcj1i88)"
        },
        text7: {
            weight: "var(--_1vqcj1i89)",
            size: "var(--_1vqcj1i8a)",
            lineHeight: "var(--_1vqcj1i8b)"
        },
        text8: {
            weight: "var(--_1vqcj1i8c)",
            size: "var(--_1vqcj1i8d)",
            lineHeight: "var(--_1vqcj1i8e)"
        },
        text9: {
            weight: "var(--_1vqcj1i8f)",
            size: "var(--_1vqcj1i8g)",
            lineHeight: "var(--_1vqcj1i8h)"
        },
        text10: {
            weight: "var(--_1vqcj1i8i)",
            size: "var(--_1vqcj1i8j)",
            lineHeight: "var(--_1vqcj1i8k)"
        }
    }
};
