// @telefonica/mistica/dist-es/button-fixed-footer-layout.js
"use client";
import { jsx as o } from "react/jsx-runtime";
import d from "./fixed-footer-layout.js";
import l from "./button-layout.js";
import { useScreenSize as f } from "./hooks.js";
import { InternalResponsiveLayout as h } from "./responsive-layout.js";
import x from "./box.js";
const F = (param)=>{
    let { isFooterVisible: i = !0, button: t, secondaryButton: e, link: s, children: n, desktopButtonAlign: r, footerBgColor: a, containerBgColor: m, onChangeFooterHeight: u } = param;
    const { isDesktopOrBigger: p } = f();
    return /* @__PURE__ */ o(d, {
        onChangeFooterHeight: u,
        isFooterVisible: (!!t || !!e) && i,
        footerBgColor: a,
        containerBgColor: m,
        footer: /* @__PURE__ */ o(h, {
            shouldExpandWhenNested: "desktop",
            children: /* @__PURE__ */ o(x, {
                paddingY: {
                    mobile: 16,
                    tablet: 32,
                    desktop: 0
                },
                children: /* @__PURE__ */ o(l, {
                    primaryButton: t,
                    secondaryButton: e,
                    align: p && r ? r : "full-width",
                    link: s
                })
            })
        }),
        children: n
    });
}, S = F;
export { S as default };
