// @telefonica/mistica/dist-es/button-group.js
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
import { jsxs as r, jsx as c } from "react/jsx-runtime";
import n from "classnames";
import { getPrefixedDataAttributes as f } from "./utils/dom.js";
import { inline as b, container as p, centerInDesktop as h, centerInTablet as u, centerInMobile as v, buttons as N, buttonChild as i } from "./button-group.css-mistica.js";
const k = (param)=>{
    let { primaryButton: t, secondaryButton: o, link: s, align: e = "left", dataAttributes: m } = param;
    var _e_mobile, _e_tablet, _ref, _e_desktop;
    const d = !!t || !!o || !!s, a = !!t && !!o, l = typeof e == "string" ? {
        mobile: e,
        tablet: e,
        desktop: e
    } : {
        mobile: (_e_mobile = e.mobile) !== null && _e_mobile !== void 0 ? _e_mobile : "left",
        tablet: (_ref = (_e_tablet = e.tablet) !== null && _e_tablet !== void 0 ? _e_tablet : e.mobile) !== null && _ref !== void 0 ? _ref : "left",
        desktop: (_e_desktop = e.desktop) !== null && _e_desktop !== void 0 ? _e_desktop : "left"
    };
    return d ? /* @__PURE__ */ r("div", _object_spread_props(_object_spread({
        className: n(b, p, {
            [h]: l.desktop === "center",
            [u]: l.tablet === "center",
            [v]: l.mobile === "center"
        })
    }, f(m, "ButtonGroup")), {
        children: [
            (t || o) && /* @__PURE__ */ r("div", {
                className: n(b, N),
                children: [
                    t && /* @__PURE__ */ c("div", {
                        className: i,
                        children: t
                    }),
                    o && /* @__PURE__ */ c("div", {
                        className: i,
                        children: o
                    })
                ]
            }),
            s && /* @__PURE__ */ c("div", {
                className: i,
                style: {
                    width: a ? "100%" : "auto"
                },
                children: s
            })
        ]
    })) : null;
}, A = k;
export { A as default };
