// @telefonica/mistica/dist-es/card.js
"use client";
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
function _object_without_properties(source, excluded) {
    if (source == null) return {};
    var target = _object_without_properties_loose(source, excluded);
    var key, i;
    if (Object.getOwnPropertySymbols) {
        var sourceSymbolKeys = Object.getOwnPropertySymbols(source);
        for(i = 0; i < sourceSymbolKeys.length; i++){
            key = sourceSymbolKeys[i];
            if (excluded.indexOf(key) >= 0) continue;
            if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
            target[key] = source[key];
        }
    }
    return target;
}
function _object_without_properties_loose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;
    for(i = 0; i < sourceKeys.length; i++){
        key = sourceKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        target[key] = source[key];
    }
    return target;
}
import { jsx as e, jsxs as l, Fragment as Le } from "react/jsx-runtime";
import * as b from "react";
import We from "./tag.js";
import te from "./stack.js";
import P from "./box.js";
import { Text as ne, textProps as ie, Text2 as D, Text6 as _e, Text3 as $e } from "./text.js";
import { Boxed as me, InternalBoxed as xe } from "./boxed.js";
import he from "./button-group.js";
import ue from "./image.js";
import { BaseTouchable as W } from "./touchable.js";
import { vars as u } from "./skins/skin-contract.css-mistica.js";
import { cardContainer as qe, touchableContainer as _, boxed as K, touchable as $, touchableMediaCardOverlay as Je, mediaCard as pe, mediaCardContent as Ke, mediaCardIcon as be, touchableNakedMediaOverlay as Ce, circularMediaOverlay as Te, nakedCardContent as Ne, touchableCardOverlay as oe, dataCard as Qe, vars as Be, dataCardTopActionsWithoutIcon as Ue, actions as Re, snapCard as Ye, displayCardContainer as ke, displayCardBackground as Ie, displayCardContent as we, displayCardGradient as Se, flexColumn as Ae, touchableCardOverlayInverse as fe, touchableCardOverlayMedia as Me } from "./card.css-mistica.js";
import { vars as L } from "./image.css-mistica.js";
import { useTheme as J } from "./hooks.js";
import { sprinkles as Ze } from "./sprinkles.css-mistica.js";
import { InternalIconButton as ea, InternalToggleIconButton as aa } from "./icon-button.js";
import ra from "./generated/mistica-icons/icon-close-regular.js";
import ta from "./generated/mistica-icons/icon-pause-filled.js";
import na from "./generated/mistica-icons/icon-play-filled.js";
import { combineRefs as ia } from "./utils/common.js";
import oa from "./spinner.js";
import da from "./video.js";
import { ThemeVariant as ve, useIsInverseVariant as De } from "./theme-variant-context.js";
import ye from "classnames";
import Oe from "./inline.js";
import { getPrefixedDataAttributes as sa } from "./utils/dom.js";
import { isRunningAcceptanceTest as la } from "./utils/platform.js";
import { applyCssVars as G } from "./utils/css.js";
const S = ()=>{
    const [a, r] = b.useState(""), t = b.useCallback((n)=>{
        n && r((process.env.NODE_ENV === "test" ? n.textContent : n.innerText) || "");
    }, []);
    return {
        text: a,
        ref: t
    };
}, ze = (a, r)=>{
    const { texts: t } = J(), n = a ? [
        ...a
    ] : [];
    return r && n.push({
        label: t.closeButtonLabel,
        onPress: r,
        Icon: ra
    }), n;
}, je = /*#__PURE__*/ b.createContext("default"), ca = (a)=>{
    const r = b.useContext(je);
    return /* @__PURE__ */ e(ve, {
        isInverse: r === "inverse",
        children: a.Icon ? /* @__PURE__ */ e(ea, _object_spread_props(_object_spread({}, a), {
            "aria-label": a.label,
            small: !0,
            isOverMedia: r === "media",
            type: "neutral",
            backgroundType: "transparent"
        })) : /* @__PURE__ */ e(aa, _object_spread_props(_object_spread({}, a), {
            checkedProps: _object_spread_props(_object_spread({}, a.checkedProps), {
                "aria-label": a.checkedProps.label,
                type: r === "media" ? "neutral" : "brand",
                backgroundType: "solid"
            }),
            uncheckedProps: _object_spread_props(_object_spread({}, a.uncheckedProps), {
                "aria-label": a.uncheckedProps.label,
                type: "neutral",
                backgroundType: "transparent"
            }),
            small: !0,
            isOverMedia: r === "media"
        }))
    });
}, Q = (param)=>{
    let { actions: a, padding: r = 16, onClose: t, type: n = "default" } = param;
    const i = ze(a, t);
    return i.length > 0 ? /* @__PURE__ */ e(je.Provider, {
        value: n,
        children: /* @__PURE__ */ e("div", {
            style: {
                position: "absolute",
                right: r,
                top: r,
                zIndex: 3
            },
            children: /* @__PURE__ */ e(Oe, {
                space: 16,
                children: i.map((o, s)=>"Icon" in o || "checkedProps" in o ? /* @__PURE__ */ e(ca, _object_spread({}, o), s) : o)
            })
        })
    }) : /* @__PURE__ */ e(Le, {});
}, ma = (a)=>a ? typeof a == "number" ? a : ({
        "1:1": 1,
        "16:9": 16 / 9,
        "7:10": 7 / 10,
        "9:10": 9 / 10,
        auto: 0
    })[a] : 0, q = /*#__PURE__*/ b.forwardRef((param, s)=>{
    let { children: a, width: r, height: t, aspectRatio: n, dataAttributes: i, className: d, "aria-label": o } = param;
    const v = r && t ? void 0 : ma(n);
    return /* @__PURE__ */ e("section", _object_spread_props(_object_spread({}, sa(i)), {
        ref: s,
        "aria-label": o,
        className: ye(d, qe),
        style: _object_spread({
            width: r || "100%",
            height: t || "100%"
        }, v ? G({
            [Be.aspectRatio]: String(v)
        }) : {}),
        children: a
    }));
}), Pe = (a)=>/* @__PURE__ */ e(ue, {
        width: "100%",
        height: "100%",
        src: a || ""
    }), ha = {
    loading: {
        showSpinner: "loadingTimeout",
        play: "playing",
        pause: "paused",
        fail: "error"
    },
    loadingTimeout: {
        play: "playing",
        pause: "paused",
        fail: "error",
        reset: "loading"
    },
    playing: {
        pause: "paused",
        reset: "loading",
        fail: "error"
    },
    paused: {
        play: "playing",
        reset: "loading",
        fail: "error"
    },
    error: {
        reset: "loading"
    }
}, ua = (a, r)=>ha[a][r] || a, pa = (param)=>{
    let { color: a } = param;
    return /* @__PURE__ */ e(oa, {
        color: a,
        size: 16,
        delay: "0"
    });
}, fa = (param)=>{
    let { color: a } = param;
    return /* @__PURE__ */ e(ta, {
        color: a,
        size: 12
    });
}, va = (param)=>{
    let { color: a } = param;
    return /* @__PURE__ */ e(na, {
        color: a,
        size: 12
    });
}, Ve = (a, r, t)=>{
    const { texts: n } = J(), i = b.useRef(null), [d, o] = b.useReducer(ua, "loading");
    b.useEffect(()=>{
        var h;
        const c = setTimeout(()=>o("showSpinner"), 2e3);
        return (h = i.current) == null || h.load(), ()=>{
            clearTimeout(c), o("reset");
        };
    }, [
        a
    ]);
    const s = b.useMemo(()=>a !== void 0 ? /* @__PURE__ */ e(da, {
            ref: ia(i, t),
            src: a,
            width: "100%",
            height: "100%",
            poster: r,
            onError: ()=>o("fail"),
            onPause: ()=>o("pause"),
            onPlay: ()=>o("play")
        }) : void 0, [
        t,
        a,
        r
    ]), v = ()=>{
        const c = i.current;
        c && (d === "loading" ? o("showSpinner") : d === "paused" ? c.play() : d === "playing" && c.pause());
    };
    if (d === "error") return {
        video: s
    };
    const m = s ? {
        uncheckedProps: {
            Icon: d === "loadingTimeout" && !la() ? pa : fa,
            label: d === "loadingTimeout" ? "" : n.pauseIconButtonLabel
        },
        checkedProps: {
            Icon: va,
            label: n.playIconButtonLabel
        },
        onChange: v,
        disabled: d === "loadingTimeout",
        checked: d === "paused"
    } : void 0;
    return {
        video: s,
        videoAction: m
    };
}, ge = (param)=>{
    let { headline: a, headlineRef: r, pretitle: t, pretitleLinesMax: n, title: i, titleAs: d = "h3", titleLinesMax: o, subtitle: s, subtitleLinesMax: v, description: m, descriptionLinesMax: c, extra: h, extraRef: y, button: p, buttonLink: g } = param;
    const { textPresets: N } = J();
    return /* @__PURE__ */ l("div", {
        className: Ze({
            display: "flex",
            flex: 1,
            justifyContent: "space-between",
            flexDirection: "column"
        }),
        children: [
            /* @__PURE__ */ l("div", {
                className: Ae,
                children: [
                    i && /* @__PURE__ */ e("div", {
                        style: {
                            paddingBottom: s || m ? 4 : 0
                        },
                        children: /* @__PURE__ */ e(ne, _object_spread_props(_object_spread({}, ie.text4), {
                            truncate: o,
                            weight: N.cardTitle.weight,
                            as: d,
                            hyphens: "auto",
                            children: i
                        }))
                    }),
                    a && // assuming that the headline will always be followed by one of: pretitle, title, subtitle, description
                    /* @__PURE__ */ e("div", {
                        ref: r,
                        style: {
                            order: -2,
                            paddingBottom: 8
                        },
                        children: typeof a == "string" ? /* @__PURE__ */ e(We, {
                            type: "promo",
                            children: a
                        }) : a
                    }),
                    t && /* @__PURE__ */ e("div", {
                        style: {
                            order: -1,
                            paddingBottom: 4
                        },
                        children: /* @__PURE__ */ e(D, {
                            truncate: n,
                            as: "div",
                            regular: !0,
                            hyphens: "auto",
                            children: t
                        })
                    }),
                    s && /* @__PURE__ */ e("div", {
                        style: {
                            paddingBottom: m ? 4 : 0
                        },
                        children: /* @__PURE__ */ e(D, {
                            truncate: v,
                            as: "div",
                            regular: !0,
                            hyphens: "auto",
                            children: s
                        })
                    }),
                    m && // this is tricky, when headline exists, the 8px padding is added by it.
                    // Otherwise, only 4px are added by title|pretitle|subtitle, so we need to add 4px more
                    /* @__PURE__ */ e("div", {
                        style: {
                            paddingTop: t || i || s ? 4 : 0
                        },
                        children: /* @__PURE__ */ e(D, {
                            truncate: c,
                            as: "p",
                            regular: !0,
                            color: u.colors.textSecondary,
                            hyphens: "auto",
                            children: m
                        })
                    }),
                    h && /* @__PURE__ */ e("div", {
                        ref: y,
                        children: h
                    })
                ]
            }),
            (p || g) && /* @__PURE__ */ e("div", {
                className: Re,
                children: /* @__PURE__ */ e(he, {
                    primaryButton: p,
                    link: g
                })
            })
        ]
    });
}, Wa = /*#__PURE__*/ b.forwardRef((_param, A)=>{
    var { media: a, icon: r, headline: t, pretitle: n, pretitleLinesMax: i, subtitle: d, subtitleLinesMax: o, title: s, titleAs: v = "h3", titleLinesMax: m, description: c, descriptionLinesMax: h, extra: y, actions: p, button: g, buttonLink: N, dataAttributes: x, "aria-label": C, onClose: T } = _param, f = _object_without_properties(_param, [
        "media",
        "icon",
        "headline",
        "pretitle",
        "pretitleLinesMax",
        "subtitle",
        "subtitleLinesMax",
        "title",
        "titleAs",
        "titleLinesMax",
        "description",
        "descriptionLinesMax",
        "extra",
        "actions",
        "button",
        "buttonLink",
        "dataAttributes",
        "aria-label",
        "onClose"
    ]);
    const B = !!(f.href || f.to || f.onPress), { text: M, ref: V } = S(), { text: O, ref: I } = S(), z = C || [
        s,
        M,
        n,
        d,
        c,
        O
    ].filter(Boolean).join(" ");
    return /* @__PURE__ */ l(q, {
        dataAttributes: _object_spread({
            "component-name": "MediaCard"
        }, x),
        ref: A,
        "aria-label": B ? void 0 : C,
        className: _,
        children: [
            /* @__PURE__ */ e(me, {
                className: K,
                width: "100%",
                height: "100%",
                children: /* @__PURE__ */ l(W, _object_spread_props(_object_spread({
                    maybe: !0
                }, f), {
                    className: $,
                    "aria-label": B ? z : void 0,
                    children: [
                        B && /* @__PURE__ */ e("div", {
                            className: Je
                        }),
                        /* @__PURE__ */ l("div", {
                            className: pe,
                            children: [
                                /* @__PURE__ */ e("div", {
                                    style: G({
                                        [L.mediaBorderRadius]: "0px"
                                    }),
                                    children: a
                                }),
                                /* @__PURE__ */ e("div", {
                                    className: Ke,
                                    children: /* @__PURE__ */ e(ge, {
                                        headline: t,
                                        headlineRef: V,
                                        pretitle: n,
                                        pretitleLinesMax: i,
                                        title: s,
                                        titleAs: v,
                                        titleLinesMax: m,
                                        subtitle: d,
                                        subtitleLinesMax: o,
                                        description: c,
                                        descriptionLinesMax: h,
                                        extra: y,
                                        extraRef: I,
                                        button: g,
                                        buttonLink: N
                                    })
                                }),
                                r && /* @__PURE__ */ e(P, {
                                    className: be,
                                    paddingLeft: {
                                        mobile: 16,
                                        desktop: 24
                                    },
                                    paddingTop: {
                                        mobile: 16,
                                        desktop: 24
                                    },
                                    children: /* @__PURE__ */ e("div", {
                                        style: G({
                                            [L.mediaBorderRadius]: u.borderRadii.mediaSmall
                                        }),
                                        children: r
                                    })
                                })
                            ]
                        })
                    ]
                }))
            }),
            /* @__PURE__ */ e(Q, {
                onClose: T,
                actions: p,
                type: "media"
            })
        ]
    });
}), _a = /*#__PURE__*/ b.forwardRef((_param, A)=>{
    var { media: a, icon: r, headline: t, pretitle: n, pretitleLinesMax: i, subtitle: d, subtitleLinesMax: o, title: s, titleAs: v = "h3", titleLinesMax: m, description: c, descriptionLinesMax: h, extra: y, actions: p, button: g, buttonLink: N, dataAttributes: x, "aria-label": C, onClose: T } = _param, f = _object_without_properties(_param, [
        "media",
        "icon",
        "headline",
        "pretitle",
        "pretitleLinesMax",
        "subtitle",
        "subtitleLinesMax",
        "title",
        "titleAs",
        "titleLinesMax",
        "description",
        "descriptionLinesMax",
        "extra",
        "actions",
        "button",
        "buttonLink",
        "dataAttributes",
        "aria-label",
        "onClose"
    ]);
    const B = !!(f.href || f.to || f.onPress), M = a && a.type === ue && a.props.circular, { text: V, ref: O } = S(), { text: I, ref: z } = S(), R = C || [
        s,
        V,
        n,
        d,
        c,
        I
    ].filter(Boolean).join(" ");
    return /* @__PURE__ */ l(q, {
        ref: A,
        dataAttributes: _object_spread({
            "component-name": "NakedCard"
        }, x),
        "aria-label": B ? void 0 : C,
        className: B ? _ : void 0,
        children: [
            /* @__PURE__ */ e(W, _object_spread_props(_object_spread({
                maybe: !0
            }, f), {
                className: $,
                "aria-label": B ? R : void 0,
                children: /* @__PURE__ */ l("div", {
                    className: pe,
                    children: [
                        /* @__PURE__ */ l("div", {
                            style: {
                                position: "relative"
                            },
                            children: [
                                B && /* @__PURE__ */ e("div", {
                                    className: ye(Ce, {
                                        [Te]: M
                                    })
                                }),
                                a
                            ]
                        }),
                        /* @__PURE__ */ e("div", {
                            className: Ne,
                            children: /* @__PURE__ */ e(ge, {
                                headline: t,
                                headlineRef: O,
                                pretitle: n,
                                pretitleLinesMax: i,
                                title: s,
                                titleAs: v,
                                titleLinesMax: m,
                                subtitle: d,
                                subtitleLinesMax: o,
                                description: c,
                                descriptionLinesMax: h,
                                extra: y,
                                extraRef: z,
                                button: g,
                                buttonLink: N
                            })
                        }),
                        r && /* @__PURE__ */ e(P, {
                            className: be,
                            paddingLeft: {
                                mobile: 16,
                                desktop: 24
                            },
                            paddingTop: {
                                mobile: 16,
                                desktop: 24
                            },
                            children: /* @__PURE__ */ e("div", {
                                style: G({
                                    [L.mediaBorderRadius]: u.borderRadii.mediaSmall
                                }),
                                children: r
                            })
                        })
                    ]
                })
            })),
            /* @__PURE__ */ e(Q, {
                onClose: T,
                actions: p,
                type: "media"
            })
        ]
    });
}), $a = /*#__PURE__*/ b.forwardRef((_param, y)=>{
    var { media: a, title: r, titleAs: t = "h3", titleLinesMax: n, subtitle: i, subtitleLinesMax: d, description: o, descriptionLinesMax: s, extra: v, dataAttributes: m, "aria-label": c } = _param, h = _object_without_properties(_param, [
        "media",
        "title",
        "titleAs",
        "titleLinesMax",
        "subtitle",
        "subtitleLinesMax",
        "description",
        "descriptionLinesMax",
        "extra",
        "dataAttributes",
        "aria-label"
    ]);
    const p = !!(h.href || h.to || h.onPress), g = a && a.type === ue && a.props.circular, { textPresets: N } = J(), { text: x, ref: C } = S(), T = c || [
        r,
        i,
        o,
        x
    ].filter(Boolean).join(" ");
    return /* @__PURE__ */ e(q, {
        ref: y,
        dataAttributes: _object_spread({
            "component-name": "SmallNakedCard"
        }, m),
        "aria-label": p ? void 0 : c,
        className: p ? _ : void 0,
        children: /* @__PURE__ */ e(W, _object_spread_props(_object_spread({
            maybe: !0
        }, h), {
            className: $,
            "aria-label": p ? T : void 0,
            children: /* @__PURE__ */ l("div", {
                className: pe,
                children: [
                    /* @__PURE__ */ l("div", {
                        style: {
                            position: "relative"
                        },
                        children: [
                            p && /* @__PURE__ */ e("div", {
                                className: ye(Ce, {
                                    [Te]: g
                                })
                            }),
                            a
                        ]
                    }),
                    /* @__PURE__ */ l("div", {
                        className: Ne,
                        children: [
                            /* @__PURE__ */ e("div", {
                                children: /* @__PURE__ */ l(te, {
                                    space: 4,
                                    children: [
                                        r && /* @__PURE__ */ e(ne, _object_spread_props(_object_spread({}, ie.text2), {
                                            truncate: n,
                                            weight: N.cardTitle.weight,
                                            as: t,
                                            hyphens: "auto",
                                            children: r
                                        })),
                                        i && /* @__PURE__ */ e(D, {
                                            truncate: d,
                                            regular: !0,
                                            as: "p",
                                            hyphens: "auto",
                                            children: i
                                        }),
                                        o && /* @__PURE__ */ e(D, {
                                            truncate: s,
                                            regular: !0,
                                            as: "p",
                                            color: u.colors.textSecondary,
                                            hyphens: "auto",
                                            children: o
                                        })
                                    ]
                                })
                            }),
                            v && /* @__PURE__ */ e("div", {
                                ref: C,
                                children: v
                            })
                        ]
                    })
                ]
            })
        }))
    });
}), qa = /*#__PURE__*/ b.forwardRef((_param, A)=>{
    var { icon: a, headline: r, pretitle: t, pretitleLinesMax: n, title: i, titleAs: d = "h3", titleLinesMax: o, subtitle: s, subtitleLinesMax: v, description: m, descriptionLinesMax: c, extra: h, actions: y, button: p, buttonLink: g, dataAttributes: N, "aria-label": x, onClose: C, aspectRatio: T } = _param, f = _object_without_properties(_param, [
        "icon",
        "headline",
        "pretitle",
        "pretitleLinesMax",
        "title",
        "titleAs",
        "titleLinesMax",
        "subtitle",
        "subtitleLinesMax",
        "description",
        "descriptionLinesMax",
        "extra",
        "actions",
        "button",
        "buttonLink",
        "dataAttributes",
        "aria-label",
        "onClose",
        "aspectRatio"
    ]);
    const B = !!a || !!r, M = !!(f.href || f.to || f.onPress), { text: V, ref: O } = S(), { text: I, ref: z } = S(), R = ze(y, C), k = x || [
        i,
        V,
        t,
        m,
        I
    ].filter(Boolean).join(" ");
    return /* @__PURE__ */ l(q, {
        dataAttributes: _object_spread({
            "component-name": "DataCard"
        }, N),
        ref: A,
        "aria-label": M ? void 0 : x,
        className: _,
        aspectRatio: T,
        children: [
            /* @__PURE__ */ e(me, {
                className: K,
                width: "100%",
                minHeight: "100%",
                children: /* @__PURE__ */ l(W, _object_spread_props(_object_spread({
                    maybe: !0
                }, f), {
                    className: $,
                    "aria-label": M ? k : void 0,
                    children: [
                        M && /* @__PURE__ */ e("div", {
                            className: oe
                        }),
                        /* @__PURE__ */ l("div", {
                            className: Qe,
                            children: [
                                /* @__PURE__ */ l(Oe, {
                                    space: 0,
                                    children: [
                                        /* @__PURE__ */ l(te, {
                                            space: 16,
                                            children: [
                                                a && /* @__PURE__ */ e("div", {
                                                    style: G({
                                                        [L.mediaBorderRadius]: u.borderRadii.mediaSmall
                                                    }),
                                                    children: a
                                                }),
                                                /* @__PURE__ */ e(ge, {
                                                    headline: r,
                                                    headlineRef: O,
                                                    pretitle: t,
                                                    pretitleLinesMax: n,
                                                    title: i,
                                                    titleAs: d,
                                                    titleLinesMax: o,
                                                    subtitle: s,
                                                    subtitleLinesMax: v,
                                                    description: m,
                                                    descriptionLinesMax: c
                                                })
                                            ]
                                        }),
                                        !B && /* @__PURE__ */ e("div", {
                                            style: G({
                                                [Be.topActionsCount]: String(R.length)
                                            }),
                                            className: Ue
                                        })
                                    ]
                                }),
                                h && /* @__PURE__ */ e("div", {
                                    ref: z,
                                    children: h
                                }),
                                (p || g) && /* @__PURE__ */ e("div", {
                                    className: Re,
                                    children: /* @__PURE__ */ e(he, {
                                        primaryButton: p,
                                        link: g
                                    })
                                })
                            ]
                        })
                    ]
                }))
            }),
            /* @__PURE__ */ e(Q, {
                onClose: C,
                actions: y,
                type: "default"
            })
        ]
    });
}), Ja = /*#__PURE__*/ b.forwardRef((_param, g)=>{
    var { icon: a, title: r, titleAs: t = "h3", titleLinesMax: n, subtitle: i, subtitleLinesMax: d, description: o, descriptionLinesMax: s, dataAttributes: v, "aria-label": m, extra: c, isInverse: h = !1, aspectRatio: y } = _param, p = _object_without_properties(_param, [
        "icon",
        "title",
        "titleAs",
        "titleLinesMax",
        "subtitle",
        "subtitleLinesMax",
        "description",
        "descriptionLinesMax",
        "dataAttributes",
        "aria-label",
        "extra",
        "isInverse",
        "aspectRatio"
    ]);
    const { textPresets: N } = J(), x = !!(p.href || p.to || p.onPress), C = h ? fe : oe, { text: T, ref: f } = S(), A = m || [
        r,
        i,
        o,
        T
    ].filter(Boolean).join(" ");
    return /* @__PURE__ */ e(q, {
        dataAttributes: _object_spread({
            "component-name": "SnapCard"
        }, v),
        ref: g,
        className: _,
        aspectRatio: y,
        "aria-label": x ? void 0 : m,
        children: /* @__PURE__ */ e(me, {
            className: K,
            isInverse: h,
            width: "100%",
            minHeight: "100%",
            children: /* @__PURE__ */ l(W, _object_spread_props(_object_spread({
                maybe: !0
            }, p), {
                className: $,
                "aria-label": x ? A : void 0,
                children: [
                    x && /* @__PURE__ */ e("div", {
                        className: C
                    }),
                    /* @__PURE__ */ l("section", {
                        className: Ye,
                        children: [
                            /* @__PURE__ */ l("div", {
                                children: [
                                    a && /* @__PURE__ */ e("div", {
                                        style: G({
                                            [L.mediaBorderRadius]: u.borderRadii.mediaSmall
                                        }),
                                        children: /* @__PURE__ */ e(P, {
                                            paddingBottom: 16,
                                            children: a
                                        })
                                    }),
                                    /* @__PURE__ */ l(te, {
                                        space: 4,
                                        children: [
                                            r && /* @__PURE__ */ e(ne, _object_spread_props(_object_spread({}, ie.text2), {
                                                truncate: n,
                                                weight: N.cardTitle.weight,
                                                as: t,
                                                hyphens: "auto",
                                                children: r
                                            })),
                                            i && /* @__PURE__ */ e(D, {
                                                truncate: d,
                                                regular: !0,
                                                color: u.colors.textPrimary,
                                                as: "p",
                                                hyphens: "auto",
                                                children: i
                                            }),
                                            o && /* @__PURE__ */ e(D, {
                                                truncate: s,
                                                regular: !0,
                                                color: u.colors.textSecondary,
                                                as: "p",
                                                hyphens: "auto",
                                                children: o
                                            })
                                        ]
                                    })
                                ]
                            }),
                            c && /* @__PURE__ */ e("div", {
                                ref: f,
                                children: c
                            })
                        ]
                    })
                ]
            }))
        })
    });
}), Ee = (param)=>{
    let { title: a, headline: r, pretitle: t, subtitle: n, description: i, extra: d, headlineRef: o, extraRef: s } = param;
    return /* @__PURE__ */ l("div", {
        className: Ae,
        children: [
            a && /* @__PURE__ */ e("div", {
                style: {
                    paddingBottom: n || i ? 4 : 0
                },
                children: a
            }),
            r && // assuming that the headline will always be followed by one of: pretitle, title, subtitle, description
            /* @__PURE__ */ e("div", {
                ref: o,
                style: {
                    order: -2,
                    paddingBottom: 16
                },
                children: r
            }),
            t && /* @__PURE__ */ e("div", {
                style: {
                    order: -1,
                    paddingBottom: 4
                },
                children: t
            }),
            n && /* @__PURE__ */ e("div", {
                style: {
                    paddingBottom: n ? 4 : 0
                },
                children: n
            }),
            i && // this is tricky, the padding between a headline and a description is 16px
            // but the padding between a title|pretitle|subtitle and a description is 8px (4px + 4px)
            /* @__PURE__ */ e("div", {
                style: {
                    paddingTop: t || a || n ? 4 : 0
                },
                children: i
            }),
            d && /* @__PURE__ */ e("div", {
                ref: s,
                children: d
            })
        ]
    });
}, Ge = /*#__PURE__*/ b.forwardRef((_param, z)=>{
    var { isInverse: a, backgroundImage: r, backgroundVideo: t, backgroundVideoRef: n, poster: i, icon: d, headline: o, pretitle: s, pretitleLinesMax: v, title: m, titleAs: c = "h3", titleLinesMax: h, description: y, descriptionLinesMax: p, extra: g, button: N, secondaryButton: x, onClose: C, actions: T, buttonLink: f, dataAttributes: A, width: B, height: M, aspectRatio: V, "aria-label": O } = _param, I = _object_without_properties(_param, [
        "isInverse",
        "backgroundImage",
        "backgroundVideo",
        "backgroundVideoRef",
        "poster",
        "icon",
        "headline",
        "pretitle",
        "pretitleLinesMax",
        "title",
        "titleAs",
        "titleLinesMax",
        "description",
        "descriptionLinesMax",
        "extra",
        "button",
        "secondaryButton",
        "onClose",
        "actions",
        "buttonLink",
        "dataAttributes",
        "width",
        "height",
        "aspectRatio",
        "aria-label"
    ]);
    const R = r !== void 0, k = t !== void 0, H = Pe(r), { video: w, videoAction: U } = Ve(t, i, n), { text: de, ref: Y } = S(), { text: se, ref: le } = S();
    k && (T = U ? [
        U
    ] : []);
    const Z = De(), E = R || k, X = E ? "0 0 16px rgba(0,0,0,0.4)" : void 0, j = (T == null ? void 0 : T.length) || C, F = !!(I.href || I.to || I.onPress), ee = R || k ? Me : a ? fe : oe, ce = O || [
        m,
        de,
        s,
        y,
        se
    ].filter(Boolean).join(" ");
    return /* @__PURE__ */ l(q, {
        dataAttributes: A,
        ref: z,
        width: B,
        height: M,
        aspectRatio: V,
        "aria-label": F ? void 0 : O,
        className: _,
        children: [
            /* @__PURE__ */ e(xe, {
                borderRadius: u.borderRadii.legacyDisplay,
                className: K,
                width: "100%",
                minHeight: "100%",
                isInverse: a,
                background: R || k ? Z ? u.colors.backgroundContainerBrandOverInverse : u.colors.backgroundContainer : void 0,
                children: /* @__PURE__ */ l(W, _object_spread_props(_object_spread({
                    maybe: !0
                }, I), {
                    className: $,
                    "aria-label": F ? ce : void 0,
                    children: [
                        F && /* @__PURE__ */ e("div", {
                            className: ee
                        }),
                        /* @__PURE__ */ l("div", {
                            className: ke,
                            children: [
                                (R || k) && /* @__PURE__ */ e(ve, {
                                    isInverse: Z,
                                    children: /* @__PURE__ */ e("div", {
                                        className: Ie,
                                        children: k ? w : H
                                    })
                                }),
                                /* @__PURE__ */ l("div", {
                                    className: we,
                                    style: {
                                        paddingTop: E && !d && !j && !k ? 0 : 24
                                    },
                                    children: [
                                        d ? /* @__PURE__ */ e("div", {
                                            style: G({
                                                [L.mediaBorderRadius]: u.borderRadii.mediaSmall
                                            }),
                                            children: /* @__PURE__ */ e(P, {
                                                paddingBottom: E ? 0 : 40,
                                                paddingX: 24,
                                                children: d
                                            })
                                        }) : /* @__PURE__ */ e(P, {
                                            paddingBottom: j || k ? E ? 24 : 64 : 0
                                        }),
                                        /* @__PURE__ */ e(P, {
                                            paddingX: 24,
                                            paddingTop: E ? 40 : 0,
                                            paddingBottom: 24,
                                            className: E ? Se : void 0,
                                            children: /* @__PURE__ */ l(te, {
                                                space: 24,
                                                children: [
                                                    /* @__PURE__ */ e(Ee, {
                                                        title: m ? /* @__PURE__ */ e(_e, {
                                                            forceMobileSizes: !0,
                                                            truncate: h,
                                                            as: c,
                                                            textShadow: X,
                                                            hyphens: "auto",
                                                            children: m
                                                        }) : void 0,
                                                        headline: o,
                                                        pretitle: s ? /* @__PURE__ */ e(D, {
                                                            forceMobileSizes: !0,
                                                            truncate: v,
                                                            as: "div",
                                                            regular: !0,
                                                            textShadow: X,
                                                            children: s
                                                        }) : void 0,
                                                        description: y ? /* @__PURE__ */ e($e, {
                                                            forceMobileSizes: !0,
                                                            truncate: p,
                                                            as: "p",
                                                            regular: !0,
                                                            color: E ? u.colors.textPrimary : u.colors.textSecondary,
                                                            textShadow: X,
                                                            hyphens: "auto",
                                                            children: y
                                                        }) : void 0,
                                                        extra: g,
                                                        headlineRef: Y,
                                                        extraRef: le
                                                    }),
                                                    (N || x || f) && /* @__PURE__ */ e(he, {
                                                        primaryButton: N,
                                                        secondaryButton: x,
                                                        link: f
                                                    })
                                                ]
                                            })
                                        })
                                    ]
                                })
                            ]
                        })
                    ]
                }))
            }),
            /* @__PURE__ */ e(Q, {
                onClose: C,
                actions: T,
                type: R || k ? "media" : a ? "inverse" : "default"
            })
        ]
    });
}), Ka = /*#__PURE__*/ b.forwardRef((_param, t)=>{
    var { dataAttributes: a } = _param, r = _object_without_properties(_param, [
        "dataAttributes"
    ]);
    return /* @__PURE__ */ e(Ge, _object_spread_props(_object_spread({}, r), {
        ref: t,
        isInverse: !0,
        dataAttributes: _object_spread({
            "component-name": "DisplayMediaCard"
        }, a)
    }));
}), Qa = /*#__PURE__*/ b.forwardRef((_param, t)=>{
    var { dataAttributes: a } = _param, r = _object_without_properties(_param, [
        "dataAttributes"
    ]);
    return /* @__PURE__ */ e(Ge, _object_spread_props(_object_spread({}, r), {
        ref: t,
        dataAttributes: _object_spread({
            "component-name": "DisplayDataCard"
        }, a)
    }));
}), Ua = /*#__PURE__*/ b.forwardRef((_param, k)=>{
    var { dataAttributes: a, backgroundImage: r, backgroundVideo: t, poster: n, backgroundVideoRef: i, width: d, height: o, aspectRatio: s = "7:10", ariaLabel: v, ["aria-label"]: m = v, actions: c, onClose: h, icon: y, headline: p, pretitle: g, pretitleLinesMax: N, title: x, titleAs: C = "h3", titleLinesMax: T, subtitle: f, subtitleLinesMax: A, description: B, descriptionLinesMax: M, extra: V, variant: O, isInverse: I, backgroundColor: z } = _param, R = _object_without_properties(_param, [
        "dataAttributes",
        "backgroundImage",
        "backgroundVideo",
        "poster",
        "backgroundVideoRef",
        "width",
        "height",
        "aspectRatio",
        "ariaLabel",
        "aria-label",
        "actions",
        "onClose",
        "icon",
        "headline",
        "pretitle",
        "pretitleLinesMax",
        "title",
        "titleAs",
        "titleLinesMax",
        "subtitle",
        "subtitleLinesMax",
        "description",
        "descriptionLinesMax",
        "extra",
        "variant",
        "isInverse",
        "backgroundColor"
    ]);
    const H = r !== void 0, w = t !== void 0, U = Pe(r), { video: de, videoAction: Y } = Ve(t, n, i), { text: se, ref: le } = S(), { text: Z, ref: E } = S();
    w && (c = Y ? [
        Y
    ] : []);
    const X = De(), j = H || w, F = j ? "0 0 16px rgba(0,0,0,0.4)" : void 0, ee = (c == null ? void 0 : c.length) || h, { textPresets: ce } = J(), ae = !!(R.href || R.to || R.onPress), re = O || (I ? "inverse" : "default"), He = ()=>z || ({
            default: u.colors.backgroundContainer,
            inverse: X ? u.colors.backgroundContainerBrandOverInverse : u.colors.backgroundBrand,
            alternative: u.colors.backgroundAlternative
        })[re], Xe = H || w ? Me : re === "inverse" ? fe : oe, Fe = m || [
        x,
        se,
        g,
        f,
        B,
        Z
    ].filter(Boolean).join(" ");
    return /* @__PURE__ */ l(q, {
        width: d,
        height: o,
        dataAttributes: _object_spread({
            "component-name": "PosterCard"
        }, a),
        ref: k,
        aspectRatio: s,
        className: _,
        "aria-label": ae ? void 0 : m,
        children: [
            /* @__PURE__ */ e(xe, {
                borderRadius: u.borderRadii.legacyDisplay,
                className: K,
                width: "100%",
                minHeight: "100%",
                isInverse: H || w || re === "inverse",
                background: H || w ? X ? u.colors.backgroundContainerBrandOverInverse : u.colors.backgroundContainer : He(),
                children: /* @__PURE__ */ l(W, _object_spread_props(_object_spread({
                    maybe: !0
                }, R), {
                    className: $,
                    "aria-label": ae ? Fe : void 0,
                    children: [
                        ae && /* @__PURE__ */ e("div", {
                            className: Xe
                        }),
                        /* @__PURE__ */ l("div", {
                            className: ke,
                            "aria-hidden": ae,
                            children: [
                                (H || w) && /* @__PURE__ */ e(ve, {
                                    isInverse: X,
                                    children: /* @__PURE__ */ e("div", {
                                        className: Ie,
                                        children: w ? de : U
                                    })
                                }),
                                /* @__PURE__ */ l(P, {
                                    className: we,
                                    paddingTop: j && !y && !ee && !w ? 0 : {
                                        mobile: y ? 16 : 24,
                                        desktop: 24
                                    },
                                    children: [
                                        y ? /* @__PURE__ */ e("div", {
                                            style: G({
                                                [L.mediaBorderRadius]: u.borderRadii.mediaSmall
                                            }),
                                            children: /* @__PURE__ */ e(P, {
                                                paddingBottom: j ? 0 : 40,
                                                paddingX: {
                                                    mobile: 16,
                                                    desktop: 24
                                                },
                                                children: y
                                            })
                                        }) : /* @__PURE__ */ e(P, {
                                            paddingBottom: ee || w ? j ? 24 : 64 : 0
                                        }),
                                        /* @__PURE__ */ e(P, {
                                            paddingX: {
                                                mobile: 16,
                                                desktop: 24
                                            },
                                            paddingTop: j ? 40 : 0,
                                            paddingBottom: 24,
                                            className: j ? Se : void 0,
                                            children: /* @__PURE__ */ e(Ee, {
                                                title: x ? /* @__PURE__ */ e(ne, _object_spread_props(_object_spread({}, ie.text4), {
                                                    truncate: T,
                                                    weight: ce.cardTitle.weight,
                                                    as: C,
                                                    children: x
                                                })) : void 0,
                                                headline: p,
                                                pretitle: g ? /* @__PURE__ */ e(D, {
                                                    forceMobileSizes: !0,
                                                    truncate: N,
                                                    as: "div",
                                                    regular: !0,
                                                    textShadow: F,
                                                    children: g
                                                }) : void 0,
                                                subtitle: f ? /* @__PURE__ */ e(D, {
                                                    forceMobileSizes: !0,
                                                    truncate: A,
                                                    as: "div",
                                                    regular: !0,
                                                    textShadow: F,
                                                    children: f
                                                }) : void 0,
                                                description: B ? /* @__PURE__ */ e(D, {
                                                    forceMobileSizes: !0,
                                                    truncate: M,
                                                    as: "p",
                                                    regular: !0,
                                                    textShadow: F,
                                                    color: j ? u.colors.textPrimary : u.colors.textSecondary,
                                                    children: B
                                                }) : void 0,
                                                headlineRef: le,
                                                extra: V,
                                                extraRef: E
                                            })
                                        })
                                    ]
                                })
                            ]
                        })
                    ]
                }))
            }),
            /* @__PURE__ */ e(Q, {
                onClose: h,
                actions: c,
                type: H || w ? "media" : re === "inverse" ? "inverse" : "default"
            })
        ]
    });
});
export { ca as CardActionIconButton, pa as CardActionSpinner, Q as CardActionsGroup, qa as DataCard, Qa as DisplayDataCard, Ka as DisplayMediaCard, Wa as MediaCard, _a as NakedCard, Ua as PosterCard, $a as SmallNakedCard, Ja as SnapCard, Ve as useVideoWithControls };
