// @telefonica/mistica/dist-es/counter.js
"use client";
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
import { jsx as r, jsxs as I } from "react/jsx-runtime";
import * as B from "react";
import { getPrefixedDataAttributes as R } from "./utils/dom.js";
import { counter as j, disabled as N, buttonContainer as M, hasTrashIcon as E, isButtonDisabled as y, button as S, buttonBackground as A, decreaseButtonIcon as O, trashButtonIcon as Q, valueContainer as Z, defaultButtonIcon as _ } from "./counter.css-mistica.js";
import { useThemeVariant as q, ThemeVariant as w } from "./theme-variant-context.js";
import { vars as c } from "./skins/skin-contract.css-mistica.js";
import F from "./inline.js";
import { Text3 as G } from "./text.js";
import { BaseTouchable as P } from "./touchable.js";
import H from "./generated/mistica-icons/icon-subtract-regular.js";
import J from "./generated/mistica-icons/icon-add-more-regular.js";
import K from "./generated/mistica-icons/icon-trash-can-regular.js";
import { useAriaId as U, useTheme as W } from "./hooks.js";
import x from "classnames";
import X from "./screen-reader-only.js";
const C = 20, Y = (param)=>{
    let { value: h, defaultValue: b, min: m, max: s, onChangeValue: a } = param;
    const i = h !== void 0, p = B.useCallback((n)=>n === void 0 ? m : Math.max(m, Math.min(s, n)), [
        m,
        s
    ]), [e, v] = B.useState(p(b)), f = (n)=>{
        i || v(n), a == null || a(n);
    };
    return [
        i ? p(h) : e,
        f
    ];
}, rr = (param)=>{
    let { value: h, defaultValue: b, onChangeValue: m, onRemove: s, min: a, max: i, dataAttributes: p, disabled: e, removeLabel: v, increaseLabel: f, decreaseLabel: n, valueLabel: T } = param;
    const D = q(), g = U(), { texts: u } = W(), l = a === void 0 ? 0 : a, d = Math.max(l, i === void 0 ? 999 : i), [t, V] = Y({
        value: h,
        defaultValue: b,
        min: l,
        max: d,
        onChangeValue: m
    }), o = !!s && t === l, $ = ()=>v === void 0 ? u.counterRemoveLabel : v, k = ()=>f === void 0 ? u.counterIncreaseLabel : f, z = ()=>n === void 0 ? u.counterDecreaseLabel : n, L = ()=>`${t}, ${T === void 0 ? u.counterQuantity : T}${a !== void 0 ? `, ${u.counterMinValue} ${a}` : ""}${i !== void 0 ? `, ${u.counterMaxValue} ${i}` : ""}`;
    return /* @__PURE__ */ r("div", _object_spread_props(_object_spread({
        className: x(j, {
            [N]: e
        })
    }, R(p, "Counter")), {
        style: {
            border: `1px solid ${D === "inverse" ? c.colors.backgroundContainer : c.colors.border}`
        },
        children: /* @__PURE__ */ I(F, {
            space: 8,
            alignItems: "center",
            children: [
                /* @__PURE__ */ r("div", {
                    className: x(M, {
                        [E]: o,
                        [y]: e || !o && t === l,
                        [N]: !e && !o && t === l
                    }),
                    children: /* @__PURE__ */ I(P, {
                        className: S,
                        disabled: !o && t === l || e,
                        onPress: ()=>o ? s == null ? void 0 : s() : V(t - 1),
                        "aria-label": o ? $() : z(),
                        "aria-controls": g,
                        children: [
                            /* @__PURE__ */ r("div", {
                                className: A
                            }),
                            /* @__PURE__ */ r("div", {
                                className: O,
                                "aria-hidden": o,
                                children: /* @__PURE__ */ r(H, {
                                    size: C,
                                    color: t === l && !o || e ? c.colors.control : c.colors.controlActivated
                                })
                            }),
                            /* @__PURE__ */ r("div", {
                                className: Q,
                                "aria-hidden": !o,
                                children: /* @__PURE__ */ r(K, {
                                    size: C,
                                    color: c.colors.controlError
                                })
                            })
                        ]
                    })
                }),
                /* @__PURE__ */ r("div", {
                    className: Z,
                    style: {
                        width: `${Math.max(3, String(d).length) * 1.25}ch`
                    },
                    id: g,
                    "aria-live": "polite",
                    children: /* @__PURE__ */ I(w, {
                        variant: "default",
                        children: [
                            /* @__PURE__ */ r("div", {
                                "aria-hidden": !0,
                                children: /* @__PURE__ */ r(G, {
                                    regular: !0,
                                    color: c.colors.textPrimary,
                                    children: t
                                })
                            }),
                            /* @__PURE__ */ r(X, {
                                children: /* @__PURE__ */ r("span", {
                                    children: L()
                                })
                            })
                        ]
                    })
                }),
                /* @__PURE__ */ r("div", {
                    className: x(M, {
                        [y]: e || t === d,
                        [N]: !e && t === d
                    }),
                    children: /* @__PURE__ */ I(P, {
                        className: S,
                        disabled: t === d || e,
                        onPress: ()=>V(t + 1),
                        "aria-label": k(),
                        "aria-controls": g,
                        children: [
                            /* @__PURE__ */ r("div", {
                                className: A
                            }),
                            /* @__PURE__ */ r("div", {
                                className: _,
                                children: /* @__PURE__ */ r(J, {
                                    size: C,
                                    color: t === d || e ? c.colors.control : c.colors.controlActivated
                                })
                            })
                        ]
                    })
                })
            ]
        })
    }));
}, vr = rr;
export { vr as default };
