// @telefonica/mistica/dist-es/cover-hero.js
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
function _object_without_properties(source, excluded) {
    if (source == null) return {};
    var target = _object_without_properties_loose(source, excluded);
    var key, i;
    if (Object.getOwnPropertySymbols) {
        var sourceSymbolKeys = Object.getOwnPropertySymbols(source);
        for(i = 0; i < sourceSymbolKeys.length; i++){
            key = sourceSymbolKeys[i];
            if (excluded.indexOf(key) >= 0) continue;
            if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
            target[key] = source[key];
        }
    }
    return target;
}
function _object_without_properties_loose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;
    for(i = 0; i < sourceKeys.length; i++){
        key = sourceKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        target[key] = source[key];
    }
    return target;
}
import { jsxs as n, jsx as r } from "react/jsx-runtime";
import * as z from "react";
import D from "./responsive-layout.js";
import v from "./box.js";
import I from "./button-group.js";
import m from "./stack.js";
import { Text3 as g, Text8 as q } from "./text.js";
import { vars as i } from "./skins/skin-contract.css-mistica.js";
import { mainContent as F, sixColumns as k, coverHeroContainer as J, minHeight as K, withAspectRatio as O, coverHero as Q, centered as U, hasSideExtra as W, sideExtra as X, vars as Y } from "./cover-hero.css-mistica.js";
import x from "classnames";
import { applyCssVars as Z } from "./utils/css.js";
import { vars as _ } from "./image.css-mistica.js";
import C from "./grid-layout.js";
import { CoverHeroMedia as E } from "./cover-hero-media.js";
import { getPrefixedDataAttributes as P } from "./utils/dom.js";
const $ = (e)=>!e || e === "auto" ? 0 : typeof e == "number" ? e : ({
        "1:1": 1,
        "16:9": 16 / 9,
        "7:10": 7 / 10,
        "4:3": 9 / 10
    })[e], rr = /*#__PURE__*/ z.forwardRef((_param, R)=>{
    var { headline: e, pretitle: d, pretitleLinesMax: H, title: y, titleLinesMax: B, titleAs: N = "h1", description: u, descriptionLinesMax: S, extra: w, sideExtra: o, button: A, secondaryButton: T, buttonLink: j, minHeight: G, aspectRatio: c = "auto", variant: p, centered: t, noPaddingY: f, dataAttributes: L, "aria-label": M } = _param, l = _object_without_properties(_param, [
        "headline",
        "pretitle",
        "pretitleLinesMax",
        "title",
        "titleLinesMax",
        "titleAs",
        "description",
        "descriptionLinesMax",
        "extra",
        "sideExtra",
        "button",
        "secondaryButton",
        "buttonLink",
        "minHeight",
        "aspectRatio",
        "variant",
        "centered",
        "noPaddingY",
        "dataAttributes",
        "aria-label"
    ]);
    const a = l.backgroundVideo || l.backgroundImage, V = a ? "none" : l.background || ({
        default: i.colors.background,
        inverse: i.colors.backgroundBrand,
        alternative: i.colors.backgroundAlternative
    })[p !== null && p !== void 0 ? p : "default"], s = a ? "0 0 15px rgba(0, 0, 0, 0.4)" : void 0, h = /* @__PURE__ */ n("div", {
        className: F,
        children: [
            e && /* @__PURE__ */ r(v, {
                paddingBottom: {
                    desktop: 8,
                    tablet: 8,
                    mobile: 16
                },
                children: e
            }),
            /* @__PURE__ */ n(m, {
                space: 16,
                children: [
                    /* @__PURE__ */ n(m, {
                        space: 8,
                        children: [
                            d && /* @__PURE__ */ r("div", {
                                className: k,
                                children: /* @__PURE__ */ r(g, {
                                    regular: !0,
                                    truncate: H,
                                    textShadow: s,
                                    children: d
                                })
                            }),
                            /* @__PURE__ */ r(q, {
                                as: N,
                                truncate: B,
                                textShadow: s,
                                children: y
                            })
                        ]
                    }),
                    u && /* @__PURE__ */ r("div", {
                        className: k,
                        children: /* @__PURE__ */ r(g, {
                            as: "p",
                            regular: !0,
                            truncate: S,
                            color: a ? i.colors.textPrimary : i.colors.textSecondary,
                            textShadow: s,
                            children: u
                        })
                    })
                ]
            }),
            w
        ]
    }), b = c && c !== "auto";
    return /* @__PURE__ */ r("section", _object_spread_props(_object_spread({}, P(L, "CoverHero")), {
        "aria-label": M,
        ref: R,
        className: x(J, {
            [K]: !f,
            [O]: b
        }),
        style: _object_spread({
            minHeight: b ? "auto" : G,
            boxSizing: "border-box",
            background: V
        }, Z({
            [Y.aspectRatio]: String($(c)),
            [_.mediaBorderRadius]: "0px"
        })),
        children: /* @__PURE__ */ n("div", {
            className: x(Q, {
                [U]: t,
                [W]: o
            }),
            children: [
                a ? /* @__PURE__ */ r(E, _object_spread({}, l)) : null,
                /* @__PURE__ */ r(D, {
                    variant: a ? "inverse" : p,
                    backgroundColor: "none",
                    children: /* @__PURE__ */ r(v, {
                        paddingY: f ? 0 : {
                            desktop: 56,
                            tablet: 56,
                            mobile: 24
                        },
                        children: /* @__PURE__ */ n(m, {
                            space: 24,
                            children: [
                                t && !o ? /* @__PURE__ */ r(C, {
                                    template: "8",
                                    children: h
                                }) : /* @__PURE__ */ r(C, {
                                    template: "8+4",
                                    collapseBreakpoint: "mobile",
                                    left: h,
                                    right: /* @__PURE__ */ r("div", {
                                        className: X,
                                        children: o
                                    })
                                }),
                                /* @__PURE__ */ r(I, {
                                    align: {
                                        mobile: t ? "center" : "left",
                                        tablet: t && !o ? "center" : "left",
                                        desktop: t && !o ? "center" : "left"
                                    },
                                    primaryButton: A,
                                    secondaryButton: T,
                                    link: j
                                })
                            ]
                        })
                    })
                })
            ]
        })
    }));
}), hr = rr;
export { hr as default };
