// @telefonica/mistica/dist-es/credit-card-expiration-field.js
"use client";
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
function _object_without_properties(source, excluded) {
    if (source == null) return {};
    var target = _object_without_properties_loose(source, excluded);
    var key, i;
    if (Object.getOwnPropertySymbols) {
        var sourceSymbolKeys = Object.getOwnPropertySymbols(source);
        for(i = 0; i < sourceSymbolKeys.length; i++){
            key = sourceSymbolKeys[i];
            if (excluded.indexOf(key) >= 0) continue;
            if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
            target[key] = source[key];
        }
    }
    return target;
}
function _object_without_properties_loose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;
    for(i = 0; i < sourceKeys.length; i++){
        key = sourceKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        target[key] = source[key];
    }
    return target;
}
import { jsx as F } from "react/jsx-runtime";
import * as k from "react";
import { useForm as v, useFieldProps as R } from "./form-context.js";
import { useTheme as $ } from "./hooks.js";
import { TextFieldBaseAutosuggest as b } from "./text-field-base.js";
const j = (_param)=>{
    var { inputRef: g, defaultValue: s, value: l } = _param, c = _object_without_properties(_param, [
        "inputRef",
        "defaultValue",
        "value"
    ]);
    const { texts: p } = $(), i = k.useRef(""), u = (r)=>{
        if (r.length === 2 && i.current === "Backspace") return r[0];
        let e = "";
        return [
            ...r
        ].forEach((x)=>{
            e = (e + x).replace(/[^0-9]/g, "").replace(/^([2-9])$/g, "0$1").replace(/^(1{1})([3-9]{1})$/g, "0$1/$2").replace(/^0{1,}/g, "0").replace(/^([0-1]{1}[0-9]{1})([0-9]{1,2}).*/g, "$1/$2");
        }), e.length === 2 ? e + "/" : e;
    }, h = (r)=>{
        i.current = r.key;
    };
    return /* @__PURE__ */ F("input", _object_spread_props(_object_spread({}, c), {
        placeholder: p.expirationDatePlaceholder,
        type: "text",
        inputMode: "decimal",
        maxLength: "5",
        onKeyDown: h,
        onInput: (r)=>{
            const e = u(r.currentTarget.value);
            r.currentTarget.value = e;
        },
        value: l === void 0 ? void 0 : u(l),
        defaultValue: s === void 0 ? void 0 : u(s),
        ref: g
    }));
}, A = (_param)=>{
    var { disabled: g, error: s, helperText: l, name: c, optional: p, validate: i, onChange: u, onChangeValue: h, onBlur: r, value: e, autoComplete: x = "cc-exp", defaultValue: D, dataAttributes: T } = _param, w = _object_without_properties(_param, [
        "disabled",
        "error",
        "helperText",
        "name",
        "optional",
        "validate",
        "onChange",
        "onChangeValue",
        "onBlur",
        "value",
        "autoComplete",
        "defaultValue",
        "dataAttributes"
    ]);
    const { texts: d } = $(), { setFormError: I, jumpToNext: M } = v(), m = (t, n)=>{
        if (!n) return p ? "" : d.formFieldErrorIsMandatory;
        const { month: a, year: o } = t;
        if (!a || !o) return d.formCreditCardExpirationError;
        const f = /* @__PURE__ */ new Date(), Y = f.getMonth() + 1, y = f.getFullYear();
        return o < y || o === y && a < Y ? d.formCreditCardExpirationError : i == null ? void 0 : i(t, n);
    }, C = (t)=>{
        const [n, a] = String(t).split("/").map((f)=>parseInt(f)), o = Number.isInteger(a) ? 2e3 + a : null;
        return {
            month: n || null,
            year: o,
            raw: t
        };
    }, E = R({
        name: c,
        value: e,
        defaultValue: D,
        processValue: C,
        helperText: l,
        optional: p,
        error: s,
        disabled: g,
        onBlur: r,
        validate: m,
        onChange: u,
        onChangeValue: h
    });
    return /* @__PURE__ */ F(b, _object_spread_props(_object_spread({}, w, E), {
        onChange: (t)=>{
            E.onChange(t);
            const n = t.currentTarget.value, a = C(n);
            if (n.length === 5) {
                const o = m == null ? void 0 : m(a, n);
                o ? I({
                    name: c,
                    error: o
                }) : M(c);
            }
        },
        autoComplete: x,
        inputComponent: j,
        dataAttributes: _object_spread({
            "component-name": "CreditCardExpirationField"
        }, T)
    }));
}, L = A;
export { L as default };
