// @telefonica/mistica/dist-es/credit-card-fields.js
"use client";
import { jsxs as i, jsx as e } from "react/jsx-runtime";
import { useForm as C } from "./form-context.js";
import { useTheme as n } from "./hooks.js";
import { getCvvLength as s } from "./utils/credit-card.js";
import c from "./double-field.js";
import f from "./stack.js";
import p from "./credit-card-expiration-field.js";
import v from "./credit-card-number-field.js";
import u from "./cvv-field.js";
const b = (param)=>{
    let { numberFieldName: t = "ccNum", expirationFieldName: m = "ccExp", cvvFieldName: a = "ccCvv", acceptedCards: o } = param;
    const { values: d } = C(), { texts: r } = n(), l = s(String(d[t]));
    return /* @__PURE__ */ i(f, {
        space: 16,
        children: [
            /* @__PURE__ */ e(v, {
                acceptedCards: o,
                name: t,
                label: r.formCreditCardNumberLabel
            }),
            /* @__PURE__ */ i(c, {
                layout: "60/40",
                children: [
                    /* @__PURE__ */ e(p, {
                        name: m,
                        label: r.formCreditCardExpirationLabel
                    }),
                    /* @__PURE__ */ e(u, {
                        acceptedCards: o,
                        name: a,
                        label: r.formCreditCardCvvLabel,
                        maxLength: l
                    })
                ]
            })
        ]
    });
}, k = b;
export { k as default };
