// @telefonica/mistica/dist-es/date-field.js
"use client";
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
function _object_without_properties(source, excluded) {
    if (source == null) return {};
    var target = _object_without_properties_loose(source, excluded);
    var key, i;
    if (Object.getOwnPropertySymbols) {
        var sourceSymbolKeys = Object.getOwnPropertySymbols(source);
        for(i = 0; i < sourceSymbolKeys.length; i++){
            key = sourceSymbolKeys[i];
            if (excluded.indexOf(key) >= 0) continue;
            if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
            target[key] = source[key];
        }
    }
    return target;
}
function _object_without_properties_loose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;
    for(i = 0; i < sourceKeys.length; i++){
        key = sourceKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        target[key] = source[key];
    }
    return target;
}
import { jsx as t } from "react/jsx-runtime";
import * as c from "react";
import { useFieldProps as z } from "./form-context.js";
import { TextFieldBaseAutosuggest as O } from "./text-field-base.js";
import { isInputTypeSupported as b } from "./utils/dom.js";
import { isServerSide as A } from "./utils/environment.js";
import N from "./generated/mistica-icons/icon-calendar-regular.js";
import { getLocalDateString as r } from "./utils/time.js";
import { useTheme as j } from "./hooks.js";
import { isFirefox as v } from "./utils/platform.js";
import { iconContainer as B } from "./date-field.css-mistica.js";
import { iconSize as C } from "./icon-button.css-mistica.js";
const E = /*#__PURE__*/ c.lazy(()=>import(/* webpackChunkName: "date-time-picker" */ "./date-time-picker.js")), L = (_param)=>{
    var { disabled: D, error: F, helperText: g, name: S, optional: f, validate: s, onChange: R, onChangeValue: n, onBlur: k, value: y, defaultValue: I, min: i, max: o, dataAttributes: d } = _param, u = _object_without_properties(_param, [
        "disabled",
        "error",
        "helperText",
        "name",
        "optional",
        "validate",
        "onChange",
        "onChangeValue",
        "onBlur",
        "value",
        "defaultValue",
        "min",
        "max",
        "dataAttributes"
    ]);
    const T = (e)=>e, h = c.useMemo(()=>v() ? !1 : b("date"), []), { texts: x } = j(), a = (e)=>!(i && e && e < r(i) || o && e && e > r(o)), l = z({
        name: S,
        value: y,
        defaultValue: I,
        processValue: T,
        helperText: g,
        optional: f,
        error: F,
        disabled: D,
        onBlur: k,
        validate: (e, m)=>a(e) ? s == null ? void 0 : s(e, m) : x.formDateOutOfRangeError,
        onChange: R,
        onChangeValue: (e, m)=>{
            a(e) && (n == null || n(e, m));
        }
    }), p = /* @__PURE__ */ t(O, _object_spread_props(_object_spread({}, u, l), {
        min: i ? r(i) : void 0,
        max: o ? r(o) : void 0,
        type: "date",
        endIconOverlay: /* @__PURE__ */ t("div", {
            className: B,
            children: /* @__PURE__ */ t(N, {
                size: C.default
            })
        }),
        dataAttributes: _object_spread({
            "component-name": "DateField"
        }, d)
    }));
    return h || A() ? p : /* @__PURE__ */ t(c.Suspense, {
        fallback: p,
        children: /* @__PURE__ */ t(E, _object_spread_props(_object_spread({}, u, l), {
            optional: f,
            isValidDate: (e)=>a(r(e.toDate())),
            dataAttributes: _object_spread({
                "component-name": "DateField"
            }, d)
        }))
    });
}, Y = L;
export { Y as default };
