// @telefonica/mistica/dist-es/decimal-field.js
"use client";
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
function _object_without_properties(source, excluded) {
    if (source == null) return {};
    var target = _object_without_properties_loose(source, excluded);
    var key, i;
    if (Object.getOwnPropertySymbols) {
        var sourceSymbolKeys = Object.getOwnPropertySymbols(source);
        for(i = 0; i < sourceSymbolKeys.length; i++){
            key = sourceSymbolKeys[i];
            if (excluded.indexOf(key) >= 0) continue;
            if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
            target[key] = source[key];
        }
    }
    return target;
}
function _object_without_properties_loose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;
    for(i = 0; i < sourceKeys.length; i++){
        key = sourceKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        target[key] = source[key];
    }
    return target;
}
import { jsx as x } from "react/jsx-runtime";
import * as S from "react";
import { useFieldProps as V } from "./form-context.js";
import { useTheme as C } from "./hooks.js";
import { TextFieldBaseAutosuggest as R } from "./text-field-base.js";
import { createChangeEvent as b } from "./utils/dom.js";
import { useRifm as I } from "rifm";
import { combineRefs as j } from "./utils/common.js";
const A = (o)=>{
    try {
        return 1.1.toLocaleString(o.replace("_", "-"))[1];
    } catch (e) {
        return ".";
    }
}, B = (o, r)=>{
    const s = String(o !== null && o !== void 0 ? o : "").replace(/[^.,\d]/g, ""), e = /[.,]/.exec(s), t = s.split(/[.,]/);
    return t.length === 0 ? "" : e && r > 0 ? t.shift() + e[0] + t.join("").slice(0, r) : t[0];
}, E = (_param)=>{
    var { inputRef: o, value: r, defaultValue: s, onChange: e, maxDecimals: t } = _param, c = _object_without_properties(_param, [
        "inputRef",
        "value",
        "defaultValue",
        "onChange",
        "maxDecimals"
    ]);
    const { i18n: u } = C(), f = A(u.locale), m = (n)=>String(n !== null && n !== void 0 ? n : "").replace(/[.,]/g, f), [p, d] = S.useState(s !== null && s !== void 0 ? s : ""), i = S.useRef(null), l = typeof r < "u", g = l ? r : p, h = S.useCallback((n)=>{
        l || d(n), i.current && (e == null || e(b(i.current, n)));
    }, [
        l,
        e
    ]), y = I({
        format: (n)=>B(n, t),
        replace: m,
        value: g,
        onChange: h,
        accept: /[\d.,]+/g
    });
    return /* @__PURE__ */ x("input", _object_spread_props(_object_spread({}, c), {
        type: "text",
        inputMode: "decimal",
        value: y.value,
        onChange: y.onChange,
        ref: j(o, i)
    }));
}, L = (_param)=>{
    var { disabled: o, error: r, helperText: s, name: e, optional: t, validate: c, onChange: u, onChangeValue: f, onBlur: m, value: p, defaultValue: d, maxDecimals: i = 1 / 0, dataAttributes: l } = _param, g = _object_without_properties(_param, [
        "disabled",
        "error",
        "helperText",
        "name",
        "optional",
        "validate",
        "onChange",
        "onChangeValue",
        "onBlur",
        "value",
        "defaultValue",
        "maxDecimals",
        "dataAttributes"
    ]);
    const { texts: h } = C(), D = V({
        name: e,
        value: p,
        defaultValue: d,
        processValue: (a)=>a.trim(),
        helperText: s,
        optional: t,
        error: r,
        disabled: o,
        onBlur: m,
        validate: (a, F)=>a ? c == null ? void 0 : c(a, F) : t ? "" : h.formFieldErrorIsMandatory,
        onChange: u,
        onChangeValue: f
    });
    return /* @__PURE__ */ x(R, _object_spread_props(_object_spread({}, g, D), {
        inputComponent: E,
        inputProps: {
            maxDecimals: i
        },
        dataAttributes: _object_spread({
            "component-name": "DecimalField"
        }, l)
    }));
}, q = L;
export { E as DecimalInput, q as default };
