// @telefonica/mistica/dist-es/dialog.js
"use client";
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
function _object_without_properties(source, excluded) {
    if (source == null) return {};
    var target = _object_without_properties_loose(source, excluded);
    var key, i;
    if (Object.getOwnPropertySymbols) {
        var sourceSymbolKeys = Object.getOwnPropertySymbols(source);
        for(i = 0; i < sourceSymbolKeys.length; i++){
            key = sourceSymbolKeys[i];
            if (excluded.indexOf(key) >= 0) continue;
            if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
            target[key] = source[key];
        }
    }
    return target;
}
function _object_without_properties_loose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;
    for(i = 0; i < sourceKeys.length; i++){
        key = sourceKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        target[key] = source[key];
    }
    return target;
}
import { jsx as e, jsxs as w, Fragment as O } from "react/jsx-runtime";
import * as n from "react";
import E from "classnames";
import { ButtonDanger as L, ButtonPrimary as S, ButtonSecondary as M } from "./button.js";
import { Portal as _ } from "./portal.js";
import j from "./focus-trap.js";
import F from "./generated/mistica-icons/icon-close-regular.js";
import { InternalIconButton as V } from "./icon-button.js";
import { isWebViewBridgeAvailable as W, nativeConfirm as z, nativeAlert as K } from "@tef-novum/webview-bridge";
import { useTheme as D } from "./hooks.js";
import U from "./button-layout.js";
import { Text5 as Y, Text4 as A, Text3 as q } from "./text.js";
import { ESC as G } from "./utils/keys.js";
import k from "./box.js";
import { isRunningAcceptanceTest as H } from "./utils/platform.js";
import { useSetModalStateEffect as J } from "./modal-context-provider.js";
import Q from "./stack.js";
import { ANIMATION_DURATION_MS as X, wrapper as Z, modalOpacityLayer as $, closedOpactityLayer as ee, modalContent as te, closedModalContent as ne, modalCloseButtonContainer as oe, variants as ae, iconContainer as re, dialogContent as ie, dialogActions as se } from "./dialog.css-mistica.js";
import { vars as le } from "./skins/skin-contract.css-mistica.js";
const ce = ()=>process.env.NODE_ENV !== "test" && !H(), de = (t)=>{
    const { texts: i } = D(), { className: d, title: r, message: f, icon: o, extra: c, showCancelButton: s, showAcceptButton: l, cancelText: g = i.dialogCancelButton, acceptText: m = i.dialogAcceptButton, onCancel: h, onAccept: C, destructive: B = !1 } = t, u = t.type === "dialog", p = u && !!t.link || l || s, y = {
        onPress: C || (()=>{}),
        children: m,
        // @deprecated - testid should be removed but many webapp tests depend on this
        dataAttributes: {
            testid: "dialog-accept-button"
        }
    };
    return /* @__PURE__ */ w("div", {
        className: E(ae[u ? "dialog" : "default"], d),
        children: [
            o && /* @__PURE__ */ e(k, {
                paddingBottom: 24,
                children: /* @__PURE__ */ e("div", {
                    className: re,
                    children: /*#__PURE__*/ n.cloneElement(o, {
                        size: "100%"
                    })
                })
            }),
            r && /* @__PURE__ */ e(k, {
                paddingBottom: 16,
                children: u ? /* @__PURE__ */ e(Y, {
                    as: "h2",
                    children: r
                }) : /* @__PURE__ */ e(A, {
                    regular: !0,
                    as: "h2",
                    children: r
                })
            }),
            t.subtitle && /* @__PURE__ */ e(k, {
                paddingBottom: 16,
                children: /* @__PURE__ */ e(A, {
                    regular: !0,
                    as: "h2",
                    children: t.subtitle
                })
            }),
            /* @__PURE__ */ e("div", {
                className: ie,
                children: /* @__PURE__ */ w(Q, {
                    space: 16,
                    children: [
                        /* @__PURE__ */ e(q, {
                            color: le.colors.textSecondary,
                            regular: !0,
                            children: f
                        }),
                        c
                    ]
                })
            }),
            p && /* @__PURE__ */ e("div", {
                className: se,
                children: /* @__PURE__ */ e(U, {
                    link: u ? t.link : void 0,
                    primaryButton: l ? B ? /* @__PURE__ */ e(L, _object_spread({
                        tabIndex: 1
                    }, y)) : /* @__PURE__ */ e(S, _object_spread({
                        tabIndex: 1
                    }, y)) : void 0,
                    secondaryButton: s ? /* @__PURE__ */ e(M, {
                        tabIndex: 2,
                        onPress: h || (()=>{}),
                        dataAttributes: {
                            testid: "dialog-cancel-button"
                        },
                        children: g
                    }) : void 0
                })
            })
        ]
    });
}, ue = (param)=>{
    let { type: t, message: i, title: d, acceptText: r, cancelText: f, onAccept: o, onCancel: c, onDestroy: s } = param;
    return t === "confirm" ? z({
        message: i,
        title: d,
        cancelText: f,
        acceptText: r
    }).then((l)=>{
        l ? o == null || o() : c == null || c(), s();
    }) : K({
        message: i,
        title: d,
        buttonText: r
    }).then(()=>{
        o == null || o(), s();
    });
}, me = (param)=>{
    let { type: t, onAccept: i, onCancel: d, onDestroy: r, acceptText: f, cancelText: o, message: c, title: s } = param;
    const { texts: l } = D(), g = n.useRef({
        type: t,
        onAccept: i,
        onCancel: d,
        onDestroy: r,
        acceptText: f || l.dialogAcceptButton,
        cancelText: o || l.dialogCancelButton,
        message: c,
        title: s
    }), m = n.useRef(!1);
    return n.useEffect(()=>{
        m.current || (m.current = !0, ue(g.current).finally(()=>{
            m.current = !1;
        }));
    }, []), /* @__PURE__ */ e(O, {});
}, Ie = (t)=>{
    J();
    const i = n.useRef(null), { texts: d } = D(), [r, f] = n.useState(!1), o = n.useRef(!1), c = n.useRef(!1), [s, l] = n.useState(!1), g = n.useRef(!1), m = n.useRef(ce() ? X : 0), h = t.type !== "dialog" && W(), C = t.type === "dialog", B = t.type === "alert", { onAccept: u, onCancel: p, onDestroy: y } = t, I = _object_without_properties(t, [
        "onAccept",
        "onCancel",
        "onDestroy"
    ]);
    n.useEffect(()=>{
        const a = setTimeout(()=>{
            o.current || l(!0);
        }, m.current);
        return ()=>{
            clearTimeout(a);
        };
    }, []);
    const x = n.useCallback(()=>{
        c.current || (c.current = !0, g.current ? u == null || u() : p == null || p(), y());
    }, [
        u,
        p,
        y
    ]), R = n.useCallback(()=>{
        let a;
        return !o.current && s && (o.current = !0, l(!1), f(!0), a = setTimeout(x, m.current)), ()=>{
            a && clearTimeout(a);
        };
    }, [
        x,
        s
    ]), N = n.useCallback(()=>{
        g.current = !0, R();
    }, [
        R
    ]), b = n.useCallback(()=>{
        g.current = !1, R();
    }, [
        R
    ]), v = n.useCallback(()=>{
        o.current || (B ? N() : b());
    }, [
        N,
        b,
        B
    ]), T = n.useCallback((a)=>{
        a.key === G && (a.stopPropagation(), a.preventDefault(), v());
    }, [
        v
    ]);
    n.useEffect(()=>{
        if (!h) return document.addEventListener("keydown", T), ()=>{
            document.removeEventListener("keydown", T);
        };
    }, [
        T,
        h
    ]), n.useEffect(()=>{
        if (!h) return window.addEventListener("popstate", v), ()=>{
            window.removeEventListener("popstate", v);
        };
    }, [
        v,
        h
    ]);
    const P = n.useCallback((a)=>{
        a.stopPropagation(), C && b();
    }, [
        C,
        b
    ]);
    return h ? /* @__PURE__ */ e(me, _object_spread({}, t)) : /* @__PURE__ */ e(_, {
        className: Z,
        children: /* @__PURE__ */ e(j, {
            children: /* @__PURE__ */ e("div", {
                onClick: P,
                className: E($, {
                    [ee]: r
                }),
                "data-testid": "dialog-overlay",
                "aria-hidden": r || !s,
                children: /* @__PURE__ */ e("div", {
                    role: "dialog",
                    onClick: (a)=>a.stopPropagation(),
                    "data-component-name": "Dialog",
                    children: /* @__PURE__ */ w("div", {
                        ref: i,
                        onAnimationEnd: (a)=>{
                            a.target === i.current && (o.current || l(!0));
                        },
                        onTransitionEnd: (a)=>{
                            a.target === i.current && o.current && x();
                        },
                        className: E(te, {
                            [ne]: r
                        }),
                        children: [
                            C && /* @__PURE__ */ e("div", {
                                className: oe,
                                children: /* @__PURE__ */ e(V, {
                                    onPress: v,
                                    "aria-label": d.modalClose || d.closeButtonLabel,
                                    bleedLeft: !0,
                                    bleedRight: !0,
                                    bleedY: !0,
                                    Icon: F
                                })
                            }),
                            /* @__PURE__ */ e(de, _object_spread_props(_object_spread({}, I), {
                                showAcceptButton: t.type !== "dialog" || !!t.onAccept,
                                showCancelButton: t.type === "confirm" || t.type === "dialog" && !!t.onCancel,
                                onCancel: b,
                                onAccept: N
                            }))
                        ]
                    })
                })
            })
        })
    });
};
export { Ie as default };
