// @telefonica/mistica/dist-es/dialog-context.js
"use client";
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
import { jsx as r, Fragment as u, jsxs as d } from "react/jsx-runtime";
import * as e from "react";
const c = /*#__PURE__*/ e.createContext({
    mounted: !1,
    dialog: null,
    setDialog: ()=>{}
}), m = /*#__PURE__*/ e.lazy(()=>import(/* webpackChunkName: "dialog" */ "./dialog.js")), a = ()=>{
    throw Error("Tried to show a dialog but the DialogRoot component was not mounted (missing ThemeContextProvider?)");
}, f = ()=>{
    const { dialog: o, setDialog: s, mounted: i } = e.useContext(c), n = e.useCallback((t)=>{
        if (i || a(), t && o) throw Error("Tried to show a dialog on top of another dialog");
        s(t);
    }, [
        s,
        o,
        i
    ]);
    return e.useMemo(()=>({
            alert: (t)=>n(_object_spread({
                    type: "alert"
                }, t)),
            confirm: (t)=>n(_object_spread({
                    type: "confirm"
                }, t)),
            dialog: (t)=>n(_object_spread({
                    type: "dialog"
                }, t))
        }), [
        n
    ]);
}, g = {
    alert: a,
    confirm: a,
    dialog: a
};
let l = g;
const D = ()=>{
    const o = f();
    return e.useEffect(()=>(l = o, ()=>{
            l = g;
        }), [
        o
    ]), /* @__PURE__ */ r(u, {});
}, p = (o)=>l.alert(o), x = (o)=>l.confirm(o), w = (o)=>l.dialog(o), C = (param)=>{
    let { children: o } = param;
    const { mounted: s } = e.useContext(c), [i, n] = e.useState(null), t = e.useMemo(()=>({
            mounted: !0,
            dialog: i,
            setDialog: n
        }), [
        i,
        n
    ]);
    return s ? /* @__PURE__ */ r(u, {
        children: o
    }) : /* @__PURE__ */ d(c.Provider, {
        value: t,
        children: [
            /* @__PURE__ */ r(D, {}),
            o,
            i && /* @__PURE__ */ r(e.Suspense, {
                fallback: null,
                children: /* @__PURE__ */ r(m, _object_spread_props(_object_spread({}, i), {
                    onDestroy: ()=>{
                        n(null);
                    }
                }))
            })
        ]
    });
};
export { C as DialogRoot, p as alert, x as confirm, w as dialog, f as useDialog };
