// @telefonica/mistica/dist-es/empty-state.js
"use client";
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
import { jsxs as s, jsx as r } from "react/jsx-runtime";
import x from "classnames";
import { useIsInverseVariant as k } from "./theme-variant-context.js";
import { ButtonPrimary as B } from "./button.js";
import { useTheme as C } from "./hooks.js";
import n from "./stack.js";
import { Text6 as I, Text3 as N } from "./text.js";
import S from "./button-group.js";
import { container as b, inverseBorder as E, vars as T, contentVariants as l, iconContainer as V, largeImageContainer as $, largeImage as j, smallImage as A } from "./empty-state.css-mistica.js";
import { vars as a } from "./skins/skin-contract.css-mistica.js";
import { AspectRatioContainer as D } from "./utils/aspect-ratio-support.js";
import { getPrefixedDataAttributes as P } from "./utils/dom.js";
import { sprinkles as R } from "./sprinkles.css-mistica.js";
import { applyCssVars as w } from "./utils/css.js";
const G = (param)=>{
    let { title: c, titleAs: p = "h1", description: d, button: e, buttonLink: f, largeImageUrl: o, imageUrl: i, icon: m, "aria-label": g, dataAttributes: h } = param;
    const { isDarkMode: u } = C(), t = k(), v = i ? /* @__PURE__ */ r("img", {
        className: A,
        alt: "",
        src: i
    }) : void 0, y = _object_spread_props(_object_spread({}, (e == null ? void 0 : e.type) === B ? {
        primaryButton: e
    } : {
        secondaryButton: e
    }), {
        link: f
    });
    return /* @__PURE__ */ s("div", _object_spread_props(_object_spread({
        className: x(b, t ? E : R({
            border: "regular"
        })),
        style: w({
            [T.backgroundColor]: t && !u ? a.colors.backgroundBrand : a.colors.backgroundContainer
        }),
        "aria-label": g,
        role: "region"
    }, P(h, "EmptyState")), {
        children: [
            /* @__PURE__ */ r("div", {
                style: {
                    flex: 1
                },
                children: /* @__PURE__ */ s(n, {
                    space: 24,
                    className: o ? l.largeImage : l.default,
                    children: [
                        v !== null && v !== void 0 ? v : m && /* @__PURE__ */ r("div", {
                            className: V,
                            children: m
                        }),
                        /* @__PURE__ */ s(n, {
                            space: 16,
                            children: [
                                /* @__PURE__ */ r(I, {
                                    as: p,
                                    children: c
                                }),
                                /* @__PURE__ */ r(N, {
                                    regular: !0,
                                    as: "p",
                                    color: t ? a.colors.inverse : a.colors.textSecondary,
                                    children: d
                                })
                            ]
                        }),
                        /* @__PURE__ */ r(S, _object_spread({}, y))
                    ]
                })
            }),
            /* @__PURE__ */ r("div", {
                style: {
                    flex: 1,
                    position: "relative"
                },
                children: o && /* @__PURE__ */ r(D, {
                    aspectRatio: 16 / 9,
                    className: $,
                    height: "100%",
                    width: "100%",
                    children: /* @__PURE__ */ r("div", {
                        className: j,
                        style: {
                            backgroundImage: `url(${o})`
                        }
                    })
                })
            })
        ]
    }));
}, _ = G;
export { _ as default };
