// @telefonica/mistica/dist-es/feedback.js
"use client";
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
function _object_without_properties(source, excluded) {
    if (source == null) return {};
    var target = _object_without_properties_loose(source, excluded);
    var key, i;
    if (Object.getOwnPropertySymbols) {
        var sourceSymbolKeys = Object.getOwnPropertySymbols(source);
        for(i = 0; i < sourceSymbolKeys.length; i++){
            key = sourceSymbolKeys[i];
            if (excluded.indexOf(key) >= 0) continue;
            if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
            target[key] = source[key];
        }
    }
    return target;
}
function _object_without_properties_loose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;
    for(i = 0; i < sourceKeys.length; i++){
        key = sourceKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        target[key] = source[key];
    }
    return target;
}
import { jsx as e, jsxs as l } from "react/jsx-runtime";
import * as K from "react";
import { useTheme as v, useScreenSize as B } from "./hooks.js";
import L from "./button-fixed-footer-layout.js";
import { VIVO_SKIN as A, VIVO_NEW_SKIN as z } from "./skins/constants.js";
import { useSetOverscrollColor as M } from "./overscroll-color-context.js";
import C from "./icons/icon-success.js";
import V from "./icons/icon-success-vivo.js";
import $ from "./icons/icon-error.js";
import G from "./icons/icon-info.js";
import { isWebViewBridgeAvailable as P, requestVibration as U } from "@tef-novum/webview-bridge";
import { isRunningAcceptanceTest as X } from "./utils/platform.js";
import { Text2 as J, Text6 as Q, Text3 as Y } from "./text.js";
import b from "./box.js";
import { InternalBoxed as Z } from "./boxed.js";
import ee from "./responsive-layout.js";
import p from "./stack.js";
import u from "classnames";
import re from "./button-group.js";
import { vars as d } from "./skins/skin-contract.css-mistica.js";
import { container as oe, innerContainer as te, innerContainerWithButtons as ce, backgroundBrand as ne, iconContainer as ie, feedbackData as ae, feedbackTextAppearFast as se, feedbackTextAppearMedium as N, feedbackTextAppearSlow as de, desktopContainer as le, desktopContent as me, desktopImage as ue } from "./feedback.css-mistica.js";
import y from "./icons/icon-success-vivo-new.js";
const O = (o)=>!X(o), D = (param)=>{
    let { primaryButton: o, secondaryButton: r } = param;
    return !!o || !!r;
}, T = (o)=>{
    P() && U(o).catch(()=>{});
}, E = (o)=>{
    K.useEffect(()=>{
        let r;
        return o === "success" && (r = setTimeout(()=>T("success"), 700)), o === "error" && (r = setTimeout(()=>T("error"), 1e3)), ()=>{
            clearTimeout(r);
        };
    }, [
        o
    ]);
}, R = (param, n)=>{
    let { icon: o, title: r, description: t, extra: c } = param;
    const i = t && Array.isArray(t) ? /* @__PURE__ */ e(p, {
        space: 16,
        children: t.map((a, m)=>/* @__PURE__ */ e("p", {
                children: a
            }, m))
    }) : t;
    return /* @__PURE__ */ l(p, {
        space: 24,
        children: [
            /* @__PURE__ */ e("div", {
                className: ie,
                children: o
            }),
            /* @__PURE__ */ l(p, {
                space: 16,
                className: u(ae),
                children: [
                    /* @__PURE__ */ e("div", {
                        className: u(n && se),
                        children: /* @__PURE__ */ e(Q, {
                            as: "h1",
                            children: r
                        })
                    }),
                    i && /* @__PURE__ */ e("div", {
                        className: u(n && N),
                        children: i && /* @__PURE__ */ e(Y, {
                            regular: !0,
                            color: d.colors.textSecondary,
                            children: i
                        })
                    }),
                    c && /* @__PURE__ */ e("div", {
                        className: u(n && (i ? de : N)),
                        children: c
                    })
                ]
            })
        ]
    });
}, W = (o, r)=>{
    const t = D(r);
    return /* @__PURE__ */ l(p, {
        space: {
            desktop: 40,
            mobile: 24
        },
        children: [
            o,
            t && /* @__PURE__ */ e(re, _object_spread({}, r))
        ]
    });
}, _ = (param)=>{
    let { isInverse: o, body: r, imageFit: t, imageUrl: c, dataAttributes: n } = param;
    return /* @__PURE__ */ e(Z, {
        borderRadius: d.borderRadii.legacyDisplay,
        desktopOnly: !0,
        isInverse: o,
        dataAttributes: n,
        children: /* @__PURE__ */ l("div", {
            className: le,
            children: [
                /* @__PURE__ */ e("div", {
                    className: me,
                    children: /* @__PURE__ */ e(b, {
                        padding: {
                            desktop: 64,
                            mobile: 0
                        },
                        paddingTop: 0,
                        children: r
                    })
                }),
                c && /* @__PURE__ */ e("div", {
                    className: ue,
                    style: {
                        backgroundImage: `url(${c})`,
                        backgroundPosition: t === "fit" ? "bottom right" : "center right",
                        backgroundRepeat: "no-repeat",
                        backgroundSize: t === "fit" ? "contain" : "cover",
                        flex: 1,
                        maxWidth: 600
                    }
                })
            ]
        })
    });
}, pe = ()=>(M({
        topColor: d.colors.backgroundBrandTop,
        bottomColor: "transparent"
    }), null), F = (param)=>{
    let { title: o, description: r, children: t, extra: c, primaryButton: n, secondaryButton: i, link: a, hapticFeedback: m, icon: f, animateText: k = !1, isInverse: s = !1, unstable_inlineInDesktop: h, imageUrl: x, imageFit: g, dataAttributes: S } = param;
    E(m);
    const { platformOverrides: q, isDarkMode: w } = v(), { isTabletOrSmaller: j } = B(), H = D({
        primaryButton: n,
        secondaryButton: i,
        link: a
    }), I = R({
        icon: f,
        title: o,
        description: r,
        extra: c !== null && c !== void 0 ? c : t
    }, k && O(q));
    return !j && h ? W(I, {
        primaryButton: n,
        secondaryButton: i,
        link: a
    }) : /* @__PURE__ */ e("div", {
        style: {
            position: "relative"
        },
        children: /* @__PURE__ */ l(ee, {
            children: [
                s && /* @__PURE__ */ e(pe, {}),
                /* @__PURE__ */ e(b, {
                    paddingTop: {
                        desktop: 64,
                        mobile: 0
                    },
                    children: _({
                        isInverse: s,
                        body: /* @__PURE__ */ e(L, {
                            isFooterVisible: H,
                            button: n,
                            secondaryButton: i,
                            link: a,
                            footerBgColor: s && !w ? d.colors.backgroundBrandBottom : void 0,
                            containerBgColor: s ? d.colors.backgroundBrand : d.colors.background,
                            children: /* @__PURE__ */ e("div", {
                                className: oe,
                                children: /* @__PURE__ */ e("div", {
                                    className: u(te, {
                                        [ce]: n || i || a
                                    }),
                                    children: I
                                })
                            })
                        }),
                        imageFit: g,
                        imageUrl: x,
                        dataAttributes: S
                    })
                })
            ]
        })
    });
}, Re = (_param)=>{
    var { dataAttributes: o } = _param, r = _object_without_properties(_param, [
        "dataAttributes"
    ]);
    const { isTabletOrSmaller: t } = B(), { skinName: c } = v();
    return /* @__PURE__ */ e(F, _object_spread_props(_object_spread({}, r), {
        isInverse: !r.unstable_inlineInDesktop || t,
        hapticFeedback: "success",
        icon: c === A ? /* @__PURE__ */ e(V, {
            size: "100%"
        }) : c === z ? /* @__PURE__ */ e(y, {
            size: "100%"
        }) : /* @__PURE__ */ e(C, {
            size: "100%"
        }),
        animateText: !0,
        imageUrl: r.imageUrl,
        imageFit: r.imageFit,
        dataAttributes: _object_spread({
            "component-name": "SuccessFeedbackScreen"
        }, o)
    }));
}, We = (_param)=>{
    var { children: o, errorReference: r, dataAttributes: t } = _param, c = _object_without_properties(_param, [
        "children",
        "errorReference",
        "dataAttributes"
    ]);
    return /* @__PURE__ */ e(F, _object_spread_props(_object_spread({}, c), {
        hapticFeedback: "error",
        icon: /* @__PURE__ */ e($, {
            size: "100%"
        }),
        animateText: !0,
        dataAttributes: _object_spread({
            "component-name": "ErrorFeedbackScreen"
        }, t),
        extra: /* @__PURE__ */ l(p, {
            space: 16,
            children: [
                o,
                r && /* @__PURE__ */ e(J, {
                    color: d.colors.textSecondary,
                    regular: !0,
                    children: r
                })
            ]
        })
    }));
}, _e = (_param)=>{
    var { dataAttributes: o, Icon: r = G } = _param, t = _object_without_properties(_param, [
        "dataAttributes",
        "Icon"
    ]);
    return /* @__PURE__ */ e(F, _object_spread({
        dataAttributes: _object_spread({
            "component-name": "InfoFeedbackScreen"
        }, o),
        icon: /* @__PURE__ */ e(r, {
            size: "100%"
        })
    }, t));
}, qe = (param)=>{
    let { title: o, description: r, children: t, extra: c, primaryButton: n, secondaryButton: i, link: a, imageUrl: m, imageFit: f, dataAttributes: k } = param;
    E("success");
    const { skinName: s, platformOverrides: h } = v(), g = R({
        icon: s === A ? /* @__PURE__ */ e(V, {
            size: "100%"
        }) : s === z ? /* @__PURE__ */ e(y, {
            size: "100%"
        }) : /* @__PURE__ */ e(C, {
            size: "100%"
        }),
        title: o,
        description: r,
        extra: c !== null && c !== void 0 ? c : t
    }, O(h)), S = W(g, {
        primaryButton: n,
        secondaryButton: i,
        link: a
    });
    return _({
        isInverse: !0,
        body: /* @__PURE__ */ e("div", {
            className: ne,
            children: /* @__PURE__ */ e(b, {
                paddingX: {
                    mobile: 16,
                    tablet: 24,
                    desktop: 0
                },
                children: /* @__PURE__ */ e(b, {
                    paddingBottom: {
                        desktop: 0,
                        mobile: 48
                    },
                    paddingTop: 64,
                    children: S
                })
            })
        }),
        imageFit: f,
        imageUrl: m,
        dataAttributes: _object_spread({
            "component-name": "SuccessFeedback"
        }, k)
    });
};
export { We as ErrorFeedbackScreen, F as FeedbackScreen, _e as InfoFeedbackScreen, qe as SuccessFeedback, Re as SuccessFeedbackScreen };
