// @telefonica/mistica/dist-es/form-context.js
"use client";
import * as c from "react";
const k = /*#__PURE__*/ c.createContext({
    values: {},
    setValue: ()=>{},
    rawValues: {},
    setRawValue: ()=>{},
    formStatus: "filling",
    register: ()=>{},
    formErrors: {},
    setFormError: ()=>{},
    jumpToNext: ()=>{},
    submit: ()=>{},
    validate: ()=>({}),
    formId: ""
}), P = ()=>c.useContext(k), y = (param)=>{
    let { name: r, value: t, defaultValue: e, onChange: u, disabled: p } = param;
    const { setRawValue: l, setValue: w, rawValues: o, setFormError: E, register: f, formStatus: R } = P();
    var _o_r;
    return c.useEffect(()=>{
        if (o[r] === void 0) {
            var _ref;
            const s = (_ref = t !== null && t !== void 0 ? t : e) !== null && _ref !== void 0 ? _ref : !1;
            w({
                name: r,
                value: s
            }), l({
                name: r,
                value: s
            });
        }
    }, [
        t,
        r,
        e,
        o,
        w,
        l
    ]), {
        name: r,
        value: t,
        defaultValue: e !== null && e !== void 0 ? e : t === void 0 ? (_o_r = o[r]) !== null && _o_r !== void 0 ? _o_r : !1 : void 0,
        focusableRef: (s)=>f(r, {
                focusableElement: s
            }),
        onChange: (s)=>{
            l({
                name: r,
                value: s
            }), w({
                name: r,
                value: s
            }), E({
                name: r,
                error: ""
            }), u == null || u(s);
        },
        disabled: R === "sending" || p
    };
}, z = (param)=>{
    let { name: r, value: t, defaultValue: e, processValue: u, helperText: p, optional: l, error: w, disabled: o, onBlur: E, validate: f, onChange: R, onChangeValue: s } = param;
    var _ref, _ref1;
    const { setRawValue: v, setValue: V, rawValues: x, values: j, formErrors: T, formStatus: q, setFormError: d, register: I } = P(), F = (_ref1 = (_ref = t !== null && t !== void 0 ? t : e) !== null && _ref !== void 0 ? _ref : x[r]) !== null && _ref1 !== void 0 ? _ref1 : "", N = c.useRef(u);
    var _x_r;
    return c.useEffect(()=>{
        v({
            name: r,
            value: F
        }), V({
            name: r,
            value: N.current(F)
        });
    }, [
        r,
        F,
        v,
        V
    ]), c.useEffect(()=>{
        o && d({
            name: r,
            error: void 0
        });
    }, [
        o,
        r,
        d
    ]), {
        value: t,
        defaultValue: e !== null && e !== void 0 ? e : t === void 0 ? (_x_r = x[r]) !== null && _x_r !== void 0 ? _x_r : "" : void 0,
        name: r,
        helperText: T[r] || p,
        required: !l,
        error: w || !!T[r],
        disabled: o || q === "sending",
        onBlur: (i)=>{
            d({
                name: r,
                error: f == null ? void 0 : f(j[r], x[r])
            }), E == null || E(i);
        },
        inputRef: (i)=>I(r, {
                input: i,
                validator: f
            }),
        onChange: (i)=>{
            const S = i.currentTarget.value, b = u(S);
            v({
                name: r,
                value: S
            }), V({
                name: r,
                value: b
            }), d({
                name: r,
                error: ""
            }), R == null || R(i), s == null || s(b, S);
        }
    };
};
export { k as FormContext, y as useControlProps, z as useFieldProps, P as useForm };
