// @telefonica/mistica/dist-es/hero.js
"use client";
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
function _object_without_properties(source, excluded) {
    if (source == null) return {};
    var target = _object_without_properties_loose(source, excluded);
    var key, i;
    if (Object.getOwnPropertySymbols) {
        var sourceSymbolKeys = Object.getOwnPropertySymbols(source);
        for(i = 0; i < sourceSymbolKeys.length; i++){
            key = sourceSymbolKeys[i];
            if (excluded.indexOf(key) >= 0) continue;
            if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
            target[key] = source[key];
        }
    }
    return target;
}
function _object_without_properties_loose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;
    for(i = 0; i < sourceKeys.length; i++){
        key = sourceKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        target[key] = source[key];
    }
    return target;
}
import { jsx as r, jsxs as p } from "react/jsx-runtime";
import * as $ from "react";
import h from "classnames";
import { useScreenSize as j } from "./hooks.js";
import { Text3 as g, Text8 as G } from "./text.js";
import A from "./grid-layout.js";
import m from "./box.js";
import R from "./stack.js";
import E from "./button-group.js";
import { vars as i } from "./skins/skin-contract.css-mistica.js";
import { container as v, containerMobile as V, containerMinHeight as C, expandedContent as _, layout as B, vars as H, containerDesktop as S, actions as z } from "./hero.css-mistica.js";
import { vars as y } from "./image.css-mistica.js";
import { useSlideshowContext as K } from "./carousel.js";
import { getPrefixedDataAttributes as T } from "./utils/dom.js";
import { sprinkles as O } from "./sprinkles.css-mistica.js";
import { useIsInverseVariant as M } from "./theme-variant-context.js";
import { applyCssVars as x } from "./utils/css.js";
import { ResetResponsiveLayout as U, InternalResponsiveLayout as W } from "./responsive-layout.js";
const k = {
    default: i.colors.background,
    alternative: i.colors.backgroundAlternative,
    brand: i.colors.backgroundBrand,
    "brand-secondary": i.colors.backgroundBrandSecondary,
    none: "transparent"
}, w = (param)=>{
    let { children: e, isInverse: o, backgroundColor: a } = param;
    return /* @__PURE__ */ r(W, {
        isInverse: o,
        className: B,
        innerDivClassName: B,
        shouldExpandWhenNested: !0,
        backgroundColor: a !== null && a !== void 0 ? a : "transparent",
        children: e
    });
}, b = (param)=>{
    let { headline: e, title: o, titleAs: a = "h1", pretitle: l, description: c, descriptionLinesMax: t, extra: n, button: d, secondaryButton: f, buttonLink: s } = param;
    return /* @__PURE__ */ p("section", {
        className: O({
            height: "100%",
            display: "flex",
            justifyContent: "space-between",
            flexDirection: "column"
        }),
        children: [
            /* @__PURE__ */ p("div", {
                children: [
                    /* @__PURE__ */ p(R, {
                        space: 16,
                        children: [
                            e && e,
                            /* @__PURE__ */ p(R, {
                                space: 8,
                                children: [
                                    l && /* @__PURE__ */ r(g, {
                                        as: "p",
                                        regular: !0,
                                        children: l
                                    }),
                                    o && /* @__PURE__ */ r(G, {
                                        as: a,
                                        children: o
                                    })
                                ]
                            }),
                            c && /* @__PURE__ */ r(g, {
                                as: "p",
                                regular: !0,
                                color: i.colors.textSecondary,
                                truncate: t,
                                children: c
                            })
                        ]
                    }),
                    n && /* @__PURE__ */ r("div", {
                        children: n
                    })
                ]
            }),
            (d || s) && /* @__PURE__ */ r("div", {
                className: z,
                children: /* @__PURE__ */ r(E, {
                    primaryButton: d,
                    secondaryButton: f,
                    link: s
                })
            })
        ]
    });
}, q = /*#__PURE__*/ $.forwardRef((_param, d)=>{
    var { height: e, background: o = "default", media: a, desktopMediaPosition: l = "left", dataAttributes: c, noPaddingY: t } = _param, n = _object_without_properties(_param, [
        "height",
        "background",
        "media",
        "desktopMediaPosition",
        "dataAttributes",
        "noPaddingY"
    ]);
    const { isTabletOrSmaller: f } = j(), s = K(), u = !!(s != null && s.withBullets), D = M(), N = o === "none" ? D : o === "brand" || o === "brand-secondary";
    if (f) return /* @__PURE__ */ r(U, {
        children: /* @__PURE__ */ r("div", {
            style: x({
                [y.mediaBorderRadius]: "0px"
            }),
            children: /* @__PURE__ */ p("div", _object_spread_props(_object_spread({}, T(_object_spread({
                "component-name": "Hero"
            }, c))), {
                ref: d,
                style: _object_spread({}, e === "100vh" ? {
                    maxHeight: "-webkit-fill-available"
                } : {}, x({
                    [H.height]: typeof e == "number" ? `${e}px` : e !== null && e !== void 0 ? e : "100%",
                    [y.mediaBorderRadius]: "0px"
                })),
                className: h(v, V, {
                    [C]: !t
                }),
                children: [
                    a,
                    /* @__PURE__ */ r(w, {
                        isInverse: N,
                        backgroundColor: k[o],
                        children: /* @__PURE__ */ r("div", {
                            className: _,
                            children: /* @__PURE__ */ r(m, {
                                paddingTop: 24,
                                paddingBottom: u ? 48 : t ? 0 : 24,
                                className: B,
                                children: /* @__PURE__ */ r(b, _object_spread({}, n))
                            })
                        })
                    })
                ]
            }))
        })
    });
    const I = l === "left" ? a : /* @__PURE__ */ r(m, {
        paddingRight: 24,
        children: /* @__PURE__ */ r(b, _object_spread({}, n))
    }), L = l === "right" ? a : /* @__PURE__ */ r(m, {
        paddingLeft: 24,
        children: /* @__PURE__ */ r(b, _object_spread({}, n))
    });
    return /* @__PURE__ */ r("div", _object_spread_props(_object_spread({}, T(_object_spread({
        "component-name": "Hero"
    }, c))), {
        ref: d,
        style: _object_spread_props(_object_spread({}, e === "100vh" ? {
            maxHeight: "-webkit-fill-available"
        } : {}, x({
            [H.height]: typeof e == "number" ? `${e}px` : e !== null && e !== void 0 ? e : "100%",
            [y.mediaBorderRadius]: i.borderRadii.container
        })), {
            background: k[o]
        }),
        className: O({
            height: "100%"
        }),
        children: /* @__PURE__ */ r(w, {
            isInverse: N,
            children: /* @__PURE__ */ r(A, {
                template: "6+6",
                left: /* @__PURE__ */ r(m, {
                    paddingTop: t ? 0 : 56,
                    paddingBottom: t && !u ? 0 : 56,
                    className: h(v, S, {
                        [C]: !t
                    }),
                    children: I
                }),
                right: /* @__PURE__ */ r(m, {
                    paddingTop: t ? 0 : 56,
                    paddingBottom: t && !u ? 0 : 56,
                    className: h(v, S),
                    children: L
                })
            })
        })
    }));
}), dr = q;
export { dr as default };
