// @telefonica/mistica/dist-es/highlighted-card.js
"use client";
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
function _object_without_properties(source, excluded) {
    if (source == null) return {};
    var target = _object_without_properties_loose(source, excluded);
    var key, i;
    if (Object.getOwnPropertySymbols) {
        var sourceSymbolKeys = Object.getOwnPropertySymbols(source);
        for(i = 0; i < sourceSymbolKeys.length; i++){
            key = sourceSymbolKeys[i];
            if (excluded.indexOf(key) >= 0) continue;
            if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
            target[key] = source[key];
        }
    }
    return target;
}
function _object_without_properties_loose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;
    for(i = 0; i < sourceKeys.length; i++){
        key = sourceKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        target[key] = source[key];
    }
    return target;
}
import { jsxs as l, jsx as i, Fragment as C } from "react/jsx-runtime";
import * as u from "react";
import { useIsInverseVariant as g } from "./theme-variant-context.js";
import y from "./stack.js";
import { BaseTouchable as k } from "./touchable.js";
import { Text as H, textProps as P, Text2 as T } from "./text.js";
import { Boxed as A } from "./boxed.js";
import M, { useIsDismissable as N } from "./maybe-dismissable.js";
import { container as R, textContainerVariant as $, imageContent as D, touchableContainer as O } from "./highlighted-card.css-mistica.js";
import { vars as S } from "./skins/skin-contract.css-mistica.js";
import { useTheme as j } from "./hooks.js";
const B = /*#__PURE__*/ u.forwardRef((n, e)=>{
    var _n_isInverse;
    const { title: s, description: r, imageUrl: a, imageFit: o, imageAlt: m, titleAs: b = "h3", width: d, dataAttributes: f, titleLinesMax: v, descriptionLinesMax: x } = n, t = _object_without_properties(n, [
        "title",
        "description",
        "imageUrl",
        "imageFit",
        "imageAlt",
        "titleAs",
        "width",
        "dataAttributes",
        "titleLinesMax",
        "descriptionLinesMax"
    ]), I = g(), p = (_n_isInverse = n.isInverse) !== null && _n_isInverse !== void 0 ? _n_isInverse : I, h = N(), { textPresets: w } = j(), c = /* @__PURE__ */ l(A, {
        ref: e,
        isInverse: p,
        className: R,
        dataAttributes: _object_spread({
            "component-name": "HighlightedCard"
        }, f),
        width: d ? `${d}px` : "100%",
        minHeight: "100%",
        children: [
            /* @__PURE__ */ l("div", {
                role: h ? void 0 : "region",
                className: $[a ? "withImage" : "withoutImage"],
                "aria-label": h ? void 0 : t["aria-label"],
                children: [
                    /* @__PURE__ */ l(y, {
                        space: 8,
                        children: [
                            !!s && /* @__PURE__ */ i(H, _object_spread_props(_object_spread({}, P.text4), {
                                truncate: v,
                                weight: w.cardTitle.weight,
                                as: b,
                                hyphens: "auto",
                                children: s
                            })),
                            !!r && /* @__PURE__ */ i(T, {
                                regular: !0,
                                color: S.colors.textSecondary,
                                truncate: x,
                                as: "p",
                                hyphens: "auto",
                                children: r
                            })
                        ]
                    }),
                    t.button && /* @__PURE__ */ l(C, {
                        children: [
                            /* @__PURE__ */ i("div", {
                                style: {
                                    minHeight: 16,
                                    flexGrow: 1
                                }
                            }),
                            t.button
                        ]
                    })
                ]
            }),
            a && /* @__PURE__ */ i("div", _object_spread_props(_object_spread({}, m ? {
                role: "img",
                "aria-label": m
            } : {}), {
                className: D,
                style: {
                    backgroundImage: `url(${a})`,
                    backgroundRepeat: "no-repeat",
                    backgroundSize: o === "fit" ? "contain" : "cover",
                    backgroundPosition: o === "fit" ? "bottom right" : `center ${o === "fill-center" ? "center" : "right"}`
                }
            }))
        ]
    });
    return t.button ? c : t.onPress || t.to || t.href ? /* @__PURE__ */ i(k, _object_spread_props(_object_spread({}, t), {
        className: O,
        children: c
    })) : c;
}), F = /*#__PURE__*/ u.forwardRef((_param, s)=>{
    var { "aria-label": n } = _param, e = _object_without_properties(_param, [
        "aria-label"
    ]);
    var _e_isInverse;
    const r = n || e.title || e.description, a = g(), o = (_e_isInverse = e.isInverse) !== null && _e_isInverse !== void 0 ? _e_isInverse : a;
    return /* @__PURE__ */ i(M, {
        onClose: e.onClose,
        "aria-label": r,
        width: e.width,
        isOverMedia: !!e.imageUrl,
        isInverse: o,
        children: /* @__PURE__ */ i(B, _object_spread_props(_object_spread({}, e), {
            "aria-label": r,
            ref: s
        }))
    });
}), W = F;
export { W as default };
