// @telefonica/mistica/dist-es/inline.js
"use client";
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
import { jsx as s } from "react/jsx-runtime";
import * as b from "react";
import y from "classnames";
import { sprinkles as g } from "./sprinkles.css-mistica.js";
import { getPrefixedDataAttributes as v } from "./utils/dom.js";
import { inline as h, wrap as I, fullWidth as x, noFullWidth as S, stringSpace as $, marginInline as k, vars as r } from "./inline.css-mistica.js";
import { applyCssVars as T } from "./utils/css.js";
import { isIos as V, isRunningAcceptanceTest as W } from "./utils/platform.js";
import { useTheme as w } from "./hooks.js";
const i = (t)=>typeof t == "number" ? `${t}px` : `space-${t}`, A = (t)=>{
    if (typeof t == "number" || typeof t == "string") return {
        [r.space]: i(t)
    };
    const e = {
        [r.spaceMobile]: i(t.mobile),
        [r.spaceDesktop]: i(t.desktop)
    };
    return t.tablet && (e[r.spaceTablet] = i(t.tablet)), e;
}, C = (param)=>{
    let { space: t, className: e, children: a, role: o, alignItems: m = "stretch", "aria-labelledby": p, fullWidth: f, wrap: c, dataAttributes: d } = param;
    const { platformOverrides: l } = w(), u = typeof t == "string";
    return /* @__PURE__ */ s("div", _object_spread_props(_object_spread({
        className: y(e, h, g({
            alignItems: m
        }), c ? I : f ? x : S, u ? $ : k),
        style: T(A(t)),
        role: o,
        "aria-labelledby": p
    }, v(d, "Inline")), {
        children: b.Children.map(a, (n)=>n || n === 0 ? /* @__PURE__ */ s("div", {
                role: o === "list" ? "listitem" : void 0,
                style: {
                    // Hack to fix https://jira.tid.es/browse/WEB-1683
                    // In iOS the inline component sometimes cuts the last line of the content
                    paddingBottom: V(l) && !W(l) ? 1 : void 0
                },
                children: n
            }) : null)
    }));
}, P = C;
export { P as default };
