// @telefonica/mistica/dist-es/loading-screen.js
"use client";
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
function _object_without_properties(source, excluded) {
    if (source == null) return {};
    var target = _object_without_properties_loose(source, excluded);
    var key, i;
    if (Object.getOwnPropertySymbols) {
        var sourceSymbolKeys = Object.getOwnPropertySymbols(source);
        for(i = 0; i < sourceSymbolKeys.length; i++){
            key = sourceSymbolKeys[i];
            if (excluded.indexOf(key) >= 0) continue;
            if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
            target[key] = source[key];
        }
    }
    return target;
}
function _object_without_properties_loose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;
    for(i = 0; i < sourceKeys.length; i++){
        key = sourceKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        target[key] = source[key];
    }
    return target;
}
import { jsxs as S, jsx as n } from "react/jsx-runtime";
import * as o from "react";
import { useSetOverscrollColor as N } from "./overscroll-color-context.js";
import { ThemeVariant as B } from "./theme-variant-context.js";
import { vars as f } from "./skins/skin-contract.css-mistica.js";
import E from "./responsive-layout.js";
import { Text4 as I, Text2 as O } from "./text.js";
import w from "./stack.js";
import { inOutAnimationMs as k, loadingScreen as V, screenBackground as j, screenBackgroundFadeOut as z, screenBackgroundAnimated as D, loadingScreenChildren as P, loadingScreenText as _, loadingScreenTextAnimatedOut as F, loadingScreenTextAnimated as y, logo as K, pulseLogo as M } from "./loading-screen.css-mistica.js";
import W from "./spinner.js";
import T from "classnames";
import { sprinkles as $ } from "./sprinkles.css-mistica.js";
import { Logo as q } from "./logo.js";
import G from "./screen-reader-only.js";
import { useTheme as R } from "./hooks.js";
import { VIVO_NEW_SKIN as H } from "./skins/constants.js";
import { getPrefixedDataAttributes as J } from "./utils/dom.js";
const L = (param)=>{
    let { isInverse: e } = param;
    const r = `body {background:${e ? f.colors.backgroundBrand : f.colors.background}}`;
    return /* @__PURE__ */ n("style", {
        children: r
    });
}, Q = (param)=>{
    let { animateText: e, isLoading: r, onClose: t, texts: i } = param;
    const [c, s] = o.useState(0), [a, l] = o.useState(!1), u = o.useCallback(()=>{
        s((c + 1) % i.length), l(!1);
    }, [
        c,
        i.length
    ]);
    o.useEffect(()=>{
        if (!r || i.length === 1) return;
        var _i_c_duration;
        const h = setTimeout(()=>{
            e ? l(!0) : u();
        }, (_i_c_duration = i[c].duration) !== null && _i_c_duration !== void 0 ? _i_c_duration : 5e3);
        return ()=>{
            clearTimeout(h);
        };
    }, [
        e,
        c,
        a,
        r,
        u,
        i
    ]);
    const m = ()=>{
        if (!r) {
            t == null || t();
            return;
        }
        a && u();
    }, { title: d, description: p } = i[c % i.length];
    return /* @__PURE__ */ n(E, {
        children: /* @__PURE__ */ n("div", {
            className: T(_, {
                [F]: e && (!r || a)
            }),
            onTransitionEnd: m,
            children: /* @__PURE__ */ S(w, {
                space: 8,
                children: [
                    d && /* @__PURE__ */ n("div", {
                        className: e ? y : void 0,
                        children: /* @__PURE__ */ n(I, {
                            textAlign: "center",
                            regular: !0,
                            as: "h1",
                            children: d
                        })
                    }),
                    p && /* @__PURE__ */ n("div", {
                        className: e ? y : void 0,
                        style: {
                            animationDelay: "200ms"
                        },
                        children: /* @__PURE__ */ n(O, {
                            textAlign: "center",
                            regular: !0,
                            as: "p",
                            color: f.colors.textSecondary,
                            children: p
                        })
                    })
                ]
            })
        }, c)
    });
}, x = /*#__PURE__*/ o.forwardRef((param, d)=>{
    let { isInverse: e, children: r, isLoading: t = !0, animateText: i, animateBackground: c, onClose: s, dataAttributes: a, title: l, description: u, texts: m = [
        {
            title: l,
            description: u
        }
    ] } = param;
    const [p, h] = o.useState(!1);
    o.useEffect(()=>{
        if (!t) return;
        const A = setTimeout(()=>{
            h(!0);
        }, k);
        return ()=>{
            clearTimeout(A);
        };
    }, [
        t
    ]);
    const v = o.useRef(!1), g = o.useCallback(()=>{
        !t && !v.current && (v.current = !0, s == null || s());
    }, [
        t,
        s
    ]);
    o.useEffect(()=>{
        if (t) return;
        if (!i && !c) {
            s == null || s();
            return;
        }
        const A = setTimeout(()=>{
            g();
        }, k);
        return ()=>{
            clearTimeout(A);
        };
    }, [
        c,
        i,
        g,
        t,
        s
    ]);
    const b = !r;
    return N(e ? {
        topColor: f.colors.backgroundBrandTop
    } : {}), /* @__PURE__ */ S(B, {
        isInverse: e,
        children: [
            /* @__PURE__ */ S("div", _object_spread_props(_object_spread({
                ref: d
            }, J(a)), {
                className: T(V, j[e ? "inverse" : "default"], {
                    [z]: !t && c,
                    [D]: c
                }, $({
                    justifyContent: b ? "center" : "space-between"
                })),
                onAnimationEnd: ()=>{
                    h(!0);
                },
                onTransitionEnd: g,
                children: [
                    r ? /* @__PURE__ */ n("div", {
                        className: P,
                        children: r
                    }) : /* @__PURE__ */ n(W, {
                        delay: "0s",
                        size: 32,
                        color: e ? f.colors.inverse : void 0
                    }),
                    /* @__PURE__ */ n(Q, {
                        animateText: i,
                        isLoading: t,
                        texts: m,
                        onClose: g
                    }),
                    !b && /* @__PURE__ */ n("div", {
                        style: {
                            height: 104
                        }
                    })
                ]
            })),
            t && p && /* @__PURE__ */ n(L, {
                isInverse: !!e
            })
        ]
    });
}), pe = /*#__PURE__*/ o.forwardRef((e, r)=>/* @__PURE__ */ n(x, _object_spread_props(_object_spread({
        ref: r
    }, e), {
        dataAttributes: _object_spread({
            "component-name": "LoadingScreen"
        }, e.dataAttributes),
        animateBackground: !0
    }))), U = (param)=>{
    let { isLoading: e, onCloseStart: r, onCloseEnd: t } = param;
    const { texts: i } = R(), [c, s] = o.useState(!0), a = ()=>{
        e || (r == null || r(), s(!1), t == null || t());
    };
    return /* @__PURE__ */ S("div", {
        className: T(K, {
            [M]: c
        }),
        onAnimationIteration: a,
        children: [
            /* @__PURE__ */ n(G, {
                children: /* @__PURE__ */ n("div", {
                    children: i.loading
                })
            }),
            /* @__PURE__ */ n(q, {
                size: 128
            })
        ]
    });
}, X = /*#__PURE__*/ o.lazy(()=>import("./vivinho-loading-animation/index.js")), Y = (param)=>{
    let { isLoading: e, onCloseStart: r, onCloseEnd: t } = param;
    const { skinName: i } = R();
    return i === H ? /* @__PURE__ */ n(o.Suspense, {
        fallback: null,
        children: /* @__PURE__ */ n(X, {
            isLoading: e,
            onCloseStart: r,
            onCloseEnd: t
        })
    }) : /* @__PURE__ */ n(U, {
        isLoading: e,
        onCloseStart: r,
        onCloseEnd: t
    });
}, he = /*#__PURE__*/ o.forwardRef((_param, c)=>{
    var { isLoading: e = !0, onClose: r, dataAttributes: t } = _param, i = _object_without_properties(_param, [
        "isLoading",
        "onClose",
        "dataAttributes"
    ]);
    const [s, a] = o.useState(!1), l = o.useRef(!1), u = o.useRef(!1), m = ()=>{
        a(!0);
    }, d = ()=>{
        l.current && u.current && (r == null || r());
    };
    return /* @__PURE__ */ n(x, _object_spread_props(_object_spread({
        ref: c,
        isInverse: !0
    }, i), {
        isLoading: e || !s,
        onClose: ()=>{
            u.current = !0, d();
        },
        animateText: !0,
        dataAttributes: _object_spread({
            "component-name": "BrandLoadingScreen"
        }, t),
        children: /* @__PURE__ */ n(Y, {
            isLoading: e,
            onCloseStart: m,
            onCloseEnd: ()=>{
                l.current = !0, d();
            }
        })
    }));
});
export { he as BrandLoadingScreen, pe as LoadingScreen };
