// @telefonica/mistica/dist-es/menu.js
"use client";
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
import { jsx as e, jsxs as E, Fragment as U } from "react/jsx-runtime";
import * as o from "react";
import ne from "classnames";
import { TAB as oe, ENTER as W, SPACE as G, ESC as se, UP as re, LEFT as ie, DOWN as K, RIGHT as ce } from "./utils/keys.js";
import { getPrefixedDataAttributes as le, cancelEvent as S } from "./utils/dom.js";
import ae from "./overlay.js";
import { menuItem as ue, menuItemEnabled as me, menuItemDisabled as fe, menuItemHovered as de, menuSectionDivider as pe, menuTransitionClasses as he, vars as b, itemContent as Y, iconContainer as j, menuContainer as ge } from "./menu.css-mistica.js";
import { useWindowSize as xe } from "./hooks.js";
import { Portal as be } from "./portal.js";
import y from "./box.js";
import Ie from "./inline.js";
import ve from "./touchable.js";
import { Text3 as X } from "./text.js";
import { vars as q } from "./skins/skin-contract.css-mistica.js";
import Me from "./divider.js";
import Oe from "./checkbox.js";
import { CSSTransition as Te } from "react-transition-group";
import { combineRefs as Se } from "./utils/common.js";
import { applyCssVars as Ee } from "./utils/css.js";
import { isRunningAcceptanceTest as Ce } from "./utils/platform.js";
const ke = 120, Q = /*#__PURE__*/ o.createContext({
    focusedItem: null,
    isMenuOpen: !1,
    setFocusedItem: ()=>{},
    closeMenu: ()=>{}
}), Ne = ()=>o.useContext(Q), D = (r)=>r ? Array.from(r.querySelectorAll("[role=menuitem],[role=menuitemcheckbox]")) : [], Re = (r, f)=>{
    if (!f) return null;
    const d = D(r).indexOf(f);
    return d < 0 ? null : d;
}, qe = (param)=>{
    let { label: r, Icon: f, destructive: d, disabled: u, onPress: C, controlType: s, checked: p } = param;
    const { focusedItem: n, setFocusedItem: v, closeMenu: i, isMenuOpen: k } = Ne(), c = o.useRef(null), h = d ? q.colors.textLinkDanger : q.colors.neutralHigh, x = c == null ? void 0 : c.current, M = (x == null ? void 0 : x.closest("[role=menu]")) || null, m = Re(M, x), l = ()=>s === "checkbox" ? /* @__PURE__ */ e(Oe, {
            ref: c,
            name: r,
            checked: p,
            onChange: ()=>{
                k && m !== null && (C(m), i());
            },
            disabled: u,
            role: "menuitemcheckbox",
            render: (param)=>{
                let { controlElement: N } = param;
                return /* @__PURE__ */ e(y, {
                    paddingX: 8,
                    paddingY: 12,
                    children: /* @__PURE__ */ E(Ie, {
                        space: "between",
                        alignItems: "center",
                        children: [
                            /* @__PURE__ */ E("div", {
                                className: Y,
                                children: [
                                    f && /* @__PURE__ */ e("div", {
                                        className: j,
                                        children: /* @__PURE__ */ e(f, {
                                            size: 24,
                                            color: h
                                        })
                                    }),
                                    /* @__PURE__ */ e(X, {
                                        regular: !0,
                                        color: h,
                                        children: r
                                    })
                                ]
                            }),
                            /* @__PURE__ */ e(y, {
                                paddingLeft: 16,
                                children: N
                            })
                        ]
                    })
                });
            }
        }) : /* @__PURE__ */ e(ve, {
            ref: c,
            onPress: ()=>{
                k && m !== null && (C(m), i());
            },
            disabled: u,
            role: "menuitem",
            children: /* @__PURE__ */ e(y, {
                paddingX: 8,
                paddingY: 12,
                children: /* @__PURE__ */ E("div", {
                    className: Y,
                    children: [
                        f && /* @__PURE__ */ e("div", {
                            className: j,
                            children: /* @__PURE__ */ e(f, {
                                size: 24,
                                color: h
                            })
                        }),
                        /* @__PURE__ */ e(X, {
                            regular: !0,
                            color: h,
                            children: r
                        })
                    ]
                })
            })
        });
    return /* @__PURE__ */ e("div", {
        className: ne(ue, {
            [me]: !u,
            [fe]: u,
            [de]: !u && m !== null && n === m
        }),
        onMouseMove: ()=>v(u ? null : m),
        onMouseLeave: ()=>v(null),
        children: l()
    });
}, Ve = (param)=>{
    let { children: r } = param;
    return r ? /* @__PURE__ */ E(U, {
        children: [
            r,
            /* @__PURE__ */ e(y, {
                paddingY: 8,
                className: pe,
                children: /* @__PURE__ */ e(Me, {})
            })
        ]
    }) : /* @__PURE__ */ e(U, {});
}, V = 12, J = 8, Je = (param)=>{
    let { renderTarget: r, renderMenu: f, width: d, position: u = "left", dataAttributes: C } = param;
    const [s, p] = o.useState(!1), [n, v] = o.useState(null), [i, k] = o.useState(null), [c, h] = o.useState(null), [x, M] = o.useState(!1), m = o.useRef(null), [l, N] = o.useState(null), R = xe();
    o.useEffect(()=>{
        const a = n == null ? void 0 : n.getBoundingClientRect();
        if (!i || !a || !s) return;
        const { top: t, right: O, left: g, bottom: A } = a, T = i.scrollHeight, I = u === "left" ? g : void 0, F = u === "right" ? R.width - O : void 0, L = t - J, $ = A + J, H = R.height - $ - V, P = L - V, B = H > T, z = P > T, te = B || !z && H > P;
        N(te ? {
            left: I,
            right: F,
            top: `${$}px`,
            bottom: "auto",
            maxHeight: B ? void 0 : H,
            transformOrigin: "center top"
        } : {
            left: I,
            right: F,
            top: "auto",
            bottom: `${R.height - L}px`,
            maxHeight: z ? void 0 : P,
            transformOrigin: "center bottom"
        });
    }, [
        u,
        s,
        i,
        n,
        d,
        R
    ]);
    const Z = o.useMemo(()=>({
            ref: v,
            onPress: ()=>{
                p(!s);
            }
        }), [
        v,
        s
    ]), ee = {
        ref: Se(k, m),
        className: ge,
        close: ()=>p(!1)
    }, w = o.useCallback(()=>{
        const t = D(i).findIndex((O)=>!O.getAttribute("aria-disabled"));
        h(t < 0 ? null : t);
    }, [
        i
    ]), _ = o.useCallback((a)=>{
        var T;
        const t = D(i);
        a && t.reverse();
        const O = c === null ? -1 : a ? t.length - 1 - c : c;
        let g = t.findIndex((I, F)=>!I.getAttribute("aria-disabled") && F > O);
        g === -1 && (g = t.findIndex((I)=>!I.getAttribute("aria-disabled")));
        const A = a && g !== -1 ? t.length - 1 - g : g;
        h(A < 0 ? null : A), (T = t[g]) == null || T.focus();
    }, [
        c,
        i
    ]);
    return o.useEffect(()=>{
        s ? x && i && (w(), M(!1)) : h(null);
    }, [
        s,
        w,
        x,
        i
    ]), o.useEffect(()=>{
        const a = (t)=>{
            if (s) switch(t.key){
                case ce:
                case K:
                    S(t), _();
                    break;
                case ie:
                case re:
                    S(t), _(!0);
                    break;
                case se:
                    p(!1);
                    break;
                case G:
                case W:
                    S(t), c !== null && D(i)[c].click();
                    break;
                case oe:
                    S(t);
                    break;
            }
            else switch(t.key){
                case W:
                case G:
                    M(!0);
                    break;
                case K:
                    n === document.activeElement && (M(!0), p(!0));
                    break;
            }
        };
        return document.addEventListener("keydown", a, !1), ()=>{
            document.removeEventListener("keydown", a, !1);
        };
    }), o.useEffect(()=>{
        n == null || n.setAttribute("aria-haspopup", "menu"), n == null || n.setAttribute("aria-expanded", String(s));
    }, [
        n,
        s
    ]), /* @__PURE__ */ E("div", _object_spread_props(_object_spread({}, le(C, "Menu")), {
        children: [
            r(_object_spread_props(_object_spread({}, Z), {
                isMenuOpen: s
            })),
            /* @__PURE__ */ e(be, {
                children: /* @__PURE__ */ e(Te, {
                    in: s,
                    nodeRef: m,
                    timeout: Ce() ? 0 : ke,
                    classNames: he,
                    mountOnEnter: !0,
                    unmountOnExit: !0,
                    onExit: ()=>n == null ? void 0 : n.focus(),
                    children: /* @__PURE__ */ e(ae, {
                        onPress: (a)=>{
                            S(a), p(!1);
                        },
                        disableScroll: !0,
                        children: /* @__PURE__ */ e("div", {
                            style: _object_spread({}, Ee(_object_spread({}, l ? _object_spread({
                                [b.top]: l.top,
                                [b.bottom]: l.bottom,
                                [b.transformOrigin]: l.transformOrigin
                            }, l.left !== void 0 && {
                                [b.left]: `${l.left}px`
                            }, l.right !== void 0 && {
                                [b.right]: `${l.right}px`
                            }, l.maxHeight !== void 0 && {
                                [b.maxHeight]: `${l.maxHeight}px`
                            }, d !== void 0 && {
                                [b.width]: d ? `${d}px` : ""
                            }) : {}))),
                            role: "menu",
                            children: /* @__PURE__ */ e(Q.Provider, {
                                value: {
                                    isMenuOpen: s,
                                    focusedItem: c,
                                    setFocusedItem: h,
                                    closeMenu: ()=>p(!1)
                                },
                                children: f(ee)
                            })
                        })
                    })
                })
            })
        ]
    }));
};
export { Je as Menu, qe as MenuItem, Ve as MenuSection };
