// @telefonica/mistica/dist-es/month-field.js
"use client";
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
function _object_without_properties(source, excluded) {
    if (source == null) return {};
    var target = _object_without_properties_loose(source, excluded);
    var key, i;
    if (Object.getOwnPropertySymbols) {
        var sourceSymbolKeys = Object.getOwnPropertySymbols(source);
        for(i = 0; i < sourceSymbolKeys.length; i++){
            key = sourceSymbolKeys[i];
            if (excluded.indexOf(key) >= 0) continue;
            if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
            target[key] = source[key];
        }
    }
    return target;
}
function _object_without_properties_loose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;
    for(i = 0; i < sourceKeys.length; i++){
        key = sourceKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        target[key] = source[key];
    }
    return target;
}
import { jsx as t } from "react/jsx-runtime";
import * as a from "react";
import { useFieldProps as z } from "./form-context.js";
import { TextFieldBaseAutosuggest as O } from "./text-field-base.js";
import { isInputTypeSupported as b } from "./utils/dom.js";
import { isServerSide as x } from "./utils/environment.js";
import A from "./generated/mistica-icons/icon-calendar-regular.js";
import { getLocalYearMonthString as r } from "./utils/time.js";
import { useTheme as N } from "./hooks.js";
import { iconContainer as j } from "./date-field.css-mistica.js";
import { iconSize as v } from "./icon-button.css-mistica.js";
const B = /*#__PURE__*/ a.lazy(()=>import(/* webpackChunkName: "date-time-picker" */ "./date-time-picker.js")), C = (_param)=>{
    var { disabled: h, error: g, helperText: D, name: F, optional: f, validate: n, onChange: S, onChangeValue: s, onBlur: y, value: R, defaultValue: k, min: o, max: i, dataAttributes: d } = _param, u = _object_without_properties(_param, [
        "disabled",
        "error",
        "helperText",
        "name",
        "optional",
        "validate",
        "onChange",
        "onChangeValue",
        "onBlur",
        "value",
        "defaultValue",
        "min",
        "max",
        "dataAttributes"
    ]);
    const I = (e)=>e, M = a.useMemo(()=>b("month"), []), { texts: T } = N(), m = (e)=>!(o && e && e < r(o) || i && e && e > r(i)), l = z({
        name: F,
        value: R,
        defaultValue: k,
        processValue: I,
        helperText: D,
        optional: f,
        error: g,
        disabled: h,
        onBlur: y,
        validate: (e, c)=>m(e) ? n == null ? void 0 : n(e, c) : T.formDateOutOfRangeError,
        onChange: S,
        onChangeValue: (e, c)=>{
            m(e) && (s == null || s(e, c));
        }
    }), p = /* @__PURE__ */ t(O, _object_spread_props(_object_spread({}, u, l), {
        min: o ? r(o) : void 0,
        max: i ? r(i) : void 0,
        type: "month",
        endIconOverlay: /* @__PURE__ */ t("div", {
            className: j,
            children: /* @__PURE__ */ t(A, {
                size: v.default
            })
        }),
        dataAttributes: _object_spread({
            "component-name": "MonthField"
        }, d)
    }));
    return M || x() ? p : /* @__PURE__ */ t(a.Suspense, {
        fallback: p,
        children: /* @__PURE__ */ t(B, _object_spread_props(_object_spread({}, u, l), {
            optional: f,
            isValidDate: (e)=>m(r(e.toDate())),
            mode: "year-month",
            dataAttributes: _object_spread({
                "component-name": "MonthField"
            }, d)
        }))
    });
}, U = C;
export { U as default };
