// @telefonica/mistica/dist-es/mosaic.js
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
import { jsx as o, jsxs as g, Fragment as M } from "react/jsx-runtime";
import { Grid as i, GridItem as e } from "./grid.js";
import { Carousel as v } from "./carousel.js";
import { singleItemRowContainer as m, squareContainer as u, fourItemsContainer as w } from "./mosaic.css-mistica.js";
const d = {
    mobile: 8,
    desktop: 16
}, C = (param)=>{
    let { items: n, isEven: r } = param;
    return /* @__PURE__ */ o("div", {
        className: n.length === 1 ? m : u,
        children: /* @__PURE__ */ o(i, {
            gap: d,
            rows: 2,
            columns: 2,
            height: "100%",
            children: n.length === 3 ? /* @__PURE__ */ g(M, {
                children: [
                    /* @__PURE__ */ o(e, {
                        columnSpan: r ? 2 : void 0,
                        children: n[0]
                    }),
                    /* @__PURE__ */ o(e, {
                        children: n[1]
                    }),
                    /* @__PURE__ */ o(e, {
                        columnSpan: r ? void 0 : 2,
                        children: n[2]
                    })
                ]
            }) : n.map((a, h)=>/* @__PURE__ */ o(e, {
                    columnSpan: n.length === 4 ? void 0 : 2,
                    rowSpan: n.length === 1 ? 2 : void 0,
                    children: a
                }, h))
        })
    });
}, N = (param)=>{
    let { items: n, withBullets: r, free: a, dataAttributes: h } = param;
    const c = n.length, t = c < 5 ? Math.min(c, 1) : Math.floor((c + 1) / 3), l = Array.from({
        length: t
    }, ()=>[]);
    return n.forEach((s, p)=>{
        const f = Math.min(t - 1, Math.floor(p / 3));
        l[f].push(s);
    }), /* @__PURE__ */ o(v, {
        dataAttributes: _object_spread_props(_object_spread({}, h), {
            "component-name": "HorizontalMosaic"
        }),
        items: l.map((s, p)=>/* @__PURE__ */ o(i, {
                children: /* @__PURE__ */ o(C, {
                    items: s,
                    isEven: p % 2 === 0
                }, p)
            })),
        withBullets: r,
        free: a
    });
}, S = (param)=>{
    let { items: n } = param;
    switch(n.length){
        case 1:
        case 2:
            return /* @__PURE__ */ o("div", {
                className: m,
                children: /* @__PURE__ */ o(i, {
                    gap: d,
                    columns: 2,
                    height: "100%",
                    children: n.map((r, a)=>/* @__PURE__ */ o(e, {
                            columnSpan: n.length === 2 ? void 0 : 2,
                            children: r
                        }, a))
                })
            });
        case 3:
            return /* @__PURE__ */ o("div", {
                className: u,
                children: /* @__PURE__ */ o(i, {
                    gap: d,
                    columns: 2,
                    flow: "column",
                    height: "100%",
                    children: n.map((r, a)=>/* @__PURE__ */ o(e, {
                            rowSpan: a === 0 ? 2 : void 0,
                            children: r
                        }, a))
                })
            });
        case 4:
        default:
            return /* @__PURE__ */ o("div", {
                className: w,
                children: /* @__PURE__ */ g(i, {
                    gap: d,
                    rows: 3,
                    columns: 2,
                    height: "100%",
                    children: [
                        /* @__PURE__ */ o(e, {
                            rowSpan: 2,
                            children: n[0]
                        }),
                        /* @__PURE__ */ o(e, {
                            children: n[1]
                        }),
                        /* @__PURE__ */ o(e, {
                            rowSpan: 2,
                            children: n[3]
                        }),
                        /* @__PURE__ */ o(e, {
                            children: n[2]
                        })
                    ]
                })
            });
    }
}, b = (param)=>{
    let { items: n, dataAttributes: r } = param;
    const a = n.length, h = Math.ceil(a / 4), c = Array.from({
        length: h
    }, ()=>[]);
    return n.forEach((t, l)=>{
        const s = Math.floor(l / 4);
        c[s].push(t);
    }), /* @__PURE__ */ o(i, {
        rows: 1,
        gap: d,
        dataAttributes: _object_spread_props(_object_spread({}, r), {
            "component-name": "VerticalMosaic"
        }),
        children: c.map((t, l)=>/* @__PURE__ */ o(S, {
                items: t
            }, l))
    });
};
export { N as HorizontalMosaic, b as VerticalMosaic };
