// @telefonica/mistica/dist-es/navigation-bar.css-mistica.js
import "./sprinkles.css.ts.vanilla.css-mistica.js";
import "./navigation-bar.css.ts.vanilla.css-mistica.js";
var y = "_1vcy5dz1 _1y2v1nfhe _1y2v1nfig _1y2v1nfiw", _ = "_1vcy5dz12 _1y2v1nfhg _1y2v1nfl2 _1y2v1nfl6 _1y2v1nfll _1y2v1nf3p", e = "_1vcy5dz10", r = {
    entering: "_1vcy5dzs",
    entered: "_1vcy5dzt",
    exiting: "_1vcy5dzu",
    exited: "_1vcy5dzv",
    unmounted: "_1vcy5dzw"
}, d = "_1vcy5dzm", a = "_1vcy5dz5 _1vcy5dz3 _1y2v1nfhf", c = "_1vcy5dz4 _1vcy5dz3 _1y2v1nfhf", f = "_1vcy5dz7 _1vcy5dz3 _1y2v1nfhf", z = "_1vcy5dz6 _1vcy5dz3 _1y2v1nfhf", i = "_1vcy5dz17", t = "_1vcy5dzz _1y2v1nfhk _1y2v1nfht", o = "_1vcy5dzx", h = {
    default: "_1vcy5dzc _1vcy5dzb _1y2v1nfhk _1y2v1nfhz _1y2v1nfi7",
    noBorder: "_1vcy5dzd _1vcy5dzb _1y2v1nfhk _1y2v1nfhz _1y2v1nfi7",
    menuOpen: "_1vcy5dze _1vcy5dzb _1y2v1nfhk _1y2v1nfhz _1y2v1nfi7"
}, l = "_1vcy5dzl", u = "_1vcy5dzn", g = "_1vcy5dzg _1y2v1nfa4 _1y2v1nfbd _1y2v1nfj3 _1y2v1nfhk _1y2v1nfhz _1y2v1nfk2 _1y2v1nf7c", s = {
    default: "_1vcy5dzh",
    inverse: "_1vcy5dzi"
}, p = "_1y2v1nfj3 _1vcy5dzo", b = "_1vcy5dzr _1vcy5dzo", k = {
    default: "_1vcy5dzj",
    inverse: "_1vcy5dzk"
}, C = "_1vcy5dz9 _1y2v1nfhg _1y2v1nfkr _1y2v1nfl6 _1y2v1nfll";
export { y as burgerIconContainer, _ as burgerMenu, e as burgerMenuButton, r as burgerMenuTransition, d as desktopOnly, a as iconCloseHidden, c as iconCloseOpen, f as iconMenuHidden, z as iconMenuOpen, i as lineHeightFix, t as logoContainer, o as mainNavbarContent, h as navbarBorderColorVariants, l as navigationBarContent, u as navigationBarContentRight, g as section, s as selectedSectionVariantes, p as spacer, b as spacerLarge, k as textWrapperVariants, C as topFixed };
