// @telefonica/mistica/dist-es/password-field.js
"use client";
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
function _object_without_properties(source, excluded) {
    if (source == null) return {};
    var target = _object_without_properties_loose(source, excluded);
    var key, i;
    if (Object.getOwnPropertySymbols) {
        var sourceSymbolKeys = Object.getOwnPropertySymbols(source);
        for(i = 0; i < sourceSymbolKeys.length; i++){
            key = sourceSymbolKeys[i];
            if (excluded.indexOf(key) >= 0) continue;
            if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
            target[key] = source[key];
        }
    }
    return target;
}
function _object_without_properties_loose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;
    for(i = 0; i < sourceKeys.length; i++){
        key = sourceKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        target[key] = source[key];
    }
    return target;
}
import { jsx as l } from "react/jsx-runtime";
import * as r from "react";
import { useFieldProps as x } from "./form-context.js";
import { TextFieldBaseAutosuggest as A, FieldEndIcon as E } from "./text-field-base.js";
import { useTheme as S } from "./hooks.js";
import k from "./generated/mistica-icons/icon-eye-off-regular.js";
import T from "./generated/mistica-icons/icon-eye-regular.js";
const j = (param)=>{
    let { isVisible: n, setVisibility: i, focus: c } = param;
    const { texts: t } = S();
    return /* @__PURE__ */ l(E, {
        checkedProps: {
            Icon: k,
            "aria-label": t.disablePasswordVisibility
        },
        uncheckedProps: {
            Icon: T,
            "aria-label": t.enablePasswordVisibility
        },
        checked: n,
        onChange: (a)=>{
            i(a), c();
        }
    });
}, q = (_param)=>{
    var { disabled: n, error: i, helperText: c, name: t, optional: a, validate: f, onChange: m, onChangeValue: p, onBlur: P, value: b, autoComplete: R = "current-password", defaultValue: w, dataAttributes: F } = _param, I = _object_without_properties(_param, [
        "disabled",
        "error",
        "helperText",
        "name",
        "optional",
        "validate",
        "onChange",
        "onChangeValue",
        "onBlur",
        "value",
        "autoComplete",
        "defaultValue",
        "dataAttributes"
    ]);
    const [u, y] = r.useState(!1), s = r.useRef(0), o = r.useRef(null), V = (e)=>e, g = ()=>{
        const e = o.current;
        e && (e.selectionStart !== null && (s.current = e.selectionStart), e.focus());
    };
    r.useEffect(()=>{
        const e = o.current;
        if (e) {
            const h = requestAnimationFrame(()=>{
                e.selectionStart = s.current, e.selectionEnd = s.current;
            });
            return ()=>{
                cancelAnimationFrame(h);
            };
        }
    }, [
        u,
        s,
        o
    ]);
    const d = x({
        name: t,
        value: b,
        defaultValue: w,
        processValue: V,
        helperText: c,
        optional: a,
        error: i,
        disabled: n,
        onBlur: P,
        validate: f,
        onChange: m,
        onChangeValue: p
    });
    return /* @__PURE__ */ l(A, _object_spread_props(_object_spread({}, I, d), {
        type: u ? "text" : "password",
        inputRef: (e)=>{
            d.inputRef(e), o.current = e;
        },
        autoComplete: R,
        endIcon: /* @__PURE__ */ l(j, {
            focus: g,
            isVisible: u,
            setVisibility: y
        }),
        dataAttributes: _object_spread({
            "component-name": "PasswordField"
        }, F)
    }));
}, D = q;
export { D as default };
