// @telefonica/mistica/dist-es/phone-number-field.js
"use client";
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
function _object_without_properties(source, excluded) {
    if (source == null) return {};
    var target = _object_without_properties_loose(source, excluded);
    var key, i;
    if (Object.getOwnPropertySymbols) {
        var sourceSymbolKeys = Object.getOwnPropertySymbols(source);
        for(i = 0; i < sourceSymbolKeys.length; i++){
            key = sourceSymbolKeys[i];
            if (excluded.indexOf(key) >= 0) continue;
            if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
            target[key] = source[key];
        }
    }
    return target;
}
function _object_without_properties_loose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;
    for(i = 0; i < sourceKeys.length; i++){
        key = sourceKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        target[key] = source[key];
    }
    return target;
}
import { jsx as R } from "react/jsx-runtime";
import * as p from "react";
import { useRifm as N } from "rifm";
import { getRegionCodeForCountryCode as x, formatToE164 as T, parse as A, formatAsYouType as k } from "@telefonica/libphonenumber";
import { useFieldProps as B } from "./form-context.js";
import { TextFieldBaseAutosuggest as E } from "./text-field-base.js";
import { useTheme as V } from "./hooks.js";
import { createChangeEvent as S } from "./utils/dom.js";
import { combineRefs as Z } from "./utils/common.js";
const b = (r, n)=>k(n.replace(/[^\d+*#]/g, ""), r), $ = (r)=>!!r.match(/^\+\d+$/), j = (_param)=>{
    var { inputRef: r, value: n, defaultValue: f, onChange: s, prefix: o, e164: y } = _param, d = _object_without_properties(_param, [
        "inputRef",
        "value",
        "defaultValue",
        "onChange",
        "prefix",
        "e164"
    ]);
    const [g, h] = p.useState(f !== null && f !== void 0 ? f : ""), l = p.useRef(null), { i18n: C } = V(), i = C.phoneNumberFormattingRegionCode, u = typeof n < "u", a = u ? n : g, P = p.useCallback((e)=>{
        u || h(e), l.current && (s == null || s(S(l.current, e)));
    }, [
        u,
        s
    ]), F = p.useCallback(// The final replacement of "-" to "@" is to workaround a bug in rifm library
    // otherwise the cursor position is incorrectly positioned
    // also note the "@" is replaced back to "-" in `replace` param in `useRifm`
    (e)=>{
        let t = "";
        if (o && $(o)) {
            const c = o + e;
            t = b(i, c), t.startsWith(o) ? t = t.slice(o.length).trim() : t = b(i, e);
        } else t = b(i, e);
        return t.replace(/-/g, "@");
    }, [
        i,
        o
    ]), m = N({
        format: F,
        value: a,
        onChange: P,
        accept: /[\d\-+#*]+/g,
        replace: (e)=>e.replace(/@/g, "-")
    });
    return /* @__PURE__ */ R("input", _object_spread_props(_object_spread({}, d), {
        value: m.value,
        onChange: m.onChange,
        type: "tel",
        ref: Z(r, l)
    }));
}, I = (_param)=>{
    var { disabled: r, error: n, helperText: f, name: s, optional: o, validate: y, onChange: d, onChangeValue: g, onBlur: h, value: l, defaultValue: C, e164: i, dataAttributes: u } = _param, a = _object_without_properties(_param, [
        "disabled",
        "error",
        "helperText",
        "name",
        "optional",
        "validate",
        "onChange",
        "onChangeValue",
        "onBlur",
        "value",
        "defaultValue",
        "e164",
        "dataAttributes"
    ]);
    const { i18n: P } = V(), m = B({
        name: s,
        value: l,
        defaultValue: C,
        processValue: (e)=>{
            if (i) try {
                var _a_prefix;
                const t = ((_a_prefix = a.prefix) !== null && _a_prefix !== void 0 ? _a_prefix : "").replace(/[^\d]/g, "");
                let c = x(t);
                return (!c || c === "ZZ") && (c = P.phoneNumberFormattingRegionCode), T(A(e, c));
            } catch (e) {
                return "";
            }
            else return e.replace(/[^\d]/g, "");
        },
        helperText: f,
        optional: o,
        error: n,
        disabled: r,
        onBlur: h,
        validate: y,
        onChange: d,
        onChangeValue: g
    });
    return /* @__PURE__ */ R(E, _object_spread_props(_object_spread({}, a, m), {
        type: "phone",
        inputProps: {
            prefix: a.prefix
        },
        inputComponent: j,
        dataAttributes: _object_spread({
            "component-name": "PhoneNumberField"
        }, u)
    }));
}, J = I;
export { J as default };
