// @telefonica/mistica/dist-es/pin-field.js
"use client";
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
import { jsxs as $, jsx as n } from "react/jsx-runtime";
import D from "classnames";
import * as m from "react";
import N from "./inline.js";
import { input as B } from "./text-field-base.css-mistica.js";
import { fieldContainer as E, disabled as L, readOnlyField as j, focusedField as O, field as W, input as q, passwordInput as z, pinInputLineHeight as h, passwordDot as K } from "./pin-field.css-mistica.js";
import { useAriaId as M, useTheme as _ } from "./hooks.js";
import G from "./screen-reader-only.js";
import { IntegerInput as J } from "./integer-field.js";
import { useFieldProps as Q } from "./form-context.js";
import { getPrefixedDataAttributes as U, createChangeEvent as X } from "./utils/dom.js";
import { HelperText as Y } from "./text-field-components.js";
import { flushSync as Z } from "react-dom";
let T = !1;
const x = (param)=>{
    let { length: l = 6, hideCode: f = !1, readSms: d, disabled: g, readOnly: y, value: F, defaultValue: b, onChange: u, inputRef: A } = param;
    var _ref;
    const { texts: C } = _(), [R, S] = m.useState((_ref = b == null ? void 0 : b.slice(0, l)) !== null && _ref !== void 0 ? _ref : ""), [w, I] = m.useState(void 0), a = m.useRef(Array.from({
        length: l
    }, ()=>null)).current, i = typeof F < "u", t = i ? F : R, c = m.useCallback((s)=>{
        if (s === t) return;
        i || S(s);
        const e = a[0];
        e && (u == null || u(X(_object_spread({}, e), s)));
    }, [
        t,
        a,
        i,
        u
    ]);
    m.useEffect(()=>{
        c(t.slice(0, l));
    }, [
        l,
        t,
        c
    ]), m.useEffect(()=>{
        if (d && "OTPCredential" in window && !T) {
            T = !0;
            const s = new AbortController();
            return navigator.credentials.get({
                // @ts-expect-error: otp is not in the types yet
                otp: {
                    transport: [
                        "sms"
                    ]
                },
                signal: s.signal
            }).then((e)=>{
                if (e) {
                    const r = e.code.slice(0, l);
                    c(r);
                }
            }).catch(()=>{}).finally(()=>{
                T = !1;
            }), ()=>{
                T = !1, s.abort();
            };
        }
    }, [
        c,
        l,
        d
    ]);
    const H = (s)=>(e)=>{
            var k;
            const r = e.target.value;
            if (r === "") return;
            const o = t[s];
            let p = r;
            !o || o === r ? p = r : o === r[0] ? p = r.slice(1) : o === r[r.length - 1] && (p = r.slice(0, -1));
            let v = s, P = t;
            if (p.length >= 2) {
                const V = p.slice(0, l - s);
                P = t.slice(0, s) + V, v = Math.min(s + V.length, l - 1);
            } else P = t.slice(0, s) + p + t.slice(s + 1), v = s + 1;
            P !== t && (Z(()=>{
                c(P);
            }), v !== s && v <= l - 1 && ((k = a[v]) == null || k.focus()));
        };
    return /* @__PURE__ */ n(N, {
        space: 8,
        children: Array.from({
            length: l
        }).map((s, e)=>/* @__PURE__ */ {
            var _t_e;
            return $("div", {
                className: y ? j : e === w ? O : W,
                children: [
                    /* @__PURE__ */ n(J, {
                        "aria-label": C.pinFieldInputLabel.replace("1$s", String(e + 1)).replace("2$s", String(l)),
                        type: f ? "password" : "text",
                        tabIndex: e > t.length ? -1 : void 0,
                        required: !0,
                        onFocus: ()=>{
                            var o;
                            const r = t.length === l ? -1 : t.length;
                            r >= 0 && r < e ? (o = a[r]) == null || o.focus() : I(e);
                        },
                        onBlur: ()=>{
                            I(void 0);
                        },
                        inputRef: (r)=>{
                            a[e] = r, e === 0 && A(r);
                        },
                        className: D(B, q, {
                            [z]: f
                        }),
                        style: {
                            marginTop: `calc(${h} - 1px)`,
                            marginBottom: `calc(${h} - 1px)`,
                            lineHeight: h,
                            fontSize: h,
                            height: h
                        },
                        disabled: g,
                        readOnly: y,
                        autoComplete: d ? "one-time-code" : void 0,
                        value: (_t_e = t[e]) !== null && _t_e !== void 0 ? _t_e : "",
                        onChange: H(e),
                        onKeyDown: (r)=>{
                            switch(r.key){
                                case "Backspace":
                                case "Delete":
                                    if (r.currentTarget.value && c(t.slice(0, e) + t.slice(e + 1)), e > 0 && e >= t.length - 1) {
                                        const o = a[e - 1];
                                        o == null || o.focus();
                                    }
                                    break;
                                case "ArrowLeft":
                                    if (e > 0) {
                                        const o = a[e - 1];
                                        o && o.focus();
                                    }
                                    break;
                                case "ArrowRight":
                                    if (e < l - 1) {
                                        const o = a[e + 1];
                                        o && o.focus();
                                    }
                                    break;
                            }
                        }
                    }),
                    f && t[e] && /* @__PURE__ */ n("div", {
                        "aria-hidden": !0,
                        className: K,
                        children: "•"
                    })
                ]
            }, e);
        })
    });
}, ee = (param)=>{
    let { length: l = 6, hideCode: f = !1, readSms: d = !f, // by default, don't read sms if the code is hidden (password input type)
    disabled: g, readOnly: y, name: F, value: b, defaultValue: u, helperText: A, error: C, onChangeValue: R, onChange: S, "aria-label": w, "aria-labelledby": I, dataAttributes: a } = param;
    const i = Q({
        name: F,
        value: b,
        defaultValue: u,
        processValue: (c)=>c,
        helperText: A,
        optional: !1,
        error: C,
        disabled: g,
        onChangeValue: R,
        onChange: S
    }), t = M();
    return /* @__PURE__ */ $("div", _object_spread_props(_object_spread({
        role: "group",
        "aria-labelledby": I !== null && I !== void 0 ? I : t,
        className: D(E, {
            [L]: g
        })
    }, U(a, "PinField")), {
        children: [
            w && !I && /* @__PURE__ */ n(G, {
                children: /* @__PURE__ */ n("div", {
                    id: t,
                    children: w
                })
            }),
            /* @__PURE__ */ n(x, {
                inputRef: i.inputRef,
                length: l,
                hideCode: f,
                readSms: d,
                value: i.value,
                defaultValue: i.defaultValue,
                disabled: i.disabled,
                onChange: i.onChange,
                readOnly: y
            }),
            /* @__PURE__ */ n(Y, {
                error: i.error,
                leftText: i.helperText
            })
        ]
    }));
}, de = ee;
export { de as default };
