// @telefonica/mistica/dist-es/radio-button.js
"use client";
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
function _object_without_properties(source, excluded) {
    if (source == null) return {};
    var target = _object_without_properties_loose(source, excluded);
    var key, i;
    if (Object.getOwnPropertySymbols) {
        var sourceSymbolKeys = Object.getOwnPropertySymbols(source);
        for(i = 0; i < sourceSymbolKeys.length; i++){
            key = sourceSymbolKeys[i];
            if (excluded.indexOf(key) >= 0) continue;
            if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
            target[key] = source[key];
        }
    }
    return target;
}
function _object_without_properties_loose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;
    for(i = 0; i < sourceKeys.length; i++){
        key = sourceKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        target[key] = source[key];
    }
    return target;
}
import { jsx as n, jsxs as S } from "react/jsx-runtime";
import * as i from "react";
import { UP as B, LEFT as q, DOWN as E, RIGHT as T, SPACE as w } from "./utils/keys.js";
import { useControlProps as G } from "./form-context.js";
import { combineRefs as O } from "./utils/common.js";
import { Text3 as k } from "./text.js";
import j from "./inline.js";
import P from "classnames";
import { useAriaId as F, useTheme as K } from "./hooks.js";
import { getPrefixedDataAttributes as D } from "./utils/dom.js";
import { outerCircleVariants as I, outerCircleCheckedVariants as $, disabled as A, innerCircle as H, innerCircleChecked as U, radioButtonContainerDisabled as W, radioButton as z } from "./radio-button.css-mistica.js";
const N = /*#__PURE__*/ i.createContext({
    disabled: !1,
    selectedValue: null,
    focusableValue: null,
    select: ()=>{},
    selectNext: ()=>{},
    selectPrev: ()=>{}
}), J = ()=>i.useContext(N), L = (_param)=>{
    var { value: t, id: s, dataAttributes: h, "aria-labelledby": v, "aria-label": x } = _param, u = _object_without_properties(_param, [
        "value",
        "id",
        "dataAttributes",
        "aria-labelledby",
        "aria-label"
    ]);
    const { disabled: l, selectedValue: C, focusableValue: p, select: c, selectNext: R, selectPrev: g } = J(), a = F(v), y = i.useRef(null), d = t === C, V = p === t ? 0 : -1, { isIos: e } = K(), r = (o)=>{
        switch(o.key){
            case w:
                c(t), o.preventDefault(), o.stopPropagation();
                break;
            case T:
            case E:
                R(), o.preventDefault(), o.stopPropagation();
                break;
            case q:
            case B:
                g(), o.preventDefault(), o.stopPropagation();
                break;
        }
    }, f = /* @__PURE__ */ n("div", {
        className: P(e ? I.ios : I.default, {
            [$[e ? "ios" : "default"]]: d,
            [A]: l
        }),
        children: !e && /* @__PURE__ */ n("div", {
            className: P(H, {
                [U]: d
            })
        })
    });
    return(// eslint-disable-next-line jsx-a11y/interactive-supports-focus
    /* @__PURE__ */ n("span", _object_spread_props(_object_spread({
        ref: y,
        id: s,
        tabIndex: l ? void 0 : V,
        role: "radio",
        "data-value": t,
        "aria-checked": d,
        "aria-disabled": l,
        "aria-label": x,
        "aria-labelledby": x ? void 0 : a,
        onClick: (o)=>{
            o.stopPropagation(), l || c(t);
        },
        onKeyDown: l ? void 0 : r,
        className: l ? W : z
    }, D(h, "RadioButton")), {
        children: u.render ? u.render({
            controlElement: f,
            disabled: !!l,
            checked: d,
            labelId: a
        }) : /* @__PURE__ */ S(j, {
            space: 16,
            children: [
                /* @__PURE__ */ n(k, {
                    regular: !0,
                    as: "div",
                    children: /* @__PURE__ */ n("div", {
                        style: {
                            position: "relative",
                            top: -2
                        },
                        children: f
                    })
                }),
                /* @__PURE__ */ n(k, {
                    regular: !0,
                    as: "div",
                    id: a,
                    children: /* @__PURE__ */ n("span", {
                        className: l ? A : "",
                        children: u.children
                    })
                })
            ]
        })
    })));
}, re = (t)=>{
    var _ref;
    const { value: s, defaultValue: h, onChange: v, focusableRef: x, disabled: u } = G({
        name: t.name,
        value: t.value,
        defaultValue: t.defaultValue,
        onChange: t.onChange,
        disabled: t.disabled
    }), l = typeof s < "u", [C, p] = i.useState((_ref = s !== null && s !== void 0 ? s : h) !== null && _ref !== void 0 ? _ref : "");
    i.useEffect(()=>{
        s !== void 0 && p(s);
    }, [
        s
    ]);
    const c = (e)=>{
        l || p(e), v(e);
    }, [R, g] = i.useState(null), a = i.useRef(null), y = ()=>{
        if (a.current) {
            const e = a.current.querySelector("[role=radio][aria-checked=true]"), r = Array.from(a.current.querySelectorAll("[role=radio]"));
            if (r.length === 0) return;
            const o = ((e ? r.indexOf(e) : 0) + 1) % r.length, m = r[o], b = m.dataset.value;
            b && (m.focus(), c(b));
        }
    }, d = ()=>{
        if (a.current) {
            const e = a.current.querySelector("[role=radio][aria-checked=true]"), r = Array.from(a.current.querySelectorAll("[role=radio]"));
            if (r.length === 0) return;
            const f = e ? r.indexOf(e) : 0, o = (r.length + f - 1) % r.length, m = r[o], b = m.dataset.value;
            b && (m.focus(), c(b));
        }
    };
    i.useEffect(()=>{
        if (a.current) {
            const e = a.current.querySelector("[role=radio]");
            var _e_dataset_value;
            e && g((_e_dataset_value = e.dataset.value) !== null && _e_dataset_value !== void 0 ? _e_dataset_value : null);
        }
    }, []);
    var _ref1;
    const V = (_ref1 = C !== null && C !== void 0 ? C : R) !== null && _ref1 !== void 0 ? _ref1 : null;
    return /* @__PURE__ */ n("div", _object_spread_props(_object_spread({
        ref: O(a, x),
        role: "radiogroup",
        "aria-labelledby": t["aria-labelledby"]
    }, D(t.dataAttributes, "RadioGroup")), {
        children: /* @__PURE__ */ n(N.Provider, {
            value: {
                disabled: u,
                selectedValue: C !== null && C !== void 0 ? C : h,
                focusableValue: V,
                select: c,
                selectNext: y,
                selectPrev: d
            },
            children: t.children
        })
    }));
}, le = L;
export { re as RadioGroup, le as default, J as useRadioContext };
