// @telefonica/mistica/dist-es/search-field.js
"use client";
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
function _object_without_properties(source, excluded) {
    if (source == null) return {};
    var target = _object_without_properties_loose(source, excluded);
    var key, i;
    if (Object.getOwnPropertySymbols) {
        var sourceSymbolKeys = Object.getOwnPropertySymbols(source);
        for(i = 0; i < sourceSymbolKeys.length; i++){
            key = sourceSymbolKeys[i];
            if (excluded.indexOf(key) >= 0) continue;
            if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
            target[key] = source[key];
        }
    }
    return target;
}
function _object_without_properties_loose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;
    for(i = 0; i < sourceKeys.length; i++){
        key = sourceKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        target[key] = source[key];
    }
    return target;
}
import { jsx as a } from "react/jsx-runtime";
import * as r from "react";
import { useFieldProps as A } from "./form-context.js";
import { TextFieldBaseAutosuggest as B, FieldEndIcon as E } from "./text-field-base.js";
import T from "./generated/mistica-icons/icon-search-regular.js";
import j from "./generated/mistica-icons/icon-close-regular.js";
import { useTheme as V } from "./hooks.js";
import { createChangeEvent as $ } from "./utils/dom.js";
import { combineRefs as q } from "./utils/common.js";
import { iconSize as w } from "./icon-button.css-mistica.js";
const D = /*#__PURE__*/ r.forwardRef((_param, x)=>{
    var { disabled: m, error: u, helperText: d, name: p, optional: n, validate: S, onChange: e, onChangeValue: t, onBlur: I, value: l, defaultValue: R, dataAttributes: F } = _param, b = _object_without_properties(_param, [
        "disabled",
        "error",
        "helperText",
        "name",
        "optional",
        "validate",
        "onChange",
        "onChangeValue",
        "onBlur",
        "value",
        "defaultValue",
        "dataAttributes"
    ]);
    const P = V(), o = r.useRef(), [h, v] = r.useState(R || ""), c = typeof l < "u", f = c ? l : h, i = r.useCallback((s, z)=>{
        c || v(s), t == null || t(s, z);
    }, [
        c,
        t
    ]), y = r.useCallback(()=>{
        i("", ""), o.current && (e == null || e($(o.current, "")), o.current.focus());
    }, [
        i,
        e
    ]), k = A({
        name: p,
        value: f,
        defaultValue: void 0,
        processValue: (s)=>s,
        helperText: d,
        optional: n,
        error: u,
        disabled: m,
        onBlur: I,
        validate: S,
        onChange: e,
        onChangeValue: i
    });
    return /* @__PURE__ */ a(B, _object_spread_props(_object_spread({
        ref: q(o, x),
        startIcon: /* @__PURE__ */ a(T, {
            size: w.default
        }),
        endIcon: f ? /* @__PURE__ */ a(E, {
            Icon: j,
            "aria-label": P.texts.formSearchClear,
            onPress: y
        }) : void 0
    }, b, k), {
        type: "search",
        dataAttributes: _object_spread({
            "component-name": "SearchField"
        }, F)
    }));
}), U = D;
export { U as default };
