// @telefonica/mistica/dist-es/select.js
"use client";
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
import { jsx as r, jsxs as _, Fragment as ke } from "react/jsx-runtime";
import * as n from "react";
import k from "classnames";
import { useForm as Ce } from "./form-context.js";
import { useAriaId as Pe, useTheme as De } from "./hooks.js";
import { SPACE as re, ENTER as se, ESC as Me, TAB as He, UP as Fe, DOWN as We } from "./utils/keys.js";
import { FieldContainer as Ge, HelperText as Le, Label as Ze } from "./text-field-components.js";
import oe from "./generated/mistica-icons/icon-chevron-down-regular.js";
import { TextFieldBaseAutosuggest as $e } from "./text-field-base.js";
import Be from "./overlay.js";
import { isAndroid as qe, isIos as ze } from "./utils/platform.js";
import { cancelEvent as p } from "./utils/dom.js";
import { Text3 as je } from "./text.js";
import { selectVariants as Ke, arrowDown as Ue, iconContainer as ie, selectContainerVariants as Xe, selectTextVariants as Je, vars as N, optionsContainer as Qe, optionsAnimationsVariants as ce, menuItem as Ye, menuItemSelected as et } from "./select.css-mistica.js";
import { inputWithLabel as ae, inputWithoutLabel as le } from "./text-field-base.css-mistica.js";
import { Portal as tt } from "./portal.js";
import { pxToRem as nt, applyCssVars as rt } from "./utils/css.js";
import { ThemeVariant as st } from "./theme-variant-context.js";
const ot = (param)=>{
    let { id: ue, label: m, helperText: de, value: fe, defaultValue: pe, onChangeValue: C, name: i, fullWidth: P, options: u, optional: D, disabled: me, error: he, onBlur: M, autoFocus: z = !1, native: Se } = param;
    var ne;
    const h = n.useRef(null), d = n.useRef(null), S = n.useRef(null), v = n.useRef(null), R = n.useRef(/* @__PURE__ */ new Map()), [ve, ge] = n.useState(!0), [g, Te] = n.useState(pe), [H, j] = n.useState(!1), [K, U] = n.useState(!1), [X, O] = n.useState(!1), [a, F] = n.useState({}), [T, W] = n.useState(), G = n.useRef(null), L = Pe(ue), { rawValues: Ie, setRawValue: Z, setValue: $, formStatus: Ee, formErrors: J, setFormError: we, register: B } = Ce(), { platformOverrides: Q } = De(), xe = Se || process.env.NODE_ENV === "test" && !process.env.SSR_TEST || qe(Q) || ze(Q), I = me || Ee === "sending", A = he || !!J[i], Y = J[i] || de, l = fe !== null && fe !== void 0 ? fe : Ie[i], ee = (e)=>{
        C == null || C(e), we({
            name: i,
            error: ""
        }), Z({
            name: i,
            value: e
        }), $({
            name: i,
            value: e
        });
    }, E = (e)=>{
        if (e) {
            if (S != null && S.current) {
                const o = S.current.getBoundingClientRect(), f = o.top, x = o.width, y = o.left, Ae = o.height, b = f + Ae, V = Math.min(u.length, 8) * 48 + 16;
                if (b + V + 12 > window.innerHeight) {
                    const Ve = window.innerHeight - b;
                    if (f > Ve) {
                        const _e = f - V;
                        F({
                            minWidth: x,
                            left: y,
                            top: Math.max(_e, 12),
                            maxHeight: Math.min(f - 12, V),
                            transformOrigin: "center bottom"
                        });
                    } else F({
                        minWidth: x,
                        top: b,
                        left: y,
                        maxHeight: window.innerHeight - b - 12,
                        transformOrigin: "center top"
                    });
                } else F({
                    minWidth: x,
                    top: b,
                    left: y,
                    maxHeight: V,
                    transformOrigin: "center top"
                });
            }
            j(!0), requestAnimationFrame(()=>{
                G.current && v.current && "scrollTop" in v.current && (v.current.scrollTop = G.current), U(!0);
            }), W(g !== null && g !== void 0 ? g : l);
        } else U(!1), j(!1), W(void 0);
    }, q = (e)=>{
        var t;
        G.current = (t = v.current) == null ? void 0 : t.scrollTop, E(!1), ee && typeof e == "string" && ee(e), typeof l > "u" && Te(e);
    }, Oe = (e)=>{
        var s;
        const t = (s = v.current) == null ? void 0 : s.getBoundingClientRect();
        if (t && e && R.current.has(e)) {
            const c = R.current.get(e), o = c == null ? void 0 : c.getBoundingClientRect();
            if (o && o.top + o.height / 2 >= t.top + t.height) {
                c == null || c.scrollIntoView();
                return;
            }
            o && o.top + o.height / 2 <= t.top && (c == null || c.scrollIntoView(!1));
        }
    }, ye = d.current, be = h.current;
    n.useEffect(()=>{
        Z({
            name: i,
            value: l
        }), $({
            name: i,
            value: l
        });
    }, [
        i,
        Z,
        $,
        l
    ]), n.useEffect(()=>(B(i, {
            input: h.current,
            focusableElement: d.current
        }), ()=>{
            B(i, {
                input: null,
                focusableElement: null
            });
        }), [
        i,
        B,
        d,
        h,
        ye,
        be
    ]), n.useEffect(()=>{
        const e = (s)=>{
            var f;
            const o = {
                [Fe]: -1,
                [We]: 1
            }[s.key];
            if (o) {
                p(s);
                var _ref;
                const x = (_ref = (f = u[u.findIndex((param)=>{
                    let { value: y } = param;
                    return y === T;
                }) + o]) == null ? void 0 : f.value) !== null && _ref !== void 0 ? _ref : T;
                W(x), Oe(x);
            }
        }, t = (s)=>{
            if (H) switch(s.key){
                case He:
                    p(s);
                    break;
                case Me:
                    E(!1);
                    break;
                case se:
                case re:
                    p(s), u.findIndex((param)=>{
                        let { value: c } = param;
                        return c === T;
                    }) !== -1 && T !== g && q(T), E(!1);
                    break;
            }
            K && e(s);
        };
        return document.addEventListener("keydown", t, !1), ()=>{
            document.removeEventListener("keydown", t, !1);
        };
    }), n.useEffect(()=>{
        z && d.current && d.current.focus();
    }, [
        z
    ]), n.useEffect(()=>{
        ge(!1);
    }, []);
    const Ne = (e)=>{
        var t;
        return e ? (t = u.find((param)=>{
            let { value: s } = param;
            return s === e;
        })) == null ? void 0 : t.text : "";
    }, Re = {
        tabIndex: 0,
        onFocus: ()=>O(!0),
        onBlur: ()=>O(!1),
        onClick: ()=>{
            E(!0), O(!0);
        },
        onKeyDown: (e)=>{
            !H && (e.key === re || e.key === se) && (p(e), E(!0));
        }
    }, te = nt(20), w = g !== null && g !== void 0 ? g : l;
    var _a_transformOrigin;
    return /* @__PURE__ */ r(st, {
        isInverse: !1,
        children: xe || ve ? /* @__PURE__ */ _(Ge, {
            disabled: I,
            helperText: /* @__PURE__ */ r(Le, {
                error: A,
                leftText: Y
            }),
            fieldRef: S,
            fullWidth: P,
            children: [
                m && /* @__PURE__ */ r(Ze, {
                    error: A,
                    forId: L,
                    inputState: X ? "focused" : (w !== null && w !== void 0 ? w : (ne = h.current) == null ? void 0 : ne.value) ? "filled" : "default",
                    optional: D,
                    children: m
                }),
                /* @__PURE__ */ _("select", {
                    className: k(Ke[I ? "disabled" : "default"], m ? ae : le),
                    id: L,
                    "aria-invalid": !!A,
                    value: w,
                    required: !D,
                    disabled: I,
                    onChange: (e)=>{
                        q(e.target.value);
                    },
                    onFocus: ()=>O(!0),
                    onBlur: (e)=>{
                        O(!1), M == null || M(e);
                    },
                    ref: (e)=>{
                        [
                            h,
                            d
                        ].forEach((t)=>{
                            t.current = e;
                        });
                    },
                    style: {
                        // Override default browser opacity when disabled. This opacity also affects the label.
                        // Without this fix, the label is invisible when disabled
                        opacity: 1,
                        // Use transparent color for the empty option (show below) to avoid showing it when the menu is closed
                        color: w ? void 0 : "transparent"
                    },
                    children: [
                        !w && u.every((param)=>{
                            let { value: e } = param;
                            return !!e;
                        }) && // If no "empty" option exists, insert a dummy empty option. Once an option is selected,
                        // this empty option is removed. This is needed to allow a native select without a selected
                        // default option.
                        // Chrome doesn't show this option when the select menu is open (because of display: none),
                        // but Safari does. So we use the select's label for this option, otherwise it would be shown
                        // as an empty option, which is weirder.
                        /* @__PURE__ */ r("option", {
                            value: "",
                            style: {
                                display: "none"
                            },
                            "aria-label": "",
                            children: m
                        }),
                        u.map((param)=>{
                            let { value: e, text: t } = param;
                            return(// Set color: 'initial' to avoid wrong text color in some browsers when using dark mode.
                            // Similar issue in another lib: https://github.com/chakra-ui/chakra-ui/issues/417#issue-566611352
                            /* @__PURE__ */ r("option", {
                                value: e,
                                style: {
                                    color: "initial"
                                },
                                children: t
                            }, e));
                        })
                    ]
                }),
                /* @__PURE__ */ r("div", {
                    className: Ue,
                    "aria-hidden": !0,
                    children: /* @__PURE__ */ r("div", {
                        className: ie,
                        children: /* @__PURE__ */ r(oe, {
                            size: te
                        })
                    })
                })
            ]
        }) : /* @__PURE__ */ _(ke, {
            children: [
                /* @__PURE__ */ _("div", _object_spread_props(_object_spread({
                    className: Xe[P ? "fullWidth" : "default"],
                    role: "button",
                    "aria-haspopup": "listbox",
                    ref: d
                }, !I && Re), {
                    children: [
                        /* @__PURE__ */ r($e, {
                            style: {
                                visibility: "hidden"
                            },
                            fullWidth: P,
                            endIcon: /* @__PURE__ */ r("div", {
                                className: ie,
                                children: /* @__PURE__ */ r(oe, {
                                    size: te
                                })
                            }),
                            focus: X,
                            label: m,
                            value: w,
                            shrinkLabel: !!(l || g),
                            name: i,
                            helperText: Y,
                            required: !D,
                            disabled: I,
                            id: L,
                            error: A,
                            inputRef: h,
                            fieldRef: S
                        }),
                        /* @__PURE__ */ r("div", {
                            className: k(Je[I ? "disabled" : "default"], m ? ae : le),
                            children: Ne(w)
                        })
                    ]
                })),
                H && /* @__PURE__ */ r(Be, {
                    onPress: (e)=>{
                        E(!1), p(e);
                    },
                    disableScroll: !0,
                    children: /* @__PURE__ */ r(tt, {
                        children: /* @__PURE__ */ r("ul", {
                            style: rt({
                                [N.top]: a.top ? `${a.top}px` : "",
                                [N.left]: a.left ? `${a.left}px` : "",
                                [N.maxHeight]: a.maxHeight ? `${a.maxHeight}px` : "",
                                [N.minWidth]: a.minWidth ? `${a.minWidth}px` : "",
                                [N.transformOrigin]: (_a_transformOrigin = a.transformOrigin) !== null && _a_transformOrigin !== void 0 ? _a_transformOrigin : ""
                            }),
                            onPointerDown: p,
                            className: k(Qe, K ? ce.show : ce.hide),
                            role: "listbox",
                            ref: v,
                            children: u.map((param)=>{
                                let { value: e, text: t } = param;
                                return /* @__PURE__ */ r("li", {
                                    role: "option",
                                    "aria-selected": e === (g !== null && g !== void 0 ? g : l),
                                    "data-value": e,
                                    className: k(Ye, {
                                        [et]: e === T || e === (g !== null && g !== void 0 ? g : l)
                                    }),
                                    onPointerDown: p,
                                    onClick: ()=>q(e),
                                    ref: (s)=>{
                                        s ? R.current.set(e, s) : R.current.delete(e);
                                    },
                                    children: /* @__PURE__ */ r(je, {
                                        regular: !0,
                                        children: t
                                    })
                                }, e);
                            })
                        })
                    })
                })
            ]
        })
    });
}, Ot = ot;
export { Ot as default };
