// @telefonica/mistica/dist-es/select.css-mistica.js
import "./sprinkles.css.ts.vanilla.css-mistica.js";
import "./icon-button.css.ts.vanilla.css-mistica.js";
import "./text-field-base.css.ts.vanilla.css-mistica.js";
import "./select.css.ts.vanilla.css-mistica.js";
var i = "_1sqpivvo _1y2v1nfhf", y = "_1sqpivvp", a = "_1sqpivvr _1y2v1nfaa _1y2v1nfbj _1y2v1nf7m _1y2v1nf8v _1y2v1nf26 _1y2v1nfhk _1y2v1nfhz _1y2v1nfkk", s = "_1y2v1nf44", r = {
    show: "_1sqpivv7",
    hide: "_1sqpivv8"
}, p = "_1sqpivv6 _1y2v1nf7d _1y2v1nf8m _1y2v1nf9v _1y2v1nfb4 _1y2v1nfhg _1y2v1nfka _1y2v1nf3q", t = {
    default: "_1sqpivvd _1sqpivvc _1y2v1nfhe _1y2v1nfkk",
    fullWidth: "_1sqpivve _1sqpivvc _1y2v1nfhe _1y2v1nfkk"
}, e = {
    default: "_1sqpivvl _1sqpivvk _1y2v1nfhf _1y2v1nfkl _1y2v1nfjz _1y2v1nf26",
    disabled: "_1sqpivvm _1sqpivvk _1y2v1nfhf _1y2v1nfkl _1y2v1nfjz _1y2v1nf26"
}, q = {
    default: "_1sqpivvh _1sqpivvg _1y2v1nf7d _1y2v1nf8m _1y2v1nfk2 _1y2v1nfkl _1y2v1nf26 _1y2v1nf7c _1y2v1nfi7",
    disabled: "_1sqpivvi _1sqpivvg _1y2v1nf7d _1y2v1nf8m _1y2v1nfk2 _1y2v1nfkl _1y2v1nf26 _1y2v1nf7c _1y2v1nfi7"
}, o = {
    top: "var(--_1sqpivv0)",
    left: "var(--_1sqpivv1)",
    transformOrigin: "var(--_1sqpivv2)",
    minWidth: "var(--_1sqpivv3)",
    maxHeight: "var(--_1sqpivv4)"
};
export { i as arrowDown, y as iconContainer, a as menuItem, s as menuItemSelected, r as optionsAnimationsVariants, p as optionsContainer, t as selectContainerVariants, e as selectTextVariants, q as selectVariants, o as vars };
