// @telefonica/mistica/dist-es/sheet.js
"use client";
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
function _object_without_properties(source, excluded) {
    if (source == null) return {};
    var target = _object_without_properties_loose(source, excluded);
    var key, i;
    if (Object.getOwnPropertySymbols) {
        var sourceSymbolKeys = Object.getOwnPropertySymbols(source);
        for(i = 0; i < sourceSymbolKeys.length; i++){
            key = sourceSymbolKeys[i];
            if (excluded.indexOf(key) >= 0) continue;
            if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
            target[key] = source[key];
        }
    }
    return target;
}
function _object_without_properties_loose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;
    for(i = 0; i < sourceKeys.length; i++){
        key = sourceKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        target[key] = source[key];
    }
    return target;
}
import { jsx as e, jsxs as v, Fragment as O } from "react/jsx-runtime";
import w from "classnames";
import * as i from "react";
import { transitionDuration as z, overlay as G, closingOverlay as H, SheetContainer as V, closingSheet as j, Sheet as F, SheetContent as K, handleContainer as $, handle as X, children as q, modalCloseButton as J, stickyTitle as Q, bodyContent as U, stickyButtons as W, sheetActionRow as Z, infoItemIcon as ee } from "./sheet.css-mistica.js";
import te from "./focus-trap.js";
import { useTheme as b, useAriaId as re, useDisableBodyScroll as ne, useIsInViewport as N, useScreenSize as C } from "./hooks.js";
import { useSetModalStateEffect as oe } from "./modal-context-provider.js";
import { Portal as ie } from "./portal.js";
import { Text5 as se, Text3 as A, Text2 as L } from "./text.js";
import { vars as S } from "./skins/skin-contract.css-mistica.js";
import { RadioGroup as ce } from "./radio-button.js";
import { RowList as ae, Row as le } from "./list.js";
import M from "./negative-box.js";
import I from "./stack.js";
import y from "./box.js";
import de from "./touchable.js";
import ue from "./inline.js";
import me from "./circle.js";
import x from "./divider.js";
import { getPrefixedDataAttributes as he, getScrollableParentElement as fe } from "./utils/dom.js";
import { ButtonPrimary as Y, ButtonSecondary as pe, ButtonLink as ge } from "./button.js";
import ve from "./generated/mistica-icons/icon-close-regular.js";
import { InternalIconButton as Re } from "./icon-button.js";
import Se from "./button-layout.js";
import _ from "./image.js";
import { safeAreaInsetBottom as De } from "./utils/css.js";
import { MOBILE_SIDE_MARGIN as ye, TABLET_SIDE_MARGIN as Ee, SMALL_DESKTOP_SIDE_MARGIN as Ie } from "./responsive-layout.css-mistica.js";
const be = process.env.NODE_ENV === "test" ? 0 : z, P = (n)=>"touches" in n ? n.touches[0].clientY : n.clientY, Te = (param)=>{
    let { closeModal: n } = param;
    const [s, l] = i.useState(0), g = i.useRef(!1), f = i.useRef(0), r = i.useRef(0), o = i.useRef(0), u = i.useRef(0), { isDesktopOrBigger: m } = C(), t = i.useCallback((h)=>{
        g.current = !0, f.current = 0, r.current = Date.now(), o.current = P(h);
    }, []), c = i.useCallback(()=>{
        g.current = !1, l(0);
    }, []);
    return i.useEffect(()=>{
        if (m) return;
        const h = (a)=>{
            if (g.current) {
                if (f.current < 3) {
                    f.current++;
                    return;
                }
                u.current = P(a), l(u.current - o.current);
            }
        }, d = ()=>{
            if (!g.current) return;
            const a = Date.now() - r.current, p = u.current - o.current, D = p / a;
            g.current = !1, l(0), p > 50 && (u.current > window.innerHeight * 0.75 || D > 0.5) && n();
        };
        return document.addEventListener("touchmove", h), document.addEventListener("touchend", d), document.addEventListener("mousemove", h), document.addEventListener("mouseup", d), ()=>{
            document.removeEventListener("touchmove", h), document.removeEventListener("touchend", d), document.removeEventListener("mousemove", h), document.removeEventListener("mouseup", d);
        };
    }, [
        n,
        m
    ]), m ? {} : {
        onTouchStart: t,
        onMouseDown: t,
        style: s ? {
            transform: `translateY(${s}px)`,
            transition: "none"
        } : void 0,
        onScroll: c,
        overlayStyle: s ? {
            // decrease opacity when dragging down the sheet
            opacity: 0.25 + 1 - s / (window.innerHeight - o.current),
            transition: "none"
        } : void 0
    };
}, ke = {
    closed: {
        open: "opening"
    },
    opening: {
        close: "closed",
        transitionEnd: "open"
    },
    open: {
        close: "closing"
    },
    closing: {
        transitionEnd: "closed"
    }
}, Le = (n, s)=>ke[n][s] || n, E = /*#__PURE__*/ i.forwardRef((param, g)=>{
    let { onClose: n, children: s, dataAttributes: l } = param;
    const { texts: f } = b(), [r, o] = i.useReducer(Le, "closed"), u = i.useRef(!1), m = re(), t = i.useCallback((p)=>{
        p.target === p.currentTarget && o("transitionEnd");
    }, []), c = ()=>{
        r === "open" && o("close");
    };
    i.useEffect(()=>{
        if (r === "opening" || r === "closing") {
            const p = setTimeout(()=>{
                o("transitionEnd");
            }, be);
            return ()=>clearTimeout(p);
        }
    }, [
        r
    ]), i.useEffect(()=>{
        r === "closed" ? u.current ? n == null || n() : o("open") : u.current = !0;
    }, [
        r,
        n
    ]);
    const _Te = Te({
        closeModal: c
    }), { onScroll: h, overlayStyle: d } = _Te, a = _object_without_properties(_Te, [
        "onScroll",
        "overlayStyle"
    ]);
    return oe(), ne(r !== "closed"), r === "closed" ? null : /* @__PURE__ */ e(ie, {
        children: /* @__PURE__ */ v(te, {
            children: [
                /* @__PURE__ */ e("div", {
                    className: w(G, {
                        [H]: r === "closing"
                    }),
                    style: d,
                    onClick: c
                }),
                /* @__PURE__ */ e("div", _object_spread_props(_object_spread({
                    className: w(V, {
                        [j]: r === "closing"
                    }),
                    onTransitionEnd: t,
                    onAnimationEnd: t
                }, a, he(l, "Sheet")), {
                    ref: g,
                    children: /* @__PURE__ */ e("div", {
                        className: F,
                        children: /* @__PURE__ */ v("div", {
                            className: K,
                            children: [
                                /* @__PURE__ */ e("div", {
                                    className: $,
                                    children: /* @__PURE__ */ e("div", {
                                        className: X
                                    })
                                }),
                                /* @__PURE__ */ v("section", {
                                    role: "dialog",
                                    "aria-modal": "true",
                                    "aria-labelledby": m,
                                    onScroll: h,
                                    className: q,
                                    style: {
                                        paddingBottom: De
                                    },
                                    children: [
                                        typeof s == "function" ? s({
                                            closeModal: c,
                                            modalTitleId: m
                                        }) : s,
                                        /* @__PURE__ */ e("div", {
                                            className: J,
                                            children: /* @__PURE__ */ e(Re, {
                                                onPress: c,
                                                "aria-label": f.modalClose || f.closeButtonLabel,
                                                Icon: ve,
                                                bleedLeft: !0,
                                                bleedRight: !0,
                                                bleedY: !0
                                            })
                                        })
                                    ]
                                })
                            ]
                        })
                    })
                }))
            ]
        })
    });
}), k = {
    mobile: ye,
    tablet: Ee,
    desktop: Ie
}, T = (param)=>{
    let { title: n, subtitle: s, description: l, modalTitleId: g, button: f, secondaryButton: r, link: o, children: u } = param;
    const m = i.useRef(null), t = i.useRef(null), c = i.useRef(null);
    i.useEffect(()=>{
        t.current && (c.current = fe(t.current));
    }, []);
    const h = !N(m, !0, {
        root: c.current
    }), d = !N(t, !0, {
        rootMargin: "1px",
        // bottomScrollSignal div has 0px height so we need a 1px margin to trigger the intersection observer
        root: c.current
    }), a = !!f || !!r || !!o;
    return /* @__PURE__ */ v(O, {
        children: [
            /* @__PURE__ */ e("div", {
                ref: m
            }),
            /* @__PURE__ */ v("div", {
                className: Q,
                children: [
                    n ? /* @__PURE__ */ e(y, {
                        paddingBottom: 8,
                        paddingTop: {
                            mobile: 0,
                            desktop: 40
                        },
                        paddingX: k,
                        children: /* @__PURE__ */ e(se, {
                            as: "h2",
                            id: g,
                            truncate: !0,
                            children: n
                        })
                    }) : /* @__PURE__ */ e(y, {
                        paddingTop: {
                            mobile: 0,
                            desktop: 40
                        }
                    }),
                    h && /* @__PURE__ */ e(x, {})
                ]
            }),
            /* @__PURE__ */ e("div", {
                className: U,
                children: /* @__PURE__ */ e(y, {
                    paddingBottom: a ? 0 : {
                        desktop: 40,
                        mobile: 0
                    },
                    paddingX: k,
                    children: /* @__PURE__ */ v(I, {
                        space: 8,
                        children: [
                            s || l ? /* @__PURE__ */ v(I, {
                                space: {
                                    mobile: 8,
                                    desktop: 16
                                },
                                children: [
                                    s && /* @__PURE__ */ e(A, {
                                        as: "p",
                                        regular: !0,
                                        children: s
                                    }),
                                    l && (Array.isArray(l) ? /* @__PURE__ */ e(L, {
                                        as: "div",
                                        regular: !0,
                                        color: S.colors.textSecondary,
                                        children: l.map((p, D)=>/* @__PURE__ */ e("p", {
                                                style: {
                                                    margin: 0,
                                                    marginBottom: D < l.length - 1 ? "1em" : void 0
                                                },
                                                children: p
                                            }, D))
                                    }) : /* @__PURE__ */ e(L, {
                                        as: "p",
                                        regular: !0,
                                        color: S.colors.textSecondary,
                                        children: l
                                    }))
                                ]
                            }) : null,
                            u
                        ]
                    })
                })
            }),
            a && /* @__PURE__ */ v("div", {
                className: W,
                children: [
                    d && /* @__PURE__ */ e(x, {}),
                    /* @__PURE__ */ e(y, {
                        paddingY: {
                            mobile: 16,
                            desktop: 40
                        },
                        paddingX: k,
                        children: /* @__PURE__ */ e(Se, {
                            align: "full-width",
                            link: o,
                            primaryButton: f,
                            secondaryButton: r
                        })
                    })
                ]
            }),
            /* @__PURE__ */ e("div", {
                ref: t
            })
        ]
    });
}, et = /*#__PURE__*/ i.forwardRef((param, t)=>{
    let { title: n, subtitle: s, description: l, items: g, selectedId: f, onClose: r, onSelect: o, button: u, dataAttributes: m } = param;
    const [c, h] = i.useState(f), d = i.useRef(!1), { isDesktopOrBigger: a } = C(), { texts: p } = b();
    return /* @__PURE__ */ e(E, {
        onClose: r,
        ref: t,
        dataAttributes: _object_spread({
            "component-name": "RadioListSheet"
        }, m),
        children: (param)=>/* @__PURE__ */ {
            let { closeModal: D, modalTitleId: B } = param;
            var _ref;
            return e(T, {
                title: n,
                subtitle: s,
                description: l,
                modalTitleId: B,
                button: a ? /* @__PURE__ */ e(Y, {
                    onPress: ()=>{
                        d.current && (o == null || o(c !== null && c !== void 0 ? c : "")), D();
                    },
                    children: (_ref = u == null ? void 0 : u.text) !== null && _ref !== void 0 ? _ref : p.sheetConfirmButton
                }) : void 0,
                children: /* @__PURE__ */ e(M, {
                    children: /* @__PURE__ */ e(ce, {
                        "aria-labelledby": B,
                        name: "sheetselection",
                        value: c,
                        onChange: (R)=>{
                            h(R), d.current = !0, !a && (o == null || o(R), setTimeout(()=>{
                                D();
                            }, 200));
                        },
                        children: /* @__PURE__ */ e(ae, {
                            children: g.map((R)=>/* @__PURE__ */ {
                                var _R_title;
                                return e(le, {
                                    title: (_R_title = R.title) !== null && _R_title !== void 0 ? _R_title : "",
                                    description: R.description,
                                    asset: R.asset,
                                    radioValue: R.id
                                }, R.id);
                            })
                        })
                    })
                })
            });
        }
    });
}), tt = /*#__PURE__*/ i.forwardRef((param, u)=>{
    let { title: n, subtitle: s, description: l, items: g, onClose: f, onSelect: r, dataAttributes: o } = param;
    const { isDarkMode: m } = b();
    return /* @__PURE__ */ e(E, {
        onClose: f,
        ref: u,
        dataAttributes: _object_spread({
            "component-name": "ActionsListSheet"
        }, o),
        children: (param)=>{
            let { closeModal: t, modalTitleId: c } = param;
            return /* @__PURE__ */ e(T, {
                title: n,
                subtitle: s,
                description: l,
                modalTitleId: c,
                children: /* @__PURE__ */ e(M, {
                    children: g.map((param)=>{
                        let { id: h, style: d, title: a, icon: p } = param;
                        return /* @__PURE__ */ e(de, {
                            onPress: ()=>{
                                r == null || r(h), t();
                            },
                            children: /* @__PURE__ */ v("div", {
                                className: Z,
                                children: [
                                    p && /* @__PURE__ */ e(y, {
                                        paddingRight: 16,
                                        children: p.Icon ? /* @__PURE__ */ e(p.Icon, {
                                            size: 24,
                                            color: d === "destructive" ? S.colors.textLinkDanger : S.colors.neutralHigh
                                        }) : /* @__PURE__ */ e(_, {
                                            circular: !0,
                                            src: m && p.urlDark ? p.urlDark : p.url,
                                            width: 40
                                        })
                                    }),
                                    /* @__PURE__ */ e(A, {
                                        regular: !0,
                                        color: d === "destructive" ? S.colors.textLinkDanger : S.colors.textPrimary,
                                        children: a
                                    })
                                ]
                            })
                        }, h);
                    })
                })
            });
        }
    });
}), rt = /*#__PURE__*/ i.forwardRef((param, o)=>{
    let { title: n, subtitle: s, description: l, items: g, onClose: f, dataAttributes: r } = param;
    const { isDarkMode: u } = b();
    return /* @__PURE__ */ e(E, {
        onClose: f,
        ref: o,
        dataAttributes: _object_spread({
            "component-name": "InfoSheet"
        }, r),
        children: (param)=>{
            let { modalTitleId: m } = param;
            return /* @__PURE__ */ e(T, {
                title: n,
                subtitle: s,
                description: l,
                modalTitleId: m,
                children: /* @__PURE__ */ e(y, {
                    paddingBottom: 16,
                    children: /* @__PURE__ */ e(I, {
                        space: 16,
                        role: "list",
                        children: g.map((t, c)=>/* @__PURE__ */ v(ue, {
                                space: 8,
                                children: [
                                    /* @__PURE__ */ e("div", {
                                        className: ee,
                                        children: t.icon.type === "bullet" ? /* @__PURE__ */ e(me, {
                                            size: 8,
                                            backgroundColor: S.colors.textPrimary
                                        }) : t.icon.Icon ? /* @__PURE__ */ e(t.icon.Icon, {
                                            size: t.icon.type === "small" ? 16 : 24
                                        }) : /* @__PURE__ */ e(_, {
                                            src: u && t.icon.urlDark ? t.icon.urlDark : t.icon.url,
                                            width: t.icon.type === "small" ? 16 : 24,
                                            height: t.icon.type === "small" ? 16 : 24
                                        })
                                    }),
                                    /* @__PURE__ */ v(I, {
                                        space: 2,
                                        children: [
                                            /* @__PURE__ */ e(A, {
                                                regular: !0,
                                                children: t.title
                                            }),
                                            t.description && /* @__PURE__ */ e(L, {
                                                regular: !0,
                                                color: S.colors.textSecondary,
                                                children: t.description
                                            })
                                        ]
                                    })
                                ]
                            }, t.id || c))
                    })
                })
            });
        }
    });
}), nt = /*#__PURE__*/ i.forwardRef((param, t)=>{
    let { title: n, subtitle: s, description: l, button: g, secondaryButton: f, buttonLink: r, onClose: o, dataAttributes: u, onPressButton: m } = param;
    const c = (d, a)=>()=>{
            m == null || m(a), d();
        }, h = (_param)=>{
        var { text: d } = _param, a = _object_without_properties(_param, [
            "text"
        ]);
        return _object_spread({
            children: d
        }, a);
    };
    return /* @__PURE__ */ e(E, {
        onClose: o,
        ref: t,
        dataAttributes: _object_spread({
            "component-name": "ActionsSheet"
        }, u),
        children: (param)=>{
            let { modalTitleId: d, closeModal: a } = param;
            return /* @__PURE__ */ e(T, {
                title: n,
                subtitle: s,
                description: l,
                modalTitleId: d,
                button: /* @__PURE__ */ e(Y, _object_spread_props(_object_spread({}, h(g)), {
                    onPress: c(a, "PRIMARY")
                })),
                secondaryButton: f ? /* @__PURE__ */ e(pe, _object_spread_props(_object_spread({}, h(f)), {
                    onPress: c(a, "SECONDARY")
                })) : void 0,
                link: r ? /* @__PURE__ */ e(ge, _object_spread_props(_object_spread({}, h(r)), {
                    onPress: c(a, "LINK")
                })) : void 0
            });
        }
    });
}), ot = E;
export { tt as ActionsListSheet, nt as ActionsSheet, rt as InfoSheet, et as RadioListSheet, T as SheetBody, ot as default };
