// @telefonica/mistica/dist-es/stacking-group.js
"use client";
import { jsx as e, jsxs as u } from "react/jsx-runtime";
import * as l from "react";
import { useIsInverseVariant as f } from "./theme-variant-context.js";
import { vars as r } from "./skins/skin-contract.css-mistica.js";
import h from "./inline.js";
import { Text4 as v } from "./text.js";
import { moreItems as b } from "./stacking-group.css-mistica.js";
import { vars as g } from "./image.css-mistica.js";
import { applyCssVars as y } from "./utils/css.js";
const C = (param)=>{
    let { moreItemsStyle: s, stacked: t = !0, maxItems: o = 1 / 0, children: n } = param;
    const c = f(), i = l.Children.count(n), m = i - o + 1, p = t ? -8 : 8, a = s.size, d = s.type === "circle" ? "50%" : r.borderRadii.mediaSmall;
    return /* @__PURE__ */ e("div", {
        style: y({
            [g.mediaBorderRadius]: d
        }),
        children: /* @__PURE__ */ u(h, {
            space: p,
            children: [
                l.Children.toArray(n).slice(0, i > o ? o - 1 : o),
                i > o && /* @__PURE__ */ e("div", {
                    className: b,
                    style: {
                        width: a,
                        height: a,
                        borderRadius: d,
                        backgroundColor: c ? r.colors.brandHigh : r.colors.brandLow,
                        border: t ? `1px solid ${r.colors.borderLow}` : "none"
                    },
                    children: /* @__PURE__ */ e(v, {
                        regular: !0,
                        color: c ? r.colors.textPrimaryInverse : r.colors.textBrand,
                        children: "+" + m
                    })
                })
            ]
        })
    });
}, B = C;
export { B as default };
