// @telefonica/mistica/dist-es/stepper.js
"use client";
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
import { jsx as e, jsxs as h } from "react/jsx-runtime";
import * as l from "react";
import d from "classnames";
import { Text1 as k, Text2 as w } from "./text.js";
import { useTheme as F, useScreenSize as P, useElementDimensions as T } from "./hooks.js";
import B from "./icons/icon-success.js";
import { stepper as D, vars as M, step as R, stepIconNumber as u, iconAnimation as $, number as j, currentNumber as z, textContainer as E, bar as H, barFilled as I, barFilledAnimation as L, barFilledReverseAnimation as O } from "./stepper.css-mistica.js";
import { applyCssVars as V, pxToRem as q } from "./utils/css.js";
import { vars as o } from "./skins/skin-contract.css-mistica.js";
import { getPrefixedDataAttributes as G } from "./utils/dom.js";
const J = (param)=>{
    let { steps: n, currentIndex: m, "aria-label": v, dataAttributes: f } = param;
    const { textPresets: g } = F(), { isDesktopOrBigger: S } = P(), { height: b, ref: N } = T(), x = b, [a, A] = l.useState(Math.ceil(m)), [i, y] = l.useState(!1);
    return l.useEffect(()=>{
        const s = Math.ceil(m);
        a !== s && (y(s < a), A(s));
    }, [
        m,
        n,
        a
    ]), /* @__PURE__ */ e("div", _object_spread_props(_object_spread({
        className: D,
        style: V({
            [M.stepperMinHeight]: q(40 + x)
        })
    }, G(f, "Stepper")), {
        children: n.map((s, t)=>{
            const r = t === a, C = t === n.length - 1, c = t < a, p = t === a - 1;
            return /* @__PURE__ */ h(l.Fragment, {
                children: [
                    /* @__PURE__ */ h("div", {
                        className: R,
                        role: "progressbar",
                        "aria-valuenow": r ? t + 1 : void 0,
                        "aria-valuemin": 1,
                        "aria-valuemax": n.length,
                        "aria-valuetext": s,
                        "aria-label": v,
                        children: [
                            c ? /* @__PURE__ */ e("div", {
                                className: d(u, {
                                    [$]: p && !i
                                }),
                                "aria-hidden": "true",
                                children: /* @__PURE__ */ e(B, {
                                    color: o.colors.controlActivated,
                                    size: "100%",
                                    skipAnimation: !p || i
                                })
                            }) : /* @__PURE__ */ e("div", {
                                className: d(u, j, {
                                    [z]: r || r
                                }),
                                children: /* @__PURE__ */ e(k, {
                                    as: "span",
                                    weight: g.indicator.weight,
                                    color: r ? o.colors.textPrimaryInverse : o.colors.textSecondary,
                                    "aria-hidden": "true",
                                    children: t + 1
                                })
                            }),
                            S && /* @__PURE__ */ e("div", {
                                className: E,
                                ref: N,
                                "aria-hidden": "true",
                                children: /* @__PURE__ */ e(w, {
                                    as: "span",
                                    regular: !0,
                                    color: c || r ? o.colors.textPrimary : o.colors.textSecondary,
                                    children: s
                                })
                            })
                        ]
                    }),
                    !C && /* @__PURE__ */ e("div", {
                        className: H,
                        "aria-hidden": "true",
                        children: (c || r) && /* @__PURE__ */ e("div", {
                            className: d({
                                [I]: i && r || c,
                                [L]: p && !i,
                                [O]: r && i
                            })
                        })
                    })
                ]
            }, t);
        })
    }));
}, re = J;
export { re as default };
