// @telefonica/mistica/dist-es/switch-component.css-mistica.js
import "./sprinkles.css.ts.vanilla.css-mistica.js";
import "./switch-component.css.ts.vanilla.css-mistica.js";
var n = {
    default: "q0hp3aw q0hp3au q0hp3at _1y2v1nfhf _1y2v1nfm0 _1y2v1nfhm _1y2v1nfk3",
    ios: "q0hp3ax q0hp3av q0hp3at _1y2v1nfhf _1y2v1nfm0 _1y2v1nfhm _1y2v1nfk3",
    checked: "q0hp3ay q0hp3au q0hp3at _1y2v1nfhf _1y2v1nfm0 _1y2v1nfhm _1y2v1nfk3",
    checkedIos: "q0hp3az q0hp3av q0hp3at _1y2v1nfhf _1y2v1nfm0 _1y2v1nfhm _1y2v1nfk3"
}, v = {
    default: "q0hp3ai q0hp3ah _1y2v1nfhm",
    ios: "q0hp3aj q0hp3ah _1y2v1nfhm",
    iosDark: "q0hp3ak q0hp3ah _1y2v1nfhm",
    checked: "q0hp3al q0hp3ah _1y2v1nfhm",
    checkedIos: "q0hp3am q0hp3ah _1y2v1nfhm"
}, f = {
    default: "q0hp3a6 _1y2v1nfho _1y2v1nfkk",
    ios: "q0hp3a7 _1y2v1nfho _1y2v1nfkk",
    disabled: "q0hp3a8 _1y2v1nfho _1y2v1nfkk",
    disabledIos: "q0hp3a9 _1y2v1nfho _1y2v1nfkk"
}, p = "q0hp3ap q0hp3an", q = "q0hp3aq q0hp3an", y = "q0hp3ar", _ = "q0hp3ao", e = {
    default: "q0hp3ac q0hp3ab _1y2v1nfhe",
    ios: "q0hp3ad q0hp3ab _1y2v1nfhe"
}, r = "q0hp3af _1y2v1nfhm _1y2v1nfkl";
export { n as ballVariants, v as barVariants, f as checkboxVariants, p as container, q as containerDisabled, y as disabled, _ as interactiveArea, e as switchCheckboxContainerVariants, r as switchCheckboxLabel };
