// @telefonica/mistica/dist-es/table.js
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
function _object_without_properties(source, excluded) {
    if (source == null) return {};
    var target = _object_without_properties_loose(source, excluded);
    var key, i;
    if (Object.getOwnPropertySymbols) {
        var sourceSymbolKeys = Object.getOwnPropertySymbols(source);
        for(i = 0; i < sourceSymbolKeys.length; i++){
            key = sourceSymbolKeys[i];
            if (excluded.indexOf(key) >= 0) continue;
            if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
            target[key] = source[key];
        }
    }
    return target;
}
function _object_without_properties_loose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;
    for(i = 0; i < sourceKeys.length; i++){
        key = sourceKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        target[key] = source[key];
    }
    return target;
}
import { jsxs as f, jsx as e } from "react/jsx-runtime";
import * as x from "react";
import { getPrefixedDataAttributes as V } from "./utils/dom.js";
import { table as X, boxed as W, collapsedRowsInMobile as E, fullWidth as M, hiddenHeadersInDesktop as ee, hiddenHeadersInMobile as te, cellTextAlign as _, verticalAlign as N, rowFirstItem as L, rowLastItem as R, rowLastCollapsedItem as H, collapsedRowHeaderItem as le, mobileCellHeading as re, collapsedRowTitle as ae, actionsTableCell as Y, topActions as oe, scrollContainer as se, scrollOverResponsiveLayout as ne, vars as ie } from "./table.css-mistica.js";
import { vars as j } from "./skins/skin-contract.css-mistica.js";
import ce, { Text1 as F, textProps as h, Text2 as me } from "./text.js";
import { InternalBoxed as pe } from "./boxed.js";
import c from "classnames";
import de from "./box.js";
import { applyCssVars as he } from "./utils/css.js";
import be from "./inline.js";
import { IconButton as ue, ToggleIconButton as ge } from "./icon-button.js";
import { iconContainerSize as G, iconSize as ye } from "./icon-button.css-mistica.js";
import { TableActionsHeader as fe } from "./table-actions-header.js";
const P = {
    default: "8px",
    boxed: "16px"
}, Z = "1px", ke = (a)=>a.Icon ? /* @__PURE__ */ e(ue, _object_spread_props(_object_spread({}, a), {
        "aria-label": a.label,
        small: !0,
        type: "neutral",
        backgroundType: "transparent",
        bleedY: !0
    })) : /* @__PURE__ */ e(ge, _object_spread_props(_object_spread({}, a), {
        checkedProps: _object_spread_props(_object_spread({}, a.checkedProps), {
            "aria-label": a.checkedProps.label,
            type: "brand",
            backgroundType: "solid"
        }),
        uncheckedProps: _object_spread_props(_object_spread({}, a.uncheckedProps), {
            "aria-label": a.uncheckedProps.label,
            type: "neutral",
            backgroundType: "transparent"
        }),
        small: !0,
        bleedY: !0
    })), q = "left", Oe = /*#__PURE__*/ x.forwardRef((_param, $)=>{
    var { dataAttributes: a, heading: m = [], content: k = [], boxed: b, responsive: J, fullWidth: u = !0, maxHeight: v, emptyCase: A, columnTextAlign: T = q, rowVerticalAlign: C = "middle", columnWidth: p, hideHeaders: g, scrollOverResponsiveLayout: K, rowHeaderIndex: s } = _param, S = _object_without_properties(_param, [
        "dataAttributes",
        "heading",
        "content",
        "boxed",
        "responsive",
        "fullWidth",
        "maxHeight",
        "emptyCase",
        "columnTextAlign",
        "rowVerticalAlign",
        "columnWidth",
        "hideHeaders",
        "scrollOverResponsiveLayout",
        "rowHeaderIndex"
    ]);
    const z = (l)=>{
        var _T_l;
        return Array.isArray(T) ? (_T_l = T[l]) !== null && _T_l !== void 0 ? _T_l : q : T;
    }, r = J === "collapse-rows", B = g === !0 || g === "mobile", Q = g === !0 || g === "desktop", y = k.some((l)=>!Array.isArray(l) && l.actions.length > 0), D = /* @__PURE__ */ f("table", {
        className: c(X, {
            [W]: b,
            [E]: r,
            [M]: u,
            [ee]: Q,
            [te]: B || r
        }),
        "aria-label": S["aria-label"],
        "aria-labelledby": S["aria-labelledby"],
        "aria-describedby": S["aria-describedby"],
        children: [
            m.length > 0 && /* @__PURE__ */ e("thead", {
                children: /* @__PURE__ */ f(F, {
                    as: "tr",
                    medium: !0,
                    transform: "uppercase",
                    color: j.colors.textSecondary,
                    wordBreak: !1,
                    children: [
                        m.map((l, o)=>/* @__PURE__ */ e("th", {
                                scope: "col",
                                className: c(_[z(o)], N[C], {
                                    [L]: o === 0,
                                    [R]: o === m.length - 1 && !y
                                }),
                                style: {
                                    minWidth: p == null ? void 0 : p[o],
                                    width: p == null ? void 0 : p[o]
                                },
                                children: l
                            }, o)),
                        y && /* @__PURE__ */ e(fe, {})
                    ]
                })
            }),
            /* @__PURE__ */ e("tbody", {
                children: k.length > 0 ? k.map((l, o)=>{
                    var _l_actions;
                    const n = Array.isArray(l) ? l : l.cells, i = Array.isArray(l) ? [] : (_l_actions = l.actions) !== null && _l_actions !== void 0 ? _l_actions : [], w = /* @__PURE__ */ e(be, {
                        space: 16,
                        children: i.map((d, t)=>"Icon" in d || "checkedProps" in d ? /* @__PURE__ */ e(ke, _object_spread({}, d), t) : /* @__PURE__ */ e(x.Fragment, {
                                children: d
                            }, t))
                    }), I = i.length ? `calc(${G.small} * ${i.length} + 16px * ${i.length - 1})` : "0px";
                    return(// Add position relative because in collapse-rows mode, actions are positioned absolutely in the row
                    /* @__PURE__ */ f("tr", {
                        style: {
                            position: "relative"
                        },
                        children: [
                            n.map((d, t)=>{
                                const U = s === n.length - 1 ? t === n.length - 2 || n.length === 1 : t === n.length - 1;
                                return /* @__PURE__ */ f(t === s ? "th" : "td", {
                                    className: c(N[C], {
                                        [L]: t === 0,
                                        [R]: t === n.length - 1 && !y,
                                        [H]: U && r,
                                        [le]: t === s && r
                                    }),
                                    style: {
                                        // add space between top actions and content
                                        marginRight: r && i.length ? `calc(${I} + 8px)` : void 0
                                    },
                                    scope: t === s ? "row" : void 0,
                                    children: [
                                        t !== s && r && m[t] && !B && // this is aria-hidden because screen readers already read the column heading from the th
                                        /* @__PURE__ */ e("div", {
                                            className: re,
                                            "aria-hidden": !0,
                                            children: /* @__PURE__ */ e(F, {
                                                medium: !0,
                                                color: j.colors.textSecondary,
                                                children: m[t]
                                            })
                                        }),
                                        /* @__PURE__ */ e(ce, _object_spread_props(_object_spread({
                                            desktopSize: h.text2.desktopSize,
                                            desktopLineHeight: h.text2.desktopLineHeight
                                        }, t === s && r ? {
                                            mobileSize: h.text4.mobileSize,
                                            mobileLineHeight: h.text4.mobileLineHeight
                                        } : {
                                            mobileSize: h.text2.mobileSize,
                                            mobileLineHeight: h.text2.mobileLineHeight
                                        }), {
                                            as: "div",
                                            wordBreak: !1,
                                            children: /* @__PURE__ */ e("div", {
                                                className: c(_[z(t)], {
                                                    [ae]: t === s && r
                                                }),
                                                children: d
                                            })
                                        }))
                                    ]
                                }, t);
                            }),
                            i.length > 0 ? /* @__PURE__ */ e("td", {
                                className: c(N[C], Y, R, {
                                    [L]: n.length === 0
                                }),
                                align: "right",
                                style: {
                                    width: `calc(${I} + 12px)`
                                },
                                children: w
                            }) : y && /* @__PURE__ */ e("td", {
                                className: Y
                            }),
                            process.env.NODE_ENV !== "test" && r && i.length > 0 && /* @__PURE__ */ e("td", {
                                className: oe,
                                style: {
                                    position: "absolute",
                                    top: `calc(${b ? P.boxed : P.default} - ${Z} + (${G.small} - ${ye.small}) / 2)`,
                                    right: b ? `calc(${P.boxed} - ${Z})` : 0,
                                    width: I
                                },
                                children: w
                            })
                        ]
                    }, o));
                }) : /* @__PURE__ */ e("tr", {
                    children: /* @__PURE__ */ e("td", {
                        colSpan: m.length,
                        className: c({
                            [H]: r
                        }),
                        children: typeof A == "string" ? /* @__PURE__ */ e(de, {
                            paddingY: 56,
                            children: /* @__PURE__ */ e(me, {
                                regular: !0,
                                textAlign: "center",
                                as: "div",
                                children: A
                            })
                        }) : A
                    })
                })
            })
        ]
    }), O = function() {
        let l = arguments.length > 0 && arguments[0] !== void 0 ? arguments[0] : !1;
        return {
            className: c(se, {
                [ne]: l,
                [E]: r,
                [M]: u
            }),
            style: he({
                [ie.maxHeight]: typeof v == "number" ? `${v}px` : v !== null && v !== void 0 ? v : "auto"
            })
        };
    };
    return b ? /* @__PURE__ */ e(pe, {
        desktopOnly: r,
        width: "fit-content",
        maxWidth: "100%",
        minWidth: {
            desktop: u ? "100%" : "auto",
            mobile: r || u ? "100%" : "auto"
        },
        ref: $,
        dataAttributes: _object_spread({
            "component-name": "Table"
        }, a),
        children: /* @__PURE__ */ e("div", _object_spread_props(_object_spread({}, O()), {
            children: D
        }))
    }) : /* @__PURE__ */ e("div", _object_spread_props(_object_spread({
        ref: $
    }, V(a, "Table"), O(K)), {
        children: D
    }));
});
export { Oe as Table };
