// @telefonica/mistica/dist-es/table.css-mistica.js
import "./sprinkles.css.ts.vanilla.css-mistica.js";
import "./responsive-layout.css.ts.vanilla.css-mistica.js";
import "./table.css.ts.vanilla.css-mistica.js";
var r = "_107ok88f", l = "_107ok88g", t = "_107ok886", k = {
    left: "_107ok888 _107ok887",
    right: "_107ok889 _107ok887",
    center: "_107ok88a _107ok887"
}, _ = "_107ok88i", v = "_107ok88k", i = "_107ok883", s = "_107ok882", d = "_107ok885", n = "_107ok884", c = "_107ok88j", p = "_107ok88l", m = "_107ok88n", b = "_107ok88m", I = "_107ok88b", g = "_107ok88c", h = "_107ok881", w = "_107ok88h", H = {
    maxHeight: "var(--_107ok880)"
}, x = {
    top: "_107ok88d",
    middle: "_107ok88e"
};
export { r as actionsHeaderText, l as actionsTableCell, t as boxed, k as cellTextAlign, _ as collapsedRowHeaderItem, v as collapsedRowTitle, i as collapsedRowsInMobile, s as fullWidth, d as hiddenHeadersInDesktop, n as hiddenHeadersInMobile, c as mobileCellHeading, p as rowFirstItem, m as rowLastCollapsedItem, b as rowLastItem, I as scrollContainer, g as scrollOverResponsiveLayout, h as table, w as topActions, H as vars, x as verticalAlign };
