// @telefonica/mistica/dist-es/tabs.js
"use client";
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
import { jsx as t, jsxs as L } from "react/jsx-runtime";
import * as f from "react";
import $ from "classnames";
import { BaseTouchable as R } from "./touchable.js";
import k from "./responsive-layout.js";
import { useTheme as w, useAriaId as x, useElementDimensions as H } from "./hooks.js";
import { Text as z } from "./text.js";
import { isRunningAcceptanceTest as B } from "./utils/platform.js";
import { getPrefixedDataAttributes as P } from "./utils/dom.js";
import { outerBorderInverse as V, outerBorder as W, outer as D, inner as _, tabsContainer as j, tabVariants as q, tabSelectionVariants as i, tabHover as y, icon as C, animatedLine as E } from "./tabs.css-mistica.js";
import M from "./inline.js";
import { useIsInverseVariant as O } from "./theme-variant-context.js";
import { vars as A } from "./skins/skin-contract.css-mistica.js";
const u = B() ? 0 : 300, F = (o)=>{
    switch(o){
        case 1:
        case 2:
        case 3:
            return "fullWidth";
        default:
            return "default";
    }
}, U = (param)=>{
    let { selectedIndex: o, onChange: I, tabs: b, dataAttributes: T } = param;
    const { textPresets: n } = w(), m = x(), { ref: N } = H(), h = f.useRef(null), p = f.useRef(null), [v, g] = f.useState(!1), r = O(), S = (c, d)=>{
        const s = document.querySelector(`[id='${m}'] [data-tabindex="${c}"]`), l = document.querySelector(`[id='${m}'] [data-tabindex="${d}"]`), e = h.current, a = p.current;
        s && l && e && a && (g(!0), e.style.display = "block", e.style.width = `${s.offsetWidth}px`, e.style.transform = `translate(${s.offsetLeft - a.scrollLeft}px, 0)`, Promise.resolve().then(()=>{
            e.style.transition = `transform ${u}ms, width ${u}ms`, e.style.width = `${l.offsetWidth}px`, e.style.transform = `translate(${l.offsetLeft - a.scrollLeft}px, 0)`;
        }), setTimeout(()=>{
            e.style.transition = "", e.style.display = "none", g(!1);
        }, u));
    };
    return /* @__PURE__ */ t("div", _object_spread_props(_object_spread({
        id: m,
        role: "tablist",
        ref: N,
        className: r ? V : W
    }, P(T, "Tabs")), {
        children: /* @__PURE__ */ t(k, {
            fullWidth: !0,
            children: /* @__PURE__ */ L("div", {
                className: D,
                children: [
                    /* @__PURE__ */ t("div", {
                        ref: p,
                        className: _,
                        children: /* @__PURE__ */ t("div", {
                            className: j,
                            children: b.map((param, e)=>{
                                let { text: c, trackingEvent: d, icon: s, "aria-controls": l } = param;
                                const a = e === o;
                                return /* @__PURE__ */ t(R, {
                                    dataAttributes: {
                                        "component-name": "Tab",
                                        tabindex: e
                                    },
                                    className: $(q[F(b.length)], a ? v ? r ? i.selectedAnimatingInverse : i.selectedAnimating : r ? i.selectedInverse : i.selected : r ? i.noSelectedInverse : i.noSelected, r ? y.inverse : y.default),
                                    onPress: ()=>{
                                        !v && o !== e && (I(e), S(o, e));
                                    },
                                    trackingEvent: d,
                                    role: "tab",
                                    "aria-controls": l,
                                    "aria-selected": a ? "true" : "false",
                                    children: /* @__PURE__ */ L(M, {
                                        space: s && c ? 8 : 0,
                                        alignItems: "center",
                                        children: [
                                            s && /* @__PURE__ */ t("div", {
                                                className: C,
                                                children: s
                                            }),
                                            /* @__PURE__ */ t(z, {
                                                as: "div",
                                                desktopSize: n.tabsLabel.size.desktop,
                                                mobileSize: n.tabsLabel.size.mobile,
                                                desktopLineHeight: n.tabsLabel.lineHeight.desktop,
                                                mobileLineHeight: n.tabsLabel.lineHeight.mobile,
                                                weight: n.tabsLabel.weight,
                                                color: "inherit",
                                                wordBreak: !1,
                                                textAlign: "center",
                                                hyphens: "auto",
                                                children: c
                                            })
                                        ]
                                    })
                                }, e);
                            })
                        })
                    }),
                    /* @__PURE__ */ t("div", {
                        ref: h,
                        className: E,
                        style: {
                            background: r ? A.colors.controlActivatedInverse : A.colors.controlActivated
                        }
                    })
                ]
            })
        })
    }));
}, ie = U;
export { ie as default };
