// @telefonica/mistica/dist-es/tag.js
"use client";
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
import { jsxs as c, jsx as r } from "react/jsx-runtime";
import g from "./box.js";
import { Text as x } from "./text.js";
import { useThemeVariant as T, ThemeVariant as h } from "./theme-variant-context.js";
import { pxToRem as v } from "./utils/css.js";
import { getPrefixedDataAttributes as k } from "./utils/dom.js";
import { tag as B } from "./tag.css-mistica.js";
import { sprinkles as a } from "./sprinkles.css-mistica.js";
import { vars as b } from "./skins/skin-contract.css-mistica.js";
import w from "classnames";
import { useTheme as A } from "./hooks.js";
import P from "./badge.js";
const { colors: t } = b, V = (param)=>{
    let { Icon: e, children: i, dataAttributes: m, type: l = "promo", badge: n } = param;
    const { textPresets: d } = A(), p = T(), o = n === !0 ? void 0 : n || 0;
    if (!i) return null;
    const u = {
        // [textColor, backgroundColor]
        promo: [
            t.tagTextPromo,
            t.tagBackgroundPromo
        ],
        active: [
            t.tagTextActive,
            t.tagBackgroundActive
        ],
        inactive: [
            t.tagTextInactive,
            // TODO: remove logic for alternative variant (https://jira.tid.es/browse/WEB-1803)
            p === "alternative" ? t.neutralLowAlternative : t.tagBackgroundInactive
        ],
        success: [
            t.tagTextSuccess,
            t.tagBackgroundSuccess
        ],
        warning: [
            t.tagTextWarning,
            t.tagBackgroundWarning
        ],
        error: [
            t.tagTextError,
            t.tagBackgroundError
        ]
    }, [s, f] = u[l];
    return /* @__PURE__ */ c("span", _object_spread_props(_object_spread({}, k(m, "Tag")), {
        className: w(B, a({
            paddingLeft: e ? 8 : 12,
            paddingRight: o !== 0 ? 8 : 12,
            background: f
        })),
        children: [
            e && /* @__PURE__ */ r(g, {
                paddingRight: 4,
                children: /* @__PURE__ */ r(e, {
                    color: s,
                    size: v(16),
                    className: a({
                        display: "block"
                    })
                })
            }),
            /* @__PURE__ */ c(h, {
                isInverse: !1,
                children: [
                    /* @__PURE__ */ r(x, {
                        color: s,
                        size: 14,
                        lineHeight: 20,
                        weight: d.indicator.weight,
                        truncate: !0,
                        children: i
                    }),
                    o !== 0 && /* @__PURE__ */ r(g, {
                        paddingLeft: 4,
                        className: a({
                            display: "inline-flex"
                        }),
                        children: /* @__PURE__ */ r(P, {
                            value: o
                        })
                    })
                ]
            })
        ]
    }));
}, I = V;
export { I as default };
