// @telefonica/mistica/dist-es/text-field.js
"use client";
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
function _object_without_properties(source, excluded) {
    if (source == null) return {};
    var target = _object_without_properties_loose(source, excluded);
    var key, i;
    if (Object.getOwnPropertySymbols) {
        var sourceSymbolKeys = Object.getOwnPropertySymbols(source);
        for(i = 0; i < sourceSymbolKeys.length; i++){
            key = sourceSymbolKeys[i];
            if (excluded.indexOf(key) >= 0) continue;
            if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
            target[key] = source[key];
        }
    }
    return target;
}
function _object_without_properties_loose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;
    for(i = 0; i < sourceKeys.length; i++){
        key = sourceKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        target[key] = source[key];
    }
    return target;
}
import { jsx as A } from "react/jsx-runtime";
import * as f from "react";
import { useFieldProps as E } from "./form-context.js";
import { TextFieldBaseAutosuggest as j } from "./text-field-base.js";
import { combineRefs as k } from "./utils/common.js";
import { createChangeEvent as w } from "./utils/dom.js";
import { useIsomorphicLayoutEffect as C } from "./hooks.js";
const H = (i, t, r, c)=>{
    C(()=>{
        i && t && r && t.length > r && c && c(w(i, t.slice(0, r)));
    }, [
        c,
        t,
        r,
        i
    ]);
}, I = /*#__PURE__*/ f.forwardRef((_param, v)=>{
    var { disabled: i, error: t, helperText: r, name: c, optional: m, validate: a, onChangeValue: p, onChange: d, value: R, defaultValue: T, onBlur: l, onFocus: u, onPress: x, dataAttributes: g } = _param, s = _object_without_properties(_param, [
        "disabled",
        "error",
        "helperText",
        "name",
        "optional",
        "validate",
        "onChangeValue",
        "onChange",
        "value",
        "defaultValue",
        "onBlur",
        "onFocus",
        "onPress",
        "dataAttributes"
    ]);
    const e = f.useRef(null), F = (o)=>o, b = (o)=>{
        s.multiline && e.current && (e.current.scrollTop = 0), l == null || l(o);
    }, h = (o)=>{
        s.multiline && setTimeout(()=>{
            if (e.current) {
                const y = e.current.value;
                e.current.value = "", e.current.value = y, e.current.scrollTop = e.current.scrollHeight;
            }
        }, 0), u == null || u(o);
    }, n = E({
        name: c,
        value: R,
        defaultValue: T,
        processValue: F,
        helperText: r,
        optional: m,
        error: t,
        disabled: i,
        onBlur: b,
        validate: a,
        onChange: d,
        onChangeValue: p
    });
    return H(e.current, n.value, s.maxLength, n.onChange), /* @__PURE__ */ A(j, _object_spread_props(_object_spread({}, s, n), {
        onClick: x,
        inputRef: k(e, n.inputRef, v),
        onFocus: h,
        type: "text",
        dataAttributes: _object_spread({
            "component-name": "TextField"
        }, g)
    }));
}), D = I;
export { D as default };
