// @telefonica/mistica/dist-es/timer.js
"use client";
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
import { jsxs as M, jsx as c } from "react/jsx-runtime";
import H from "classnames";
import * as N from "react";
import B from "./box.js";
import { useTheme as C, useAriaId as x, useIsomorphicLayoutEffect as z } from "./hooks.js";
import G from "./inline.js";
import F from "./screen-reader-only.js";
import { Text2 as J, Text6 as K } from "./text.js";
import { useThemeVariant as Q, ThemeVariant as X } from "./theme-variant-context.js";
import { inlineText as O, timerWrapper as Z, shortLabelText as P, boxedTimerValueContainer as U, boxedTimerValueContainerInverse as ee, timerDisplayValue as te, boxedTimerDisplayValue as ne, unitContainer as j } from "./timer.css-mistica.js";
import { getPrefixedDataAttributes as w } from "./utils/dom.js";
import { isEqual as re } from "./utils/helpers.js";
import { isRunningAcceptanceTest as W } from "./utils/platform.js";
const V = 24, $ = 60, T = 60, _ = 1e3, Y = T * _, k = $ * Y, se = V * k, R = (n, s, a, u)=>{
    s = u === "none" && s === "days" ? "hours" : s, a = u === "none" && a === "days" ? "hours" : a;
    const i = [
        "seconds",
        "minutes",
        "hours",
        "days"
    ], d = i.indexOf(s), g = Math.max(i.indexOf(a), d), e = i.indexOf(n);
    return d <= e && e <= g;
}, E = (n, s, a, u)=>[
        {
            unit: "days",
            value: n.days
        },
        {
            unit: "hours",
            value: n.hours
        },
        {
            unit: "minutes",
            value: n.minutes
        },
        {
            unit: "seconds",
            value: n.seconds
        }
    ].filter((i)=>R(i.unit, s, a, u)), A = (n)=>W() ? 0 : Math.max(0, (typeof n == "object" ? n : new Date(n)).valueOf() - Date.now()), q = (param)=>{
    let { endTimestamp: n, labelType: s, minTimeUnit: a, maxTimeUnit: u, onProgress: i } = param;
    const [d, g] = N.useState(A(n));
    z(()=>{
        let v;
        const D = ()=>{
            const I = A(n);
            g(I), I || clearInterval(v);
        };
        if (!W()) return D(), v = setInterval(D, _), ()=>clearInterval(v);
    }, [
        n
    ]);
    const e = R("days", a, u, s), S = R("hours", a, u, s), l = R("minutes", a, u, s), m = e ? "days" : S ? "hours" : l ? "minutes" : "seconds", y = Math.floor(d / k) % 24, p = Math.floor(d / Y) % 60, f = Math.floor(d / _) % 60, h = Math.floor(d / se), L = m === "hours" ? y + h * V : y, t = m === "minutes" ? p + $ * (h * V + L) : p, r = m === "seconds" ? f + T * (h * V * $ + L * $ + t) : f, [o, b] = N.useState(E({
        days: h,
        hours: L,
        minutes: t,
        seconds: r
    }, a, u, s));
    return N.useEffect(()=>{
        const v = E({
            days: h,
            hours: L,
            minutes: t,
            seconds: r
        }, a, u, s);
        if (!re(v, o)) {
            b(v);
            const D = {};
            v.forEach((I)=>D[I.unit] = I.value), i == null || i(D);
        }
    }, [
        h,
        L,
        t,
        r,
        s,
        a,
        u,
        o,
        i
    ]), o;
}, fe = (param)=>{
    let { endTimestamp: n, labelType: s = "none", minTimeUnit: a, maxTimeUnit: u, onProgress: i, dataAttributes: d, "aria-label": g } = param;
    const { texts: e } = C(), S = x(), l = q({
        endTimestamp: n,
        labelType: s,
        minTimeUnit: a,
        maxTimeUnit: u,
        onProgress: i
    }), m = {
        days: e.timerDaysShortLabel,
        hours: e.timerHoursShortLabel,
        minutes: e.timerMinutesShortLabel,
        seconds: e.timerSecondsShortLabel
    }, y = {
        days: e.timerDayLongLabel,
        hours: e.timerHourLongLabel,
        minutes: e.timerMinuteLongLabel,
        seconds: e.timerSecondLongLabel
    }, p = {
        days: e.timerDaysLongLabel,
        hours: e.timerHoursLongLabel,
        minutes: e.timerMinutesLongLabel,
        seconds: e.timerSecondsLongLabel
    }, f = (t)=>{
        const r = Math.max(String(t).length, s === "long" ? 1 : 2);
        return /* @__PURE__ */ c("span", {
            className: j,
            style: {
                minWidth: `${r}ch`
            },
            children: String(t).padStart(r, "0")
        });
    }, h = ()=>{
        switch(s){
            case "none":
                return l.map((t, r)=>/* @__PURE__ */ M(N.Fragment, {
                        children: [
                            r > 0 && ":",
                            f(t.value)
                        ]
                    }, r));
            case "short":
                return l.map((t, r)=>/* @__PURE__ */ M(N.Fragment, {
                        children: [
                            r > 0 && " ",
                            /* @__PURE__ */ M("span", {
                                className: P,
                                children: [
                                    f(t.value),
                                    ` ${m[t.unit]}`
                                ]
                            })
                        ]
                    }, r));
            case "long":
            default:
                return l.map((t, r)=>/* @__PURE__ */ M(N.Fragment, {
                        children: [
                            r > 0 && " ",
                            f(t.value),
                            ` ${t.value === 1 ? y[t.unit] : p[t.unit]}`,
                            r === l.length - 2 && ` ${e.timerAnd}`,
                            r < l.length - 2 && ","
                        ]
                    }, r));
        }
    }, L = l.map((t, r)=>`${t.value} ${t.value === 1 ? y[t.unit] : p[t.unit]}${r === l.length - 1 ? "" : r === l.length - 2 ? ` ${e.timerAnd} ` : ", "}`).join("");
    return /* @__PURE__ */ M("span", _object_spread_props(_object_spread({
        role: "timer",
        "aria-labelledby": S,
        className: O,
        style: {
            // try to keep the timer in the same line without wrapping only when label is not long
            display: s !== "long" ? "inline-block" : void 0
        }
    }, w(d, "TextTimer")), {
        children: [
            /* @__PURE__ */ c(F, {
                children: /* @__PURE__ */ c("span", {
                    id: S,
                    children: g ? `${g}. ${L}` : L
                })
            }),
            /* @__PURE__ */ c("span", {
                "aria-hidden": !0,
                className: O,
                children: h()
            })
        ]
    }));
}, Se = (param)=>{
    let { boxed: n, endTimestamp: s, minTimeUnit: a, maxTimeUnit: u, onProgress: i, dataAttributes: d, "aria-label": g } = param;
    const { texts: e } = C(), S = x(), l = Q(), m = q({
        endTimestamp: s,
        minTimeUnit: a,
        maxTimeUnit: u,
        onProgress: i
    }), y = {
        days: e.timerDayLongLabel,
        hours: e.timerHourLongLabel,
        minutes: e.timerDisplayMinutesLabel,
        seconds: e.timerDisplaySecondsLabel
    }, p = {
        days: e.timerDaysLongLabel,
        hours: e.timerHoursLongLabel,
        minutes: e.timerDisplayMinutesLabel,
        seconds: e.timerDisplaySecondsLabel
    }, f = {
        days: e.timerDayLongLabel,
        hours: e.timerHourLongLabel,
        minutes: e.timerMinuteLongLabel,
        seconds: e.timerSecondLongLabel
    }, h = {
        days: e.timerDaysLongLabel,
        hours: e.timerHoursLongLabel,
        minutes: e.timerMinutesLongLabel,
        seconds: e.timerSecondsLongLabel
    }, L = (o)=>{
        const b = Math.max(String(o).length, 2);
        return /* @__PURE__ */ c(K, {
            children: /* @__PURE__ */ c("div", {
                className: j,
                style: {
                    minWidth: `${b}ch`
                },
                children: String(o).padStart(b, "0")
            })
        });
    }, t = m.map((o, b)=>`${o.value} ${o.value === 1 ? f[o.unit] : h[o.unit]}${b === m.length - 1 ? "" : b === m.length - 2 ? ` ${e.timerAnd} ` : ", "}`).join(""), r = ()=>m.map((o, b)=>/* @__PURE__ */ c(B, {
                className: H({
                    [U]: l === "default" && n,
                    [ee]: l !== "default" && n
                }),
                children: /* @__PURE__ */ c(X, {
                    variant: n ? "default" : l,
                    children: /* @__PURE__ */ M("div", {
                        className: H(te, {
                            [ne]: n
                        }),
                        children: [
                            L(o.value),
                            /* @__PURE__ */ c(J, {
                                regular: !0,
                                children: o.value === 1 ? y[o.unit] : p[o.unit]
                            })
                        ]
                    })
                })
            }, b));
    return /* @__PURE__ */ M("div", _object_spread_props(_object_spread({
        role: "timer",
        "aria-labelledby": S,
        className: Z
    }, w(d, "Timer")), {
        children: [
            /* @__PURE__ */ c(F, {
                children: /* @__PURE__ */ c("span", {
                    id: S,
                    children: g ? `${g}. ${t}` : t
                })
            }),
            /* @__PURE__ */ c("div", {
                "aria-hidden": !0,
                children: /* @__PURE__ */ c(G, {
                    space: n ? 8 : 16,
                    wrap: !0,
                    children: r()
                })
            })
        ]
    }));
};
export { fe as TextTimer, Se as Timer };
