// @telefonica/mistica/dist-es/title.js
"use client";
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
import { jsx as e, jsxs as a } from "react/jsx-runtime";
import c, { Text1 as p, Text6 as h, Text2 as d } from "./text.js";
import x from "./inline.js";
import T from "./box.js";
import { vars as f } from "./skins/skin-contract.css-mistica.js";
import { useTheme as l } from "./hooks.js";
import { getPrefixedDataAttributes as g } from "./utils/dom.js";
const m = (param)=>{
    let { title: t, right: i, dataAttributes: o } = param;
    const { textPresets: r } = l();
    return i ? /* @__PURE__ */ a(x, {
        space: "between",
        alignItems: "baseline",
        dataAttributes: o,
        children: [
            t,
            /* @__PURE__ */ e(T, {
                paddingLeft: 16,
                children: /* @__PURE__ */ e(d, {
                    weight: r.link.weight,
                    children: i
                })
            })
        ]
    }) : /* @__PURE__ */ e("div", _object_spread_props(_object_spread({}, g(o)), {
        children: t
    }));
}, P = (param)=>{
    let { children: t, as: i = "h3", id: o, right: r, dataAttributes: s } = param;
    const { textPresets: n } = l();
    return /* @__PURE__ */ e(m, {
        title: /* @__PURE__ */ e(p, {
            color: f.colors.textSecondary,
            transform: "uppercase",
            weight: n.title1.weight,
            as: i,
            id: o,
            wordBreak: !1,
            children: t
        }),
        right: r,
        dataAttributes: _object_spread({
            "component-name": "Title1"
        }, s)
    });
}, A = (param)=>{
    let { children: t, as: i = "h3", id: o, right: r, dataAttributes: s } = param;
    const { textPresets: n } = l();
    return /* @__PURE__ */ e(m, {
        title: /* @__PURE__ */ e(c, {
            as: i,
            id: o,
            mobileSize: n.title2.size.mobile,
            desktopSize: n.title2.size.desktop,
            weight: n.title2.weight,
            mobileLineHeight: n.title2.lineHeight.mobile,
            desktopLineHeight: n.title2.lineHeight.desktop,
            children: t
        }),
        right: r,
        dataAttributes: _object_spread({
            "component-name": "Title2"
        }, s)
    });
}, S = (param)=>{
    let { children: t, as: i = "h3", id: o, right: r, dataAttributes: s } = param;
    return /* @__PURE__ */ e(m, {
        title: /* @__PURE__ */ e(h, {
            as: i,
            id: o,
            children: t
        }),
        right: r,
        dataAttributes: _object_spread({
            "component-name": "Title3"
        }, s)
    });
};
export { P as Title1, A as Title2, S as Title3 };
