// @telefonica/mistica/dist/community/advanced-data-card.css-mistica.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    MIN_HEIGHT: function() {
        return a;
    },
    actions: function() {
        return _;
    },
    actionsVariants: function() {
        return n;
    },
    adjustButtonLink: function() {
        return t;
    },
    button: function() {
        return u;
    },
    buttonMobile: function() {
        return o;
    },
    cardContentStyle: function() {
        return i;
    },
    dataCard: function() {
        return x;
    },
    divider: function() {
        return y;
    },
    footerDirection: function() {
        return e;
    },
    footerText: function() {
        return h;
    },
    marginRightAuto: function() {
        return d;
    },
    marginRightButton: function() {
        return b;
    },
    minHeight: function() {
        return c;
    },
    paddingX: function() {
        return l;
    },
    topActionsWithoutIcon: function() {
        return m;
    },
    touchable: function() {
        return p;
    },
    touchableCardHoverOverlay: function() {
        return s;
    },
    vars: function() {
        return g;
    },
    zindex: function() {
        return j;
    }
});
require("../sprinkles.css.ts.vanilla.css-mistica.js");
require("../icon-button.css.ts.vanilla.css-mistica.js");
require("./advanced-data-card.css.ts.vanilla.css-mistica.js");
var a = 216, _ = "_1x00fru3 _1x00fru1 _1y2v1nfaa _1y2v1nfbj _1y2v1nfhk _1y2v1nfhw _1y2v1nf97", n = {
    default: "_1x00frui",
    mobile: "_1x00fruj"
}, t = "_1x00frup", u = "_1x00frul", o = "_1x00frum _1x00frul", i = "_1x00fru7 _1x00fru1 _1y2v1nfaa _1y2v1nfbj _1y2v1nf7m _1y2v1nf97", x = "_1x00frub _1y2v1nfhe _1y2v1nfhk _1y2v1nfi6 _1y2v1nfhq", y = "_1x00frug", e = "_1x00fruo _1y2v1nfhk _1y2v1nfhp", h = "_1y2v1nfhk _1y2v1nfi6 _1y2v1nfhz _1y2v1nfjt", d = "_1x00fruk", b = "_1x00fruq", c = "_1x00fru9", l = "_1x00fru1 _1y2v1nfaa _1y2v1nfbj", m = "_1x00frus", p = "_1x00fru4", s = "_1x00fru5", g = {
    topActionsCount: "var(--_1x00frur)"
}, j = "_1x00fru8";
