// @telefonica/mistica/dist/community/advanced-data-card.js
"use client";
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    AdvancedDataCard: function() {
        return Tr;
    },
    default: function() {
        return kr;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _react = /*#__PURE__*/ _interop_require_wildcard(require("react"));
const _boxed = require("../boxed.js");
const _sprinklescssmistica = require("../sprinkles.css-mistica.js");
const _stack = /*#__PURE__*/ _interop_require_default(require("../stack.js"));
const _advanceddatacardcssmistica = require("./advanced-data-card.css-mistica.js");
const _imagecssmistica = require("../image.css-mistica.js");
const _divider = /*#__PURE__*/ _interop_require_default(require("../divider.js"));
const _text = require("../text.js");
const _skincontractcssmistica = require("../skins/skin-contract.css-mistica.js");
const _box = /*#__PURE__*/ _interop_require_default(require("../box.js"));
const _touchable = /*#__PURE__*/ _interop_require_default(require("../touchable.js"));
const _classnames = /*#__PURE__*/ _interop_require_default(require("classnames"));
const _card = require("../card.js");
const _hooks = require("../hooks.js");
const _dom = require("../utils/dom.js");
const _inline = /*#__PURE__*/ _interop_require_default(require("../inline.js"));
const _css = require("../utils/css.js");
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
function _getRequireWildcardCache(nodeInterop) {
    if (typeof WeakMap !== "function") return null;
    var cacheBabelInterop = new WeakMap();
    var cacheNodeInterop = new WeakMap();
    return (_getRequireWildcardCache = function(nodeInterop) {
        return nodeInterop ? cacheNodeInterop : cacheBabelInterop;
    })(nodeInterop);
}
function _interop_require_wildcard(obj, nodeInterop) {
    if (!nodeInterop && obj && obj.__esModule) {
        return obj;
    }
    if (obj === null || typeof obj !== "object" && typeof obj !== "function") {
        return {
            default: obj
        };
    }
    var cache = _getRequireWildcardCache(nodeInterop);
    if (cache && cache.has(obj)) {
        return cache.get(obj);
    }
    var newObj = {
        __proto__: null
    };
    var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor;
    for(var key in obj){
        if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) {
            var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null;
            if (desc && (desc.get || desc.set)) {
                Object.defineProperty(newObj, key, desc);
            } else {
                newObj[key] = obj[key];
            }
        }
    }
    newObj.default = obj;
    if (cache) {
        cache.set(obj, newObj);
    }
    return newObj;
}
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
function _object_without_properties(source, excluded) {
    if (source == null) return {};
    var target = _object_without_properties_loose(source, excluded);
    var key, i;
    if (Object.getOwnPropertySymbols) {
        var sourceSymbolKeys = Object.getOwnPropertySymbols(source);
        for(i = 0; i < sourceSymbolKeys.length; i++){
            key = sourceSymbolKeys[i];
            if (excluded.indexOf(key) >= 0) continue;
            if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
            target[key] = source[key];
        }
    }
    return target;
}
function _object_without_properties_loose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;
    for(i = 0; i < sourceKeys.length; i++){
        key = sourceKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        target[key] = source[key];
    }
    return target;
}
const xr = (param)=>{
    let { headline: a, pretitle: i, pretitleAs: s = "p", pretitleLinesMax: m, title: n, titleAs: d = "h3", titleLinesMax: c, subtitle: l, subtitleLinesMax: o, description: t, descriptionLinesMax: h } = param;
    const { textPresets: x } = (0, _hooks.useTheme)();
    return /* @__PURE__ */ (0, _jsxruntime.jsxs)(_stack.default, {
        space: 4,
        children: [
            a,
            i && /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text2, {
                color: _skincontractcssmistica.vars.colors.textPrimary,
                truncate: m,
                as: s,
                regular: !0,
                hyphens: "auto",
                children: i
            }),
            /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text, _object_spread_props(_object_spread({}, _text.textProps.text4), {
                truncate: c,
                weight: x.cardTitle.weight,
                as: d,
                hyphens: "auto",
                children: n
            })),
            /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text2, {
                color: _skincontractcssmistica.vars.colors.textPrimary,
                truncate: o,
                as: "p",
                regular: !0,
                hyphens: "auto",
                children: l
            }),
            t && /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text2, {
                truncate: h,
                as: "p",
                regular: !0,
                color: _skincontractcssmistica.vars.colors.textSecondary,
                hyphens: "auto",
                children: t
            })
        ]
    });
}, Cr = (param)=>{
    let { button: a, footerImage: i, footerText: s, footerTextLinesMax: m, buttonLink: n } = param;
    const d = !!a, c = !!i, l = !!s, o = !!n, t = d && (c || l) && o, h = o && !t ? "178px" : "";
    return /* @__PURE__ */ (0, _jsxruntime.jsxs)(_jsxruntime.Fragment, {
        children: [
            /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                className: _advanceddatacardcssmistica.divider,
                children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_divider.default, {})
            }),
            /* @__PURE__ */ (0, _jsxruntime.jsxs)("div", {
                className: (0, _classnames.default)(_advanceddatacardcssmistica.actions, _advanceddatacardcssmistica.actionsVariants[t ? "mobile" : "default"]),
                children: [
                    d && /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                        className: (0, _classnames.default)(_advanceddatacardcssmistica.marginRightButton, _advanceddatacardcssmistica.buttonMobile),
                        children: a
                    }),
                    /* @__PURE__ */ (0, _jsxruntime.jsxs)("div", {
                        className: (0, _classnames.default)(_advanceddatacardcssmistica.footerDirection, t ? _advanceddatacardcssmistica.marginRightAuto : (0, _sprinklescssmistica.sprinkles)({
                            paddingTop: 16
                        })),
                        children: [
                            c && /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                                paddingRight: 16,
                                className: (0, _sprinklescssmistica.sprinkles)({
                                    alignItems: "center",
                                    display: "flex"
                                }),
                                children: /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                                    style: (0, _css.applyCssVars)({
                                        [_imagecssmistica.vars.mediaBorderRadius]: _skincontractcssmistica.vars.borderRadii.mediaSmall
                                    }),
                                    children: i
                                })
                            }),
                            l && /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                                style: {
                                    maxWidth: h
                                },
                                className: _advanceddatacardcssmistica.footerText,
                                children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text2, {
                                    truncate: m,
                                    regular: !0,
                                    children: s
                                })
                            })
                        ]
                    }),
                    o && /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                        className: (0, _classnames.default)(_advanceddatacardcssmistica.adjustButtonLink, t ? _advanceddatacardcssmistica.button : _advanceddatacardcssmistica.buttonMobile),
                        children: n
                    })
                ]
            })
        ]
    });
}, Tr = /*#__PURE__*/ _react.forwardRef((_param, k)=>{
    var { stackingGroup: a, headline: i, pretitle: s, pretitleAs: m, pretitleLinesMax: n, title: d, titleAs: c, titleLinesMax: l, subtitle: o, subtitleLinesMax: t, description: h, descriptionLinesMax: x, extra: p, extraDividerPadding: G = 24, noExtraDivider: M = !1, button: A, footerImage: B, footerText: b, footerTextLinesMax: V, buttonLink: R, dataAttributes: W, actions: v, "aria-label": w, onClose: D } = _param, f = _object_without_properties(_param, [
        "stackingGroup",
        "headline",
        "pretitle",
        "pretitleAs",
        "pretitleLinesMax",
        "title",
        "titleAs",
        "titleLinesMax",
        "subtitle",
        "subtitleLinesMax",
        "description",
        "descriptionLinesMax",
        "extra",
        "extraDividerPadding",
        "noExtraDivider",
        "button",
        "footerImage",
        "footerText",
        "footerTextLinesMax",
        "buttonLink",
        "dataAttributes",
        "actions",
        "aria-label",
        "onClose"
    ]);
    const C = !!f.href || !!f.onPress || !!f.to, z = {
        button: A,
        footerImage: B,
        footerText: b,
        footerTextLinesMax: V,
        buttonLink: R
    }, F = !!A || !!B || !!b || !!R, H = !!(p != null && p.length), O = ((v == null ? void 0 : v.length) || 0) + (D ? 1 : 0);
    return /* @__PURE__ */ (0, _jsxruntime.jsxs)("section", _object_spread_props(_object_spread({
        className: (0, _sprinklescssmistica.sprinkles)({
            position: "relative",
            height: "100%"
        })
    }, (0, _dom.getPrefixedDataAttributes)(W, "AdvancedDataCard")), {
        ref: k,
        "aria-label": C ? void 0 : w,
        children: [
            /* @__PURE__ */ (0, _jsxruntime.jsxs)(_boxed.Boxed, {
                className: _advanceddatacardcssmistica.dataCard,
                width: "100%",
                height: "100%",
                minHeight: _advanceddatacardcssmistica.MIN_HEIGHT,
                children: [
                    /* @__PURE__ */ (0, _jsxruntime.jsxs)(_touchable.default, _object_spread_props(_object_spread({
                        maybe: !0
                    }, f), {
                        "aria-label": C ? w : void 0,
                        className: _advanceddatacardcssmistica.touchable,
                        children: [
                            C && /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                                className: _advanceddatacardcssmistica.touchableCardHoverOverlay
                            }),
                            /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                                className: (0, _classnames.default)(_advanceddatacardcssmistica.cardContentStyle, !F && !H ? _advanceddatacardcssmistica.minHeight : ""),
                                children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                                    paddingTop: 8,
                                    children: /* @__PURE__ */ (0, _jsxruntime.jsxs)(_inline.default, {
                                        space: 0,
                                        children: [
                                            /* @__PURE__ */ (0, _jsxruntime.jsxs)(_stack.default, {
                                                space: 8,
                                                children: [
                                                    a,
                                                    /* @__PURE__ */ (0, _jsxruntime.jsx)(xr, {
                                                        headline: i,
                                                        pretitle: s,
                                                        pretitleAs: m,
                                                        pretitleLinesMax: n,
                                                        title: d,
                                                        titleAs: c,
                                                        titleLinesMax: l,
                                                        subtitle: o,
                                                        subtitleLinesMax: t,
                                                        description: h,
                                                        descriptionLinesMax: x
                                                    })
                                                ]
                                            }),
                                            !a && /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                                                style: (0, _css.applyCssVars)({
                                                    [_advanceddatacardcssmistica.vars.topActionsCount]: String(O)
                                                }),
                                                className: _advanceddatacardcssmistica.topActionsWithoutIcon
                                            })
                                        ]
                                    })
                                })
                            }),
                            /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                                style: {
                                    flexGrow: 1
                                }
                            }),
                            H && /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                                paddingTop: 16,
                                paddingBottom: 24,
                                width: "100%",
                                className: _advanceddatacardcssmistica.zindex,
                                children: p.map((X, I)=>/* @__PURE__ */ (0, _jsxruntime.jsxs)("div", {
                                        children: [
                                            /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                                                className: _advanceddatacardcssmistica.paddingX,
                                                children: X
                                            }),
                                            I + 1 !== p.length && /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                                                paddingY: G,
                                                children: !M && /* @__PURE__ */ (0, _jsxruntime.jsx)(_divider.default, {})
                                            })
                                        ]
                                    }, I))
                            })
                        ]
                    })),
                    F && /* @__PURE__ */ (0, _jsxruntime.jsx)(Cr, _object_spread({}, z))
                ]
            }),
            /* @__PURE__ */ (0, _jsxruntime.jsx)(_card.CardActionsGroup, {
                actions: v,
                onClose: D
            })
        ]
    }));
}), kr = Tr;
