// @telefonica/mistica/dist/community/blocks.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    HighlightedValueBlock: function() {
        return K;
    },
    InformationBlock: function() {
        return J;
    },
    ProgressBlock: function() {
        return O;
    },
    RowBlock: function() {
        return E;
    },
    SimpleBlock: function() {
        return F;
    },
    ValueBlock: function() {
        return M;
    }
});
const _jsxruntime = require("react/jsx-runtime");
const _stack = /*#__PURE__*/ _interop_require_default(require("../stack.js"));
const _blockscssmistica = require("./blocks.css-mistica.js");
const _imagecssmistica = require("../image.css-mistica.js");
const _text = require("../text.js");
const _skincontractcssmistica = require("../skins/skin-contract.css-mistica.js");
const _inline = /*#__PURE__*/ _interop_require_default(require("../inline.js"));
const _box = /*#__PURE__*/ _interop_require_default(require("../box.js"));
const _progressbar = require("../progress-bar.js");
const _classnames = /*#__PURE__*/ _interop_require_default(require("classnames"));
const _css = require("../utils/css.js");
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
const g = (param)=>{
    let { title: a, description: l } = param;
    const o = typeof l == "string" ? [
        l
    ] : l;
    return /* @__PURE__ */ (0, _jsxruntime.jsxs)("div", {
        className: _blockscssmistica.column,
        children: [
            /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text3, {
                regular: !0,
                color: _skincontractcssmistica.vars.colors.textPrimary,
                children: a
            }),
            o && o.map((t, i)=>/* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text2, {
                    regular: !0,
                    color: _skincontractcssmistica.vars.colors.textSecondary,
                    as: "p",
                    children: t
                }, i))
        ]
    });
}, E = (param)=>{
    let { title: a, stackingGroup: l, description: o, "aria-label": t } = param;
    return /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
        "aria-label": t,
        children: /* @__PURE__ */ (0, _jsxruntime.jsxs)(_inline.default, {
            space: "between",
            alignItems: "center",
            children: [
                a && /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                    paddingRight: 32,
                    children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text2, {
                        regular: !0,
                        children: a
                    })
                }),
                l || /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text2, {
                    regular: !0,
                    color: _skincontractcssmistica.vars.colors.textSecondary,
                    textAlign: "right",
                    as: "div",
                    children: o
                })
            ]
        })
    });
}, F = (param)=>{
    let { image: a, description: l, "aria-label": o } = param;
    return /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
        "aria-label": o,
        children: /* @__PURE__ */ (0, _jsxruntime.jsxs)(_inline.default, {
            space: 16,
            alignItems: "center",
            children: [
                /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
                    style: (0, _css.applyCssVars)({
                        [_imagecssmistica.vars.mediaBorderRadius]: _skincontractcssmistica.vars.borderRadii.mediaSmall
                    }),
                    children: a
                }),
                /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text2, {
                    regular: !0,
                    color: _skincontractcssmistica.vars.colors.textSecondary,
                    children: l
                })
            ]
        })
    });
}, J = (param)=>{
    let { title: a, description: l, secondaryValue: o, value: t, valueColor: i = _skincontractcssmistica.vars.colors.textPrimary, "aria-label": s } = param;
    return /* @__PURE__ */ (0, _jsxruntime.jsxs)(_inline.default, {
        space: "between",
        alignItems: "flex-end",
        "aria-label": s,
        children: [
            /* @__PURE__ */ (0, _jsxruntime.jsx)(g, {
                title: a,
                description: l
            }),
            /* @__PURE__ */ (0, _jsxruntime.jsxs)("div", {
                className: (0, _classnames.default)(_blockscssmistica.column, _blockscssmistica.rightContent),
                children: [
                    /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text2, {
                        regular: !0,
                        color: _skincontractcssmistica.vars.colors.textSecondary,
                        decoration: "line-through",
                        children: o
                    }),
                    /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text5, {
                        color: i,
                        children: t
                    })
                ]
            })
        ]
    });
}, K = (param)=>{
    let { headline: a, headings: l, mainHeading: o, secondHeading: t, secondaryValue: i, title: s, description: h, valueColor: b = _skincontractcssmistica.vars.colors.textPrimary, "aria-label": v } = param;
    return /* @__PURE__ */ (0, _jsxruntime.jsxs)("div", {
        "aria-label": v,
        children: [
            a && /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                paddingBottom: 24,
                children: a
            }),
            /* @__PURE__ */ (0, _jsxruntime.jsxs)(_stack.default, {
                space: 2,
                children: [
                    l && l.map((x, f)=>/* @__PURE__ */ {
                        var _x_valueColor;
                        return (0, _jsxruntime.jsxs)(_inline.default, {
                            space: 8,
                            alignItems: "baseline",
                            children: [
                                /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text8, {
                                    color: (_x_valueColor = x.valueColor) !== null && _x_valueColor !== void 0 ? _x_valueColor : _skincontractcssmistica.vars.colors.textPrimary,
                                    children: x.value
                                }),
                                /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text2, {
                                    regular: !0,
                                    color: _skincontractcssmistica.vars.colors.textSecondary,
                                    children: x.text
                                })
                            ]
                        }, f);
                    }),
                    o && /* @__PURE__ */ (0, _jsxruntime.jsxs)(_inline.default, {
                        space: 8,
                        alignItems: "baseline",
                        children: [
                            /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text8, {
                                color: b,
                                children: o.value
                            }),
                            /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text2, {
                                regular: !0,
                                color: _skincontractcssmistica.vars.colors.textSecondary,
                                children: o.text
                            })
                        ]
                    }),
                    t && /* @__PURE__ */ (0, _jsxruntime.jsxs)(_inline.default, {
                        space: 8,
                        alignItems: "baseline",
                        children: [
                            /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text8, {
                                color: b,
                                children: t.value
                            }),
                            /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text2, {
                                regular: !0,
                                color: _skincontractcssmistica.vars.colors.textSecondary,
                                children: t.text
                            })
                        ]
                    }),
                    i && /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text8, {
                        color: _skincontractcssmistica.vars.colors.textSecondary,
                        children: i
                    })
                ]
            }),
            s || h ? /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                paddingTop: 8,
                children: /* @__PURE__ */ (0, _jsxruntime.jsx)(g, {
                    title: s,
                    description: h
                })
            }) : null
        ]
    });
}, M = (param)=>{
    let { title: a, value: l, description: o, valueColor: t = _skincontractcssmistica.vars.colors.textPrimary, "aria-label": i } = param;
    return /* @__PURE__ */ (0, _jsxruntime.jsxs)("div", {
        "aria-label": i,
        className: _blockscssmistica.column,
        children: [
            /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text2, {
                regular: !0,
                color: _skincontractcssmistica.vars.colors.textPrimary,
                children: a
            }),
            /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text8, {
                color: t,
                children: l
            }),
            /* @__PURE__ */ (0, _jsxruntime.jsx)(g, {
                description: o
            })
        ]
    });
}, O = (param)=>{
    let { title: a, stackingGroup: l, progressPercent: o, reverse: t, heading: i, description: s, "aria-label": h } = param;
    return /* @__PURE__ */ (0, _jsxruntime.jsx)("div", {
        "aria-label": h,
        children: /* @__PURE__ */ (0, _jsxruntime.jsxs)(_stack.default, {
            space: 8,
            children: [
                /* @__PURE__ */ (0, _jsxruntime.jsxs)(_inline.default, {
                    space: "between",
                    alignItems: "flex-end",
                    children: [
                        /* @__PURE__ */ (0, _jsxruntime.jsx)(_box.default, {
                            paddingRight: 32,
                            children: /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text2, {
                                regular: !0,
                                children: a
                            })
                        }),
                        l
                    ]
                }),
                o !== void 0 && /* @__PURE__ */ (0, _jsxruntime.jsx)(_progressbar.ProgressBar, {
                    "aria-hidden": !0,
                    progressPercent: o,
                    reverse: t
                }),
                /* @__PURE__ */ (0, _jsxruntime.jsxs)(_inline.default, {
                    space: 8,
                    alignItems: "baseline",
                    children: [
                        /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text8, {
                            color: i.valueColor || _skincontractcssmistica.vars.colors.textPrimary,
                            children: i.value
                        }),
                        /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text2, {
                            regular: !0,
                            color: _skincontractcssmistica.vars.colors.textSecondary,
                            children: i.text
                        })
                    ]
                }),
                s && /* @__PURE__ */ (0, _jsxruntime.jsx)(_text.Text2, {
                    regular: !0,
                    color: _skincontractcssmistica.vars.colors.textSecondary,
                    children: s
                })
            ]
        })
    });
};
