// @telefonica/mistica/dist/community/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    AdvancedDataCard: function() {
        return _advanceddatacard.default;
    },
    ExampleComponent: function() {
        return _examplecomponent.default;
    },
    HighlightedValueBlock: function() {
        return _blocks.HighlightedValueBlock;
    },
    InformationBlock: function() {
        return _blocks.InformationBlock;
    },
    ProgressBlock: function() {
        return _blocks.ProgressBlock;
    },
    RowBlock: function() {
        return _blocks.RowBlock;
    },
    SimpleBlock: function() {
        return _blocks.SimpleBlock;
    },
    ValueBlock: function() {
        return _blocks.ValueBlock;
    }
});
const _examplecomponent = /*#__PURE__*/ _interop_require_default(require("./example-component.js"));
const _advanceddatacard = /*#__PURE__*/ _interop_require_default(require("./advanced-data-card.js"));
const _blocks = require("./blocks.js");
function _interop_require_default(obj) {
    return obj && obj.__esModule ? obj : {
        default: obj
    };
}
