// @telefonica/mistica/dist-es/community/advanced-data-card.js
"use client";
function _define_property(obj, key, value) {
    if (key in obj) {
        Object.defineProperty(obj, key, {
            value: value,
            enumerable: true,
            configurable: true,
            writable: true
        });
    } else {
        obj[key] = value;
    }
    return obj;
}
function _object_spread(target) {
    for(var i = 1; i < arguments.length; i++){
        var source = arguments[i] != null ? arguments[i] : {};
        var ownKeys = Object.keys(source);
        if (typeof Object.getOwnPropertySymbols === "function") {
            ownKeys = ownKeys.concat(Object.getOwnPropertySymbols(source).filter(function(sym) {
                return Object.getOwnPropertyDescriptor(source, sym).enumerable;
            }));
        }
        ownKeys.forEach(function(key) {
            _define_property(target, key, source[key]);
        });
    }
    return target;
}
function ownKeys(object, enumerableOnly) {
    var keys = Object.keys(object);
    if (Object.getOwnPropertySymbols) {
        var symbols = Object.getOwnPropertySymbols(object);
        if (enumerableOnly) {
            symbols = symbols.filter(function(sym) {
                return Object.getOwnPropertyDescriptor(object, sym).enumerable;
            });
        }
        keys.push.apply(keys, symbols);
    }
    return keys;
}
function _object_spread_props(target, source) {
    source = source != null ? source : {};
    if (Object.getOwnPropertyDescriptors) {
        Object.defineProperties(target, Object.getOwnPropertyDescriptors(source));
    } else {
        ownKeys(Object(source)).forEach(function(key) {
            Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key));
        });
    }
    return target;
}
function _object_without_properties(source, excluded) {
    if (source == null) return {};
    var target = _object_without_properties_loose(source, excluded);
    var key, i;
    if (Object.getOwnPropertySymbols) {
        var sourceSymbolKeys = Object.getOwnPropertySymbols(source);
        for(i = 0; i < sourceSymbolKeys.length; i++){
            key = sourceSymbolKeys[i];
            if (excluded.indexOf(key) >= 0) continue;
            if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
            target[key] = source[key];
        }
    }
    return target;
}
function _object_without_properties_loose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;
    for(i = 0; i < sourceKeys.length; i++){
        key = sourceKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        target[key] = source[key];
    }
    return target;
}
import { jsxs as e, jsx as r, Fragment as Y } from "react/jsx-runtime";
import * as _ from "react";
import { Boxed as $ } from "../boxed.js";
import { sprinkles as T } from "../sprinkles.css-mistica.js";
import P from "../stack.js";
import { dataCard as q, MIN_HEIGHT as J, touchable as K, touchableCardHoverOverlay as Q, cardContentStyle as U, minHeight as Z, topActionsWithoutIcon as L, zindex as rr, paddingX as er, divider as tr, actions as ar, actionsVariants as ir, marginRightButton as or, buttonMobile as S, footerDirection as sr, marginRightAuto as nr, footerText as dr, adjustButtonLink as cr, button as lr, vars as mr } from "./advanced-data-card.css-mistica.js";
import { vars as hr } from "../image.css-mistica.js";
import j from "../divider.js";
import { Text2 as g, Text as pr, textProps as ur } from "../text.js";
import { vars as y } from "../skins/skin-contract.css-mistica.js";
import N from "../box.js";
import vr from "../touchable.js";
import u from "classnames";
import { CardActionsGroup as fr } from "../card.js";
import { useTheme as gr } from "../hooks.js";
import { getPrefixedDataAttributes as yr } from "../utils/dom.js";
import Nr from "../inline.js";
import { applyCssVars as E } from "../utils/css.js";
const xr = (param)=>{
    let { headline: a, pretitle: i, pretitleAs: s = "p", pretitleLinesMax: m, title: n, titleAs: d = "h3", titleLinesMax: c, subtitle: l, subtitleLinesMax: o, description: t, descriptionLinesMax: h } = param;
    const { textPresets: x } = gr();
    return /* @__PURE__ */ e(P, {
        space: 4,
        children: [
            a,
            i && /* @__PURE__ */ r(g, {
                color: y.colors.textPrimary,
                truncate: m,
                as: s,
                regular: !0,
                hyphens: "auto",
                children: i
            }),
            /* @__PURE__ */ r(pr, _object_spread_props(_object_spread({}, ur.text4), {
                truncate: c,
                weight: x.cardTitle.weight,
                as: d,
                hyphens: "auto",
                children: n
            })),
            /* @__PURE__ */ r(g, {
                color: y.colors.textPrimary,
                truncate: o,
                as: "p",
                regular: !0,
                hyphens: "auto",
                children: l
            }),
            t && /* @__PURE__ */ r(g, {
                truncate: h,
                as: "p",
                regular: !0,
                color: y.colors.textSecondary,
                hyphens: "auto",
                children: t
            })
        ]
    });
}, Cr = (param)=>{
    let { button: a, footerImage: i, footerText: s, footerTextLinesMax: m, buttonLink: n } = param;
    const d = !!a, c = !!i, l = !!s, o = !!n, t = d && (c || l) && o, h = o && !t ? "178px" : "";
    return /* @__PURE__ */ e(Y, {
        children: [
            /* @__PURE__ */ r("div", {
                className: tr,
                children: /* @__PURE__ */ r(j, {})
            }),
            /* @__PURE__ */ e("div", {
                className: u(ar, ir[t ? "mobile" : "default"]),
                children: [
                    d && /* @__PURE__ */ r("div", {
                        className: u(or, S),
                        children: a
                    }),
                    /* @__PURE__ */ e("div", {
                        className: u(sr, t ? nr : T({
                            paddingTop: 16
                        })),
                        children: [
                            c && /* @__PURE__ */ r(N, {
                                paddingRight: 16,
                                className: T({
                                    alignItems: "center",
                                    display: "flex"
                                }),
                                children: /* @__PURE__ */ r("div", {
                                    style: E({
                                        [hr.mediaBorderRadius]: y.borderRadii.mediaSmall
                                    }),
                                    children: i
                                })
                            }),
                            l && /* @__PURE__ */ r("div", {
                                style: {
                                    maxWidth: h
                                },
                                className: dr,
                                children: /* @__PURE__ */ r(g, {
                                    truncate: m,
                                    regular: !0,
                                    children: s
                                })
                            })
                        ]
                    }),
                    o && /* @__PURE__ */ r("div", {
                        className: u(cr, t ? lr : S),
                        children: n
                    })
                ]
            })
        ]
    });
}, Tr = /*#__PURE__*/ _.forwardRef((_param, k)=>{
    var { stackingGroup: a, headline: i, pretitle: s, pretitleAs: m, pretitleLinesMax: n, title: d, titleAs: c, titleLinesMax: l, subtitle: o, subtitleLinesMax: t, description: h, descriptionLinesMax: x, extra: p, extraDividerPadding: G = 24, noExtraDivider: M = !1, button: A, footerImage: B, footerText: b, footerTextLinesMax: V, buttonLink: R, dataAttributes: W, actions: v, "aria-label": w, onClose: D } = _param, f = _object_without_properties(_param, [
        "stackingGroup",
        "headline",
        "pretitle",
        "pretitleAs",
        "pretitleLinesMax",
        "title",
        "titleAs",
        "titleLinesMax",
        "subtitle",
        "subtitleLinesMax",
        "description",
        "descriptionLinesMax",
        "extra",
        "extraDividerPadding",
        "noExtraDivider",
        "button",
        "footerImage",
        "footerText",
        "footerTextLinesMax",
        "buttonLink",
        "dataAttributes",
        "actions",
        "aria-label",
        "onClose"
    ]);
    const C = !!f.href || !!f.onPress || !!f.to, z = {
        button: A,
        footerImage: B,
        footerText: b,
        footerTextLinesMax: V,
        buttonLink: R
    }, F = !!A || !!B || !!b || !!R, H = !!(p != null && p.length), O = ((v == null ? void 0 : v.length) || 0) + (D ? 1 : 0);
    return /* @__PURE__ */ e("section", _object_spread_props(_object_spread({
        className: T({
            position: "relative",
            height: "100%"
        })
    }, yr(W, "AdvancedDataCard")), {
        ref: k,
        "aria-label": C ? void 0 : w,
        children: [
            /* @__PURE__ */ e($, {
                className: q,
                width: "100%",
                height: "100%",
                minHeight: J,
                children: [
                    /* @__PURE__ */ e(vr, _object_spread_props(_object_spread({
                        maybe: !0
                    }, f), {
                        "aria-label": C ? w : void 0,
                        className: K,
                        children: [
                            C && /* @__PURE__ */ r("div", {
                                className: Q
                            }),
                            /* @__PURE__ */ r("div", {
                                className: u(U, !F && !H ? Z : ""),
                                children: /* @__PURE__ */ r(N, {
                                    paddingTop: 8,
                                    children: /* @__PURE__ */ e(Nr, {
                                        space: 0,
                                        children: [
                                            /* @__PURE__ */ e(P, {
                                                space: 8,
                                                children: [
                                                    a,
                                                    /* @__PURE__ */ r(xr, {
                                                        headline: i,
                                                        pretitle: s,
                                                        pretitleAs: m,
                                                        pretitleLinesMax: n,
                                                        title: d,
                                                        titleAs: c,
                                                        titleLinesMax: l,
                                                        subtitle: o,
                                                        subtitleLinesMax: t,
                                                        description: h,
                                                        descriptionLinesMax: x
                                                    })
                                                ]
                                            }),
                                            !a && /* @__PURE__ */ r("div", {
                                                style: E({
                                                    [mr.topActionsCount]: String(O)
                                                }),
                                                className: L
                                            })
                                        ]
                                    })
                                })
                            }),
                            /* @__PURE__ */ r("div", {
                                style: {
                                    flexGrow: 1
                                }
                            }),
                            H && /* @__PURE__ */ r(N, {
                                paddingTop: 16,
                                paddingBottom: 24,
                                width: "100%",
                                className: rr,
                                children: p.map((X, I)=>/* @__PURE__ */ e("div", {
                                        children: [
                                            /* @__PURE__ */ r("div", {
                                                className: er,
                                                children: X
                                            }),
                                            I + 1 !== p.length && /* @__PURE__ */ r(N, {
                                                paddingY: G,
                                                children: !M && /* @__PURE__ */ r(j, {})
                                            })
                                        ]
                                    }, I))
                            })
                        ]
                    })),
                    F && /* @__PURE__ */ r(Cr, _object_spread({}, z))
                ]
            }),
            /* @__PURE__ */ r(fr, {
                actions: v,
                onClose: D
            })
        ]
    }));
}), kr = Tr;
export { Tr as AdvancedDataCard, kr as default };
