// @telefonica/mistica/dist-es/community/blocks.js
import { jsx as r, jsxs as c } from "react/jsx-runtime";
import y from "../stack.js";
import { column as p, rightContent as S } from "./blocks.css-mistica.js";
import { vars as B } from "../image.css-mistica.js";
import { Text2 as n, Text5 as I, Text8 as m, Text3 as P } from "../text.js";
import { vars as e } from "../skins/skin-contract.css-mistica.js";
import d from "../inline.js";
import u from "../box.js";
import { ProgressBar as k } from "../progress-bar.js";
import R from "classnames";
import { applyCssVars as T } from "../utils/css.js";
const g = (param)=>{
    let { title: a, description: l } = param;
    const o = typeof l == "string" ? [
        l
    ] : l;
    return /* @__PURE__ */ c("div", {
        className: p,
        children: [
            /* @__PURE__ */ r(P, {
                regular: !0,
                color: e.colors.textPrimary,
                children: a
            }),
            o && o.map((t, i)=>/* @__PURE__ */ r(n, {
                    regular: !0,
                    color: e.colors.textSecondary,
                    as: "p",
                    children: t
                }, i))
        ]
    });
}, E = (param)=>{
    let { title: a, stackingGroup: l, description: o, "aria-label": t } = param;
    return /* @__PURE__ */ r("div", {
        "aria-label": t,
        children: /* @__PURE__ */ c(d, {
            space: "between",
            alignItems: "center",
            children: [
                a && /* @__PURE__ */ r(u, {
                    paddingRight: 32,
                    children: /* @__PURE__ */ r(n, {
                        regular: !0,
                        children: a
                    })
                }),
                l || /* @__PURE__ */ r(n, {
                    regular: !0,
                    color: e.colors.textSecondary,
                    textAlign: "right",
                    as: "div",
                    children: o
                })
            ]
        })
    });
}, F = (param)=>{
    let { image: a, description: l, "aria-label": o } = param;
    return /* @__PURE__ */ r("div", {
        "aria-label": o,
        children: /* @__PURE__ */ c(d, {
            space: 16,
            alignItems: "center",
            children: [
                /* @__PURE__ */ r("div", {
                    style: T({
                        [B.mediaBorderRadius]: e.borderRadii.mediaSmall
                    }),
                    children: a
                }),
                /* @__PURE__ */ r(n, {
                    regular: !0,
                    color: e.colors.textSecondary,
                    children: l
                })
            ]
        })
    });
}, J = (param)=>{
    let { title: a, description: l, secondaryValue: o, value: t, valueColor: i = e.colors.textPrimary, "aria-label": s } = param;
    return /* @__PURE__ */ c(d, {
        space: "between",
        alignItems: "flex-end",
        "aria-label": s,
        children: [
            /* @__PURE__ */ r(g, {
                title: a,
                description: l
            }),
            /* @__PURE__ */ c("div", {
                className: R(p, S),
                children: [
                    /* @__PURE__ */ r(n, {
                        regular: !0,
                        color: e.colors.textSecondary,
                        decoration: "line-through",
                        children: o
                    }),
                    /* @__PURE__ */ r(I, {
                        color: i,
                        children: t
                    })
                ]
            })
        ]
    });
}, K = (param)=>{
    let { headline: a, headings: l, mainHeading: o, secondHeading: t, secondaryValue: i, title: s, description: h, valueColor: b = e.colors.textPrimary, "aria-label": v } = param;
    return /* @__PURE__ */ c("div", {
        "aria-label": v,
        children: [
            a && /* @__PURE__ */ r(u, {
                paddingBottom: 24,
                children: a
            }),
            /* @__PURE__ */ c(y, {
                space: 2,
                children: [
                    l && l.map((x, f)=>/* @__PURE__ */ {
                        var _x_valueColor;
                        return c(d, {
                            space: 8,
                            alignItems: "baseline",
                            children: [
                                /* @__PURE__ */ r(m, {
                                    color: (_x_valueColor = x.valueColor) !== null && _x_valueColor !== void 0 ? _x_valueColor : e.colors.textPrimary,
                                    children: x.value
                                }),
                                /* @__PURE__ */ r(n, {
                                    regular: !0,
                                    color: e.colors.textSecondary,
                                    children: x.text
                                })
                            ]
                        }, f);
                    }),
                    o && /* @__PURE__ */ c(d, {
                        space: 8,
                        alignItems: "baseline",
                        children: [
                            /* @__PURE__ */ r(m, {
                                color: b,
                                children: o.value
                            }),
                            /* @__PURE__ */ r(n, {
                                regular: !0,
                                color: e.colors.textSecondary,
                                children: o.text
                            })
                        ]
                    }),
                    t && /* @__PURE__ */ c(d, {
                        space: 8,
                        alignItems: "baseline",
                        children: [
                            /* @__PURE__ */ r(m, {
                                color: b,
                                children: t.value
                            }),
                            /* @__PURE__ */ r(n, {
                                regular: !0,
                                color: e.colors.textSecondary,
                                children: t.text
                            })
                        ]
                    }),
                    i && /* @__PURE__ */ r(m, {
                        color: e.colors.textSecondary,
                        children: i
                    })
                ]
            }),
            s || h ? /* @__PURE__ */ r(u, {
                paddingTop: 8,
                children: /* @__PURE__ */ r(g, {
                    title: s,
                    description: h
                })
            }) : null
        ]
    });
}, M = (param)=>{
    let { title: a, value: l, description: o, valueColor: t = e.colors.textPrimary, "aria-label": i } = param;
    return /* @__PURE__ */ c("div", {
        "aria-label": i,
        className: p,
        children: [
            /* @__PURE__ */ r(n, {
                regular: !0,
                color: e.colors.textPrimary,
                children: a
            }),
            /* @__PURE__ */ r(m, {
                color: t,
                children: l
            }),
            /* @__PURE__ */ r(g, {
                description: o
            })
        ]
    });
}, O = (param)=>{
    let { title: a, stackingGroup: l, progressPercent: o, reverse: t, heading: i, description: s, "aria-label": h } = param;
    return /* @__PURE__ */ r("div", {
        "aria-label": h,
        children: /* @__PURE__ */ c(y, {
            space: 8,
            children: [
                /* @__PURE__ */ c(d, {
                    space: "between",
                    alignItems: "flex-end",
                    children: [
                        /* @__PURE__ */ r(u, {
                            paddingRight: 32,
                            children: /* @__PURE__ */ r(n, {
                                regular: !0,
                                children: a
                            })
                        }),
                        l
                    ]
                }),
                o !== void 0 && /* @__PURE__ */ r(k, {
                    "aria-hidden": !0,
                    progressPercent: o,
                    reverse: t
                }),
                /* @__PURE__ */ c(d, {
                    space: 8,
                    alignItems: "baseline",
                    children: [
                        /* @__PURE__ */ r(m, {
                            color: i.valueColor || e.colors.textPrimary,
                            children: i.value
                        }),
                        /* @__PURE__ */ r(n, {
                            regular: !0,
                            color: e.colors.textSecondary,
                            children: i.text
                        })
                    ]
                }),
                s && /* @__PURE__ */ r(n, {
                    regular: !0,
                    color: e.colors.textSecondary,
                    children: s
                })
            ]
        })
    });
};
export { K as HighlightedValueBlock, J as InformationBlock, O as ProgressBlock, E as RowBlock, F as SimpleBlock, M as ValueBlock };
